<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.52">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>14&nbsp; Models for non-independence – linear mixed models – Statistics for the Experimental Bench Biologist</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../chapters/two-factors.html" rel="prev">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script src="../site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="../site_libs/lightable-0.0.1/lightable.css" rel="stylesheet">

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../chapters/lmm.html"><span class="chapter-number">14</span>&nbsp; <span class="chapter-title">Models for non-independence – linear mixed models</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../">Statistics for the Experimental Bench Biologist</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Preface</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../intro.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Analyzing experimental data with a linear model</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/setup.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Getting Started – R Projects and R Markdown</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/reading-writing-wrangling.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Data – Reading, Wrangling, and Writing</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/plotting-models.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Plotting models</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/uncertainty-sd-and-se.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Uncertainty – Standard Deviations, Standard Errors, and Confidence Intervals</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/p-values.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">P-values</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/inference-errors.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Errors in inference</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/intro-stat-modeling.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">An introduction to linear models</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/continuous_x.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">Linear models with a single, continuous <em>X</em> (“linear regression”)</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/one-factor.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">Linear models with a single, categorical <em>X</em> (“t-tests” and “ANOVA”)</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/model-checking.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">11</span>&nbsp; <span class="chapter-title">Model checking</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/multiple-testing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">12</span>&nbsp; <span class="chapter-title">Multiple tests – why and when to adjust <em>p</em>-values</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/two-factors.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">13</span>&nbsp; <span class="chapter-title">Linear models with two categorical <span class="math inline">\(X\)</span> – Factorial linear models (“two-way ANOVA”)</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/lmm.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">14</span>&nbsp; <span class="chapter-title">Models for non-independence – linear mixed models</span></span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#liberal-inference-from-pseudoreplication" id="toc-liberal-inference-from-pseudoreplication" class="nav-link active" data-scroll-target="#liberal-inference-from-pseudoreplication"><span class="header-section-number">14.1</span> Liberal inference from pseudoreplication</a></li>
  <li><a href="#conservative-inference-from-failure-to-identify-blocks" id="toc-conservative-inference-from-failure-to-identify-blocks" class="nav-link" data-scroll-target="#conservative-inference-from-failure-to-identify-blocks"><span class="header-section-number">14.2</span> Conservative inference from failure to identify blocks</a></li>
  <li><a href="#introduction-to-models-for-non-independent-data-linear-mixed-models" id="toc-introduction-to-models-for-non-independent-data-linear-mixed-models" class="nav-link" data-scroll-target="#introduction-to-models-for-non-independent-data-linear-mixed-models"><span class="header-section-number">14.3</span> Introduction to models for non-independent data (linear mixed models)</a></li>
  <li><a href="#experimental-designs-in-experimental-bench-biology" id="toc-experimental-designs-in-experimental-bench-biology" class="nav-link" data-scroll-target="#experimental-designs-in-experimental-bench-biology"><span class="header-section-number">14.4</span> Experimental designs in experimental bench biology</a>
  <ul>
  <li><a href="#completely-randomized-design-crd" id="toc-completely-randomized-design-crd" class="nav-link" data-scroll-target="#completely-randomized-design-crd"><span class="header-section-number">14.4.1</span> Completely Randomized Design (CRD)</a></li>
  <li><a href="#completely-randomized-design-with-subsampling-crds-or-nested-design" id="toc-completely-randomized-design-with-subsampling-crds-or-nested-design" class="nav-link" data-scroll-target="#completely-randomized-design-with-subsampling-crds-or-nested-design"><span class="header-section-number">14.4.2</span> Completely Randomized Design with Subsampling (CRDS) or “Nested Design”</a></li>
  <li><a href="#randomized-complete-block-design-rcbd" id="toc-randomized-complete-block-design-rcbd" class="nav-link" data-scroll-target="#randomized-complete-block-design-rcbd"><span class="header-section-number">14.4.3</span> Randomized Complete Block Design (RCBD)</a></li>
  <li><a href="#randomized-split-plot-design-rspd" id="toc-randomized-split-plot-design-rspd" class="nav-link" data-scroll-target="#randomized-split-plot-design-rspd"><span class="header-section-number">14.4.4</span> Randomized Split Plot Design (RSPD)</a></li>
  <li><a href="#generalized-randomized-complete-block-design-grcbd" id="toc-generalized-randomized-complete-block-design-grcbd" class="nav-link" data-scroll-target="#generalized-randomized-complete-block-design-grcbd"><span class="header-section-number">14.4.5</span> Generalized Randomized Complete Block Design (GRCBD)</a></li>
  <li><a href="#nested-randomized-complete-block-design-nrcbd" id="toc-nested-randomized-complete-block-design-nrcbd" class="nav-link" data-scroll-target="#nested-randomized-complete-block-design-nrcbd"><span class="header-section-number">14.4.6</span> Nested Randomized Complete Block Design (NRCBD)</a></li>
  <li><a href="#longitudinal-randomized-complete-block-design-lrcbd" id="toc-longitudinal-randomized-complete-block-design-lrcbd" class="nav-link" data-scroll-target="#longitudinal-randomized-complete-block-design-lrcbd"><span class="header-section-number">14.4.7</span> Longitudinal Randomized Complete Block Design (LRCBD)</a></li>
  <li><a href="#variations-due-to-multiple-measures-of-the-response-variable" id="toc-variations-due-to-multiple-measures-of-the-response-variable" class="nav-link" data-scroll-target="#variations-due-to-multiple-measures-of-the-response-variable"><span class="header-section-number">14.4.8</span> Variations due to multiple measures of the response variable</a></li>
  </ul></li>
  <li><a href="#building-the-linear-mixed-model-for-clustered-data" id="toc-building-the-linear-mixed-model-for-clustered-data" class="nav-link" data-scroll-target="#building-the-linear-mixed-model-for-clustered-data"><span class="header-section-number">14.5</span> Building the linear (mixed) model for clustered data</a></li>
  <li><a href="#statistical-models-for-experimental-designs" id="toc-statistical-models-for-experimental-designs" class="nav-link" data-scroll-target="#statistical-models-for-experimental-designs"><span class="header-section-number">14.6</span> Statistical models for experimental designs</a>
  <ul>
  <li><a href="#models-for-completely-randomized-designs-crd" id="toc-models-for-completely-randomized-designs-crd" class="nav-link" data-scroll-target="#models-for-completely-randomized-designs-crd"><span class="header-section-number">14.6.1</span> Models for Completely Randomized Designs (CRD)</a></li>
  <li><a href="#models-for-batched-data-crds-rcbd-rspd-grcbd-nrcbd" id="toc-models-for-batched-data-crds-rcbd-rspd-grcbd-nrcbd" class="nav-link" data-scroll-target="#models-for-batched-data-crds-rcbd-rspd-grcbd-nrcbd"><span class="header-section-number">14.6.2</span> Models for batched data (CRDS, RCBD, RSPD, GRCBD, NRCBD)</a>
  <ul class="collapse">
  <li><a href="#linear-models" id="toc-linear-models" class="nav-link" data-scroll-target="#linear-models"><span class="header-section-number">14.6.2.1</span> Linear models</a></li>
  <li><a href="#linear-mixed-models-random-effects-models" id="toc-linear-mixed-models-random-effects-models" class="nav-link" data-scroll-target="#linear-mixed-models-random-effects-models"><span class="header-section-number">14.6.2.2</span> linear mixed models (“random effects” models)</a></li>
  <li><a href="#anova-models-mixed-models-repeated-measures-anova" id="toc-anova-models-mixed-models-repeated-measures-anova" class="nav-link" data-scroll-target="#anova-models-mixed-models-repeated-measures-anova"><span class="header-section-number">14.6.2.3</span> ANOVA models (“mixed models”, “repeated measures ANOVA”)</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#which-model-and-why" id="toc-which-model-and-why" class="nav-link" data-scroll-target="#which-model-and-why"><span class="header-section-number">14.7</span> Which model and why?</a>
  <ul>
  <li><a href="#crd" id="toc-crd" class="nav-link" data-scroll-target="#crd"><span class="header-section-number">14.7.1</span> CRD</a></li>
  <li><a href="#crds" id="toc-crds" class="nav-link" data-scroll-target="#crds"><span class="header-section-number">14.7.2</span> CRDS</a></li>
  <li><a href="#rcbd" id="toc-rcbd" class="nav-link" data-scroll-target="#rcbd"><span class="header-section-number">14.7.3</span> RCBD</a>
  <ul class="collapse">
  <li><a href="#so-which-model-and-why-a-simulation-to-help-decide" id="toc-so-which-model-and-why-a-simulation-to-help-decide" class="nav-link" data-scroll-target="#so-which-model-and-why-a-simulation-to-help-decide"><span class="header-section-number">14.7.3.1</span> So, which model, and why? A simulation to help decide</a></li>
  <li><a href="#so-which-model-and-why-a-simulators-guide-to-best-practices" id="toc-so-which-model-and-why-a-simulators-guide-to-best-practices" class="nav-link" data-scroll-target="#so-which-model-and-why-a-simulators-guide-to-best-practices"><span class="header-section-number">14.7.3.2</span> So, which model, and why? A simulator’s guide to best practices</a></li>
  </ul></li>
  <li><a href="#rspd" id="toc-rspd" class="nav-link" data-scroll-target="#rspd"><span class="header-section-number">14.7.4</span> RSPD</a></li>
  <li><a href="#rcbds" id="toc-rcbds" class="nav-link" data-scroll-target="#rcbds"><span class="header-section-number">14.7.5</span> RCBDS</a></li>
  <li><a href="#grcbd" id="toc-grcbd" class="nav-link" data-scroll-target="#grcbd"><span class="header-section-number">14.7.6</span> GRCBD</a></li>
  <li><a href="#grcbds" id="toc-grcbds" class="nav-link" data-scroll-target="#grcbds"><span class="header-section-number">14.7.7</span> GRCBDS</a></li>
  </ul></li>
  <li><a href="#example-1-crd-with-subsampling-classical-equivalent-nested-anova" id="toc-example-1-crd-with-subsampling-classical-equivalent-nested-anova" class="nav-link" data-scroll-target="#example-1-crd-with-subsampling-classical-equivalent-nested-anova"><span class="header-section-number">14.8</span> Example 1: CRD with subsampling (classical equivalent: Nested ANOVA)</a>
  <ul>
  <li><a href="#model-fit-and-inference" id="toc-model-fit-and-inference" class="nav-link" data-scroll-target="#model-fit-and-inference"><span class="header-section-number">14.8.1</span> Model fit and inference</a>
  <ul class="collapse">
  <li><a href="#model-fit" id="toc-model-fit" class="nav-link" data-scroll-target="#model-fit"><span class="header-section-number">14.8.1.1</span> Model fit</a></li>
  <li><a href="#inference" id="toc-inference" class="nav-link" data-scroll-target="#inference"><span class="header-section-number">14.8.1.2</span> Inference</a></li>
  <li><a href="#plot-the-model" id="toc-plot-the-model" class="nav-link" data-scroll-target="#plot-the-model"><span class="header-section-number">14.8.1.3</span> Plot the model</a></li>
  </ul></li>
  <li><a href="#what-the-researchers-did-classical-nested-anova" id="toc-what-the-researchers-did-classical-nested-anova" class="nav-link" data-scroll-target="#what-the-researchers-did-classical-nested-anova"><span class="header-section-number">14.8.2</span> What the researchers did: Classical nested ANOVA</a></li>
  <li><a href="#what-if-the-researchers-had-ignored-the-non-independence" id="toc-what-if-the-researchers-had-ignored-the-non-independence" class="nav-link" data-scroll-target="#what-if-the-researchers-had-ignored-the-non-independence"><span class="header-section-number">14.8.3</span> What if the researchers had ignored the non-independence?</a></li>
  <li><a href="#a-deeper-dive-into-the-random-intercepts" id="toc-a-deeper-dive-into-the-random-intercepts" class="nav-link" data-scroll-target="#a-deeper-dive-into-the-random-intercepts"><span class="header-section-number">14.8.4</span> A deeper dive into the random intercepts</a></li>
  </ul></li>
  <li><a href="#example-2-rcbd-with-2-groups-classical-equivalent-paired-t-test" id="toc-example-2-rcbd-with-2-groups-classical-equivalent-paired-t-test" class="nav-link" data-scroll-target="#example-2-rcbd-with-2-groups-classical-equivalent-paired-t-test"><span class="header-section-number">14.9</span> Example 2: RCBD with 2 groups (classical equivalent: paired <em>t</em>-test)</a>
  <ul>
  <li><a href="#lmm1-random-intercept-model" id="toc-lmm1-random-intercept-model" class="nav-link" data-scroll-target="#lmm1-random-intercept-model"><span class="header-section-number">14.9.1</span> lmm1 – random intercept model</a></li>
  <li><a href="#aov2-repeated-measures-anova-rm-anova" id="toc-aov2-repeated-measures-anova-rm-anova" class="nav-link" data-scroll-target="#aov2-repeated-measures-anova-rm-anova"><span class="header-section-number">14.9.2</span> aov2 – repeated measures ANOVA (RM-ANOVA)</a></li>
  <li><a href="#pptt-pairwise-paired-t-test" id="toc-pptt-pairwise-paired-t-test" class="nav-link" data-scroll-target="#pptt-pairwise-paired-t-test"><span class="header-section-number">14.9.3</span> pptt – pairwise paired t-test</a></li>
  <li><a href="#notes-on-the-three-models" id="toc-notes-on-the-three-models" class="nav-link" data-scroll-target="#notes-on-the-three-models"><span class="header-section-number">14.9.4</span> Notes on the three models</a></li>
  <li><a href="#plot-the-model-1" id="toc-plot-the-model-1" class="nav-link" data-scroll-target="#plot-the-model-1"><span class="header-section-number">14.9.5</span> Plot the model</a></li>
  <li><a href="#better-know-the-linear-mixed-model" id="toc-better-know-the-linear-mixed-model" class="nav-link" data-scroll-target="#better-know-the-linear-mixed-model"><span class="header-section-number">14.9.6</span> Better Know the linear mixed model</a></li>
  </ul></li>
  <li><a href="#sec-lmm-example2" id="toc-sec-lmm-example2" class="nav-link" data-scroll-target="#sec-lmm-example2"><span class="header-section-number">14.10</span> Example 3: RCBD with &gt; 2 groups (classical equivalent: repeated measures ANOVA)</a>
  <ul>
  <li><a href="#lmm1-random-intercept-model-1" id="toc-lmm1-random-intercept-model-1" class="nav-link" data-scroll-target="#lmm1-random-intercept-model-1"><span class="header-section-number">14.10.1</span> lmm1 – random intercept model</a></li>
  <li><a href="#lmm2-linear-model-for-correlated-error" id="toc-lmm2-linear-model-for-correlated-error" class="nav-link" data-scroll-target="#lmm2-linear-model-for-correlated-error"><span class="header-section-number">14.10.2</span> lmm2 – linear model for correlated error</a></li>
  <li><a href="#aov2-repeated-measures-anova-rm-anova-multivariate-model" id="toc-aov2-repeated-measures-anova-rm-anova-multivariate-model" class="nav-link" data-scroll-target="#aov2-repeated-measures-anova-rm-anova-multivariate-model"><span class="header-section-number">14.10.3</span> aov2 – repeated measures ANOVA (RM-ANOVA), multivariate model</a></li>
  <li><a href="#aov1-repeated-measures-anova-rm-anova-univariate-model" id="toc-aov1-repeated-measures-anova-rm-anova-univariate-model" class="nav-link" data-scroll-target="#aov1-repeated-measures-anova-rm-anova-univariate-model"><span class="header-section-number">14.10.4</span> aov1 – repeated measures ANOVA (RM-ANOVA), univariate model</a></li>
  <li><a href="#pptt-pairwise-paired-t-test-1" id="toc-pptt-pairwise-paired-t-test-1" class="nav-link" data-scroll-target="#pptt-pairwise-paired-t-test-1"><span class="header-section-number">14.10.5</span> pptt – pairwise paired t-test</a></li>
  <li><a href="#notes-on-the-models" id="toc-notes-on-the-models" class="nav-link" data-scroll-target="#notes-on-the-models"><span class="header-section-number">14.10.6</span> Notes on the models</a></li>
  <li><a href="#plot-the-model-2" id="toc-plot-the-model-2" class="nav-link" data-scroll-target="#plot-the-model-2"><span class="header-section-number">14.10.7</span> Plot the model</a></li>
  </ul></li>
  <li><a href="#example-4-rcbd-with-two-factors-exp5c" id="toc-example-4-rcbd-with-two-factors-exp5c" class="nav-link" data-scroll-target="#example-4-rcbd-with-two-factors-exp5c"><span class="header-section-number">14.11</span> Example 4 – RCBD with two factors (exp5c)</a>
  <ul>
  <li><a href="#understand-the-data" id="toc-understand-the-data" class="nav-link" data-scroll-target="#understand-the-data"><span class="header-section-number">14.11.1</span> Understand the data</a></li>
  <li><a href="#examine-the-data" id="toc-examine-the-data" class="nav-link" data-scroll-target="#examine-the-data"><span class="header-section-number">14.11.2</span> Examine the data</a></li>
  <li><a href="#lmm-exp5c-m1" id="toc-lmm-exp5c-m1" class="nav-link" data-scroll-target="#lmm-exp5c-m1"><span class="header-section-number">14.11.3</span> Model fit and inference</a>
  <ul class="collapse">
  <li><a href="#check-the-model" id="toc-check-the-model" class="nav-link" data-scroll-target="#check-the-model"><span class="header-section-number">14.11.3.1</span> Check the model</a></li>
  <li><a href="#inference-from-the-model" id="toc-inference-from-the-model" class="nav-link" data-scroll-target="#inference-from-the-model"><span class="header-section-number">14.11.3.2</span> Inference from the model</a></li>
  <li><a href="#lmm-exp5c-plotthemodel" id="toc-lmm-exp5c-plotthemodel" class="nav-link" data-scroll-target="#lmm-exp5c-plotthemodel"><span class="header-section-number">14.11.3.3</span> Plot the model</a></li>
  <li><a href="#alternaplot-the-model" id="toc-alternaplot-the-model" class="nav-link" data-scroll-target="#alternaplot-the-model"><span class="header-section-number">14.11.3.4</span> Alternaplot the model</a></li>
  </ul></li>
  <li><a href="#why-we-care-about-modeling-batch-in-exp5c" id="toc-why-we-care-about-modeling-batch-in-exp5c" class="nav-link" data-scroll-target="#why-we-care-about-modeling-batch-in-exp5c"><span class="header-section-number">14.11.4</span> Why we care about modeling batch in exp5c</a></li>
  </ul></li>
  <li><a href="#sec-lmm-example4" id="toc-sec-lmm-example4" class="nav-link" data-scroll-target="#sec-lmm-example4"><span class="header-section-number">14.12</span> Example 4 – RCBDS Experiments with subsampling replication (exp1g)</a>
  <ul>
  <li><a href="#understand-the-data-1" id="toc-understand-the-data-1" class="nav-link" data-scroll-target="#understand-the-data-1"><span class="header-section-number">14.12.1</span> Understand the data</a></li>
  <li><a href="#examine-the-data-1" id="toc-examine-the-data-1" class="nav-link" data-scroll-target="#examine-the-data-1"><span class="header-section-number">14.12.2</span> Examine the data</a></li>
  <li><a href="#fit-the-model" id="toc-fit-the-model" class="nav-link" data-scroll-target="#fit-the-model"><span class="header-section-number">14.12.3</span> Fit the model</a></li>
  <li><a href="#inference-from-the-model-1" id="toc-inference-from-the-model-1" class="nav-link" data-scroll-target="#inference-from-the-model-1"><span class="header-section-number">14.12.4</span> Inference from the model</a></li>
  <li><a href="#plot-the-model-3" id="toc-plot-the-model-3" class="nav-link" data-scroll-target="#plot-the-model-3"><span class="header-section-number">14.12.5</span> Plot the model</a></li>
  <li><a href="#alternaplot-the-model-1" id="toc-alternaplot-the-model-1" class="nav-link" data-scroll-target="#alternaplot-the-model-1"><span class="header-section-number">14.12.6</span> Alternaplot the model</a></li>
  <li><a href="#understanding-the-alternative-models" id="toc-understanding-the-alternative-models" class="nav-link" data-scroll-target="#understanding-the-alternative-models"><span class="header-section-number">14.12.7</span> Understanding the alternative models</a></li>
  <li><a href="#the-varcorr-matrix-of-models-exp1g_m1a-and-exp1g_m1b" id="toc-the-varcorr-matrix-of-models-exp1g_m1a-and-exp1g_m1b" class="nav-link" data-scroll-target="#the-varcorr-matrix-of-models-exp1g_m1a-and-exp1g_m1b"><span class="header-section-number">14.12.8</span> The VarCorr matrix of models exp1g_m1a and exp1g_m1b</a></li>
  <li><a href="#the-linear-mixed-model-has-more-precision-and-power-than-the-fixed-effect-model-of-batch-means" id="toc-the-linear-mixed-model-has-more-precision-and-power-than-the-fixed-effect-model-of-batch-means" class="nav-link" data-scroll-target="#the-linear-mixed-model-has-more-precision-and-power-than-the-fixed-effect-model-of-batch-means"><span class="header-section-number">14.12.9</span> The linear mixed model has more precision and power than the fixed effect model of batch means</a></li>
  <li><a href="#fixed-effect-models-and-pseudoreplication" id="toc-fixed-effect-models-and-pseudoreplication" class="nav-link" data-scroll-target="#fixed-effect-models-and-pseudoreplication"><span class="header-section-number">14.12.10</span> Fixed effect models and pseudoreplication</a></li>
  <li><a href="#mixed-effect-anova" id="toc-mixed-effect-anova" class="nav-link" data-scroll-target="#mixed-effect-anova"><span class="header-section-number">14.12.11</span> Mixed-effect ANOVA</a></li>
  </ul></li>
  <li><a href="#working-in-r" id="toc-working-in-r" class="nav-link" data-scroll-target="#working-in-r"><span class="header-section-number">14.13</span> Working in R</a>
  <ul>
  <li><a href="#plotting-models-fit-to-batched-data" id="toc-plotting-models-fit-to-batched-data" class="nav-link" data-scroll-target="#plotting-models-fit-to-batched-data"><span class="header-section-number">14.13.1</span> Plotting models fit to batched data</a>
  <ul class="collapse">
  <li><a href="#models-without-subsampling---example-3-experiment-2c-right-panel" id="toc-models-without-subsampling---example-3-experiment-2c-right-panel" class="nav-link" data-scroll-target="#models-without-subsampling---example-3-experiment-2c-right-panel"><span class="header-section-number">14.13.1.1</span> Models without subsampling - Example 3 (Experiment 2c right panel)</a></li>
  <li><a href="#models-with-subsampling---experiment-1g" id="toc-models-with-subsampling---experiment-1g" class="nav-link" data-scroll-target="#models-with-subsampling---experiment-1g"><span class="header-section-number">14.13.1.2</span> Models with subsampling - Experiment 1g</a></li>
  </ul></li>
  <li><a href="#repeated-measures-anova-randomized-complete-block-with-no-subsampling" id="toc-repeated-measures-anova-randomized-complete-block-with-no-subsampling" class="nav-link" data-scroll-target="#repeated-measures-anova-randomized-complete-block-with-no-subsampling"><span class="header-section-number">14.13.2</span> Repeated measures ANOVA (randomized complete block with no subsampling)</a>
  <ul class="collapse">
  <li><a href="#univariate-vs.-multivariate-repeated-measures-anova" id="toc-univariate-vs.-multivariate-repeated-measures-anova" class="nav-link" data-scroll-target="#univariate-vs.-multivariate-repeated-measures-anova"><span class="header-section-number">14.13.2.1</span> univariate vs.&nbsp;multivariate repeated measures ANOVA</a></li>
  <li><a href="#the-anova-table" id="toc-the-anova-table" class="nav-link" data-scroll-target="#the-anova-table"><span class="header-section-number">14.13.2.2</span> The ANOVA table</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#hidden-code" id="toc-hidden-code" class="nav-link" data-scroll-target="#hidden-code"><span class="header-section-number">14.14</span> Hidden code</a>
  <ul>
  <li><a href="#import-exp5c" id="toc-import-exp5c" class="nav-link" data-scroll-target="#import-exp5c"><span class="header-section-number">14.14.1</span> Import exp5c</a></li>
  <li><a href="#import-exp1g" id="toc-import-exp1g" class="nav-link" data-scroll-target="#import-exp1g"><span class="header-section-number">14.14.2</span> Import exp1g</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title"><span id="sec-lmm" class="quarto-section-identifier"><span class="chapter-number">14</span>&nbsp; <span class="chapter-title">Models for non-independence – linear mixed models</span></span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<p>Probably no chapter in this book is more important for the best-practice analysis of experimental data than this chapter. Why? Because many if not most experimental data violates the assumption of independence and any analysis using standard <em>t</em>-tests and ANOVA will always lead to quantitative error in inference (confidence intervals and <em>p</em>-values) and often lead to qualitative errors in inference (statements about “significance”). Standard analysis of non-independent data can lead to absurdly liberal inference (the <em>p</em>-values are far lower than the data support), but can also lead to moderately conservative inference (the <em>p</em>-values are higher than the data supports). Liberal inference generates false discovery and lures researchers down dead-end research pathways. Conservative inference steers researchers away from true discovery.</p>
<p><strong>Linear Mixed Models</strong> are an extension of linear models that appropriately adjust inferential statistics for non-independent data. <strong>Paired t-tests</strong> and <strong>Repeated Measures ANOVA</strong> are classical tests that are special cases of linear mixed models. Linear mixed models are more flexible than these classical tests because the models can include added covariates or more complex models generally. And, linear mixed models can be extended to <strong>Generalized Linear Mixed Models</strong> for counts, binary responses, skewed responses, and ratios.</p>
<p>Before introducing experimental designs that generate non-independent data and the models used to analyze these, let’s explore two ubiquitous examples, one that leads to liberal inference and one that leads to conservative inference.</p>
<section id="liberal-inference-from-pseudoreplication" class="level2" data-number="14.1">
<h2 data-number="14.1" class="anchored" data-anchor-id="liberal-inference-from-pseudoreplication"><span class="header-section-number">14.1</span> Liberal inference from pseudoreplication</h2>
<p>Researchers are interested in regulation and repair of DNA double-stranded breaks and use a proximity ligation assay (PLA) of HeLa cells to investigate the number of damage response events (“foci”) per cell with and without an inhibitor of transcription elongation (DRB). The number of foci in each of fifty cells per treatment is measured. The experiment is replicated three times. The researchers use a <em>t</em>-test to compare the effect of DRB on foci count and naively include all measures in the analysis.</p>
<p>What is naive about the analysis? The fifty measures per cell are <strong>technical replicates</strong> and the values within a cell are not independent of each other because they share aspects of cell environment not shared by values in other cells. Including technical replicates in an analysis without accounting for this non-independence is a kind of <strong>pseudoreplication</strong></p>
<p>To show how this naive analysis results in extremely liberal inference and an increase in false discovery, I simulate this experiment using a case in which there is no effect of DRB treatment, so a low <em>p</em>-value indicates a false-discovery. The simulation is simplified with the two following conditions: 1) the model pretends that count data are normally distributed (this is because we want to focus on pseudoreplication and not a misspecified distribution) and 2) the model pretends that values from each treatment within an experiment are independent (this is because we want to focus on pseudoreplication).</p>
<p>In this naive analysis of the experiment, the researcher finds an effect of treatment with a <em>p</em>-value of 0.000073 and uses this small <em>p</em>-value to justify a decision to move forward with follow-up experiments. But this low <em>p</em>-value is not supported by the data – this discovery is false. This very small <em>p</em>-value is not an example of a “rare event”. In fact, if the researcher repeats the experiment 1000 times, then the median <em>p</em>-value is 0.000195 and 72.3% of the 1000 <em>p</em>-values lead to the same false discovery if 0.05 is used to make the decision to move forward.</p>
<p>While this example is fake, I see this naive analysis a lot: tumor area of multiple tumors per mouse, islet area of multiple islets per mouse, number of vesicles docked to a membrane in multiple cells per mouse, the number of mitochondria in multiple cells per mouse, the number of neurites in multiple neurons in multiple cells per mouse, etc. etc. Indeed, when I’m looking for examples of pseudoreplication to teach, I just look for figures with a bunch of points per treatment – something similar to this plot of the fake data experiment. Regardless, this is a huge source of false discovery that could disappear overnight.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-lmm-pseudoreplication-fake-example" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-lmm-pseudoreplication-fake-example-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="lmm_files/figure-html/fig-lmm-pseudoreplication-fake-example-1.png" class="img-fluid figure-img" width="384">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-lmm-pseudoreplication-fake-example-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;14.1: A fake example of pseudoreplication motivated by a real experiment.
</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="conservative-inference-from-failure-to-identify-blocks" class="level2" data-number="14.2">
<h2 data-number="14.2" class="anchored" data-anchor-id="conservative-inference-from-failure-to-identify-blocks"><span class="header-section-number">14.2</span> Conservative inference from failure to identify blocks</h2>
<p>Researchers are investigating the regulation of the differentiation of stem blood cells into osteoclasts, which can cause osteoporosis if overactivated. The researchers randomly sample three mice from 6 litters and use a <strong>littermate control</strong> design: within a litter, one sib is assigned to control (CN), one to glucortacoid (GC) treatment, and one to buthionine sulphoximine (BSO) treatment. The researchers are investigating the mechanism of GC-induced osteoporosis and if the researcher’s model is correct, then BSO should block this mechanism. The response variable is Bone Mineral Density. Following an ANOVA, the researchers report the unadjusted <em>p</em>-value for each pairwise comparison with the expectation that BSO will reverse the effect of GC on Bone Mineral Density.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-lmm-failed-discovery-example" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-lmm-failed-discovery-example-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="lmm_files/figure-html/fig-lmm-failed-discovery-example-1.png" class="img-fluid figure-img" width="768">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-lmm-failed-discovery-example-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;14.2: Results of the fake experiment. A) results using ANOVA using on initial six litters, B) published results with four additional litters, C) results using LMM on initial six litters
</figcaption>
</figure>
</div>
</div>
</div>
<p>What is naive about the analysis? Any value measured from sibling mice within a litter are not independent of each other because the sibling mice share aspects of genetics and maternal environment not shared by mice in other litters. This shared variance adds correlated noise to data and failure to account for the shared variance will almost always lead to more conservative inference.</p>
<p>In the experiment, the results “looked” like the expected results if the model GC-induced osteoporosis were correct, but the <em>p</em>-value of the focal contrast (BSO - GC) was not &lt; 0.05 (<a href="#fig-lmm-failed-discovery-example" class="quarto-xref">Figure&nbsp;<span>14.2</span></a> A). The researchers expanded the experiment, adding data from four more litters (<a href="#fig-lmm-failed-discovery-example" class="quarto-xref">Figure&nbsp;<span>14.2</span></a> B). Had the researchers analyzed the initial set of data using a statistical model that accounts for the shared variance within a litter (<a href="#fig-lmm-failed-discovery-example" class="quarto-xref">Figure&nbsp;<span>14.2</span></a> C), the researchers would have been satisfied and fewer mice would have been killed, fewer resources used, and more time to pursue continued probing of the mechanism. The statistical model used to take advantage of the littermate control design is a <strong>linear-mixed model</strong>. An experiment with littermate control is known as a <strong>blocked design</strong>.</p>
<div class="callout callout-style-default callout-warning callout-titled" title="NHST Blues">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
NHST Blues
</div>
</div>
<div class="callout-body-container callout-body">
<p>NHST encourages peeking at the data to see if p &lt; 0.05 for a focal contrast, and collecting more data if this isn’t the case. Don’t do this. Peeking makes the <em>p</em>-value non-valid and will increase the false discovery rate. Peeking is a variation of the multiple test problem. In clinical trials there are statistically rigorous methods for peeking, which allows a trial to stop early.</p>
</div>
</div>
<p>The small <em>p</em>-value in the BSO - GC contrast using a linear mixed model is not an example of a “rare event”. In fact, if the researchers repeated the experiment 1000 times, 89.1% of the BSO - GC <em>p</em>-values using the linear mixed model are less than 0.05 while only 29.9% of the <em>p</em>-values following classical ANOVA are less than 0.05.</p>
<p>While this example is fake, I see this naive analysis a lot – littermate controls is very, very common but other examples are too, including replicated experiments (each experiment is a block). There are certain instances where researchers do recognize non-independence and do use a paired <em>t</em>-test but the vast majority of blocked designs (occurring in almost all experimental biology papers) go unrecognized. This is a huge source of failed discovery that could disappear overnight.</p>
</section>
<section id="introduction-to-models-for-non-independent-data-linear-mixed-models" class="level2" data-number="14.3">
<h2 data-number="14.3" class="anchored" data-anchor-id="introduction-to-models-for-non-independent-data-linear-mixed-models"><span class="header-section-number">14.3</span> Introduction to models for non-independent data (linear mixed models)</h2>
<p>This chapter is about models for <strong>correlated error</strong>, including linear models with added <strong>random factors</strong>, which are known as <strong>linear mixed models</strong>. In classical hypothesis testing, a <strong>paired t-test</strong>, <strong>repeated measures ANOVA</strong>, and <strong>mixed-effect ANOVA</strong> are equivalent to specific cases of linear mixed models. Linear mixed models are used for analyzing data composed of subsets – or <strong>batches</strong> – of data that were measured from the “same thing”, such as multiple measures within a mouse, or multiple mice within a litter. Batched data results in <strong>correlated error</strong>, which violates a key assumption of linear models (and their “which test” equivalents) and muddles statistical inference unless the correlated error is modeled, explicitly or implicitly. In some experimental designs (<strong>blocked designs</strong>), failure to model the correlated error reduces precision and power, contributing to reduced rates of discovery or confirmation. In other designs (<strong>nested designs</strong>), failure to model the correlated error results in falsely high precision and low <em>p</em>-values, leading to increased rates of false discovery. The falsely high precision is due to <strong>pseudoreplication</strong>. I think it’s fair to infer from the experimental biology literature, that experimental biologists don’t recognize the ubiquitousness of batched data and correlated error. This is probably the biggest issue in inference in the field (far more of an issue than say, a <em>t</em>-test on non-normal data).</p>
<p>What do I mean by “batch” and how can correlated error both increase and decrease false discovery? Consider an experiment to measure pancreatic islet area with the experimental factor <span class="math inline">\(\texttt{treatment}\)</span> with levels “Cn” and “Tr”, where “Cn” is the administration of saline and “Tr” is the administration of a drug believed to disrupt the system. While it may seem like the data from this experiment should be analyzed using a <em>t</em>-test, the best practice statistical model actually depends on the <strong>experimental design</strong>. Experimental design matters because different designs introduce different patterns of <strong>correlated error</strong> due to shared genetics and environment. Recall that inference from a linear model (including <em>t</em>-tests and ANOVA) assumes <em>independence</em> (<a href="intro-stat-modeling.html" class="quarto-xref"><span>Chapter 8</span></a>) – that is, each response value has no relationship to any other value, other than that due to treatment. Lack of independence results in patterns of correlation among the residuals, or correlated error.</p>
<p>Something like the first experiment below (Design 1) is the necessary design to use the statistics that have been covered in this book to this point, without extreme violation of the independence assumption. But many (most?) experiments in experimental bench biology do not look like the design in Design 1 below. Instead, many (most?) experiments are variants of Designs 2 and 3, both of which have extreme violations of the independence assumption. Design 2 and its variants result in liberal statistics and increased, false discovery rate but, interestingly, Design 3 and its variants (generally) result in conservative statistics and reduced, true discovery rate.</p>
<div id="fig-lmm-design-1" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-lmm-design-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="../images/experimental_designs/CRD.png" class="img-fluid figure-img" width="300">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-lmm-design-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;14.3: Design 1: Completely Randomized Design. Each treatment replicate (mouse) is separately housed. There is a single measure of islet area per mouse. Image created with BioRender.com
</figcaption>
</figure>
</div>
<p>Design 1. The design in <a href="#fig-lmm-design-1" class="quarto-xref">Figure&nbsp;<span>14.3</span></a> has a single factor <span class="math inline">\(\texttt{genotype}\)</span> with two levels (Cn and Tr). Five mice of the same sex, each from a different litter from a unique dam and sire mating, are randomly sampled and assigned to either Cn or Tr. All mice are housed individually (10 cages). The pancreatic tissue from all mice is prepared in a single batch and the area of a single islet is measured from each mouse. The entire experiment is carried out at the same time and each component (tissue preparation, measuring) is carried out by the same person (these could be different people for each component). This is a <strong>Completely Randomized Design</strong> (CRD). The five replicate mice per treatment are <strong>treatment replicates</strong> (often called <strong>biological replicates</strong> in experimental biology). A CRD does not have batched data.</p>
<div id="fig-lmm-design-2" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-lmm-design-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="../images/experimental_designs/CRDS.png" class="img-fluid figure-img" width="300">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-lmm-design-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;14.4: Design 2: Completely Randomized Design with subsampling. Each treatment replicate (mouse) is separately housed. There are ten measures of islet area per mouse. Image created with BioRender.com
</figcaption>
</figure>
</div>
<p>Design 2. The design in <a href="#fig-lmm-design-2" class="quarto-xref">Figure&nbsp;<span>14.4</span></a> is exactly like that in Design 1, except that the researchers take ten measures of iselet area per mouse. The ten measures are <strong>subsampled replicates</strong>. Experimental biologists often call these <strong>technical replicates</strong>, especially when the multiple measures are taken from the same preparation. Subsampling is a kind of <strong>nested design</strong> in which one variable is nested within (as opposed to crossed with) another variable. Here, the subsampled variable (subsample_id) is nested within the mouse_id variable. Each mouse is a batch. Each mouse has a unique set of factors that contribute to the error variance of the measures of the response in that mouse. All response measures within a mouse share the component of the error variance unique to that mouse and, as a consequence, the error (residuals) within a mouse are more similar to each other than they are to the residuals between mice</p>
<div id="fig-lmm-design-3" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-lmm-design-3-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="../images/experimental_designs/RCBD.png" class="img-fluid figure-img" width="300">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-lmm-design-3-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;14.5: Design 3: Randomized Complete Block Design. Pairs of treatment replicates (mouse) are siblings and housed together with no other mice. There is a single measure of islet area per mouse. Image created with BioRender.com
</figcaption>
</figure>
</div>
<p>Design 3. In the design in <a href="#fig-lmm-design-3" class="quarto-xref">Figure&nbsp;<span>14.5</span></a>, two littermates are randomly sampled from five litters, each with a different dam and sire. Within each litter, one mouse is assigned to Cn and a second is assigned to Tr. Each litter is randomly assigned to cage with only a single litter per cage. All other aspects of this design are as in Design 1. This is a <strong>Randomized Complete Block Design</strong>. The five replicate mice per treatment are the treatment replicates. Each litter/cage combination is a type of batch called a <strong>block</strong>. A blocked design typically functions to reduce noise in the model fit (this increases power) and to reduce the number of litters and cages needed for an experiment. The two measures of Islet Area within a litter/cage (one per mouse) are not independent of each other. Each cage has two mice from the same litter and these mice share genetic and maternal factors that contribute to mouse anatomy and physiology that are not shared by mice in other litters. Additionally, each cage has a unique set of environmental factors that contribute to the error variance of the measure of the response. Each cage shares a cage-specific history of temperature, humidity, food, light, interactions with animal facilities staff, and behavioral interactions among the mice. All response measures within a litter/cage share the component of the error variance unique to that litter/cage and, as a consequence, the error (residuals) within a litter/cage are more similar to each other than they are to the residuals among litters/cages.</p>
<p>In each of these experiments, there is systematic variation at multiple levels: among treatments due to treatment effects and among batches due to <strong>batch effects</strong>. Batches come in lots of flavors, including experiment, litter, cage, flask, plate, slide, donor, and individual. The among-treatment differences in means are the <strong>fixed effects</strong>. The among-batch differences are the <strong>random effects</strong>. An assumption of modeling random effects is that the batches are a random sample of the batches that could have been sampled. This is often not strictly true as batches are often <strong>convenience samples</strong> (example: the human donors of the Type 2 diabetes beta cells are those that were in the hospital).</p>
<p>The variation among batches/lack of independence within batches has different consequences on the uncertainty of the estimate of a treatment effect. The batches in Experiment 3 contain both treatments (Cn and Tr). The researcher is interested in the treatment effect but not the variation due to differences among the batches. The batches are nuissance factors that add additional variance to the response, with the consequence that estimates of treatment effects are less precise, unless the variance due to the batches is explicitly modeled. <strong>Modeling a batch that contains some or all treatment combinations will increase precision and power</strong>.</p>
<p>Batches that contain more than one treatment combination are known as <strong>blocks</strong>. A block that contains all treatment combinations is a <strong>complete block</strong>. A block that contains fewer than all combinations is an <strong>incomplete block</strong>. Including block structure in the design is known as <strong>blocking</strong>. Blocks are non-experimental factors. Adding a blocking factor to a statistical model is used to increase the precision of an estimated treatment effect. Design 3 is an example of a <strong>randomized complete block design</strong>.</p>
<p>In Design 2, there are multiple measures per mouse and the design is a <strong>Completely Randomized Design with subsampling</strong>. The subsampling is not the kind of replication that can be used to infer the among treatment effect because the treatment assignment was not at the level of the subsamples. The <strong>treatment replicates</strong> are the mice, because <em>it was at this level that treatment assignment was randomized</em>. A statistical analysis of all measures from a subsampled design without modeling the correlated error due to the subsampling is a kind of <a href="https://en.wikipedia.org/wiki/Pseudoreplication" target="_blank">pseudoreplication</a>. Pseudoreplication results in falsely precise standard errors and false small <em>p</em>-values and, consequently, increased rates of false discovery.</p>
<p>In all of these designs, it is important for the researcher to identify the <strong>experimental unit</strong> and the <strong>measurement</strong> unit. The experimental unit is the entity that was randomly assigned the treatment. In all the designs above, the experimental unit is the mouse. In designs 1 and 3, the measurement unit is the mouse. In design 2, the measurement unit is the specific islet that was measured.</p>
<div class="callout callout-style-default callout-note callout-titled" title="Pseudoreplication">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Pseudoreplication
</div>
</div>
<div class="callout-body-container callout-body">
<p>In pseudoreplication, the degrees of freedom used to compute the test statistic and the <em>p</em>-value are inflated given the experimental design and research question. An example: A researcher wants to investigate the effect of some protein on mitochondrial biogenesis and designs an experiment with a wildtype (WT) and a conditional knockout (KO) mouse. Mitochondrial counts from twenty cells in one WT mouse and one KO mouse are measured and the researcher uses a <em>t</em>-test to compare counts. The sample size used to compute the standard error in the denominator of the <em>t</em>-value is 20. The <em>t</em>-distribution used to compute the <em>p</em>-value uses 38 df (20 measures times two groups minus two estimated parameters). This is wrong. The df are inflated and the estimate of the standard error of the difference (the denominator of the t-value) is falsely small. The correct sample size for this design is 1 and the correct df is zero. The sample size and df are inflated for this design because the treatment was randomized to mouse and not to cell. Mouse is the experimental unit – the number of experimental units is what gives the degrees of freedom. The df are correct for inference about the two individuals (how compatible are the data and a model of sampling from the same individual?), but not for inference about the effect of genotype. We cannot infer anything about genotype with a sample size of 1, even with 20 measures per mouse, because any effect of treatment is completely confounded with other differences between the two mice.</p>
</div>
</div>
</section>
<section id="experimental-designs-in-experimental-bench-biology" class="level2" data-number="14.4">
<h2 data-number="14.4" class="anchored" data-anchor-id="experimental-designs-in-experimental-bench-biology"><span class="header-section-number">14.4</span> Experimental designs in experimental bench biology</h2>
<p>Given the basic principles above, let’s consider the kinds of experimental designs seen in experimental bench biology (<a href="#fig-lmm-designs" class="quarto-xref">Figure&nbsp;<span>14.6</span></a>). For each of the experimental designs, I give examples with a certain number of treatments (t), blocks (b), treatment replications (r), and subsampled replications (s).</p>
<p>Notation for models</p>
<p>i = 1..t (treatments)</p>
<p>j = 1..b (blocks)</p>
<p>k = 1..r (treatment replications within a block or within a CRD with no block structure)</p>
<p>m = 1..s (subsamples or technical replicates)</p>
<p><img src="../images/experimental_designs/designs_1.png" class="img-fluid" width="800"></p>
<div id="fig-lmm-designs" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-lmm-designs-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="../images/experimental_designs/designs_2.png" class="img-fluid figure-img" width="550">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-lmm-designs-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;14.6: Experimental designs in experimental bench biology. Images created with BioRender.com
</figcaption>
</figure>
</div>
<section id="completely-randomized-design-crd" class="level3" data-number="14.4.1">
<h3 data-number="14.4.1" class="anchored" data-anchor-id="completely-randomized-design-crd"><span class="header-section-number">14.4.1</span> Completely Randomized Design (CRD)</h3>
<p>The <strong>Completely Randomized Design</strong> experiment in <a href="#fig-lmm-designs" class="quarto-xref">Figure&nbsp;<span>14.6</span></a> A has a single factor, <span class="math inline">\(\texttt{treatment}\)</span> with two levels (“Cn” and “Tr”). Five mice are randomly assigned to each treatment level. Each mouse is bred from a different litter and housed in a separate cage. The researchers measure a single value of the response variable from each mouse. The five replicate mice per treatment are the <strong>treatment (biological) replicates</strong>. The design is completely randomized because there is no subgrouping due to batches. What kinds of subgrouping does this design avoid?</p>
<p>By using a single mouse per litter, there are no litter batches and subsets of mice don’t share <strong>litter effects</strong> – common litter responses to the Cn or Tr treatments. Each litter has a unique set of factors that contribute to the error variance of the measure of the response. Siblings from the same dam and sire share more genetic variation than non-siblings and this shared genetic variation contributes to phenotypes (including the response to treatment) that are more likely to be similar to each other than to non-siblings. Siblings from the same litter share the same history of maternal factors (<strong>maternal effects</strong>, including <strong>epigenetic effects</strong>) specific to the pregnancy and even the history of events leading up to the pregnancy. This shared non-genetic and epigenetic variation contributes to phenotypes (including the response to treatment) that are more likely to be similar to each other than to non-siblings. All response measures within a litter share the genetic, maternal environmental, and epigenetic components of the error variance unique to that litter and, as a consequence, the error (residuals) within a litter are more similar to each other than they are to the residuals between litters.</p>
<p>By housing each mouse in it’s own cage, there is no cage batch and subsets of mice don’t share <strong>cage effects</strong> – common cage responses to the Cn or Tr treatments. As stated earlier, each cage has a unique set of factors that contribute to the error variance of the measure of the response. Each cage shares a cage-specific history of temperature, humidity, food, light, interactions with animal facilities staff and behavioral interactions among the mice. All response measures within a cage share the component of the error variance unique to that cage and, as a consequence, the error (residuals) within a cage are more similar to each other than they are to the residuals between cages.</p>
<p>Examples:</p>
<ol type="1">
<li>Ten mice from separate litters are sampled. Five mice are randomly assigned to control. Five mice are randomly assigned to treatment. A single measure per mouse is taken. Mouse is the experimental unit. <span class="math inline">\(t=2\)</span>, <span class="math inline">\(b=0\)</span>, <span class="math inline">\(r=5\)</span>, and <span class="math inline">\(s=1\)</span>.</li>
<li>Ten cell cultures are created. Five cultures are randomly assigned to control and five to treatment. A single measure per culture is taken. Culture is the experimental unit. <span class="math inline">\(t=2\)</span>, <span class="math inline">\(b=0\)</span>, <span class="math inline">\(r=5\)</span>, and <span class="math inline">\(s=1\)</span>.</li>
</ol>
<div class="callout callout-style-default callout-note callout-titled" title="Known Unknowns">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Known Unknowns
</div>
</div>
<div class="callout-body-container callout-body">
<p>While most data from experiments in bench biology are analyzed as if the experimental design is a CRD, a good question is, what fraction of these actually are CRD? We know that many (most?) mouse experiments will have batched measures and correlated responses because most experiments are conducted with multiple mice per litter and/or cage (or equivalents in other model systems). Many cell culture data come from multiple replicates of the whole experiment – the experiment functions as a block. And time series experiments include multiple measures on the same experimental unit over time. Except for time series experiments, researchers in experimental bench biology are using almost exclusively statistical tests that assume independence of errors (the tests appropriate for CRDs). How this mismatch between experimental design and statistical practice affects the rate of false and true discovery in cell and molecular biology is a <a href="https://en.wikipedia.org/wiki/There_are_unknown_unknowns" target="_blank">known unknown</a>.</p>
</div>
</div>
</section>
<section id="completely-randomized-design-with-subsampling-crds-or-nested-design" class="level3" data-number="14.4.2">
<h3 data-number="14.4.2" class="anchored" data-anchor-id="completely-randomized-design-with-subsampling-crds-or-nested-design"><span class="header-section-number">14.4.2</span> Completely Randomized Design with Subsampling (CRDS) or “Nested Design”</h3>
<p>The <strong>Completely Randomized Design with Subsampling</strong> experiment in <a href="#fig-lmm-designs" class="quarto-xref">Figure&nbsp;<span>14.6</span></a> B is exactly like the CRD except that the researchers measure multiple values of the response variable from each mouse <em>under the same condition</em> (that is, not different in treatment or time). The multiple measures are <strong>subsampled (technical) replicates</strong>.</p>
<ol type="1">
<li>Do not confuse subsampled replicates with measures of the response under different conditions in the same mouse, for example a measure from one brain slice under the control treatment and a measure from a second brain slice under the drug treatment. This example is a kind of Randomized Complete Block Design, which is outlined next and the core design in this chapter.</li>
<li>Do not confuse subsampled replicates with a measures of the response at different times in the same mouse, for example, the plasma glucose levels at baseline and at five post-baseline time points. This example is a kind of longitudinal design, which is outlined below and more thoroughly in the chapter <a href="#pre-post">Linear models for longitudinal experiments</a>.</li>
<li>Do not confuse subsampled replicates with measures of different response variables from the same mouse, for example measures of the weights of five different skeletal muscles. This example is a kind of <strong>multiple response</strong> which is addressed xxx.</li>
</ol>
<p>The technical replicates are a kind of <strong>pseudoreplication</strong>. The general linear model <code>y ~ treatment</code> fit to these data, including t-tests and traditional ANOVA, will have falsely high precision and falsely low <em>p</em>-values.</p>
<p>Examples:</p>
<ol type="1">
<li>Ten mice from separate litters are sampled. Five mice are randomly assigned to control. Five mice are randomly assigned to treatment. Multiple measures per mouse are taken. Example: five measures of Islet Area are measured in each pancreas. Mouse is the experimental unit. Each of the measures is a technical replicate because the treatment is not randomly assigned to each islet but to the whole mouse. <span class="math inline">\(t=2\)</span>, <span class="math inline">\(b=0\)</span>, <span class="math inline">\(r=5\)</span>, and <span class="math inline">\(s=5\)</span>.</li>
<li>Ten cell cultures are created. Five cultures are randomly assigned to control and five to treatment. Multiple measures per culture are taken. Example: Mitochondrial counts are measured from five cells in each culture. Culture is the experimental unit. Each of the counts is a technical replicate because the treatment is not randomly assigned to each cell but to the whole culture. <span class="math inline">\(t=2\)</span>, <span class="math inline">\(b=0\)</span>, <span class="math inline">\(r=5\)</span>, and <span class="math inline">\(s=5\)</span>.</li>
</ol>
<p>Notes:</p>
<ol type="1">
<li>Subsampling can occur at multiple levels. Example: Ten mice from separate litters are sampled. Five mice are randomly assigned to control. Five mice are randomly assigned to treatment. Five neurons in a slice of brain tissue are identified in each mouse. From each neuron, the length of five dendrite spines are measured. The five measures of spine length are “nested within” neuron and the five neurons are “nested within” mouse. Nested subsampling can quickly lead to massive pseudoreplication and false discovery.</li>
</ol>
</section>
<section id="randomized-complete-block-design-rcbd" class="level3" data-number="14.4.3">
<h3 data-number="14.4.3" class="anchored" data-anchor-id="randomized-complete-block-design-rcbd"><span class="header-section-number">14.4.3</span> Randomized Complete Block Design (RCBD)</h3>
<p>The <strong>Randomized Complete Block Design</strong> experiment in <a href="#fig-lmm-designs" class="quarto-xref">Figure&nbsp;<span>14.6</span></a> C is similar to the CRD except that all treatment combinations (two here) are randomly assigned to sibling mice within a litter. Here, two mice from each litter are randomly selected and one is randomly assigned to “Cn” and the other to “Tr”. Each litter is randomly assigned to a unique cage. The researchers measure a single value of the response variable from each mouse. The five replicate mice per treatment are the treatment replicates. The litters (or cage) are the blocks. In this design, litter and cage effects are confounded but this has no consequence on the statistical model and inference unless the researchers want to explicitly estimate these effects separately. Compared to the CRD, this design requires fewer resources (five litters instead of ten, five cages instead of ten). Compared to the general linear model fit to data from the CRD (<code>y ~ treatment</code>), including t-tests and traditional ANOVA, the linear mixed model fit to the RCBD has increased precision and power. While many researchers seem to be designing experiments similar to this (“littermate controls”), most are failing to fit a statistical model that accounts for the batching and taking advantage of the increased precision and power.</p>
<p>Examples:</p>
<ol type="1">
<li>Ten mice are sampled. In each mouse, one forelimb is assigned to control and the other forelimb is assigned to treatment. Only a single measure on each side is taken. Limb is the experimental unit. Mouse is a block. <span class="math inline">\(t=2\)</span>, <span class="math inline">\(b=10\)</span>, <span class="math inline">\(r=1\)</span>, and <span class="math inline">\(s=1\)</span>.</li>
<li>Ten litters are sampled. In each litter, one sib is assigned to control and the other sib is assigned to treatment. Only a single measure on each sib is taken. Mouse is the experimental unit. Litter is a block. <span class="math inline">\(t=2\)</span>, <span class="math inline">\(b=10\)</span>, <span class="math inline">\(r=1\)</span>, and <span class="math inline">\(s=1\)</span>.</li>
<li>Two mice, each from a separate litter are sampled. One is randomly assigned to control and the other to treatment. Only a single measure of the response variable is taken per mouse. The experiment is replicated five times (five different days, each with a newly made set of reagents and machine calibrations). Mouse is the experimental unit. Experiment is a block. <span class="math inline">\(t=2\)</span>, <span class="math inline">\(b=5\)</span>, <span class="math inline">\(r=1\)</span>, and <span class="math inline">\(s=1\)</span>.</li>
</ol>
</section>
<section id="randomized-split-plot-design-rspd" class="level3" data-number="14.4.4">
<h3 data-number="14.4.4" class="anchored" data-anchor-id="randomized-split-plot-design-rspd"><span class="header-section-number">14.4.4</span> Randomized Split Plot Design (RSPD)</h3>
<p>The <strong>Randomized Split Plot Design</strong> experiment in <a href="#fig-lmm-designs" class="quarto-xref">Figure&nbsp;<span>14.6</span></a> D is similar to the RCBD except that there is now a second experimental factor that is crossed with the first experimental factor. An individual mouse acts as single experimental unit for one factor (here, <span class="math inline">\(\texttt{genotype}\)</span> with levels “WT” and “KO”) but acts as a block for the second experimental factor (here, <span class="math inline">\(\texttt{treatment}\)</span> with levels “Cn” and “Tr”). The first factor (<span class="math inline">\(\texttt{genotype}\)</span>) is the <strong>main plot</strong> – the levels of the factor are randomly assigned to the main plots. The second factor (<span class="math inline">\(\texttt{treatment}\)</span>) is the <strong>subplot</strong> – the levels of this factor are randomly assigned to the subplots. <span class="math inline">\(\texttt{Litter}\)</span> is a replicated block. Also in <a href="#fig-lmm-designs" class="quarto-xref">Figure&nbsp;<span>14.6</span></a> D, a 2 x 2 RCBD with the same two experimental factors is shown for comparison. In the 2 x 2 RCBD, four mice per block (litter) are each randomly assigned one of the 2 x 2 combinations of <span class="math inline">\(\texttt{genotype}\)</span> and <span class="math inline">\(\texttt{treatment}\)</span>).</p>
</section>
<section id="generalized-randomized-complete-block-design-grcbd" class="level3" data-number="14.4.5">
<h3 data-number="14.4.5" class="anchored" data-anchor-id="generalized-randomized-complete-block-design-grcbd"><span class="header-section-number">14.4.5</span> Generalized Randomized Complete Block Design (GRCBD)</h3>
<p>The <strong>Generalized Randomized Complete Block Design</strong> experiment in <a href="#fig-lmm-designs" class="quarto-xref">Figure&nbsp;<span>14.6</span></a> E is similar to the RCBD except that two treatment replicates per block (litter/cage) are assigned.</p>
<p><strong>Important and somewhat not intuitive</strong> Because the treatment replicates within a litter share common error variance, these do not act like independent replicates. One consequence of this is, the sample size (<span class="math inline">\(n\)</span>) is five and not ten (<span class="math inline">\(litters \times treatment replicates\)</span>). The general linear model <code>y ~ treatment</code> fit to these data, including t-tests and traditional ANOVA, will generally have falsely high precision and falsely low <em>p</em>-values.</p>
</section>
<section id="nested-randomized-complete-block-design-nrcbd" class="level3" data-number="14.4.6">
<h3 data-number="14.4.6" class="anchored" data-anchor-id="nested-randomized-complete-block-design-nrcbd"><span class="header-section-number">14.4.6</span> Nested Randomized Complete Block Design (NRCBD)</h3>
<p>The <strong>Nested Randomized Complete Block Design</strong> experiment in <a href="#fig-lmm-designs" class="quarto-xref">Figure&nbsp;<span>14.6</span></a> F is similar to the RCBD except that there are now replicated experiments. This is a nested block design with litter (a block) nested within experiment (a block).</p>
</section>
<section id="longitudinal-randomized-complete-block-design-lrcbd" class="level3" data-number="14.4.7">
<h3 data-number="14.4.7" class="anchored" data-anchor-id="longitudinal-randomized-complete-block-design-lrcbd"><span class="header-section-number">14.4.7</span> Longitudinal Randomized Complete Block Design (LRCBD)</h3>
<p>The <strong>Longitudinal Randomized Complete Block Design</strong> experiment in <a href="#fig-lmm-designs" class="quarto-xref">Figure&nbsp;<span>14.6</span></a> G is similar to the RCBD except that there are multiple measures of the response variable taken, each taken at a different time point, including <strong>baseline</strong> (time zero).</p>
</section>
<section id="variations-due-to-multiple-measures-of-the-response-variable" class="level3" data-number="14.4.8">
<h3 data-number="14.4.8" class="anchored" data-anchor-id="variations-due-to-multiple-measures-of-the-response-variable"><span class="header-section-number">14.4.8</span> Variations due to multiple measures of the response variable</h3>
<p>Similar to the CRDS above, the other basic designs can include subsampling, resulting in, for example, RCBDS or RSPDS. If there is subsampling within subsampled units, then we can designated these with “SS”, for example RCBDSS.</p>
<p>Similar to the LRCBD above, the other basic designs can include longitudinal sampling, resulting in, for example, LCRD or LRSPD.</p>
</section>
</section>
<section id="building-the-linear-mixed-model-for-clustered-data" class="level2" data-number="14.5">
<h2 data-number="14.5" class="anchored" data-anchor-id="building-the-linear-mixed-model-for-clustered-data"><span class="header-section-number">14.5</span> Building the linear (mixed) model for clustered data</h2>
<p>Notation</p>
<ol type="1">
<li>i = 1..t (treatments)</li>
<li>j = 1..b (blocks)</li>
<li>k = 1..r (experimental replications within a block)</li>
<li>m = 1..s (subsamples or technical replicates)</li>
</ol>
<p><span class="math display">\[y_i = \beta_0 + \beta_i treatment_i + (\gamma_j block_j) + (\gamma_{ij} block_j treatment_i) + \varepsilon\]</span></p>
</section>
<section id="statistical-models-for-experimental-designs" class="level2" data-number="14.6">
<h2 data-number="14.6" class="anchored" data-anchor-id="statistical-models-for-experimental-designs"><span class="header-section-number">14.6</span> Statistical models for experimental designs</h2>
<section id="models-for-completely-randomized-designs-crd" class="level3" data-number="14.6.1">
<h3 data-number="14.6.1" class="anchored" data-anchor-id="models-for-completely-randomized-designs-crd"><span class="header-section-number">14.6.1</span> Models for Completely Randomized Designs (CRD)</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r fold-show code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>lm0 <span class="ot">&lt;-</span> <span class="fu">lm</span>(y <span class="sc">~</span> treatment,</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>             <span class="at">data =</span> figx)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="models-for-batched-data-crds-rcbd-rspd-grcbd-nrcbd" class="level3" data-number="14.6.2">
<h3 data-number="14.6.2" class="anchored" data-anchor-id="models-for-batched-data-crds-rcbd-rspd-grcbd-nrcbd"><span class="header-section-number">14.6.2</span> Models for batched data (CRDS, RCBD, RSPD, GRCBD, NRCBD)</h3>
<section id="linear-models" class="level4" data-number="14.6.2.1">
<h4 data-number="14.6.2.1" class="anchored" data-anchor-id="linear-models"><span class="header-section-number">14.6.2.1</span> Linear models</h4>
<section id="fixed-effect-model" class="level5" data-number="14.6.2.1.1">
<h5 data-number="14.6.2.1.1" class="anchored" data-anchor-id="fixed-effect-model"><span class="header-section-number">14.6.2.1.1</span> fixed effect model</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r fold-show code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>lm1 <span class="ot">&lt;-</span> <span class="fu">lm</span>(y <span class="sc">~</span> treatment <span class="sc">+</span> block,</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>             <span class="at">data =</span> figx)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="linear-model-of-batch-means" class="level5" data-number="14.6.2.1.2">
<h5 data-number="14.6.2.1.2" class="anchored" data-anchor-id="linear-model-of-batch-means"><span class="header-section-number">14.6.2.1.2</span> linear model of batch means</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r fold-show code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>figx_means <span class="ot">&lt;-</span> figx[, .(<span class="at">y =</span> <span class="fu">mean</span>(y)),</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>               by <span class="ot">=</span> .(treatment, batch)]</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>lm2 <span class="ot">&lt;-</span> <span class="fu">lm</span>(y <span class="sc">~</span> treatment,</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>             <span class="at">data =</span> figx_means)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="linear-mixed-models-random-effects-models" class="level4" data-number="14.6.2.2">
<h4 data-number="14.6.2.2" class="anchored" data-anchor-id="linear-mixed-models-random-effects-models"><span class="header-section-number">14.6.2.2</span> linear mixed models (“random effects” models)</h4>
<section id="random-intercept-model" class="level5" data-number="14.6.2.2.1">
<h5 data-number="14.6.2.2.1" class="anchored" data-anchor-id="random-intercept-model"><span class="header-section-number">14.6.2.2.1</span> Random intercept model</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r fold-show code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>lmm1 <span class="ot">&lt;-</span> <span class="fu">lmer</span>(y <span class="sc">~</span> treatment <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> block),</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>              <span class="at">data =</span> figx)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="lmm-for-correlated-error" class="level5" data-number="14.6.2.2.2">
<h5 data-number="14.6.2.2.2" class="anchored" data-anchor-id="lmm-for-correlated-error"><span class="header-section-number">14.6.2.2.2</span> lmm for correlated error</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r fold-show code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>lmm2 <span class="ot">&lt;-</span> <span class="fu">lme</span>(y <span class="sc">~</span> treatment,</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>             <span class="at">random =</span> <span class="sc">~</span><span class="dv">1</span> <span class="sc">|</span> block,</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>             <span class="at">correlation =</span> <span class="fu">corSymm</span>(<span class="at">form =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> block),</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>             <span class="at">weights =</span> <span class="fu">varIdent</span>(<span class="at">form =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> treatment),</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>             <span class="at">data =</span> figx)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="random-intercept-and-slope-model" class="level5" data-number="14.6.2.2.3">
<h5 data-number="14.6.2.2.3" class="anchored" data-anchor-id="random-intercept-and-slope-model"><span class="header-section-number">14.6.2.2.3</span> random intercept and slope model</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r fold-show code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>lmm3 <span class="ot">&lt;-</span>  <span class="fu">lmer</span>(y <span class="sc">~</span> treatment <span class="sc">+</span> (treatment <span class="sc">|</span> block),</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>               <span class="at">data =</span> figx)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="random-interaction-intercept-model" class="level5" data-number="14.6.2.2.4">
<h5 data-number="14.6.2.2.4" class="anchored" data-anchor-id="random-interaction-intercept-model"><span class="header-section-number">14.6.2.2.4</span> random interaction intercept model</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r fold-show code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>lmm4 <span class="ot">&lt;-</span>  <span class="fu">lmer</span>(y <span class="sc">~</span> treatment <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> block) <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> block<span class="sc">:</span>treatment),</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>               <span class="at">data =</span> figx)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="random-interaction-model-with-subsampling" class="level5" data-number="14.6.2.2.5">
<h5 data-number="14.6.2.2.5" class="anchored" data-anchor-id="random-interaction-model-with-subsampling"><span class="header-section-number">14.6.2.2.5</span> random interaction model with subsampling</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r fold-show code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>lmm5 <span class="ot">&lt;-</span>  <span class="fu">lmer</span>(y <span class="sc">~</span> treatment <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> block) <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> block<span class="sc">:</span>treatment) <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> block<span class="sc">:</span>treatment<span class="sc">:</span>replicate),</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>               <span class="at">data =</span> figx)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="crd-split-plot-model" class="level5" data-number="14.6.2.2.6">
<h5 data-number="14.6.2.2.6" class="anchored" data-anchor-id="crd-split-plot-model"><span class="header-section-number">14.6.2.2.6</span> CRD split plot model</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r fold-show code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># tr1 is the main plot</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="co"># tr2 is the subplot</span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="co"># main_plot is tr1:rep, where rep is the rep id</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>lmm6 <span class="ot">&lt;-</span>  <span class="fu">lmer</span>(y <span class="sc">~</span> tr1 <span class="sc">*</span> tr2 <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> main_plot),</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>               <span class="at">data =</span> figx)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="rcbd-split-plot-model" class="level5" data-number="14.6.2.2.7">
<h5 data-number="14.6.2.2.7" class="anchored" data-anchor-id="rcbd-split-plot-model"><span class="header-section-number">14.6.2.2.7</span> RCBD split plot model</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r fold-show code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co"># tr1 is the main plot</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="co"># tr2 is the subplot</span></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="co"># main_plot is tr1:block</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>lmm7 <span class="ot">&lt;-</span>  <span class="fu">lmer</span>(y <span class="sc">~</span> tr1 <span class="sc">*</span> tr2 <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> block) <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> block<span class="sc">:</span>tr1),</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>               <span class="at">data =</span> figx)</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>lmm7 <span class="ot">&lt;-</span>  <span class="fu">lmer</span>(y <span class="sc">~</span> tr1 <span class="sc">*</span> tr2 <span class="sc">+</span> block <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> block<span class="sc">:</span>tr1),</span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>               <span class="at">data =</span> figx)</span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a><span class="co"># the two versions are numerically equivalent</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="anova-models-mixed-models-repeated-measures-anova" class="level4" data-number="14.6.2.3">
<h4 data-number="14.6.2.3" class="anchored" data-anchor-id="anova-models-mixed-models-repeated-measures-anova"><span class="header-section-number">14.6.2.3</span> ANOVA models (“mixed models”, “repeated measures ANOVA”)</h4>
<section id="multivariate-repeated-measures-anova" class="level5" data-number="14.6.2.3.1">
<h5 data-number="14.6.2.3.1" class="anchored" data-anchor-id="multivariate-repeated-measures-anova"><span class="header-section-number">14.6.2.3.1</span> Multivariate repeated measures ANOVA</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r fold-show code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>aov2 <span class="ot">&lt;-</span> <span class="fu">aov_4</span>(y <span class="sc">~</span> treatment <span class="sc">+</span> (treatment <span class="sc">|</span> block),</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>             <span class="at">data =</span> figx)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="univariate-repeated-measures-anova" class="level5" data-number="14.6.2.3.2">
<h5 data-number="14.6.2.3.2" class="anchored" data-anchor-id="univariate-repeated-measures-anova"><span class="header-section-number">14.6.2.3.2</span> Univariate repeated measures ANOVA</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r fold-show code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>aov1 <span class="ot">&lt;-</span> <span class="fu">aov_4</span>(y <span class="sc">~</span> treatment <span class="sc">+</span> (treatment <span class="sc">|</span> block),</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>           <span class="at">include_aov =</span> <span class="cn">TRUE</span>,</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>             <span class="at">data =</span> figx)</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a><span class="co"># requires the argument model = "univariate" in the emmeans function</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="pairwise-paired-t-tests" class="level5" data-number="14.6.2.3.3">
<h5 data-number="14.6.2.3.3" class="anchored" data-anchor-id="pairwise-paired-t-tests"><span class="header-section-number">14.6.2.3.3</span> Pairwise paired, t-tests</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r fold-show code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>pptt <span class="ot">&lt;-</span> <span class="fu">aov_4</span>(y <span class="sc">~</span> treatment <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> block),</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>             <span class="at">data =</span> figx)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
</section>
</section>
<section id="which-model-and-why" class="level2" data-number="14.7">
<h2 data-number="14.7" class="anchored" data-anchor-id="which-model-and-why"><span class="header-section-number">14.7</span> Which model and why?</h2>
<section id="crd" class="level3" data-number="14.7.1">
<h3 data-number="14.7.1" class="anchored" data-anchor-id="crd"><span class="header-section-number">14.7.1</span> CRD</h3>
<ol type="1">
<li>No batch to model!</li>
</ol>
</section>
<section id="crds" class="level3" data-number="14.7.2">
<h3 data-number="14.7.2" class="anchored" data-anchor-id="crds"><span class="header-section-number">14.7.2</span> CRDS</h3>
<ol type="1">
<li>lmm1 – random intercept model</li>
<li>lm2 – linear model of batch means</li>
</ol>
<p>Notes</p>
<ol type="1">
<li>These are numerically equivalent and should result in same estimates, SE, CIs, and p-values</li>
<li>These are equivalent to a <strong>Nested t-test</strong> or <strong>Nested ANOVA</strong></li>
</ol>
</section>
<section id="rcbd" class="level3" data-number="14.7.3">
<h3 data-number="14.7.3" class="anchored" data-anchor-id="rcbd"><span class="header-section-number">14.7.3</span> RCBD</h3>
<p>Reasonable models:</p>
<ol type="1">
<li>lmm1 – random intercept model</li>
<li>lmm2 – lmm for correlated error</li>
<li>lm1 – fixed effect model</li>
<li>aov2 – multivariate RM-ANOVA</li>
<li>aov1 – univariate RM-ANOVA</li>
<li>pptt – pairwise, paired t-test</li>
</ol>
<p>(Cannot use lmm3 or lmm4 because there is no replication of each block:treatment combination.)</p>
<p>Assumptions:</p>
<ol type="1">
<li>lmm1, lm1, and aov1 assume
<ul>
<li><strong>compound symmetry</strong>.</li>
<li><strong>sphericity</strong>.</li>
</ul></li>
<li>lmm2, aov2, and pptt do not assume compound symmetry and sphericity.</li>
</ol>
<p>Notes:</p>
<ol type="1">
<li>lmm1 is the standard. If the number of treatments = 2, then lmm1 is eqivalent with a paired t-test.</li>
<li>If the design is balanced AND the number of treatments = 2 then <em>all six methods are numerically equivalent</em>.</li>
<li>If the design is balanced AND the number of treatments &gt; 2, then
<ul>
<li>lmm1, lm1, and aov1 are numerically equivalent</li>
<li>lmm2, aov2, pptt result in the same estimates and SE but lmm2 has more df, so the CIs, and p-values of lmm2 are less conservative.</li>
</ul></li>
<li>If the number of treatments = 2 AND the design is not balanced (at least one block is missing value for one treatment level) then the linear models (lm1, aov2, aov1, pptt) are equivalent but the linear mixed models (lmm1, lmm2) differ from each other and from the linear models.</li>
<li>If a treatment within a block is missing, the whole block is deleted in the RM-ANOVA models. This reduces power (the loss of power depends partly on the number of missing values).</li>
<li>If a treatment within a block is missing, the block is deleted only in the comparisons including the missing treatment in the pairwise, paired t-tests. This makes pptt more powerful than aov2.</li>
<li>the lmm models are very flexible – covariates can be added or these can be used as generalized lmms for non-normal distributions.</li>
<li>the lmm models and especially lmm2 sometimes (often?) fail to converge, especially with small samples (small number of blocks) or if the among-block component of variance is small.</li>
</ol>
<section id="so-which-model-and-why-a-simulation-to-help-decide" class="level4" data-number="14.7.3.1">
<h4 data-number="14.7.3.1" class="anchored" data-anchor-id="so-which-model-and-why-a-simulation-to-help-decide"><span class="header-section-number">14.7.3.1</span> So, which model, and why? A simulation to help decide</h4>
<p>Two linear mixed modeling strategies would be</p>
<ol type="1">
<li>Fit the most complete model first, if the model fails or has boundary warning because of too little variance at some level, then fit the next most complete model, and so on. This order would be lmm2, lmm1, lm1 (lmm1 and lm1 are equivalent if the design is balanced).</li>
<li>Fit all linear mixed models (lmm2, lmm1) and use a model selection method to choose the best model that balances completeness (controling type I error) and overfitting.</li>
</ol>
<p>I suggest a third strategy, which I prefer. This is,</p>
<ol start="3" type="1">
<li>use simulation to compare the performance (Type I error and Power) of each model under differ combinations of the structure of the variances of the levels of the random factor. Use these results to guide the strategy</li>
</ol>
<p>Here I use a simulation to measure the performance of the RCBD statistical models under four conditions of correlated error among the blocks between two treatment levels.</p>
<ol type="1">
<li>High correlated error. There is heterogenous correlated error among the three pairs of treatments, with one high, one intermediate, and one low correlation</li>
<li>Low correlated error. There is heterogenous correlated error among the three pairs of treatments, all with low correlation.</li>
<li>Homogenous correlated error. There is high correlated errror among the three pairs of treatment, all with the same expected level</li>
<li>There is no correlated error.</li>
</ol>
<div class="cell">
<div id="tbl-lmm-rcbd-type1" class="cell quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-lmm-rcbd-type1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;14.1: Type I error rate the RCBD statistical models under different levels of correlation structure among the observations. The Cor Error column is the average correlated error of the residuals fit by a simple linear model y ~ treatment. Model abbreviations as in text.
</figcaption>
<div aria-describedby="tbl-lmm-rcbd-type1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<table class="table do-not-create-environment cell caption-top table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">Contrast</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Cor Error</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">lm0</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">lm1</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">aov2</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">aov1</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">pptt</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">lmm2</th>
</tr>
</thead>
<tbody>
<tr class="odd" data-grouplength="3">
<td colspan="8" style="border-bottom: 1px solid"><strong>sim 1 - high correlated error</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Tr1 - Cn</td>
<td style="text-align: right;">0.300</td>
<td style="text-align: right;">0.044</td>
<td style="text-align: right;">0.106</td>
<td style="text-align: right;">0.106</td>
<td style="text-align: right;">0.057</td>
<td style="text-align: right;">0.057</td>
<td style="text-align: right;">0.0640</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Tr2 - Cn</td>
<td style="text-align: right;">0.416</td>
<td style="text-align: right;">0.024</td>
<td style="text-align: right;">0.082</td>
<td style="text-align: right;">0.082</td>
<td style="text-align: right;">0.048</td>
<td style="text-align: right;">0.048</td>
<td style="text-align: right;">0.0635</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Tr2 - Tr1</td>
<td style="text-align: right;">0.662</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: right;">0.004</td>
<td style="text-align: right;">0.004</td>
<td style="text-align: right;">0.050</td>
<td style="text-align: right;">0.050</td>
<td style="text-align: right;">0.0600</td>
</tr>
<tr class="odd" data-grouplength="3">
<td colspan="8" style="border-bottom: 1px solid"><strong>sim 2 - low correlated error</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Tr1 - Cn</td>
<td style="text-align: right;">0.161</td>
<td style="text-align: right;">0.048</td>
<td style="text-align: right;">0.070</td>
<td style="text-align: right;">0.070</td>
<td style="text-align: right;">0.054</td>
<td style="text-align: right;">0.054</td>
<td style="text-align: right;">0.0645</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Tr2 - Cn</td>
<td style="text-align: right;">0.178</td>
<td style="text-align: right;">0.048</td>
<td style="text-align: right;">0.072</td>
<td style="text-align: right;">0.072</td>
<td style="text-align: right;">0.055</td>
<td style="text-align: right;">0.055</td>
<td style="text-align: right;">0.0705</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Tr2 - Tr1</td>
<td style="text-align: right;">0.230</td>
<td style="text-align: right;">0.014</td>
<td style="text-align: right;">0.025</td>
<td style="text-align: right;">0.025</td>
<td style="text-align: right;">0.052</td>
<td style="text-align: right;">0.052</td>
<td style="text-align: right;">0.0705</td>
</tr>
<tr class="odd" data-grouplength="3">
<td colspan="8" style="border-bottom: 1px solid"><strong>sim 3 - equal correlated error</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Tr1 - Cn</td>
<td style="text-align: right;">0.771</td>
<td style="text-align: right;">0.002</td>
<td style="text-align: right;">0.052</td>
<td style="text-align: right;">0.052</td>
<td style="text-align: right;">0.052</td>
<td style="text-align: right;">0.052</td>
<td style="text-align: right;">0.0635</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Tr2 - Cn</td>
<td style="text-align: right;">0.774</td>
<td style="text-align: right;">0.001</td>
<td style="text-align: right;">0.050</td>
<td style="text-align: right;">0.050</td>
<td style="text-align: right;">0.052</td>
<td style="text-align: right;">0.052</td>
<td style="text-align: right;">0.0685</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Tr2 - Tr1</td>
<td style="text-align: right;">0.771</td>
<td style="text-align: right;">0.002</td>
<td style="text-align: right;">0.048</td>
<td style="text-align: right;">0.048</td>
<td style="text-align: right;">0.048</td>
<td style="text-align: right;">0.048</td>
<td style="text-align: right;">0.0620</td>
</tr>
<tr class="odd" data-grouplength="3">
<td colspan="8" style="border-bottom: 1px solid"><strong>sim 4 - zero correlated error</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Tr1 - Cn</td>
<td style="text-align: right;">0.011</td>
<td style="text-align: right;">0.046</td>
<td style="text-align: right;">0.043</td>
<td style="text-align: right;">0.043</td>
<td style="text-align: right;">0.040</td>
<td style="text-align: right;">0.040</td>
<td style="text-align: right;">0.0560</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Tr2 - Cn</td>
<td style="text-align: right;">0.016</td>
<td style="text-align: right;">0.049</td>
<td style="text-align: right;">0.049</td>
<td style="text-align: right;">0.049</td>
<td style="text-align: right;">0.046</td>
<td style="text-align: right;">0.046</td>
<td style="text-align: right;">0.0580</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Tr2 - Tr1</td>
<td style="text-align: right;">-0.005</td>
<td style="text-align: right;">0.052</td>
<td style="text-align: right;">0.050</td>
<td style="text-align: right;">0.050</td>
<td style="text-align: right;">0.051</td>
<td style="text-align: right;">0.051</td>
<td style="text-align: right;">0.0710</td>
</tr>
</tbody>
</table>


</div>
</div>
</figure>
</div>
</div>
<p>Notes</p>
<ol type="1">
<li>When there is zero correlated error, all methods have reasonable type I error control. Perhaps lmm2 is very slightly anticonservative (one of the error rates is 0.07).</li>
<li>When there is equal correlated error, all methods have reasonable type I error control (lmm2 is a very slightly anti-conservative).</li>
<li>When there is heterogenous correlated error, only aov2/pptt/lmm2 have reasonable type I error control (again, lmm2 is very slightly anticonservative). lmm1/aov2 have liberal Type I control in contrasts with low correlated error and conservative Type I control in contrasts with high correlated error.</li>
</ol>
<p>Combined, these results suggest that aov2/pptt should be our preferred model, if we only take Type I control into consideration.</p>
<div class="cell">
<div id="tbl-lmm-rcbd-power" class="cell quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-lmm-rcbd-power-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;14.2: Power of the RCBD statistical models under different levels of correlation structure among the observations. The Cor Error column is the average correlated error of the residuals fit by a simple linear model y ~ treatment. Model abbreviations as in text.
</figcaption>
<div aria-describedby="tbl-lmm-rcbd-power-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<table class="table do-not-create-environment cell caption-top table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">Contrast</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Cor Error</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">lm0</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">lm1</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">aov2</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">aov1</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">pptt</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">lmm2</th>
</tr>
</thead>
<tbody>
<tr class="odd" data-grouplength="3">
<td colspan="8" style="border-bottom: 1px solid"><strong>sim 1 - high correlated error</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Tr1 - Cn</td>
<td style="text-align: right;">0.296</td>
<td style="text-align: right;">0.048</td>
<td style="text-align: right;">0.103</td>
<td style="text-align: right;">0.103</td>
<td style="text-align: right;">0.051</td>
<td style="text-align: right;">0.051</td>
<td style="text-align: right;">0.0626</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Tr2 - Cn</td>
<td style="text-align: right;">0.411</td>
<td style="text-align: right;">0.342</td>
<td style="text-align: right;">0.497</td>
<td style="text-align: right;">0.497</td>
<td style="text-align: right;">0.388</td>
<td style="text-align: right;">0.388</td>
<td style="text-align: right;">0.4414</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Tr2 - Tr1</td>
<td style="text-align: right;">0.660</td>
<td style="text-align: right;">0.247</td>
<td style="text-align: right;">0.501</td>
<td style="text-align: right;">0.501</td>
<td style="text-align: right;">0.900</td>
<td style="text-align: right;">0.900</td>
<td style="text-align: right;">0.9246</td>
</tr>
<tr class="odd" data-grouplength="3">
<td colspan="8" style="border-bottom: 1px solid"><strong>sim 2 - low correlated error</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Tr1 - Cn</td>
<td style="text-align: right;">0.155</td>
<td style="text-align: right;">0.048</td>
<td style="text-align: right;">0.072</td>
<td style="text-align: right;">0.072</td>
<td style="text-align: right;">0.051</td>
<td style="text-align: right;">0.051</td>
<td style="text-align: right;">0.0652</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Tr2 - Cn</td>
<td style="text-align: right;">0.171</td>
<td style="text-align: right;">0.679</td>
<td style="text-align: right;">0.733</td>
<td style="text-align: right;">0.733</td>
<td style="text-align: right;">0.663</td>
<td style="text-align: right;">0.663</td>
<td style="text-align: right;">0.7116</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Tr2 - Tr1</td>
<td style="text-align: right;">0.230</td>
<td style="text-align: right;">0.712</td>
<td style="text-align: right;">0.771</td>
<td style="text-align: right;">0.771</td>
<td style="text-align: right;">0.811</td>
<td style="text-align: right;">0.811</td>
<td style="text-align: right;">0.8544</td>
</tr>
<tr class="odd" data-grouplength="3">
<td colspan="8" style="border-bottom: 1px solid"><strong>sim 3 - equal correlated error</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Tr1 - Cn</td>
<td style="text-align: right;">0.771</td>
<td style="text-align: right;">0.001</td>
<td style="text-align: right;">0.049</td>
<td style="text-align: right;">0.049</td>
<td style="text-align: right;">0.051</td>
<td style="text-align: right;">0.051</td>
<td style="text-align: right;">0.0626</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Tr2 - Cn</td>
<td style="text-align: right;">0.773</td>
<td style="text-align: right;">0.264</td>
<td style="text-align: right;">0.917</td>
<td style="text-align: right;">0.917</td>
<td style="text-align: right;">0.878</td>
<td style="text-align: right;">0.878</td>
<td style="text-align: right;">0.9118</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Tr2 - Tr1</td>
<td style="text-align: right;">0.771</td>
<td style="text-align: right;">0.265</td>
<td style="text-align: right;">0.918</td>
<td style="text-align: right;">0.918</td>
<td style="text-align: right;">0.883</td>
<td style="text-align: right;">0.883</td>
<td style="text-align: right;">0.9136</td>
</tr>
<tr class="odd" data-grouplength="3">
<td colspan="8" style="border-bottom: 1px solid"><strong>sim 4 - zero correlated error</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Tr1 - Cn</td>
<td style="text-align: right;">0.007</td>
<td style="text-align: right;">0.045</td>
<td style="text-align: right;">0.044</td>
<td style="text-align: right;">0.044</td>
<td style="text-align: right;">0.043</td>
<td style="text-align: right;">0.043</td>
<td style="text-align: right;">0.0582</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Tr2 - Cn</td>
<td style="text-align: right;">0.005</td>
<td style="text-align: right;">0.276</td>
<td style="text-align: right;">0.265</td>
<td style="text-align: right;">0.265</td>
<td style="text-align: right;">0.247</td>
<td style="text-align: right;">0.247</td>
<td style="text-align: right;">0.2896</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Tr2 - Tr1</td>
<td style="text-align: right;">-0.004</td>
<td style="text-align: right;">0.284</td>
<td style="text-align: right;">0.277</td>
<td style="text-align: right;">0.277</td>
<td style="text-align: right;">0.245</td>
<td style="text-align: right;">0.245</td>
<td style="text-align: right;">0.2900</td>
</tr>
</tbody>
</table>


</div>
</div>
</figure>
</div>
</div>
<p>Notes</p>
<ol type="1">
<li>When there is zero correlated error, lmm2 has slightly higher power than lmm1/aov1/lm1. And lmm1/aov1/lm1 have slightly higher power than aov2/pptt.</li>
<li>When there is equal correlated error, lmm2 and lmm1/aov1/lm1 have slightly higher power than aov2/pptt.</li>
<li>When there is heterogenous correlated error, lmm1/aov1/lm1 have non-trivially higher power than aov2/pptt in the contrast with the lower correlated error is but non-trivially lower power in the contrast with the higher correlated error. lmm2 has relatively high power regardless of the correlated error of the contrast. The major issue in experimental bench biology is that sample sizes are generally far too low to have good estimates of these correlations or of variance components.</li>
</ol>
<p>These results suggest lmm2 is the preferred model. If lmm2 fails, then it’s a toss-up between aov2/pptt and lmm1/aov1/lmm1.</p>
<p>But, combining the Type I and power simulations, lmm1/aov1/lmm1 have poor type I error control under some conditions while lmm2 and aov2/pptt have good error control under all conditions.</p>
</section>
<section id="so-which-model-and-why-a-simulators-guide-to-best-practices" class="level4" data-number="14.7.3.2">
<h4 data-number="14.7.3.2" class="anchored" data-anchor-id="so-which-model-and-why-a-simulators-guide-to-best-practices"><span class="header-section-number">14.7.3.2</span> So, which model, and why? A simulator’s guide to best practices</h4>
<ol type="1">
<li>If the number of treatments = 2 and
<ul>
<li>the design is balanced: it doesn’t matter which method you use.</li>
<li>the design is not balanced: use lmm1, since this will use all of the data</li>
</ul></li>
<li>If the number of treatments &gt; 2
<ul>
<li>use lmm2. If the model fit fails then</li>
<li>if the design is balanced: use aov2/pptt.</li>
<li>the design is not balanced: pptt is attractive if there are few missing values.</li>
</ul></li>
<li>If there are covariates, then start with lmm2. Use lmeControl if lmm2 fails. If this fails, use pairwise lmm1, which is the same as pairwise t-tests but allows covariates to be added to the model. Note that the covariate will be modeled independently in each pair, which effectively models the consequence of a treatment:covariate interaction.</li>
</ol>
</section>
</section>
<section id="rspd" class="level3" data-number="14.7.4">
<h3 data-number="14.7.4" class="anchored" data-anchor-id="rspd"><span class="header-section-number">14.7.4</span> RSPD</h3>
<p>Reasonable models:</p>
<ol type="1">
<li>lmm4 – random interaction intercept model</li>
</ol>
<p>Assumptions:</p>
<ol type="1">
<li></li>
</ol>
<p>Notes:</p>
<ol type="1">
<li></li>
</ol>
<p>Performance:</p>
<div class="cell">
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<caption>Type I error rate for the RSPD statistical models under different models of random variance. The design is 2 (main plot: WT, KO) x 3 (subplot: Cn, Tr1, Tr2) with ten blocks. All four simulations have a component of variance due to block. Sim 1 includes a block:main plot component, Sim 2 includes a block:sub plot component, Sim 3 includes both block:main and block:sub components. Type I error rates for the 9 simple effect contrasts were averaged within the two sets: contrast = main is the aggregate of the single main plot contrast (KO - WT) at each level of the subplot factor. contrast = sub is the aggregate of the three subplot contrasts (Tr1 - Cn, Tr2 - Cn, Tr2 - Tr1) at each level of the main plot factor. The Cor Error column is the average correlated error of the residuals fit by a simple linear model y ~ treatment. Model abbreviations as in text.</caption>
<thead>
<tr class="header">
<th style="text-align: right;" data-quarto-table-cell-role="th">sim_id</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">sig_block</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">sig_ss</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">sig_main</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">sig_sub</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">contrast</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">r</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">lm0</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">lm1</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">lmm1</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">lmm3</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">lmm4</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">lmm5</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">lmm6</th>
</tr>
</thead>
<tbody>
<tr class="odd" data-grouplength="2">
<td colspan="14" style="border-bottom: 1px solid"><strong>sim 1 - block:main plot</strong></td>
</tr>
<tr class="even">
<td style="text-align: right; padding-left: 2em;" data-indentlevel="1">1</td>
<td style="text-align: right;">0.8</td>
<td style="text-align: right;">0.5</td>
<td style="text-align: right;">0.6</td>
<td style="text-align: right;">0.0</td>
<td style="text-align: left;">main</td>
<td style="text-align: right;">0.494</td>
<td style="text-align: right;">0.010</td>
<td style="text-align: right;">0.199</td>
<td style="text-align: right;">0.087</td>
<td style="text-align: right;">0.051</td>
<td style="text-align: right;">0.052</td>
<td style="text-align: right;">0.083</td>
<td style="text-align: right;">0.054</td>
</tr>
<tr class="odd">
<td style="text-align: right; padding-left: 2em;" data-indentlevel="1">1</td>
<td style="text-align: right;">0.8</td>
<td style="text-align: right;">0.5</td>
<td style="text-align: right;">0.6</td>
<td style="text-align: right;">0.0</td>
<td style="text-align: left;">sub</td>
<td style="text-align: right;">0.783</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: right;">0.052</td>
<td style="text-align: right;">0.011</td>
<td style="text-align: right;">0.052</td>
<td style="text-align: right;">0.052</td>
<td style="text-align: right;">0.011</td>
<td style="text-align: right;">0.052</td>
</tr>
<tr class="even" data-grouplength="2">
<td colspan="14" style="border-bottom: 1px solid"><strong>sim 2 - block:sub plot</strong></td>
</tr>
<tr class="odd">
<td style="text-align: right; padding-left: 2em;" data-indentlevel="1">2</td>
<td style="text-align: right;">0.8</td>
<td style="text-align: right;">0.5</td>
<td style="text-align: right;">0.0</td>
<td style="text-align: right;">0.6</td>
<td style="text-align: left;">main</td>
<td style="text-align: right;">0.782</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: right;">0.002</td>
<td style="text-align: right;">0.003</td>
<td style="text-align: right;">0.002</td>
<td style="text-align: right;">0.003</td>
<td style="text-align: right;">0.042</td>
<td style="text-align: right;">0.041</td>
</tr>
<tr class="even">
<td style="text-align: right; padding-left: 2em;" data-indentlevel="1">2</td>
<td style="text-align: right;">0.8</td>
<td style="text-align: right;">0.5</td>
<td style="text-align: right;">0.0</td>
<td style="text-align: right;">0.6</td>
<td style="text-align: left;">sub</td>
<td style="text-align: right;">0.497</td>
<td style="text-align: right;">0.008</td>
<td style="text-align: right;">0.053</td>
<td style="text-align: right;">0.066</td>
<td style="text-align: right;">0.068</td>
<td style="text-align: right;">0.065</td>
<td style="text-align: right;">0.050</td>
<td style="text-align: right;">0.053</td>
</tr>
<tr class="odd" data-grouplength="2">
<td colspan="14" style="border-bottom: 1px solid"><strong>sim 3 - block:main + block:sub</strong></td>
</tr>
<tr class="even">
<td style="text-align: right; padding-left: 2em;" data-indentlevel="1">3</td>
<td style="text-align: right;">0.8</td>
<td style="text-align: right;">0.5</td>
<td style="text-align: right;">0.6</td>
<td style="text-align: right;">0.4</td>
<td style="text-align: left;">main</td>
<td style="text-align: right;">0.549</td>
<td style="text-align: right;">0.007</td>
<td style="text-align: right;">0.111</td>
<td style="text-align: right;">0.062</td>
<td style="text-align: right;">0.038</td>
<td style="text-align: right;">0.039</td>
<td style="text-align: right;">0.068</td>
<td style="text-align: right;">0.054</td>
</tr>
<tr class="odd">
<td style="text-align: right; padding-left: 2em;" data-indentlevel="1">3</td>
<td style="text-align: right;">0.8</td>
<td style="text-align: right;">0.5</td>
<td style="text-align: right;">0.6</td>
<td style="text-align: right;">0.4</td>
<td style="text-align: left;">sub</td>
<td style="text-align: right;">0.689</td>
<td style="text-align: right;">0.001</td>
<td style="text-align: right;">0.056</td>
<td style="text-align: right;">0.026</td>
<td style="text-align: right;">0.057</td>
<td style="text-align: right;">0.056</td>
<td style="text-align: right;">0.024</td>
<td style="text-align: right;">0.054</td>
</tr>
<tr class="even" data-grouplength="2">
<td colspan="14" style="border-bottom: 1px solid"><strong>sim 4 - block only</strong></td>
</tr>
<tr class="odd">
<td style="text-align: right; padding-left: 2em;" data-indentlevel="1">4</td>
<td style="text-align: right;">0.8</td>
<td style="text-align: right;">0.5</td>
<td style="text-align: right;">0.0</td>
<td style="text-align: right;">0.0</td>
<td style="text-align: left;">main</td>
<td style="text-align: right;">0.707</td>
<td style="text-align: right;">0.002</td>
<td style="text-align: right;">0.049</td>
<td style="text-align: right;">0.049</td>
<td style="text-align: right;">0.044</td>
<td style="text-align: right;">0.045</td>
<td style="text-align: right;">0.056</td>
<td style="text-align: right;">0.054</td>
</tr>
<tr class="even">
<td style="text-align: right; padding-left: 2em;" data-indentlevel="1">4</td>
<td style="text-align: right;">0.8</td>
<td style="text-align: right;">0.5</td>
<td style="text-align: right;">0.0</td>
<td style="text-align: right;">0.0</td>
<td style="text-align: left;">sub</td>
<td style="text-align: right;">0.704</td>
<td style="text-align: right;">0.002</td>
<td style="text-align: right;">0.057</td>
<td style="text-align: right;">0.055</td>
<td style="text-align: right;">0.061</td>
<td style="text-align: right;">0.060</td>
<td style="text-align: right;">0.052</td>
<td style="text-align: right;">0.059</td>
</tr>
</tbody>
</table>


</div>
</div>
</section>
<section id="rcbds" class="level3" data-number="14.7.5">
<h3 data-number="14.7.5" class="anchored" data-anchor-id="rcbds"><span class="header-section-number">14.7.5</span> RCBDS</h3>
<ol type="1">
<li>lmm3 – random intercept and slopes model</li>
<li>lmm4 – random interaction intercept model</li>
<li>lmm1 – random intercept model on batch means</li>
<li>lmm2 – lmm for correlated error on batch means</li>
<li>aov2 – multivariate RM-ANOVA</li>
<li>aov1 – univariate RM-ANOVA</li>
<li>pptt – pairwise, paired t-test</li>
</ol>
<p>Notes</p>
<ol type="1">
<li>lmm3 is the “maximal model” – it fits the most parameters and has the fewest assumptions. lmm1 on the full data is the minimal model. I’m not recommending this at all as this is pseudoreplication and will result in very anti-conservative inference. lmm4 is less conservative than lmm3 (and makes more assumptions).</li>
<li>lmm1 and lmm2 <em>here</em> are the models for RCBD designs but using the aggregated data - that is the subsampled replicates within a batch (block:treatment combination) are averaged.</li>
<li>If design is balanced
<ul>
<li>lmm3 and aov2 result in same estimates, SE, CIs, and p-values</li>
<li>lmm4, lmm1, and aov1 result in same estimates, SE, CIs, and p-values</li>
<li>lmm2 has same estimates and SE as lmm3/aov2 but more df so less conservative</li>
</ul></li>
<li>If a treatment within a block is missing, the whole block is deleted in the RM-ANOVA models. This reduces power</li>
<li>the lmm models are very flexible – covariates can be added or these can be used as generalized lmms for non-normal distributions.</li>
<li>the lmm models, and especially lmm2 and lmm3, sometimes (often?) fail to converge, especially with small samples (small number of blocks) or if within block correlation is small</li>
<li>lmm1, lmm4, and aov1 will be less conservative (more power but at higher type I error/false positive rate)</li>
</ol>
<p>Best practices</p>
<ol type="1">
<li>If balanced and no covariates then use lmm3/lmm2/aov2 if discovery is expensive (want to avoid false positives) or lmm4/aov1/lmm1 if discovery is cheap (can afford false positives). lmm2 is less conservative than lmm3/aov2. Use lmm3 if interested in the variance at different levels and/or covariance structure.</li>
<li>If unbalanced, or if there are covariates, then start with lmm3 or lmm2. Use lmeControl if lmm2 fails. If these fail, go to 1.</li>
</ol>
</section>
<section id="grcbd" class="level3" data-number="14.7.6">
<h3 data-number="14.7.6" class="anchored" data-anchor-id="grcbd"><span class="header-section-number">14.7.6</span> GRCBD</h3>
<ol type="1">
<li>lmm3 – random intercept and slopes model</li>
<li>lmm4 – random interaction intercept model</li>
<li>lmm1 – random intercept model on block:treatment means</li>
<li>lmm2 – lmm for correlated error on block:treatment means</li>
<li>aov2 – multivariate RM-ANOVA</li>
<li>aov1 – univariate RM-ANOVA</li>
<li>lm1 – fixed effect model on all data</li>
<li>pptt – pairwise, paired t-test on block:treatment means</li>
</ol>
</section>
<section id="grcbds" class="level3" data-number="14.7.7">
<h3 data-number="14.7.7" class="anchored" data-anchor-id="grcbds"><span class="header-section-number">14.7.7</span> GRCBDS</h3>
<ol type="1">
<li>lmm3 – random intercept and slopes model</li>
<li>lmm4 – random interaction intercept model</li>
<li>lmm1 – random intercept model on block:treatment means</li>
<li>lmm2 – lmm for correlated error on block:treatment means</li>
<li>aov2 – multivariate RM-ANOVA</li>
<li>aov1 – univariate RM-ANOVA</li>
<li>lm1 – fixed effect model on all block:treatment:rep means</li>
<li>pptt – pairwise, paired t-test on block:treatment:rep means</li>
</ol>
</section>
</section>
<section id="example-1-crd-with-subsampling-classical-equivalent-nested-anova" class="level2" data-number="14.8">
<h2 data-number="14.8" class="anchored" data-anchor-id="example-1-crd-with-subsampling-classical-equivalent-nested-anova"><span class="header-section-number">14.8</span> Example 1: CRD with subsampling (classical equivalent: Nested ANOVA)</h2>
<p>Source article: <a href="https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1010477" target="_blank">A new mouse model of Charcot-Marie-Tooth 2J neuropathy replicates human axonopathy and suggest alteration in axo-glia communication</a></p>
<p>Published methods: Nested ANOVA</p>
<p>Fig: 5b</p>
<p>Design: 3 x 1 CRDS</p>
<div class="cell">
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<caption>Number of paranodes measured per mouse. Each measurement is a subsampled replicate. Each mouse is a treatment replicate.</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">genotype</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">mouse</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">N</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">WT</td>
<td style="text-align: left;">WT 1</td>
<td style="text-align: right;">90</td>
</tr>
<tr class="even">
<td style="text-align: left;">WT</td>
<td style="text-align: left;">WT 2</td>
<td style="text-align: right;">58</td>
</tr>
<tr class="odd">
<td style="text-align: left;">WT</td>
<td style="text-align: left;">WT 3</td>
<td style="text-align: right;">60</td>
</tr>
<tr class="even">
<td style="text-align: left;">WT</td>
<td style="text-align: left;">WT 4</td>
<td style="text-align: right;">92</td>
</tr>
<tr class="odd">
<td style="text-align: left;">TM/+</td>
<td style="text-align: left;">TM/+ 1</td>
<td style="text-align: right;">76</td>
</tr>
<tr class="even">
<td style="text-align: left;">TM/+</td>
<td style="text-align: left;">TM/+ 2</td>
<td style="text-align: right;">52</td>
</tr>
<tr class="odd">
<td style="text-align: left;">TM/+</td>
<td style="text-align: left;">TM/+ 3</td>
<td style="text-align: right;">66</td>
</tr>
<tr class="even">
<td style="text-align: left;">TM/TM</td>
<td style="text-align: left;">TM/TM 1</td>
<td style="text-align: right;">78</td>
</tr>
<tr class="odd">
<td style="text-align: left;">TM/TM</td>
<td style="text-align: left;">TM/TM 2</td>
<td style="text-align: right;">74</td>
</tr>
<tr class="even">
<td style="text-align: left;">TM/TM</td>
<td style="text-align: left;">TM/TM 3</td>
<td style="text-align: right;">70</td>
</tr>
<tr class="odd">
<td style="text-align: left;">TM/TM</td>
<td style="text-align: left;">TM/TM 4</td>
<td style="text-align: right;">48</td>
</tr>
<tr class="even">
<td style="text-align: left;">TM/TM</td>
<td style="text-align: left;">TM/TM 5</td>
<td style="text-align: right;">54</td>
</tr>
</tbody>
</table>


</div>
</div>
<section id="model-fit-and-inference" class="level3" data-number="14.8.1">
<h3 data-number="14.8.1" class="anchored" data-anchor-id="model-fit-and-inference"><span class="header-section-number">14.8.1</span> Model fit and inference</h3>
<section id="model-fit" class="level4" data-number="14.8.1.1">
<h4 data-number="14.8.1.1" class="anchored" data-anchor-id="model-fit"><span class="header-section-number">14.8.1.1</span> Model fit</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>fig5b_lmm <span class="ot">&lt;-</span> <span class="fu">lmer</span>(caspr_length <span class="sc">~</span> genotype <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> mouse), <span class="at">data =</span> fig5b)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Notes</p>
<ol type="1">
<li>This model is called a “random intercept” model. The factor <span class="math inline">\(\texttt{mouse}\)</span> is a random factor and is added to the model as a random intercept. Random factors are added inside parentheses. A random intercept is specified with a 1 in front of the bar.</li>
</ol>
</section>
<section id="inference" class="level4" data-number="14.8.1.2">
<h4 data-number="14.8.1.2" class="anchored" data-anchor-id="inference"><span class="header-section-number">14.8.1.2</span> Inference</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>fig5b_lmm_emm <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(fig5b_lmm, <span class="at">specs =</span> <span class="st">"genotype"</span>)</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>fig5b_lmm_pairs <span class="ot">&lt;-</span> <span class="fu">contrast</span>(fig5b_lmm_emm, <span class="at">method =</span> <span class="st">"revpairwise"</span>, <span class="at">adjust =</span> <span class="st">"tukey"</span>) <span class="sc">|&gt;</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>fig5b_lmm_pairs <span class="sc">|&gt;</span></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>(<span class="at">digits =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">1</span>,<span class="dv">4</span>)) <span class="sc">|&gt;</span></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">contrast</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">estimate</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">SE</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">df</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">lower.CL</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">upper.CL</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">t.ratio</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">p.value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">(TM/+) - WT</td>
<td style="text-align: right;">0.48</td>
<td style="text-align: right;">0.16</td>
<td style="text-align: right;">8.9</td>
<td style="text-align: right;">0.03</td>
<td style="text-align: right;">0.93</td>
<td style="text-align: right;">3.0</td>
<td style="text-align: right;">0.0387</td>
</tr>
<tr class="even">
<td style="text-align: left;">(TM/TM) - WT</td>
<td style="text-align: right;">0.49</td>
<td style="text-align: right;">0.14</td>
<td style="text-align: right;">8.8</td>
<td style="text-align: right;">0.09</td>
<td style="text-align: right;">0.88</td>
<td style="text-align: right;">3.4</td>
<td style="text-align: right;">0.0192</td>
</tr>
<tr class="odd">
<td style="text-align: left;">(TM/TM) - (TM/+)</td>
<td style="text-align: right;">0.01</td>
<td style="text-align: right;">0.16</td>
<td style="text-align: right;">9.2</td>
<td style="text-align: right;">-0.43</td>
<td style="text-align: right;">0.44</td>
<td style="text-align: right;">0.0</td>
<td style="text-align: right;">0.9990</td>
</tr>
</tbody>
</table>


</div>
</div>
<p>Notes</p>
<ol type="1">
<li>. I would not adjust the <em>p</em>-values (see <a href="multiple-testing.html" class="quarto-xref"><span>Chapter 12</span></a>) but I have adjusted using Tukey HSD to compare to researcher’s results in next section</li>
</ol>
</section>
<section id="plot-the-model" class="level4" data-number="14.8.1.3">
<h4 data-number="14.8.1.3" class="anchored" data-anchor-id="plot-the-model"><span class="header-section-number">14.8.1.3</span> Plot the model</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1</span>)</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggptm</span>(fig5b_lmm, fig5b_lmm_emm, fig5b_lmm_pairs,</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>      <span class="at">show_nest_data =</span> <span class="cn">TRUE</span>,</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>      <span class="at">nest_id =</span> <span class="st">"mouse"</span>,</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>      <span class="at">hide_pairs =</span> <span class="dv">3</span>,</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>      <span class="at">y_label =</span> <span class="st">"Caspr Length (µm)"</span>,</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>      <span class="at">palette =</span> <span class="st">"pal_okabe_ito_blue"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="lmm_files/figure-html/lmm5b-plot-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="what-the-researchers-did-classical-nested-anova" class="level3" data-number="14.8.2">
<h3 data-number="14.8.2" class="anchored" data-anchor-id="what-the-researchers-did-classical-nested-anova"><span class="header-section-number">14.8.2</span> What the researchers did: Classical nested ANOVA</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>fig5b_aov <span class="ot">&lt;-</span> <span class="fu">aov_4</span>(caspr_length <span class="sc">~</span> genotype <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> mouse), </span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>                   <span class="at">fun_aggregate =</span> mean,</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>                   <span class="at">data =</span> fig5b)</span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>fig5b_aov_emm <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(fig5b_aov, <span class="at">specs =</span> <span class="st">"genotype"</span>)</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>fig5b_aov_pairs <span class="ot">&lt;-</span> <span class="fu">contrast</span>(fig5b_aov_emm, <span class="at">method =</span> <span class="st">"revpairwise"</span>, <span class="at">adjust =</span> <span class="st">"tukey"</span>) <span class="sc">|&gt;</span></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.table</span>()</span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>fig5b_aov_pairs <span class="sc">|&gt;</span></span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>(<span class="at">digits =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">1</span>,<span class="dv">4</span>)) <span class="sc">|&gt;</span></span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">contrast</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">estimate</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">SE</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">df</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">lower.CL</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">upper.CL</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">t.ratio</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">p.value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">(TM/+) - WT</td>
<td style="text-align: right;">0.49</td>
<td style="text-align: right;">0.17</td>
<td style="text-align: right;">9</td>
<td style="text-align: right;">0.02</td>
<td style="text-align: right;">0.95</td>
<td style="text-align: right;">2.9</td>
<td style="text-align: right;">0.0401</td>
</tr>
<tr class="even">
<td style="text-align: left;">(TM/TM) - WT</td>
<td style="text-align: right;">0.49</td>
<td style="text-align: right;">0.15</td>
<td style="text-align: right;">9</td>
<td style="text-align: right;">0.08</td>
<td style="text-align: right;">0.90</td>
<td style="text-align: right;">3.4</td>
<td style="text-align: right;">0.0204</td>
</tr>
<tr class="odd">
<td style="text-align: left;">(TM/TM) - (TM/+)</td>
<td style="text-align: right;">0.00</td>
<td style="text-align: right;">0.16</td>
<td style="text-align: right;">9</td>
<td style="text-align: right;">-0.44</td>
<td style="text-align: right;">0.45</td>
<td style="text-align: right;">0.0</td>
<td style="text-align: right;">0.9996</td>
</tr>
</tbody>
</table>


</div>
</div>
<p>Notes</p>
<ol type="1">
<li>If the design is balanced (all units have the same sample size), classical nested ANOVA and the linear mixed model with a random intercept are equivalent.</li>
<li>In general, ANOVA doesn’t handle lack of balance well (different sample sizes per unit). If there were the same number of measures per mouse, The <em>t</em> and <em>p</em> values of this nested anova would be the same as these values fit by the linear mixed model. As is, the <em>t</em> value is slightly smaller and the <em>p</em>-value is slightly larger in the nested ANOVA.</li>
</ol>
</section>
<section id="what-if-the-researchers-had-ignored-the-non-independence" class="level3" data-number="14.8.3">
<h3 data-number="14.8.3" class="anchored" data-anchor-id="what-if-the-researchers-had-ignored-the-non-independence"><span class="header-section-number">14.8.3</span> What if the researchers had ignored the non-independence?</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>fig5b_lm <span class="ot">&lt;-</span> <span class="fu">lm</span>(caspr_length <span class="sc">~</span> genotype, <span class="at">data =</span> fig5b)</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>fig5b_lm_emm <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(fig5b_lm, <span class="at">specs =</span> <span class="st">"genotype"</span>)</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>fig5b_lm_pairs <span class="ot">&lt;-</span> <span class="fu">contrast</span>(fig5b_lm_emm, <span class="at">method =</span> <span class="st">"revpairwise"</span>, <span class="at">adjust =</span> <span class="st">"tukey"</span>) <span class="sc">|&gt;</span></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span></span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.table</span>()</span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>fig5b_lm_pairs[, p.value <span class="sc">:</span><span class="er">=</span> <span class="fu">format</span>(p.value, <span class="at">digits =</span> <span class="dv">3</span>)]</span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>fig5b_lm_pairs <span class="sc">|&gt;</span></span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>(<span class="at">digits =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">1</span>,<span class="dv">2</span>)) <span class="sc">|&gt;</span></span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">contrast</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">estimate</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">SE</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">df</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">lower.CL</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">upper.CL</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">t.ratio</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">p.value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">(TM/+) - WT</td>
<td style="text-align: right;">0.46</td>
<td style="text-align: right;">0.08</td>
<td style="text-align: right;">815</td>
<td style="text-align: right;">0.27</td>
<td style="text-align: right;">0.66</td>
<td style="text-align: right;">5.6</td>
<td style="text-align: left;">7.29e-08</td>
</tr>
<tr class="even">
<td style="text-align: left;">(TM/TM) - WT</td>
<td style="text-align: right;">0.48</td>
<td style="text-align: right;">0.07</td>
<td style="text-align: right;">815</td>
<td style="text-align: right;">0.31</td>
<td style="text-align: right;">0.65</td>
<td style="text-align: right;">6.7</td>
<td style="text-align: left;">1.15e-10</td>
</tr>
<tr class="odd">
<td style="text-align: left;">(TM/TM) - (TM/+)</td>
<td style="text-align: right;">0.02</td>
<td style="text-align: right;">0.08</td>
<td style="text-align: right;">815</td>
<td style="text-align: right;">-0.17</td>
<td style="text-align: right;">0.21</td>
<td style="text-align: right;">0.2</td>
<td style="text-align: left;">9.79e-01</td>
</tr>
</tbody>
</table>


</div>
</div>
<p>Notes</p>
<ol type="1">
<li>The <em>p</em>-values are extremely small because the model “thinks” the data are independent. This is an example of pseudoreplication – using technical replicates as treatment replicates.</li>
</ol>
</section>
<section id="a-deeper-dive-into-the-random-intercepts" class="level3" data-number="14.8.4">
<h3 data-number="14.8.4" class="anchored" data-anchor-id="a-deeper-dive-into-the-random-intercepts"><span class="header-section-number">14.8.4</span> A deeper dive into the random intercepts</h3>
<p>Remember, intercepts are means. A random intercept model computes a mean for each level of the random factor, so here, the model is computing a mean for each mouse. The mouse means <em>are not</em> the mean you would compute using a calculator. Instead the modeled mean of each mouse within a treatment is a little closer to the treatment mean than the sample mean of each mouse. That is, the deviation between the mouse mean and the treatment mean are “shrunk toward zero”.</p>
<p>It is easy to show these shrunken estimates of the cluster means. Here are the modeled means of each WT mouse, and the deviation from the modeled treatment mean.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>lmm_means_table <span class="ot">&lt;-</span> <span class="fu">data.table</span>(</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">mouse =</span> <span class="fu">unique</span>(fig5b[genotype <span class="sc">==</span> <span class="st">"WT"</span>, mouse]),</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">intercept =</span> <span class="fu">coef</span>(fig5b_lmm)<span class="sc">$</span>mouse[<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>, <span class="st">"(Intercept)"</span>])</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>lmm_means_table[, deviation <span class="sc">:</span><span class="er">=</span> intercept <span class="sc">-</span> <span class="fu">coef</span>(<span class="fu">summary</span>(fig5b_lmm))[<span class="dv">1</span>, <span class="st">"Estimate"</span>]]</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>lmm_means_table</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>    mouse intercept   deviation
   &lt;fctr&gt;     &lt;num&gt;       &lt;num&gt;
1:   WT 1  3.872542 -0.08586488
2:   WT 2  4.137722  0.17931496
3:   WT 3  3.839612 -0.11879463
4:   WT 4  3.983751  0.02534454</code></pre>
</div>
</div>
<p>Here are the sample means of each WT mouse and the deviation from the treatment mean</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>sample_means_table <span class="ot">&lt;-</span> fig5b[genotype <span class="sc">==</span> <span class="st">"WT"</span>, .(<span class="at">mean =</span> (<span class="fu">mean</span>(caspr_length))), by <span class="ot">=</span> .(mouse)]</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>sample_means_table[, deviation <span class="sc">:</span><span class="er">=</span> mean <span class="sc">-</span> <span class="fu">mean</span>(mean)]</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>sample_means_table</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>    mouse     mean   deviation
   &lt;fctr&gt;    &lt;num&gt;       &lt;num&gt;
1:   WT 1 3.850525 -0.11038077
2:   WT 2 4.209068  0.24816221
3:   WT 3 3.793921 -0.16698438
4:   WT 4 3.990108  0.02920293</code></pre>
</div>
</div>
</section>
</section>
<section id="example-2-rcbd-with-2-groups-classical-equivalent-paired-t-test" class="level2" data-number="14.9">
<h2 data-number="14.9" class="anchored" data-anchor-id="example-2-rcbd-with-2-groups-classical-equivalent-paired-t-test"><span class="header-section-number">14.9</span> Example 2: RCBD with 2 groups (classical equivalent: paired <em>t</em>-test)</h2>
<p>source: <a href="https://www.nature.com/articles/s41586-019-0985-x">Chen, J., López-Moyado, I.F., Seo, H., Lio, C.W.J., Hempleman, L.J., Sekiya, T., Yoshimura, A., Scott-Browne, J.P. and Rao, A., 2019. NR4A transcription factors limit CAR T cell function in solid tumours. Nature, 567(7749), pp.530-534.</a></p>
<p>Source Figure: Fig 2C upper left (NR4A1 using CAR)</p>
<p>Response variable: NR4A expression measured as MFI (mean flourescence intensity)</p>
<p>Treatment: CAR PD-1hi TIM3hi TILs, CAR PD-1hi TIM3lo TILs (so differ in density of TIM3 receptors)</p>
<p>Experimental Design: Randomized Complete Block Design (RCBD). Three preparations (experiments). Experiment is the block.</p>
<p>Background: CAR-T are T cells expressing chimeric antigen receptors and target/kill solid tumor cells. TILs are tumor infiltrating lymphocytes. NR are nuclear receptors that function as transcription factors. PD-1 and TIM3 are inhibitory receptors. CAR PD-1hi TIM3hi TILs are “highly exhausted”, CAR PD-1hi TIM3lo TILs are “antigen-specific memory precursor”</p>
<section id="lmm1-random-intercept-model" class="level3" data-number="14.9.1">
<h3 data-number="14.9.1" class="anchored" data-anchor-id="lmm1-random-intercept-model"><span class="header-section-number">14.9.1</span> lmm1 – random intercept model</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>lmm1 <span class="ot">&lt;-</span> <span class="fu">lmer</span>(nr4a1 <span class="sc">~</span> genotype <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> experiment_id), <span class="at">data =</span> fig2c)</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>lmm1_emm <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(lmm1, <span class="at">specs =</span> <span class="st">"genotype"</span>)</span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>lmm1_pairs <span class="ot">&lt;-</span> <span class="fu">contrast</span>(lmm1_emm, <span class="at">method =</span> <span class="st">"revpairwise"</span>) <span class="sc">|&gt;</span></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>lmm1_pairs</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> contrast          estimate   SE df lower.CL upper.CL t.ratio p.value
 (TIM3-) - (TIM3+)    -81.3 12.2  2     -134      -29  -6.683  0.0217

Degrees-of-freedom method: kenward-roger 
Confidence level used: 0.95 </code></pre>
</div>
</div>
</section>
<section id="aov2-repeated-measures-anova-rm-anova" class="level3" data-number="14.9.2">
<h3 data-number="14.9.2" class="anchored" data-anchor-id="aov2-repeated-measures-anova-rm-anova"><span class="header-section-number">14.9.2</span> aov2 – repeated measures ANOVA (RM-ANOVA)</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>aov2 <span class="ot">&lt;-</span> <span class="fu">aov_4</span>(nr4a1 <span class="sc">~</span> genotype <span class="sc">+</span> (genotype <span class="sc">|</span> experiment_id), <span class="at">data =</span> fig2c)</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>aov2_emm <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(aov2, <span class="at">specs =</span> <span class="st">"genotype"</span>)</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>aov2_pairs <span class="ot">&lt;-</span> <span class="fu">contrast</span>(aov2_emm, <span class="at">method =</span> <span class="st">"revpairwise"</span>) <span class="sc">|&gt;</span></span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>aov2_pairs</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> contrast        estimate   SE df lower.CL upper.CL t.ratio p.value
 TIM3..1 - TIM3.    -81.3 12.2  2     -134      -29  -6.683  0.0217

Confidence level used: 0.95 </code></pre>
</div>
</div>
</section>
<section id="pptt-pairwise-paired-t-test" class="level3" data-number="14.9.3">
<h3 data-number="14.9.3" class="anchored" data-anchor-id="pptt-pairwise-paired-t-test"><span class="header-section-number">14.9.3</span> pptt – pairwise paired t-test</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co"># t.test(fig2c[genotype == "TIM3-", nr4a1], fig2c[genotype == "TIM3+", nr4a1], paired = TRUE)</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>pptt1 <span class="ot">&lt;-</span> <span class="fu">pptt</span>(nr4a1 <span class="sc">~</span> genotype <span class="sc">+</span> (genotype <span class="sc">|</span> experiment_id), <span class="at">data =</span> fig2c)</span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(pptt1)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>        contrast  estimate       SE    df lower.CL  upper.CL   t.ratio
          &lt;char&gt;     &lt;num&gt;    &lt;num&gt; &lt;num&gt;    &lt;num&gt;     &lt;num&gt;     &lt;num&gt;
1: TIM3- - TIM3+ -81.33333 12.17009     2 -133.697 -28.96966 -6.683051
      p.value
        &lt;num&gt;
1: 0.02166486</code></pre>
</div>
</div>
</section>
<section id="notes-on-the-three-models" class="level3" data-number="14.9.4">
<h3 data-number="14.9.4" class="anchored" data-anchor-id="notes-on-the-three-models"><span class="header-section-number">14.9.4</span> Notes on the three models</h3>
<ol type="1">
<li>All inferential statistics are the same in all three models.</li>
<li>A RM-ANOVA with two groups is equivalent to a random intercept linear mixed model. A paired <em>t</em>-test is a special case of a linear mixed model.</li>
<li>aov2 fits an ANOVA model even though there are only two groups. Yes ANOVA can be used with only two groups, and the statistics are exactly the same as with a <em>t</em>-test, but most people choose <em>t</em>-test and not ANOVA when there are only two groups.</li>
<li>The random component of the model formula for the linear mixed model using <code>lmer</code> is <code>(1 | experiment_id)</code>. This is the notation for adding a random intercept. <span class="math inline">\(\texttt{experiment\_id\)</span> is a random factor and is added as a random intercept.</li>
<li>There is potential for confusion in the formula for the RM-ANOVA model using <code>aov_4</code>. The afex package has three equivalent functions for fitting repeated measures ANOVA models, each function is targeted to users with different statistics backgrounds. This text uses the <code>aov_4</code> function which uses the <code>lmer</code> formula notation. For an RCBD experimental design, <code>aov_4</code> uses the <code>lmer</code> formula notation for a random intercept AND a random slope, which is <code>(genotype | experiment_id)</code>. No random intercept or slope is estimated in an ANOVA model but the notation is necessary to aggregate multiple values of <span class="math inline">\(\texttt{experiment\_id}\)</span> only within and not among <span class="math inline">\(\texttt{genotype}\)</span>.</li>
</ol>
</section>
<section id="plot-the-model-1" class="level3" data-number="14.9.5">
<h3 data-number="14.9.5" class="anchored" data-anchor-id="plot-the-model-1"><span class="header-section-number">14.9.5</span> Plot the model</h3>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="lmm_files/figure-html/lmm-fig2c-ggplot-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="better-know-the-linear-mixed-model" class="level3" data-number="14.9.6">
<h3 data-number="14.9.6" class="anchored" data-anchor-id="better-know-the-linear-mixed-model"><span class="header-section-number">14.9.6</span> Better Know the linear mixed model</h3>
<p>Intraclass correlation</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="fu">VarCorr</span>(lmm1)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> Groups        Name        Std.Dev.
 experiment_id (Intercept) 52.602  
 Residual                  14.905  </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>lmm1_vc <span class="ot">&lt;-</span> <span class="fu">VarCorr</span>(lmm1) <span class="sc">|&gt;</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">|&gt;</span></span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.table</span>()</span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a>among_var <span class="ot">&lt;-</span> lmm1_vc[<span class="dv">1</span>, vcov]</span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a>within_var <span class="ot">&lt;-</span> lmm1_vc[<span class="dv">2</span>, vcov]</span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a>icc <span class="ot">&lt;-</span> among_var<span class="sc">/</span>(among_var <span class="sc">+</span> within_var)</span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a>icc</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.9256761</code></pre>
</div>
</div>
</section>
</section>
<section id="sec-lmm-example2" class="level2" data-number="14.10">
<h2 data-number="14.10" class="anchored" data-anchor-id="sec-lmm-example2"><span class="header-section-number">14.10</span> Example 3: RCBD with &gt; 2 groups (classical equivalent: repeated measures ANOVA)</h2>
<p>source: <a href="https://www.nature.com/articles/s41586-019-0985-x">Chen, J., López-Moyado, I.F., Seo, H., Lio, C.W.J., Hempleman, L.J., Sekiya, T., Yoshimura, A., Scott-Browne, J.P. and Rao, A., 2019. NR4A transcription factors limit CAR T cell function in solid tumours. Nature, 567(7749), pp.530-534.</a></p>
<p>Source Figure: Fig 2C upper right (NR4A1)</p>
<p>Response variable: NR4A1 expression measured as MFI (mean flourescence intensity)</p>
<p>Treatment: CAR PD-1hi TIM3hi TILs, CAR PD-1hi TIM3lo TILs (so differ in density of TIM3 receptors)</p>
<p>Experimental Design: Randomized Complete Block Design (RCBD). Three preparations (experiments). Experiment is the block.</p>
<p>Background: CAR-T are T cells expressing chimeric antigen receptors and target/kill solid tumor cells. TILs are tumor infiltrating lymphocytes. NR are nuclear receptors that function as transcription factors. PD-1 and TIM3 are inhibitory receptors. CAR PD-1hi TIM3hi TILs are “highly exhausted”, CAR PD-1hi TIM3lo TILs are “antigen-specific memory precursor”</p>
<section id="lmm1-random-intercept-model-1" class="level3" data-number="14.10.1">
<h3 data-number="14.10.1" class="anchored" data-anchor-id="lmm1-random-intercept-model-1"><span class="header-section-number">14.10.1</span> lmm1 – random intercept model</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>lmm1 <span class="ot">&lt;-</span> <span class="fu">lmer</span>(nr4a1 <span class="sc">~</span> genotype <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> experiment_id), <span class="at">data =</span> fig2c_r)</span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>lmm1_emm <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(lmm1, <span class="at">specs =</span> <span class="st">"genotype"</span>)</span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>lmm1_pairs <span class="ot">&lt;-</span> <span class="fu">contrast</span>(lmm1_emm, <span class="at">method =</span> <span class="st">"revpairwise"</span>, <span class="at">adjust =</span> <span class="st">"none"</span>) <span class="sc">|&gt;</span></span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>lmm1_pairs</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> contrast                  estimate   SE df lower.CL upper.CL t.ratio p.value
 (PD1+TIM3-) - (PD1+TIM3+)    -69.8 11.7  4   -102.2    -37.3  -5.966  0.0040
 (PD1-TIM3-) - (PD1+TIM3+)    -82.8 11.7  4   -115.3    -50.3  -7.081  0.0021
 (PD1-TIM3-) - (PD1+TIM3-)    -13.0 11.7  4    -45.5     19.4  -1.115  0.3275

Degrees-of-freedom method: kenward-roger 
Confidence level used: 0.95 </code></pre>
</div>
</div>
</section>
<section id="lmm2-linear-model-for-correlated-error" class="level3" data-number="14.10.2">
<h3 data-number="14.10.2" class="anchored" data-anchor-id="lmm2-linear-model-for-correlated-error"><span class="header-section-number">14.10.2</span> lmm2 – linear model for correlated error</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="co"># lmm3 &lt;- lme(nr4a1 ~ genotype,</span></span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a><span class="co">#                 random = ~1|experiment_id,</span></span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a><span class="co">#                 correlation = corSymm(form = ~ 1 | experiment_id),</span></span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a><span class="co">#                 weights = varIdent(form = ~ 1 | genotype),</span></span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a><span class="co">#                 data = fig2c_r)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="aov2-repeated-measures-anova-rm-anova-multivariate-model" class="level3" data-number="14.10.3">
<h3 data-number="14.10.3" class="anchored" data-anchor-id="aov2-repeated-measures-anova-rm-anova-multivariate-model"><span class="header-section-number">14.10.3</span> aov2 – repeated measures ANOVA (RM-ANOVA), multivariate model</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>aov2 <span class="ot">&lt;-</span> <span class="fu">aov_4</span>(nr4a1 <span class="sc">~</span> genotype <span class="sc">+</span> (genotype <span class="sc">|</span> experiment_id), <span class="at">data =</span> fig2c_r)</span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>aov2_emm <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(aov2, <span class="at">specs =</span> <span class="st">"genotype"</span>)</span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>aov2_pairs <span class="ot">&lt;-</span> <span class="fu">contrast</span>(aov2_emm, <span class="at">method =</span> <span class="st">"revpairwise"</span>, <span class="at">adjust =</span> <span class="st">"none"</span>) <span class="sc">|&gt;</span></span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a>aov2_pairs</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> contrast                  estimate    SE df lower.CL upper.CL t.ratio p.value
 PD1.TIM3..1 - PD1.TIM3.      -69.8  9.44  2   -110.4    -29.1  -7.390  0.0178
 PD1.TIM3..2 - PD1.TIM3.      -82.8 15.70  2   -150.4    -15.2  -5.273  0.0341
 PD1.TIM3..2 - PD1.TIM3..1    -13.0  8.63  2    -50.2     24.1  -1.510  0.2701

Confidence level used: 0.95 </code></pre>
</div>
</div>
</section>
<section id="aov1-repeated-measures-anova-rm-anova-univariate-model" class="level3" data-number="14.10.4">
<h3 data-number="14.10.4" class="anchored" data-anchor-id="aov1-repeated-measures-anova-rm-anova-univariate-model"><span class="header-section-number">14.10.4</span> aov1 – repeated measures ANOVA (RM-ANOVA), univariate model</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a>aov1 <span class="ot">&lt;-</span> <span class="fu">aov_4</span>(nr4a1 <span class="sc">~</span> genotype <span class="sc">+</span> (genotype <span class="sc">|</span> experiment_id),</span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a>              <span class="at">data =</span> fig2c_r,</span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a>              <span class="at">include_aov =</span> <span class="cn">TRUE</span>)</span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a>aov1_emm <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(aov1,</span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">specs =</span> <span class="st">"genotype"</span>,</span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">model =</span> <span class="st">"univariate"</span>)</span>
<span id="cb38-7"><a href="#cb38-7" aria-hidden="true" tabindex="-1"></a>aov1_pairs <span class="ot">&lt;-</span> <span class="fu">contrast</span>(aov1_emm,</span>
<span id="cb38-8"><a href="#cb38-8" aria-hidden="true" tabindex="-1"></a>                       <span class="at">method =</span> <span class="st">"revpairwise"</span>,</span>
<span id="cb38-9"><a href="#cb38-9" aria-hidden="true" tabindex="-1"></a>                       <span class="at">adjust =</span> <span class="st">"none"</span>) <span class="sc">|&gt;</span></span>
<span id="cb38-10"><a href="#cb38-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span>
<span id="cb38-11"><a href="#cb38-11" aria-hidden="true" tabindex="-1"></a>aov1_pairs</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> contrast                  estimate   SE df lower.CL upper.CL t.ratio p.value
 PD1.TIM3..1 - PD1.TIM3.      -69.8 11.7  4   -102.2    -37.3  -5.966  0.0040
 PD1.TIM3..2 - PD1.TIM3.      -82.8 11.7  4   -115.3    -50.3  -7.081  0.0021
 PD1.TIM3..2 - PD1.TIM3..1    -13.0 11.7  4    -45.5     19.4  -1.115  0.3275

Confidence level used: 0.95 </code></pre>
</div>
</div>
</section>
<section id="pptt-pairwise-paired-t-test-1" class="level3" data-number="14.10.5">
<h3 data-number="14.10.5" class="anchored" data-anchor-id="pptt-pairwise-paired-t-test-1"><span class="header-section-number">14.10.5</span> pptt – pairwise paired t-test</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a>pptt1 <span class="ot">&lt;-</span> <span class="fu">pptt</span>(nr4a1 <span class="sc">~</span> genotype <span class="sc">+</span> (genotype <span class="sc">|</span> experiment_id), <span class="at">data =</span> fig2c_r)</span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(pptt1)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                contrast  estimate        SE    df   lower.CL  upper.CL
                  &lt;char&gt;     &lt;num&gt;     &lt;num&gt; &lt;num&gt;      &lt;num&gt;     &lt;num&gt;
1: PD1+TIM3- - PD1+TIM3+ -69.76667  9.440398     2 -110.38542 -29.14791
2: PD1-TIM3- - PD1+TIM3+ -82.80000 15.702654     2 -150.36307 -15.23693
3: PD1-TIM3- - PD1+TIM3- -13.03333  8.631403     2  -50.17126  24.10459
     t.ratio    p.value
       &lt;num&gt;      &lt;num&gt;
1: -7.390225 0.01782184
2: -5.272994 0.03413460
3: -1.509990 0.27012469</code></pre>
</div>
</div>
</section>
<section id="notes-on-the-models" class="level3" data-number="14.10.6">
<h3 data-number="14.10.6" class="anchored" data-anchor-id="notes-on-the-models"><span class="header-section-number">14.10.6</span> Notes on the models</h3>
<ol type="1">
<li>lmm2 fails.</li>
<li>aov2 and pptt1 are equivalent. This is generally true if there are no missing data (genotype:experiment_id combinations). Both aov2 and pptt1 require balance but pptt1 is a pairwise procedure while aov2 analyzes all treatment levels simultaneously. Consequently pptt1 will exclude fewer measures and will have slightly more power than aov2.</li>
<li>aov2 and lmm1 are not equivalent and this is generally true.</li>
<li>aov1 and lmm1 are equivalent and this is generally true if there are no missing data.</li>
<li>The lmm equivalent of aov2 is lmm3 (random intercept and slopte model) but this model cannot be fit without replication with genotype:experiment_id combinations.</li>
<li>One reasonable conclusion of item 4 is that aov2 effectively overfits the data and we should use aov1/lmm1 as the model of choice. The simulation under RCBD above argues against this. Instead, aov2/lmm3 is preferred (although lmm3 cannot be fit without replication)</li>
<li>Item 5 raises the question, why bother with random intercept model in a RCBD? The random intercept model (lmm1) should be used if there are covariates or if the response type (count, proportion, amount) and model check suggest a generalized linear model (glm).</li>
</ol>
</section>
<section id="plot-the-model-2" class="level3" data-number="14.10.7">
<h3 data-number="14.10.7" class="anchored" data-anchor-id="plot-the-model-2"><span class="header-section-number">14.10.7</span> Plot the model</h3>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="lmm_files/figure-html/lmm-fig2c-r-ggplot-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Notes</p>
<ol type="1">
<li>The blocks are connected by the gray lines</li>
<li>The modeled SE of the treatment means are large relative to the spread of the treatment replicates. The modeled SE of the treatment means are not a good guide to differences among the treatments in a model with random factors if the sample size (number of blocks) is small.</li>
</ol>
</section>
</section>
<section id="example-4-rcbd-with-two-factors-exp5c" class="level2" data-number="14.11">
<h2 data-number="14.11" class="anchored" data-anchor-id="example-4-rcbd-with-two-factors-exp5c"><span class="header-section-number">14.11</span> Example 4 – RCBD with two factors (exp5c)</h2>
<p>Example 3 is similar to example 2 in that there is no subsampling replication and we cannot add random slopes to the linear mixed model. But, because example 3 is a factorial design with two factors, there is replication within each factor within a block. Consequently, there are several alternative models with different sets of random intercepts. The reported model includes two random intercepts, one of which models differences in batch effects among treatment levels (treatment by batch interactions). This <strong>interaction intercept</strong> is an alternative to random slope for modeling treatment by batch interactions.</p>
<p><a href="https://www.nature.com/articles/s41467-019-13869-w" target="_blank">Transcriptomic profiling of skeletal muscle adaptations to exercise and inactivity</a></p>
<p>Source figure: <a href="https://www.nature.com/articles/s41467-019-13869-w#Fig5" target="_blank">Fig. 5c</a></p>
<p>Source data: <a href="https://www.nature.com/articles/s41467-019-13869-w#Sec27" target="_blank">Source Data Fig. 5</a></p>
<div class="cell">
<div class="cell-output cell-output-stderr">
<pre><code>New names:
• `` -&gt; `...1`
• `` -&gt; `...2`
• `` -&gt; `...3`
• `` -&gt; `...4`
• `` -&gt; `...5`
• `` -&gt; `...6`
• `` -&gt; `...7`
• `` -&gt; `...8`
• `` -&gt; `...9`
• `` -&gt; `...10`
• `` -&gt; `...11`
• `` -&gt; `...12`
• `` -&gt; `...13`</code></pre>
</div>
</div>
<section id="understand-the-data" class="level3" data-number="14.11.1">
<h3 data-number="14.11.1" class="anchored" data-anchor-id="understand-the-data"><span class="header-section-number">14.11.1</span> Understand the data</h3>
<p>The data for Example 1 are from Figure 5c. Six muscle source cells were used to start six independent cultures. Cells from each culture were treated with either a negative control (“Scr”) or a siRNA (“siNR4A3”) that “silences” expression of the NR4A3 gene product by <a href="https://en.wikipedia.org/wiki/Small_interfering_RNA" target="_blank">cleaving the mRNA</a>. Glucose uptake in the two cell types was measured at rest (“Basal”) and during electrical pulse stimulation (“EPS”).</p>
<p>The design is a <span class="math inline">\(2 \times 2\)</span> <a href="https://en.wikipedia.org/wiki/Blocking_(statistics)#Randomized_block_design" target="_blank">Randomized complete block with no subsampling</a>. There are two factors each with two levels: <span class="math inline">\(\texttt{treatment}\)</span> (“Scr”, “siNR4A3”) and <span class="math inline">\(\texttt{activity}\)</span> (“Basal”, “EPS”). Each source cell is a block. All four treatment combinations were measured once per block.</p>
</section>
<section id="examine-the-data" class="level3" data-number="14.11.2">
<h3 data-number="14.11.2" class="anchored" data-anchor-id="examine-the-data"><span class="header-section-number">14.11.2</span> Examine the data</h3>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="lmm_files/figure-html/lmm-exp5c-examine-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>The plot shows a strong donor effect.</p>
</section>
<section id="lmm-exp5c-m1" class="level3" data-number="14.11.3">
<h3 data-number="14.11.3" class="anchored" data-anchor-id="lmm-exp5c-m1"><span class="header-section-number">14.11.3</span> Model fit and inference</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a>exp5c_m1a <span class="ot">&lt;-</span> <span class="fu">lmer</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor),</span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp5c)</span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-5"><a href="#cb43-5" aria-hidden="true" tabindex="-1"></a>exp5c_m1b <span class="ot">&lt;-</span> <span class="fu">lmer</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb43-6"><a href="#cb43-6" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor) <span class="sc">+</span></span>
<span id="cb43-7"><a href="#cb43-7" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor<span class="sc">:</span>treatment) <span class="sc">+</span></span>
<span id="cb43-8"><a href="#cb43-8" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor<span class="sc">:</span>activity),</span>
<span id="cb43-9"><a href="#cb43-9" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp5c)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>boundary (singular) fit: see help('isSingular')</code></pre>
</div>
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a>exp5c_m1c <span class="ot">&lt;-</span> <span class="fu">lmer</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor) <span class="sc">+</span></span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor<span class="sc">:</span>treatment),</span>
<span id="cb45-4"><a href="#cb45-4" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp5c)</span>
<span id="cb45-5"><a href="#cb45-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-6"><a href="#cb45-6" aria-hidden="true" tabindex="-1"></a>exp5c_m1d <span class="ot">&lt;-</span> <span class="fu">lme</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity,</span>
<span id="cb45-7"><a href="#cb45-7" aria-hidden="true" tabindex="-1"></a>                 <span class="at">random =</span>  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> donor,</span>
<span id="cb45-8"><a href="#cb45-8" aria-hidden="true" tabindex="-1"></a>                 <span class="at">correlation =</span> <span class="fu">corSymm</span>(<span class="at">form =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> donor),</span>
<span id="cb45-9"><a href="#cb45-9" aria-hidden="true" tabindex="-1"></a>                 <span class="at">weights =</span> <span class="fu">varIdent</span>(<span class="at">form =</span> <span class="sc">~</span> <span class="dv">1</span><span class="sc">|</span>t.by.a),</span>
<span id="cb45-10"><a href="#cb45-10" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp5c)</span>
<span id="cb45-11"><a href="#cb45-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-12"><a href="#cb45-12" aria-hidden="true" tabindex="-1"></a><span class="co"># check AIC</span></span>
<span id="cb45-13"><a href="#cb45-13" aria-hidden="true" tabindex="-1"></a><span class="fu">AIC</span>(exp5c_m1a, exp5c_m1b, exp5c_m1c, exp5c_m1d)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in AIC.default(exp5c_m1a, exp5c_m1b, exp5c_m1c, exp5c_m1d): models are
not all fitted to the same number of observations</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>          df       AIC
exp5c_m1a  6  8.153813
exp5c_m1b  8 10.052676
exp5c_m1c  7  8.052676
exp5c_m1d 15 14.763024</code></pre>
</div>
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="co"># check VarCorr model c</span></span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a><span class="fu">VarCorr</span>(exp5c_m1c) <span class="co"># fine</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> Groups          Name        Std.Dev.
 donor:treatment (Intercept) 0.089519
 donor           (Intercept) 0.352097
 Residual                    0.090685</code></pre>
</div>
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="co"># report 1c (based on AIC and VarCorr check)</span></span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a>exp5c_m1 <span class="ot">&lt;-</span> exp5c_m1c</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>exp5c_m1b (equivalent to univariate repeated measures ANOVA) is singular fit. don’t use. Trivial difference in AIC between exp5c_m1a and exp5c_m1c. Nothing in VarCorr with exp5c_m1c raises red flags. Report exp5c_m1c.</p>
<section id="check-the-model" class="level4" data-number="14.11.3.1">
<h4 data-number="14.11.3.1" class="anchored" data-anchor-id="check-the-model"><span class="header-section-number">14.11.3.1</span> Check the model</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggcheck_the_model</span>(exp5c_m1)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>boundary (singular) fit: see help('isSingular')
boundary (singular) fit: see help('isSingular')
boundary (singular) fit: see help('isSingular')
boundary (singular) fit: see help('isSingular')
boundary (singular) fit: see help('isSingular')
boundary (singular) fit: see help('isSingular')
boundary (singular) fit: see help('isSingular')
boundary (singular) fit: see help('isSingular')</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in rlm.default(x, y, weights, method = method, wt.method = wt.method, :
'rlm' failed to converge in 20 steps</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>boundary (singular) fit: see help('isSingular')
boundary (singular) fit: see help('isSingular')
boundary (singular) fit: see help('isSingular')
boundary (singular) fit: see help('isSingular')
boundary (singular) fit: see help('isSingular')
boundary (singular) fit: see help('isSingular')
boundary (singular) fit: see help('isSingular')
boundary (singular) fit: see help('isSingular')</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in rlm.default(x, y, weights, method = method, wt.method = wt.method, :
'rlm' failed to converge in 20 steps</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>boundary (singular) fit: see help('isSingular')
boundary (singular) fit: see help('isSingular')
boundary (singular) fit: see help('isSingular')</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Model failed to converge with 1 negative eigenvalue: -8.3e-02</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>boundary (singular) fit: see help('isSingular')</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.
ℹ Please use `linewidth` instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>`geom_smooth()` using formula = 'y ~ x'</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="lmm_files/figure-html/lmm-exp5c_m1-check-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>fine.</p>
</section>
<section id="inference-from-the-model" class="level4" data-number="14.11.3.2">
<h4 data-number="14.11.3.2" class="anchored" data-anchor-id="inference-from-the-model"><span class="header-section-number">14.11.3.2</span> Inference from the model</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb61"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a>exp5c_m1_coef <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">coef</span>(<span class="fu">summary</span>(exp5c_m1)),</span>
<span id="cb61-2"><a href="#cb61-2" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">confint</span>(exp5c_m1)[<span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>),])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a>exp5c_m1_coef <span class="sc">|&gt;</span></span>
<span id="cb62-2"><a href="#cb62-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>(<span class="at">digits =</span> <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">4</span>,<span class="dv">2</span>,<span class="dv">2</span>)) <span class="sc">|&gt;</span></span>
<span id="cb62-3"><a href="#cb62-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Estimate</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Std. Error</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">df</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">t value</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Pr(&gt;&amp;#124;t&amp;#124;)</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">2.5 %</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">97.5 %</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">(Intercept)</td>
<td style="text-align: right;">1.34</td>
<td style="text-align: right;">0.153</td>
<td style="text-align: right;">5.8</td>
<td style="text-align: right;">8.8</td>
<td style="text-align: right;">0.0001</td>
<td style="text-align: right;">1.02</td>
<td style="text-align: right;">1.66</td>
</tr>
<tr class="even">
<td style="text-align: left;">treatmentsiNR4A3</td>
<td style="text-align: right;">0.08</td>
<td style="text-align: right;">0.074</td>
<td style="text-align: right;">8.5</td>
<td style="text-align: right;">1.1</td>
<td style="text-align: right;">0.2994</td>
<td style="text-align: right;">-0.07</td>
<td style="text-align: right;">0.23</td>
</tr>
<tr class="odd">
<td style="text-align: left;">activityEPS</td>
<td style="text-align: right;">0.15</td>
<td style="text-align: right;">0.052</td>
<td style="text-align: right;">10.0</td>
<td style="text-align: right;">2.9</td>
<td style="text-align: right;">0.0163</td>
<td style="text-align: right;">0.05</td>
<td style="text-align: right;">0.25</td>
</tr>
<tr class="even">
<td style="text-align: left;">treatmentsiNR4A3:activityEPS</td>
<td style="text-align: right;">-0.27</td>
<td style="text-align: right;">0.074</td>
<td style="text-align: right;">10.0</td>
<td style="text-align: right;">-3.6</td>
<td style="text-align: right;">0.0048</td>
<td style="text-align: right;">-0.41</td>
<td style="text-align: right;">-0.12</td>
</tr>
</tbody>
</table>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb63"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a>exp5c_m1_emm <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(exp5c_m1, <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">"treatment"</span>, <span class="st">"activity"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">treatment</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">activity</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">emmean</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">SE</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">df</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">lower.CL</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">upper.CL</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Scr</td>
<td style="text-align: left;">Basal</td>
<td style="text-align: right;">1.34</td>
<td style="text-align: right;">0.153</td>
<td style="text-align: right;">5.8</td>
<td style="text-align: right;">0.96</td>
<td style="text-align: right;">1.72</td>
</tr>
<tr class="even">
<td style="text-align: left;">siNR4A3</td>
<td style="text-align: left;">Basal</td>
<td style="text-align: right;">1.42</td>
<td style="text-align: right;">0.153</td>
<td style="text-align: right;">5.8</td>
<td style="text-align: right;">1.04</td>
<td style="text-align: right;">1.80</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Scr</td>
<td style="text-align: left;">EPS</td>
<td style="text-align: right;">1.49</td>
<td style="text-align: right;">0.153</td>
<td style="text-align: right;">5.8</td>
<td style="text-align: right;">1.11</td>
<td style="text-align: right;">1.87</td>
</tr>
<tr class="even">
<td style="text-align: left;">siNR4A3</td>
<td style="text-align: left;">EPS</td>
<td style="text-align: right;">1.31</td>
<td style="text-align: right;">0.153</td>
<td style="text-align: right;">5.8</td>
<td style="text-align: right;">0.93</td>
<td style="text-align: right;">1.68</td>
</tr>
</tbody>
</table>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="co"># exp5c_emm # print in console to get row numbers</span></span>
<span id="cb64-2"><a href="#cb64-2" aria-hidden="true" tabindex="-1"></a><span class="co"># set the mean as the row number from the emmeans table</span></span>
<span id="cb64-3"><a href="#cb64-3" aria-hidden="true" tabindex="-1"></a>scr_basal <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>)</span>
<span id="cb64-4"><a href="#cb64-4" aria-hidden="true" tabindex="-1"></a>siNR4A3_basal <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">0</span>,<span class="dv">0</span>)</span>
<span id="cb64-5"><a href="#cb64-5" aria-hidden="true" tabindex="-1"></a>scr_eps <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb64-6"><a href="#cb64-6" aria-hidden="true" tabindex="-1"></a>siNR4A3_eps <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>)</span>
<span id="cb64-7"><a href="#cb64-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-8"><a href="#cb64-8" aria-hidden="true" tabindex="-1"></a>exp5c_m1_planned <span class="ot">&lt;-</span> <span class="fu">contrast</span>(exp5c_m1_emm,</span>
<span id="cb64-9"><a href="#cb64-9" aria-hidden="true" tabindex="-1"></a>                       <span class="at">method =</span> <span class="fu">list</span>(</span>
<span id="cb64-10"><a href="#cb64-10" aria-hidden="true" tabindex="-1"></a>                         <span class="st">"(Scr EPS) - (Scr Basal)"</span> <span class="ot">=</span> <span class="fu">c</span>(scr_eps <span class="sc">-</span> scr_basal),</span>
<span id="cb64-11"><a href="#cb64-11" aria-hidden="true" tabindex="-1"></a>                         <span class="st">"(siNR4A3 EPS) - (siNR4A3 Basal)"</span> <span class="ot">=</span> <span class="fu">c</span>(siNR4A3_eps <span class="sc">-</span> siNR4A3_basal),</span>
<span id="cb64-12"><a href="#cb64-12" aria-hidden="true" tabindex="-1"></a>                         <span class="st">"Interaction"</span> <span class="ot">=</span> <span class="fu">c</span>(siNR4A3_eps <span class="sc">-</span> siNR4A3_basal) <span class="sc">-</span></span>
<span id="cb64-13"><a href="#cb64-13" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">c</span>(scr_eps <span class="sc">-</span> scr_basal)</span>
<span id="cb64-14"><a href="#cb64-14" aria-hidden="true" tabindex="-1"></a>                           </span>
<span id="cb64-15"><a href="#cb64-15" aria-hidden="true" tabindex="-1"></a>                       ),</span>
<span id="cb64-16"><a href="#cb64-16" aria-hidden="true" tabindex="-1"></a>                       <span class="at">adjust =</span> <span class="st">"none"</span></span>
<span id="cb64-17"><a href="#cb64-17" aria-hidden="true" tabindex="-1"></a>) <span class="sc">|&gt;</span></span>
<span id="cb64-18"><a href="#cb64-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">contrast</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">estimate</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">SE</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">df</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">lower.CL</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">upper.CL</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">t.ratio</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">p.value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">(Scr EPS) - (Scr Basal)</td>
<td style="text-align: right;">0.15</td>
<td style="text-align: right;">0.052</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">0.03</td>
<td style="text-align: right;">0.27</td>
<td style="text-align: right;">2.88</td>
<td style="text-align: right;">0.016</td>
</tr>
<tr class="even">
<td style="text-align: left;">(siNR4A3 EPS) - (siNR4A3 Basal)</td>
<td style="text-align: right;">-0.12</td>
<td style="text-align: right;">0.052</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">-0.23</td>
<td style="text-align: right;">0.00</td>
<td style="text-align: right;">-2.22</td>
<td style="text-align: right;">0.051</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Interaction</td>
<td style="text-align: right;">-0.27</td>
<td style="text-align: right;">0.074</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">-0.43</td>
<td style="text-align: right;">-0.10</td>
<td style="text-align: right;">-3.61</td>
<td style="text-align: right;">0.005</td>
</tr>
</tbody>
</table>


</div>
</div>
</section>
<section id="lmm-exp5c-plotthemodel" class="level4" data-number="14.11.3.3">
<h4 data-number="14.11.3.3" class="anchored" data-anchor-id="lmm-exp5c-plotthemodel"><span class="header-section-number">14.11.3.3</span> Plot the model</h4>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="lmm_files/figure-html/unnamed-chunk-41-1.png" class="img-fluid figure-img" width="576"></p>
<figcaption>Treatment effect on glucose uptake. Gray dots connected by lines are individual donors.</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="alternaplot-the-model" class="level4" data-number="14.11.3.4">
<h4 data-number="14.11.3.4" class="anchored" data-anchor-id="alternaplot-the-model"><span class="header-section-number">14.11.3.4</span> Alternaplot the model</h4>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="lmm_files/figure-html/lmm-exp5c-alternaplot-1.png" class="img-fluid figure-img" width="576"></p>
<figcaption>Glucose response to different treatments. Gray dots connected by lines are individual donors. Dashed gray line is expected additive mean of “siNR4A3 EPS”.</figcaption>
</figure>
</div>
</div>
</div>
<p>Notes</p>
<ol type="1">
<li>Many researchers might look at the wide confidence intervals relative to the short distance between the means and think “no effect”. The confidence intervals are correct, they simply are <em>not</em> meant to be tools for inferring anything about differences in means. This is one of many reasons why plots of means and error bars can be misleading for inference, despite the ubiquity of their use for communicating results. And, its why I prefer the <a href="#lmm-exp5c-plotthemodel">effects-and-response plots</a>, which explicitly communicate correct inference about effects.</li>
</ol>
</section>
</section>
<section id="why-we-care-about-modeling-batch-in-exp5c" class="level3" data-number="14.11.4">
<h3 data-number="14.11.4" class="anchored" data-anchor-id="why-we-care-about-modeling-batch-in-exp5c"><span class="header-section-number">14.11.4</span> Why we care about modeling batch in exp5c</h3>
<p>Figure @ref(fig:lmm-exp5c-why) shows the modeled means of the four treatment combinations and the individual values colored by donor. It is pretty easy to see that the glucose uptake values for donors 4 and 5 are well above the mean for all four treatments. And, the values for donors 1, 2, and 3 are well below the mean for all four treatments. The values for donor 6 are near the mean for all four treatments.</p>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="lmm_files/figure-html/lmm-exp5c-why-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption>Why we care about blocking. The black dots are the modeled means of each treatment combination. The colored dots are the measured values of the response for each donor. The position of a donor relative to the mean is easy to see with these data.</figcaption>
</figure>
</div>
</div>
</div>
<p>Let’s compare the effects estimated by the linear mixed model <code>exp5c_m1</code> with a linear model that ignores donor.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb65"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a>exp5c_m2 <span class="ot">&lt;-</span> <span class="fu">lm</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity,</span>
<span id="cb65-2"><a href="#cb65-2" aria-hidden="true" tabindex="-1"></a>           <span class="at">data =</span> exp5c)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="lmm_files/figure-html/lmm-exp5c-why2-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption>A. Inference from a linear mixed model with blocking factor (donor) added as a random intercept. B. Inference from a fixed effects model.</figcaption>
</figure>
</div>
</div>
</div>
<p>Figure @ref(fig:lmm-exp5c-why2)A is a plot of the effects from the linear mixed model that models the correlated error due to donor. Figure @ref(fig:lmm-exp5c-why2)B is a plot of the effects from the fixed effect model that ignores the correlated error due to donor. Adding <span class="math inline">\(\texttt{donor}\)</span> as a factor to the linear model increases the precision of the estimate of the treatment effects by eliminating the among-donor component of variance from the error variance.</p>
</section>
</section>
<section id="sec-lmm-example4" class="level2" data-number="14.12">
<h2 data-number="14.12" class="anchored" data-anchor-id="sec-lmm-example4"><span class="header-section-number">14.12</span> Example 4 – RCBDS Experiments with subsampling replication (exp1g)</h2>
<p>This example is from a design with batches (independent experiments) that contain all treatment levels of a single factor <em>and</em> subsampling replication. These data were used to introduce linear mixed models in Example 1. The design of the experiment is <span class="math inline">\(2 \times 2\)</span> factorial. Example 1 flattened the analysis to simplify explanation of random intercepts and random slopes. Here, the data are analyzed with a factorial model.</p>
<p><a href="https://www.nature.com/articles/s41586-019-1873-0" target="_blank">A GPR174–CCL21 module imparts sexual dimorphism to humoral immunity</a></p>
<p><a href="https://pubmed.ncbi.nlm.nih.gov/31875850/" target="_blank">Public source</a></p>
<p>Source figure: <a href="https://www.nature.com/articles/s41586-019-1873-0/figures/1" target="_blank">Fig. 1g</a></p>
<p>Source data: <a href="https://www.nature.com/articles/s41586-019-1873-0#MOESM3" target="_blank">Source Data Fig. 1</a></p>
<section id="understand-the-data-1" class="level3" data-number="14.12.1">
<h3 data-number="14.12.1" class="anchored" data-anchor-id="understand-the-data-1"><span class="header-section-number">14.12.1</span> Understand the data</h3>
<p>The researchers in this paper are interested in discovering mechanisms causing the lower <a href="https://en.wikipedia.org/wiki/Humoral_immunity" target="_blank">antibody-mediated immune response</a> in males relative to females. The data in Fig. 1 are from a set of experiments on mice to investigate how the G-protein coupled receptor protein GPR174 regulates formation of the <a href="https://en.wikipedia.org/wiki/Germinal_center" target="_blank">B-cell germinal center in secondary lymph tissue</a>. GPR174 is a X-linked gene.</p>
<p><strong>Response variable</strong> <span class="math inline">\(\texttt{gc}\)</span> – germinal center size (%). The units are the percent of cells expressing germinal center markers.</p>
<p><strong>Factor 1</strong> – <span class="math inline">\(\texttt{sex}\)</span> (“M”, “F”). Male (“M”) is the reference level.</p>
<p><strong>Factor 2</strong> – <span class="math inline">\(\texttt{chromosome}\)</span> (“Gpr174+”, “Gpr174-”). “Gpr174-” is a GPR174 knockout. The wildtype (“Gpr174+”) condition is the reference level.</p>
<p><strong>Design</strong> – <span class="math inline">\(2 \times 2\)</span>, that is, two crossed factors each with two levels. This results in four groups, each with a unique combination of the levels from each factor. “M Gpr174+” is the control. “M Gpr174+” is the knockout genotype in males (“knockout added”). “F Gpr174+” is the wildtype female (“X chromosome added”). “F Gpr174-” is the knockout female (“knockout and X chromosome added”.</p>
</section>
<section id="examine-the-data-1" class="level3" data-number="14.12.2">
<h3 data-number="14.12.2" class="anchored" data-anchor-id="examine-the-data-1"><span class="header-section-number">14.12.2</span> Examine the data</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> exp1g,</span>
<span id="cb66-2"><a href="#cb66-2" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x =</span> treatment,</span>
<span id="cb66-3"><a href="#cb66-3" aria-hidden="true" tabindex="-1"></a>           <span class="at">y =</span> gc,</span>
<span id="cb66-4"><a href="#cb66-4" aria-hidden="true" tabindex="-1"></a>           <span class="at">color =</span> experiment_id)) <span class="sc">+</span></span>
<span id="cb66-5"><a href="#cb66-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="fl">0.4</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="lmm_files/figure-html/lmm-exp1g-examine-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="fit-the-model" class="level3" data-number="14.12.3">
<h3 data-number="14.12.3" class="anchored" data-anchor-id="fit-the-model"><span class="header-section-number">14.12.3</span> Fit the model</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb67"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1" aria-hidden="true" tabindex="-1"></a><span class="co"># three slope parameters</span></span>
<span id="cb67-2"><a href="#cb67-2" aria-hidden="true" tabindex="-1"></a>exp1g_m1a <span class="ot">&lt;-</span> <span class="fu">lmer</span>(gc <span class="sc">~</span> genotype <span class="sc">*</span> sex <span class="sc">+</span></span>
<span id="cb67-3"><a href="#cb67-3" aria-hidden="true" tabindex="-1"></a>                   (genotype <span class="sc">*</span> sex <span class="sc">|</span> experiment_id),</span>
<span id="cb67-4"><a href="#cb67-4" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp1g)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>boundary (singular) fit: see help('isSingular')</code></pre>
</div>
<div class="sourceCode cell-code" id="cb69"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a><span class="fu">VarCorr</span>(exp1g_m1a) <span class="co"># looks fine</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> Groups        Name                 Std.Dev. Corr                
 experiment_id (Intercept)          1.77088                      
               genotypeGpr174-      0.96193  -0.023              
               sexF                 2.90154   0.459  0.878       
               genotypeGpr174-:sexF 1.33170  -0.314 -0.625 -0.706
 Residual                           1.93155                      </code></pre>
</div>
<div class="sourceCode cell-code" id="cb71"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a><span class="co"># one slope parameter but capturing all treatment combinations</span></span>
<span id="cb71-2"><a href="#cb71-2" aria-hidden="true" tabindex="-1"></a>exp1g_m1b <span class="ot">&lt;-</span> <span class="fu">lmer</span>(gc <span class="sc">~</span> genotype <span class="sc">*</span> sex <span class="sc">+</span></span>
<span id="cb71-3"><a href="#cb71-3" aria-hidden="true" tabindex="-1"></a>                   (treatment <span class="sc">|</span> experiment_id),</span>
<span id="cb71-4"><a href="#cb71-4" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp1g)</span>
<span id="cb71-5"><a href="#cb71-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-6"><a href="#cb71-6" aria-hidden="true" tabindex="-1"></a><span class="co"># intercept interactions</span></span>
<span id="cb71-7"><a href="#cb71-7" aria-hidden="true" tabindex="-1"></a>exp1g_m1c <span class="ot">&lt;-</span> <span class="fu">lmer</span>(gc <span class="sc">~</span> genotype <span class="sc">*</span> sex <span class="sc">+</span></span>
<span id="cb71-8"><a href="#cb71-8" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> experiment_id) <span class="sc">+</span></span>
<span id="cb71-9"><a href="#cb71-9" aria-hidden="true" tabindex="-1"></a>                    (<span class="dv">1</span> <span class="sc">|</span> experiment_id<span class="sc">:</span>genotype) <span class="sc">+</span></span>
<span id="cb71-10"><a href="#cb71-10" aria-hidden="true" tabindex="-1"></a>                    (<span class="dv">1</span> <span class="sc">|</span> experiment_id<span class="sc">:</span>sex) <span class="sc">+</span></span>
<span id="cb71-11"><a href="#cb71-11" aria-hidden="true" tabindex="-1"></a>                    (<span class="dv">1</span> <span class="sc">|</span> experiment_id<span class="sc">:</span>genotype<span class="sc">:</span>sex),</span>
<span id="cb71-12"><a href="#cb71-12" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp1g)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>boundary (singular) fit: see help('isSingular')</code></pre>
</div>
<div class="sourceCode cell-code" id="cb73"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a><span class="fu">VarCorr</span>(exp1g_m1c) <span class="co"># id:genotype is low</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> Groups                     Name        Std.Dev. 
 experiment_id:genotype:sex (Intercept) 0.4046670
 experiment_id:sex          (Intercept) 1.6927499
 experiment_id:genotype     (Intercept) 0.0001216
 experiment_id              (Intercept) 2.5487069
 Residual                               1.9792428</code></pre>
</div>
<div class="sourceCode cell-code" id="cb75"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb75-1"><a href="#cb75-1" aria-hidden="true" tabindex="-1"></a><span class="co"># drop id:genotype which has low variance</span></span>
<span id="cb75-2"><a href="#cb75-2" aria-hidden="true" tabindex="-1"></a>exp1g_m1d <span class="ot">&lt;-</span> <span class="fu">lmer</span>(gc <span class="sc">~</span> genotype <span class="sc">*</span> sex <span class="sc">+</span></span>
<span id="cb75-3"><a href="#cb75-3" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> experiment_id) <span class="sc">+</span></span>
<span id="cb75-4"><a href="#cb75-4" aria-hidden="true" tabindex="-1"></a>                    (<span class="dv">1</span> <span class="sc">|</span> experiment_id<span class="sc">:</span>sex) <span class="sc">+</span></span>
<span id="cb75-5"><a href="#cb75-5" aria-hidden="true" tabindex="-1"></a>                    (<span class="dv">1</span> <span class="sc">|</span> experiment_id<span class="sc">:</span>sex<span class="sc">:</span>genotype),</span>
<span id="cb75-6"><a href="#cb75-6" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp1g)</span>
<span id="cb75-7"><a href="#cb75-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb75-8"><a href="#cb75-8" aria-hidden="true" tabindex="-1"></a><span class="fu">AIC</span>(exp1g_m1a, exp1g_m1b, exp1g_m1c, exp1g_m1d)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>          df      AIC
exp1g_m1a 15 344.1579
exp1g_m1b 15 344.1529
exp1g_m1c  9 337.7602
exp1g_m1d  8 335.7602</code></pre>
</div>
<div class="sourceCode cell-code" id="cb77"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb77-1"><a href="#cb77-1" aria-hidden="true" tabindex="-1"></a><span class="co"># go with exp1g_m1d.</span></span>
<span id="cb77-2"><a href="#cb77-2" aria-hidden="true" tabindex="-1"></a>exp1g_m1 <span class="ot">&lt;-</span> exp1g_m1d</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="inference-from-the-model-1" class="level3" data-number="14.12.4">
<h3 data-number="14.12.4" class="anchored" data-anchor-id="inference-from-the-model-1"><span class="header-section-number">14.12.4</span> Inference from the model</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a>exp1g_m1_coef <span class="ot">&lt;-</span> <span class="fu">coef</span>(<span class="fu">summary</span>(exp1g_m1))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb79"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a>exp1g_m1_coef</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                      Estimate Std. Error       df   t value    Pr(&gt;|t|)
(Intercept)           8.435718   1.612602 4.638595  5.231123 0.004202015
genotypeGpr174-       2.568365   0.712122 5.760582  3.606637 0.012094327
sexF                  5.583860   1.397388 4.024010  3.995927 0.015995349
genotypeGpr174-:sexF -5.304752   1.013552 5.918159 -5.233823 0.002033657</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb81"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb81-1"><a href="#cb81-1" aria-hidden="true" tabindex="-1"></a><span class="co"># order of factors reversed in specs because I want sex to be</span></span>
<span id="cb81-2"><a href="#cb81-2" aria-hidden="true" tabindex="-1"></a><span class="co"># main x-axis variable in plot</span></span>
<span id="cb81-3"><a href="#cb81-3" aria-hidden="true" tabindex="-1"></a>exp1g_m1_emm <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(exp1g_m1, <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">"sex"</span>, <span class="st">"genotype"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb82"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a><span class="co"># exp1g_m1_emm # print in console to get row numbers</span></span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a><span class="co"># set the mean as the row number from the emmeans table</span></span>
<span id="cb82-3"><a href="#cb82-3" aria-hidden="true" tabindex="-1"></a>wt_m <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>)</span>
<span id="cb82-4"><a href="#cb82-4" aria-hidden="true" tabindex="-1"></a>wt_f <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">0</span>,<span class="dv">0</span>)</span>
<span id="cb82-5"><a href="#cb82-5" aria-hidden="true" tabindex="-1"></a>ko_m <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb82-6"><a href="#cb82-6" aria-hidden="true" tabindex="-1"></a>ko_f <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>)</span>
<span id="cb82-7"><a href="#cb82-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-8"><a href="#cb82-8" aria-hidden="true" tabindex="-1"></a><span class="co"># simple effects within males and females + interaction </span></span>
<span id="cb82-9"><a href="#cb82-9" aria-hidden="true" tabindex="-1"></a><span class="co"># 1. (ko_m - wt_m) </span></span>
<span id="cb82-10"><a href="#cb82-10" aria-hidden="true" tabindex="-1"></a><span class="co"># 2. (ko_f - wt_f)</span></span>
<span id="cb82-11"><a href="#cb82-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-12"><a href="#cb82-12" aria-hidden="true" tabindex="-1"></a>exp1g_contrasts <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb82-13"><a href="#cb82-13" aria-hidden="true" tabindex="-1"></a>  <span class="st">"(Gpr174- M) - (Gpr174+ M)"</span> <span class="ot">=</span> <span class="fu">c</span>(ko_m <span class="sc">-</span> wt_m),</span>
<span id="cb82-14"><a href="#cb82-14" aria-hidden="true" tabindex="-1"></a>  <span class="st">"(Gpr174- F) - (Gpr174+ F)"</span> <span class="ot">=</span> <span class="fu">c</span>(ko_f <span class="sc">-</span> wt_f),</span>
<span id="cb82-15"><a href="#cb82-15" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Interaction"</span> <span class="ot">=</span> <span class="fu">c</span>(ko_f <span class="sc">-</span> wt_f) <span class="sc">-</span></span>
<span id="cb82-16"><a href="#cb82-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">c</span>(ko_m <span class="sc">-</span> wt_m)</span>
<span id="cb82-17"><a href="#cb82-17" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb82-18"><a href="#cb82-18" aria-hidden="true" tabindex="-1"></a>exp1g_m1_planned <span class="ot">&lt;-</span> <span class="fu">contrast</span>(exp1g_m1_emm,</span>
<span id="cb82-19"><a href="#cb82-19" aria-hidden="true" tabindex="-1"></a>                       <span class="at">method =</span> exp1g_contrasts,</span>
<span id="cb82-20"><a href="#cb82-20" aria-hidden="true" tabindex="-1"></a>                       <span class="at">adjust =</span> <span class="st">"none"</span></span>
<span id="cb82-21"><a href="#cb82-21" aria-hidden="true" tabindex="-1"></a>) <span class="sc">|&gt;</span></span>
<span id="cb82-22"><a href="#cb82-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">contrast</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">estimate</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">SE</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">df</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">lower.CL</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">upper.CL</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">t.ratio</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">p.value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">(Gpr174- M) - (Gpr174+ M)</td>
<td style="text-align: right;">2.57</td>
<td style="text-align: right;">0.714</td>
<td style="text-align: right;">5.784</td>
<td style="text-align: right;">0.81</td>
<td style="text-align: right;">4.33</td>
<td style="text-align: right;">3.60</td>
<td style="text-align: right;">0.012</td>
</tr>
<tr class="even">
<td style="text-align: left;">(Gpr174- F) - (Gpr174+ F)</td>
<td style="text-align: right;">-2.74</td>
<td style="text-align: right;">0.722</td>
<td style="text-align: right;">6.100</td>
<td style="text-align: right;">-4.50</td>
<td style="text-align: right;">-0.98</td>
<td style="text-align: right;">-3.79</td>
<td style="text-align: right;">0.009</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Interaction</td>
<td style="text-align: right;">-5.30</td>
<td style="text-align: right;">1.016</td>
<td style="text-align: right;">5.942</td>
<td style="text-align: right;">-7.80</td>
<td style="text-align: right;">-2.81</td>
<td style="text-align: right;">-5.22</td>
<td style="text-align: right;">0.002</td>
</tr>
</tbody>
</table>


</div>
</div>
<p>Notes</p>
<ol type="1">
<li>The direction of the estimated effect is opposite in males and females</li>
</ol>
</section>
<section id="plot-the-model-3" class="level3" data-number="14.12.5">
<h3 data-number="14.12.5" class="anchored" data-anchor-id="plot-the-model-3"><span class="header-section-number">14.12.5</span> Plot the model</h3>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="lmm_files/figure-html/unnamed-chunk-46-1.png" class="img-fluid figure-img" width="576"></p>
<figcaption>Treatment effect on germinal center (GC) formation. Small, pale, colored dots are independent experiments. Intermediate size colored dots are experiment means.</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="alternaplot-the-model-1" class="level3" data-number="14.12.6">
<h3 data-number="14.12.6" class="anchored" data-anchor-id="alternaplot-the-model-1"><span class="header-section-number">14.12.6</span> Alternaplot the model</h3>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="lmm_files/figure-html/lmm-exp1g-alternaplot-1.png" class="img-fluid figure-img" width="576"></p>
<figcaption>Germinal center (GC) formation in response to treatment. Small, pale, colored dots are independent experiments. Intermediate size colored dots are experiment means. Dashed gray line is expected additive mean of Gpr174- F</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="understanding-the-alternative-models" class="level3" data-number="14.12.7">
<h3 data-number="14.12.7" class="anchored" data-anchor-id="understanding-the-alternative-models"><span class="header-section-number">14.12.7</span> Understanding the alternative models</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb83"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb83-1"><a href="#cb83-1" aria-hidden="true" tabindex="-1"></a>exp1g_m1a <span class="ot">&lt;-</span> <span class="fu">lmer</span>(gc <span class="sc">~</span> genotype <span class="sc">*</span> sex <span class="sc">+</span></span>
<span id="cb83-2"><a href="#cb83-2" aria-hidden="true" tabindex="-1"></a>                   (genotype <span class="sc">*</span> sex <span class="sc">|</span> experiment_id),</span>
<span id="cb83-3"><a href="#cb83-3" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp1g)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>boundary (singular) fit: see help('isSingular')</code></pre>
</div>
<div class="sourceCode cell-code" id="cb85"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb85-1"><a href="#cb85-1" aria-hidden="true" tabindex="-1"></a>exp1g_m1b <span class="ot">&lt;-</span> <span class="fu">lmer</span>(gc <span class="sc">~</span> genotype <span class="sc">*</span> sex <span class="sc">+</span></span>
<span id="cb85-2"><a href="#cb85-2" aria-hidden="true" tabindex="-1"></a>                   (treatment <span class="sc">|</span> experiment_id),</span>
<span id="cb85-3"><a href="#cb85-3" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp1g)</span>
<span id="cb85-4"><a href="#cb85-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb85-5"><a href="#cb85-5" aria-hidden="true" tabindex="-1"></a>exp1g_m1c <span class="ot">&lt;-</span> <span class="fu">lmer</span>(gc <span class="sc">~</span> genotype <span class="sc">*</span> sex <span class="sc">+</span></span>
<span id="cb85-6"><a href="#cb85-6" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> experiment_id) <span class="sc">+</span></span>
<span id="cb85-7"><a href="#cb85-7" aria-hidden="true" tabindex="-1"></a>                    (<span class="dv">1</span> <span class="sc">|</span> experiment_id<span class="sc">:</span>genotype) <span class="sc">+</span></span>
<span id="cb85-8"><a href="#cb85-8" aria-hidden="true" tabindex="-1"></a>                    (<span class="dv">1</span> <span class="sc">|</span> experiment_id<span class="sc">:</span>sex) <span class="sc">+</span></span>
<span id="cb85-9"><a href="#cb85-9" aria-hidden="true" tabindex="-1"></a>                    (<span class="dv">1</span> <span class="sc">|</span> experiment_id<span class="sc">:</span>genotype<span class="sc">:</span>sex),</span>
<span id="cb85-10"><a href="#cb85-10" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp1g)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>boundary (singular) fit: see help('isSingular')</code></pre>
</div>
<div class="sourceCode cell-code" id="cb87"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb87-1"><a href="#cb87-1" aria-hidden="true" tabindex="-1"></a>exp1g_m1d <span class="ot">&lt;-</span> <span class="fu">lmer</span>(gc <span class="sc">~</span> genotype <span class="sc">*</span> sex <span class="sc">+</span></span>
<span id="cb87-2"><a href="#cb87-2" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> experiment_id) <span class="sc">+</span></span>
<span id="cb87-3"><a href="#cb87-3" aria-hidden="true" tabindex="-1"></a>                    (<span class="dv">1</span> <span class="sc">|</span> experiment_id<span class="sc">:</span>sex) <span class="sc">+</span></span>
<span id="cb87-4"><a href="#cb87-4" aria-hidden="true" tabindex="-1"></a>                    (<span class="dv">1</span> <span class="sc">|</span> experiment_id<span class="sc">:</span>genotype<span class="sc">:</span>sex),</span>
<span id="cb87-5"><a href="#cb87-5" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp1g)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Notes</p>
<ol type="1">
<li>All four models model the same fixed effects. The models differ only in how they model the random effects.</li>
<li>Model <code>exp1g_m1a</code> fits one random intercept, two random slopes, and one random interaction.</li>
</ol>
<ul>
<li><span class="math inline">\(\gamma_{0j}\)</span> – a random intercept modeling batch effects of <span class="math inline">\(\texttt{experiment\_id}\)</span> on the intercept</li>
<li><span class="math inline">\(\gamma_{1j}\)</span> – a random slope modeling the effect of the non-reference level of “Gpr174-” on the batch effect of <span class="math inline">\(\texttt{experiment\_id}\)</span>. This is a <span class="math inline">\(\texttt{experiment\_id} \times \texttt{genotype}\)</span> interaction.</li>
<li><span class="math inline">\(\gamma_{2j}\)</span> – a random slope modeling the effect of “F” (female) on the batch effect of <span class="math inline">\(\texttt{experiment\_id}\)</span>. This is a <span class="math inline">\(\texttt{experiment\_id} \times \texttt{sex}\)</span> interaction.</li>
<li><span class="math inline">\(\gamma_{3j}\)</span> – a random slope modeling the effect of the interaction effect of “Gpr174-” and “F” on the batch effect of <span class="math inline">\(\texttt{experiment\_id}\)</span>. This is a <span class="math inline">\(\texttt{experiment\_id} \times \texttt{sex}  \times \texttt{genotype}\)</span> interaction.</li>
</ul>
<ol start="3" type="1">
<li>Model <code>exp1g_m1b</code> fits one random intercept and three random slopes</li>
</ol>
<ul>
<li><span class="math inline">\(\gamma_{0j}\)</span> – a random intercept modeling batch effects of <span class="math inline">\(\texttt{experiment\_id}\)</span> on the intercept. This is modeling the same thing as <span class="math inline">\(\gamma_{0j}\)</span> in Model <code>exp1g_m1a</code>.</li>
<li><span class="math inline">\(\gamma_{1j}\)</span> – a random slope modeling the effect of “Gpr174- M” on the batch effect of <span class="math inline">\(\texttt{experiment\_id}\)</span>. This is modeling the same thing as <span class="math inline">\(\gamma_{1j}\)</span> in Model <code>exp1g_m1a</code>.</li>
<li><span class="math inline">\(\gamma_{2j}\)</span> – a random slope modeling the effect of “Gpr174+ F” on the batch effect of <span class="math inline">\(\texttt{experiment\_id}\)</span>. This is modeling the same thing as <span class="math inline">\(\gamma_{2j}\)</span> in Model <code>exp1g_m1a</code>.</li>
<li><span class="math inline">\(\gamma_{3j}\)</span> – a random slope modeling the effect of “M Gpr174- F” on the batch effect of <span class="math inline">\(\texttt{experiment\_id}\)</span>. This is modeling the added variance accounted for by the random interaction <span class="math inline">\(\gamma_{2j}\)</span> in Model <code>exp1g_m1a</code> but in a different way.</li>
</ul>
<ol start="4" type="1">
<li>Model <code>exp1g_m1c</code> fits four random intercepts</li>
</ol>
<ul>
<li><span class="math inline">\(\gamma_{0j}\)</span> – a random intercept modeling batch effects of <span class="math inline">\(\texttt{experiment\_id}\)</span> on the intercept. This is modeling the same thing as <span class="math inline">\(\gamma_{0j}\)</span> in Model <code>exp1g_m1a</code>.</li>
<li><span class="math inline">\(\gamma_{0jk}\)</span> – a random intercept modeling the effects of the combination of <span class="math inline">\(\texttt{experiment\_id}\)</span> and <span class="math inline">\(\texttt{genotype}\)</span>. This is very similar to the variance modeled by <span class="math inline">\(\gamma_{1j}\)</span> in Model <code>exp1g_m1a</code> except the draws from <span class="math inline">\(\gamma_{0jk}\)</span> are independent (uncorrelated) of draws from the other random intercepts.</li>
<li><span class="math inline">\(\gamma_{0jl}\)</span> – a random intercept modeling the effects of the combination of <span class="math inline">\(\texttt{experiment\_id}\)</span> and <span class="math inline">\(\texttt{sex}\)</span>. This is very similar to the variance modeled by <span class="math inline">\(\gamma_{2j}\)</span> in Model <code>exp1g_m1a</code> except the draws from <span class="math inline">\(\gamma_{0jl}\)</span> are independent (uncorrelated) of draws from the other random intercepts (review <a href="#lmm-varcorr">The correlation among random intercepts and slopes</a> if this doesn’t make sense).</li>
<li><span class="math inline">\(\gamma_{0jkl}\)</span> – a random intercept modeling the effects of the combination of <span class="math inline">\(\texttt{experiment\_id}\)</span>, <span class="math inline">\(\texttt{genotype}\)</span> and <span class="math inline">\(\texttt{sex}\)</span>. This is very similar to the variance modeled by <span class="math inline">\(\gamma_{3j}\)</span> in Model <code>exp1g_m1a</code> except the draws from <span class="math inline">\(\gamma_{0jkl}\)</span> are independent (uncorrelated) of draws from the other random intercepts.</li>
</ul>
<ol start="5" type="1">
<li>Model <code>exp1g_m1d</code> fits the same random intercepts as Model <code>exp1g_m1c</code> but excludes <span class="math inline">\(\gamma_{0jl}\)</span> (the random intercept for the experiment_id by genotyp combination. This was excluded because of the low variance of this component in the fit model.</li>
</ol>
</section>
<section id="the-varcorr-matrix-of-models-exp1g_m1a-and-exp1g_m1b" class="level3" data-number="14.12.8">
<h3 data-number="14.12.8" class="anchored" data-anchor-id="the-varcorr-matrix-of-models-exp1g_m1a-and-exp1g_m1b"><span class="header-section-number">14.12.8</span> The VarCorr matrix of models exp1g_m1a and exp1g_m1b</h3>
<p>The random effect similarity of models <code>exp1g_m1a</code> and <code>exp1g_m1b</code> can be seen in the estimated variance components and correlations among the random effects.</p>
<div class="cell">
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<caption>The Varcorr matrix. Standard deviations of random effects on the diagonal. Correlations of random effects on the off-diagonal.</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: right;" data-quarto-table-cell-role="th"></th>
<th style="text-align: right;" data-quarto-table-cell-role="th"></th>
<th style="text-align: right;" data-quarto-table-cell-role="th"></th>
<th style="text-align: right;" data-quarto-table-cell-role="th"></th>
</tr>
</thead>
<tbody>
<tr class="odd" data-grouplength="4">
<td colspan="5" style="border-bottom: 1px solid"><strong>exp1g_m1a</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(Intercept)</td>
<td style="text-align: right;">1.7709</td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">genotypeGpr174-</td>
<td style="text-align: right;">-0.0232</td>
<td style="text-align: right;">0.9619</td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">sexF</td>
<td style="text-align: right;">0.4589</td>
<td style="text-align: right;">0.8776</td>
<td style="text-align: right;">2.9015</td>
<td style="text-align: right;"></td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">genotypeGpr174-:sexF</td>
<td style="text-align: right;">-0.3137</td>
<td style="text-align: right;">-0.6247</td>
<td style="text-align: right;">-0.7056</td>
<td style="text-align: right;">1.3317</td>
</tr>
<tr class="even" data-grouplength="4">
<td colspan="5" style="border-bottom: 1px solid"><strong>exp1g_m1b</strong></td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(Intercept)</td>
<td style="text-align: right;">1.7711</td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">treatmentGpr174- M</td>
<td style="text-align: right;">-0.0250</td>
<td style="text-align: right;">0.9641</td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">treatmentGpr174+ F</td>
<td style="text-align: right;">0.4585</td>
<td style="text-align: right;">0.8761</td>
<td style="text-align: right;">2.9024</td>
<td style="text-align: right;"></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">treatmentGpr174- F</td>
<td style="text-align: right;">0.2999</td>
<td style="text-align: right;">0.8793</td>
<td style="text-align: right;">0.9384</td>
<td style="text-align: right;">2.9909</td>
</tr>
</tbody>
</table>


</div>
</div>
</section>
<section id="the-linear-mixed-model-has-more-precision-and-power-than-the-fixed-effect-model-of-batch-means" class="level3" data-number="14.12.9">
<h3 data-number="14.12.9" class="anchored" data-anchor-id="the-linear-mixed-model-has-more-precision-and-power-than-the-fixed-effect-model-of-batch-means"><span class="header-section-number">14.12.9</span> The linear mixed model has more precision and power than the fixed effect model of batch means</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb88"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a><span class="co"># means pooling model</span></span>
<span id="cb88-2"><a href="#cb88-2" aria-hidden="true" tabindex="-1"></a>exp1g_m2 <span class="ot">&lt;-</span> <span class="fu">lm</span>(gc <span class="sc">~</span> sex <span class="sc">*</span> genotype,</span>
<span id="cb88-3"><a href="#cb88-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">data =</span> exp1g_means)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<caption>Planned contrasts for the linear mixed model exp1g_m1 and the fixed effects model of experiment means exp1g_m2.</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">contrast</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">estimate</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">SE</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">df</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">lower.CL</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">upper.CL</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">t.ratio</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">p.value</th>
</tr>
</thead>
<tbody>
<tr class="odd" data-grouplength="3">
<td colspan="8" style="border-bottom: 1px solid"><strong>exp1g_m1 (lmm)</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(Gpr174- M) - (Gpr174+ M)</td>
<td style="text-align: right;">2.57</td>
<td style="text-align: right;">0.714</td>
<td style="text-align: right;">5.8</td>
<td style="text-align: right;">0.81</td>
<td style="text-align: right;">4.33</td>
<td style="text-align: right;">3.60</td>
<td style="text-align: right;">0.012</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(Gpr174- F) - (Gpr174+ F)</td>
<td style="text-align: right;">-2.74</td>
<td style="text-align: right;">0.722</td>
<td style="text-align: right;">6.1</td>
<td style="text-align: right;">-4.50</td>
<td style="text-align: right;">-0.98</td>
<td style="text-align: right;">-3.79</td>
<td style="text-align: right;">0.009</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Interaction</td>
<td style="text-align: right;">-5.30</td>
<td style="text-align: right;">1.016</td>
<td style="text-align: right;">5.9</td>
<td style="text-align: right;">-7.80</td>
<td style="text-align: right;">-2.81</td>
<td style="text-align: right;">-5.22</td>
<td style="text-align: right;">0.002</td>
</tr>
<tr class="odd" data-grouplength="3">
<td colspan="8" style="border-bottom: 1px solid"><strong>exp1g_m2 (lm means pooling)</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(Gpr174- M) - (Gpr174+ M)</td>
<td style="text-align: right;">2.50</td>
<td style="text-align: right;">2.257</td>
<td style="text-align: right;">12.0</td>
<td style="text-align: right;">-2.41</td>
<td style="text-align: right;">7.42</td>
<td style="text-align: right;">1.11</td>
<td style="text-align: right;">0.289</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(Gpr174- F) - (Gpr174+ F)</td>
<td style="text-align: right;">-2.73</td>
<td style="text-align: right;">2.257</td>
<td style="text-align: right;">12.0</td>
<td style="text-align: right;">-7.65</td>
<td style="text-align: right;">2.18</td>
<td style="text-align: right;">-1.21</td>
<td style="text-align: right;">0.249</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Interaction</td>
<td style="text-align: right;">-5.24</td>
<td style="text-align: right;">3.192</td>
<td style="text-align: right;">12.0</td>
<td style="text-align: right;">-12.20</td>
<td style="text-align: right;">1.72</td>
<td style="text-align: right;">-1.64</td>
<td style="text-align: right;">0.127</td>
</tr>
</tbody>
</table>


</div>
</div>
<p>Notes</p>
<ol type="1">
<li>A fixed effects model fit to batch-means pooled data is strongly conservative and will result in less discovery.</li>
<li>Means pooling does not make the data independent in a randomized complete block design. A linear mixed model of batch-means pooled data is a <strong>mixed-effect ANOVA</strong> (Next section. Also see Section @ref(issues-exp4d) in the Issues chapter).</li>
</ol>
</section>
<section id="fixed-effect-models-and-pseudoreplication" class="level3" data-number="14.12.10">
<h3 data-number="14.12.10" class="anchored" data-anchor-id="fixed-effect-models-and-pseudoreplication"><span class="header-section-number">14.12.10</span> Fixed effect models and pseudoreplication</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb89"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb89-1"><a href="#cb89-1" aria-hidden="true" tabindex="-1"></a><span class="co"># complete pooling model</span></span>
<span id="cb89-2"><a href="#cb89-2" aria-hidden="true" tabindex="-1"></a>exp1g_m3 <span class="ot">&lt;-</span> <span class="fu">lm</span>(gc <span class="sc">~</span> sex <span class="sc">*</span> genotype,</span>
<span id="cb89-3"><a href="#cb89-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">data =</span> exp1g)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<caption>Planned contrasts for the linear mixed model exp1g_m1 and the fixed effects model exp1g_m3 with complete pooling.</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">contrast</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">estimate</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">SE</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">df</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">lower.CL</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">upper.CL</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">t.ratio</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">p.value</th>
</tr>
</thead>
<tbody>
<tr class="odd" data-grouplength="3">
<td colspan="8" style="border-bottom: 1px solid"><strong>exp1g_m1 (lmm)</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(Gpr174- M) - (Gpr174+ M)</td>
<td style="text-align: right;">2.57</td>
<td style="text-align: right;">0.714</td>
<td style="text-align: right;">5.8</td>
<td style="text-align: right;">0.81</td>
<td style="text-align: right;">4.33</td>
<td style="text-align: right;">3.60</td>
<td style="text-align: right;">0.012</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(Gpr174- F) - (Gpr174+ F)</td>
<td style="text-align: right;">-2.74</td>
<td style="text-align: right;">0.722</td>
<td style="text-align: right;">6.1</td>
<td style="text-align: right;">-4.50</td>
<td style="text-align: right;">-0.98</td>
<td style="text-align: right;">-3.79</td>
<td style="text-align: right;">0.009</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Interaction</td>
<td style="text-align: right;">-5.30</td>
<td style="text-align: right;">1.016</td>
<td style="text-align: right;">5.9</td>
<td style="text-align: right;">-7.80</td>
<td style="text-align: right;">-2.81</td>
<td style="text-align: right;">-5.22</td>
<td style="text-align: right;">0.002</td>
</tr>
<tr class="odd" data-grouplength="3">
<td colspan="8" style="border-bottom: 1px solid"><strong>exp1g_m3 (lm complete pooling)</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(Gpr174- M) - (Gpr174+ M)</td>
<td style="text-align: right;">2.44</td>
<td style="text-align: right;">1.111</td>
<td style="text-align: right;">69.0</td>
<td style="text-align: right;">0.22</td>
<td style="text-align: right;">4.65</td>
<td style="text-align: right;">2.20</td>
<td style="text-align: right;">0.032</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(Gpr174- F) - (Gpr174+ F)</td>
<td style="text-align: right;">-2.53</td>
<td style="text-align: right;">1.125</td>
<td style="text-align: right;">69.0</td>
<td style="text-align: right;">-4.77</td>
<td style="text-align: right;">-0.28</td>
<td style="text-align: right;">-2.25</td>
<td style="text-align: right;">0.028</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Interaction</td>
<td style="text-align: right;">-4.97</td>
<td style="text-align: right;">1.581</td>
<td style="text-align: right;">69.0</td>
<td style="text-align: right;">-8.12</td>
<td style="text-align: right;">-1.81</td>
<td style="text-align: right;">-3.14</td>
<td style="text-align: right;">0.002</td>
</tr>
</tbody>
</table>


</div>
</div>
<p>Notes</p>
<ol type="1">
<li>Complete pooling is strongly anti-conservative and will result in increased false discovery. Complete pooling is a type of pseudoreplication – the subsamples have been analyzed as if they are independent replicates. Subsamples are not independent.</li>
<li>For the Experiment 1g data, the 95% confidence intervals are wider and the <em>p</em>-values are larger in the complete pool model <code>exp1g_m3</code> compared to the linear mixed model <code>exp1g_m1</code>. This is an unusual result.</li>
</ol>
</section>
<section id="mixed-effect-anova" class="level3" data-number="14.12.11">
<h3 data-number="14.12.11" class="anchored" data-anchor-id="mixed-effect-anova"><span class="header-section-number">14.12.11</span> Mixed-effect ANOVA</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb90"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a>exp1g_m1_aov <span class="ot">&lt;-</span> <span class="fu">aov_4</span>(gc <span class="sc">~</span> sex <span class="sc">*</span> genotype <span class="sc">+</span></span>
<span id="cb90-2"><a href="#cb90-2" aria-hidden="true" tabindex="-1"></a>                  (sex <span class="sc">*</span> genotype <span class="sc">|</span> experiment_id),</span>
<span id="cb90-3"><a href="#cb90-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">data =</span> exp1g)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: More than one observation per design cell, aggregating data using `fun_aggregate = mean`.
To turn off this warning, pass `fun_aggregate = mean` explicitly.</code></pre>
</div>
</div>
<p>Notes</p>
<ol type="1">
<li>The formula has the same format as that in Example 3 for repeated measures ANOVA on RCB designs with no subsampling. In biology, this is often called a <strong>mixed effect ANOVA</strong> with two fixed factors and one random factor.</li>
<li>The data are aggregated prior to fitting the model – this means the subsamples are averaged within each batch by treatment combination.</li>
<li>The mixed-effect ANOVA is equivalent to the linear mixed model <code>exp1g_m1c</code> if there are the same number of replicates in each treatment combination and the same number of subsamples in all treatment by <span class="math inline">\(\texttt{experiment\_id}\)</span> combinations. The design is not balanced in this example.</li>
</ol>
<div class="cell">
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<caption>Planned contrasts from mixed ANOVA compared to lmm equivalent of mixed ANOVA and lowest AIC lmm.</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">contrast</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">estimate</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">SE</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">df</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">lower.CL</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">upper.CL</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">t.ratio</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">p.value</th>
</tr>
</thead>
<tbody>
<tr class="odd" data-grouplength="3">
<td colspan="8" style="border-bottom: 1px solid"><strong>exp1g_m1_aov (mixed ANOVA)</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(Gpr174- M) - (Gpr174+ M)</td>
<td style="text-align: right;">2.50</td>
<td style="text-align: right;">0.599</td>
<td style="text-align: right;">3.000</td>
<td style="text-align: right;">0.60</td>
<td style="text-align: right;">4.41</td>
<td style="text-align: right;">4.18</td>
<td style="text-align: right;">0.025</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(Gpr174- F) - (Gpr174+ F)</td>
<td style="text-align: right;">-2.73</td>
<td style="text-align: right;">0.808</td>
<td style="text-align: right;">3.000</td>
<td style="text-align: right;">-5.30</td>
<td style="text-align: right;">-0.16</td>
<td style="text-align: right;">-3.39</td>
<td style="text-align: right;">0.043</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Interaction</td>
<td style="text-align: right;">-5.24</td>
<td style="text-align: right;">1.087</td>
<td style="text-align: right;">3.000</td>
<td style="text-align: right;">-8.70</td>
<td style="text-align: right;">-1.78</td>
<td style="text-align: right;">-4.82</td>
<td style="text-align: right;">0.017</td>
</tr>
<tr class="odd" data-grouplength="3">
<td colspan="8" style="border-bottom: 1px solid"><strong>exp1g_m1_c (lmm equivalent of mixed ANOVA)</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(Gpr174- M) - (Gpr174+ M)</td>
<td style="text-align: right;">2.57</td>
<td style="text-align: right;">0.715</td>
<td style="text-align: right;">5.784</td>
<td style="text-align: right;">0.80</td>
<td style="text-align: right;">4.33</td>
<td style="text-align: right;">3.59</td>
<td style="text-align: right;">0.012</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(Gpr174- F) - (Gpr174+ F)</td>
<td style="text-align: right;">-2.74</td>
<td style="text-align: right;">0.724</td>
<td style="text-align: right;">6.100</td>
<td style="text-align: right;">-4.50</td>
<td style="text-align: right;">-0.97</td>
<td style="text-align: right;">-3.78</td>
<td style="text-align: right;">0.009</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Interaction</td>
<td style="text-align: right;">-5.30</td>
<td style="text-align: right;">1.017</td>
<td style="text-align: right;">2.972</td>
<td style="text-align: right;">-8.56</td>
<td style="text-align: right;">-2.05</td>
<td style="text-align: right;">-5.21</td>
<td style="text-align: right;">0.014</td>
</tr>
<tr class="odd" data-grouplength="3">
<td colspan="8" style="border-bottom: 1px solid"><strong>exp1g_md (lmm with min AIC)</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(Gpr174- M) - (Gpr174+ M)</td>
<td style="text-align: right;">2.57</td>
<td style="text-align: right;">0.714</td>
<td style="text-align: right;">5.784</td>
<td style="text-align: right;">0.81</td>
<td style="text-align: right;">4.33</td>
<td style="text-align: right;">3.60</td>
<td style="text-align: right;">0.012</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(Gpr174- F) - (Gpr174+ F)</td>
<td style="text-align: right;">-2.74</td>
<td style="text-align: right;">0.722</td>
<td style="text-align: right;">6.100</td>
<td style="text-align: right;">-4.50</td>
<td style="text-align: right;">-0.98</td>
<td style="text-align: right;">-3.79</td>
<td style="text-align: right;">0.009</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Interaction</td>
<td style="text-align: right;">-5.30</td>
<td style="text-align: right;">1.016</td>
<td style="text-align: right;">5.942</td>
<td style="text-align: right;">-7.80</td>
<td style="text-align: right;">-2.81</td>
<td style="text-align: right;">-5.22</td>
<td style="text-align: right;">0.002</td>
</tr>
</tbody>
</table>


</div>
</div>
</section>
</section>
<section id="working-in-r" class="level2" data-number="14.13">
<h2 data-number="14.13" class="anchored" data-anchor-id="working-in-r"><span class="header-section-number">14.13</span> Working in R</h2>
<section id="plotting-models-fit-to-batched-data" class="level3" data-number="14.13.1">
<h3 data-number="14.13.1" class="anchored" data-anchor-id="plotting-models-fit-to-batched-data"><span class="header-section-number">14.13.1</span> Plotting models fit to batched data</h3>
<section id="models-without-subsampling---example-3-experiment-2c-right-panel" class="level4" data-number="14.13.1.1">
<h4 data-number="14.13.1.1" class="anchored" data-anchor-id="models-without-subsampling---example-3-experiment-2c-right-panel"><span class="header-section-number">14.13.1.1</span> Models without subsampling - Example 3 (Experiment 2c right panel)</h4>
<p>Data wrangling necessary for plot:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb92"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1" aria-hidden="true" tabindex="-1"></a><span class="co"># aov_4 didn't like the group names, so switching these back</span></span>
<span id="cb92-2"><a href="#cb92-2" aria-hidden="true" tabindex="-1"></a>aov2_emm_dt <span class="ot">&lt;-</span> aov2_emm <span class="sc">|&gt;</span></span>
<span id="cb92-3"><a href="#cb92-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">|&gt;</span></span>
<span id="cb92-4"><a href="#cb92-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.table</span>()</span>
<span id="cb92-5"><a href="#cb92-5" aria-hidden="true" tabindex="-1"></a>aov2_emm_dt[, genotype <span class="sc">:</span><span class="er">=</span> <span class="fu">levels</span>(fig2c_r<span class="sc">$</span>genotype)]</span>
<span id="cb92-6"><a href="#cb92-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-7"><a href="#cb92-7" aria-hidden="true" tabindex="-1"></a><span class="co"># create a pretty p-value column</span></span>
<span id="cb92-8"><a href="#cb92-8" aria-hidden="true" tabindex="-1"></a>aov2_pairs_dt <span class="ot">&lt;-</span> <span class="fu">data.table</span>(aov2_pairs)</span>
<span id="cb92-9"><a href="#cb92-9" aria-hidden="true" tabindex="-1"></a>aov2_pairs_dt[, pretty_p <span class="sc">:</span><span class="er">=</span> lazyWeave<span class="sc">::</span><span class="fu">pvalString</span>(p.value)]</span>
<span id="cb92-10"><a href="#cb92-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-11"><a href="#cb92-11" aria-hidden="true" tabindex="-1"></a><span class="co"># add group1 and group2 columns to aov2_pairs_dt</span></span>
<span id="cb92-12"><a href="#cb92-12" aria-hidden="true" tabindex="-1"></a><span class="co"># again, aov_4 didn't like group names so replace contrasts</span></span>
<span id="cb92-13"><a href="#cb92-13" aria-hidden="true" tabindex="-1"></a><span class="co"># with those from the lmm1 fit</span></span>
<span id="cb92-14"><a href="#cb92-14" aria-hidden="true" tabindex="-1"></a>aov2_pairs_dt[, contrast <span class="sc">:</span><span class="er">=</span> lmm1_pairs<span class="sc">$</span>contrast]</span>
<span id="cb92-15"><a href="#cb92-15" aria-hidden="true" tabindex="-1"></a><span class="co"># now copy the first part of each contrast into group 1 and 2nd part into group 2</span></span>
<span id="cb92-16"><a href="#cb92-16" aria-hidden="true" tabindex="-1"></a>aov2_pairs_dt[, group1 <span class="sc">:</span><span class="er">=</span> <span class="fu">c</span>(<span class="st">"PD1+TIM3-"</span>, <span class="st">"PD1-TIM3-"</span>, <span class="st">"PD1-TIM3-"</span>)]</span>
<span id="cb92-17"><a href="#cb92-17" aria-hidden="true" tabindex="-1"></a>aov2_pairs_dt[, group2 <span class="sc">:</span><span class="er">=</span> <span class="fu">c</span>(<span class="st">"PD1+TIM3+"</span>, <span class="st">"PD1+TIM3+"</span>, <span class="st">"PD1+TIM3-"</span>)]</span>
<span id="cb92-18"><a href="#cb92-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-19"><a href="#cb92-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-20"><a href="#cb92-20" aria-hidden="true" tabindex="-1"></a>gg1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> fig2c_r,</span>
<span id="cb92-21"><a href="#cb92-21" aria-hidden="true" tabindex="-1"></a>              <span class="fu">aes</span>(<span class="at">x =</span> genotype,</span>
<span id="cb92-22"><a href="#cb92-22" aria-hidden="true" tabindex="-1"></a>                  <span class="at">y =</span> nr4a1)) <span class="sc">+</span></span>
<span id="cb92-23"><a href="#cb92-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">group =</span> experiment_id),</span>
<span id="cb92-24"><a href="#cb92-24" aria-hidden="true" tabindex="-1"></a>             <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.2</span>),</span>
<span id="cb92-25"><a href="#cb92-25" aria-hidden="true" tabindex="-1"></a>             <span class="at">size =</span> <span class="dv">2</span>,</span>
<span id="cb92-26"><a href="#cb92-26" aria-hidden="true" tabindex="-1"></a>             <span class="at">color =</span> <span class="st">"gray40"</span>) <span class="sc">+</span></span>
<span id="cb92-27"><a href="#cb92-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group =</span> experiment_id),</span>
<span id="cb92-28"><a href="#cb92-28" aria-hidden="true" tabindex="-1"></a>            <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.2</span>),</span>
<span id="cb92-29"><a href="#cb92-29" aria-hidden="true" tabindex="-1"></a>            <span class="at">color =</span> <span class="st">"gray40"</span>) <span class="sc">+</span></span>
<span id="cb92-30"><a href="#cb92-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> aov2_emm_dt,</span>
<span id="cb92-31"><a href="#cb92-31" aria-hidden="true" tabindex="-1"></a>             <span class="fu">aes</span>(<span class="at">y =</span> emmean,</span>
<span id="cb92-32"><a href="#cb92-32" aria-hidden="true" tabindex="-1"></a>                 <span class="at">color =</span> genotype),</span>
<span id="cb92-33"><a href="#cb92-33" aria-hidden="true" tabindex="-1"></a>             <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb92-34"><a href="#cb92-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_errorbar</span>(<span class="at">data =</span> aov2_emm_dt,</span>
<span id="cb92-35"><a href="#cb92-35" aria-hidden="true" tabindex="-1"></a>                <span class="fu">aes</span>(<span class="at">y =</span> emmean,</span>
<span id="cb92-36"><a href="#cb92-36" aria-hidden="true" tabindex="-1"></a>                    <span class="at">ymin =</span> lower.CL,</span>
<span id="cb92-37"><a href="#cb92-37" aria-hidden="true" tabindex="-1"></a>                    <span class="at">ymax =</span> upper.CL,</span>
<span id="cb92-38"><a href="#cb92-38" aria-hidden="true" tabindex="-1"></a>                    <span class="at">color =</span> genotype),</span>
<span id="cb92-39"><a href="#cb92-39" aria-hidden="true" tabindex="-1"></a>                <span class="at">width =</span> .<span class="dv">05</span>) <span class="sc">+</span></span>
<span id="cb92-40"><a href="#cb92-40" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb92-41"><a href="#cb92-41" aria-hidden="true" tabindex="-1"></a>  <span class="co">#pvalue brackets</span></span>
<span id="cb92-42"><a href="#cb92-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_pvalue_manual</span>(aov2_pairs_dt,</span>
<span id="cb92-43"><a href="#cb92-43" aria-hidden="true" tabindex="-1"></a>                     <span class="at">label =</span> <span class="st">"pretty_p"</span>,</span>
<span id="cb92-44"><a href="#cb92-44" aria-hidden="true" tabindex="-1"></a>                     <span class="at">y.position =</span> <span class="fu">c</span>(<span class="dv">305</span>,<span class="dv">320</span>,<span class="dv">290</span>),</span>
<span id="cb92-45"><a href="#cb92-45" aria-hidden="true" tabindex="-1"></a>                     <span class="at">size =</span> <span class="dv">4</span>,</span>
<span id="cb92-46"><a href="#cb92-46" aria-hidden="true" tabindex="-1"></a>                     <span class="at">tip.length =</span> <span class="fl">0.01</span>) <span class="sc">+</span></span>
<span id="cb92-47"><a href="#cb92-47" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb92-48"><a href="#cb92-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"MFI"</span>) <span class="sc">+</span></span>
<span id="cb92-49"><a href="#cb92-49" aria-hidden="true" tabindex="-1"></a><span class="co">#  scale_color_manual(values = pal_okabe_ito_3) +</span></span>
<span id="cb92-50"><a href="#cb92-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#E69F00"</span>, <span class="st">"#D55E00"</span>, <span class="st">"#0072B2"</span>)) <span class="sc">+</span></span>
<span id="cb92-51"><a href="#cb92-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_pubr</span>() <span class="sc">+</span></span>
<span id="cb92-52"><a href="#cb92-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb92-53"><a href="#cb92-53" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(), <span class="co"># no x-axis title</span></span>
<span id="cb92-54"><a href="#cb92-54" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">"none"</span></span>
<span id="cb92-55"><a href="#cb92-55" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span> </span>
<span id="cb92-56"><a href="#cb92-56" aria-hidden="true" tabindex="-1"></a>  <span class="cn">NULL</span></span>
<span id="cb92-57"><a href="#cb92-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-58"><a href="#cb92-58" aria-hidden="true" tabindex="-1"></a>gg1</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="lmm_files/figure-html/unnamed-chunk-49-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="models-with-subsampling---experiment-1g" class="level4" data-number="14.13.1.2">
<h4 data-number="14.13.1.2" class="anchored" data-anchor-id="models-with-subsampling---experiment-1g"><span class="header-section-number">14.13.1.2</span> Models with subsampling - Experiment 1g</h4>
<p>Data wrangling necessary for plot:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb93"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb93-1"><a href="#cb93-1" aria-hidden="true" tabindex="-1"></a><span class="co"># add treatment column to emmeans table</span></span>
<span id="cb93-2"><a href="#cb93-2" aria-hidden="true" tabindex="-1"></a>exp1g_m1_emm_dt <span class="ot">&lt;-</span> <span class="fu">summary</span>(exp1g_m1_emm) <span class="sc">|&gt;</span></span>
<span id="cb93-3"><a href="#cb93-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.table</span>()</span>
<span id="cb93-4"><a href="#cb93-4" aria-hidden="true" tabindex="-1"></a>exp1g_m1_emm_dt[, treatment <span class="sc">:</span><span class="er">=</span> <span class="fu">paste</span>(genotype, sex)]</span>
<span id="cb93-5"><a href="#cb93-5" aria-hidden="true" tabindex="-1"></a>exp1g_m1_emm_dt[, treatment <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(treatment,</span>
<span id="cb93-6"><a href="#cb93-6" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">levels =</span> <span class="fu">levels</span>(exp1g<span class="sc">$</span>treatment))]</span>
<span id="cb93-7"><a href="#cb93-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb93-8"><a href="#cb93-8" aria-hidden="true" tabindex="-1"></a><span class="co"># create table of means for each treatment * experiment_id combination</span></span>
<span id="cb93-9"><a href="#cb93-9" aria-hidden="true" tabindex="-1"></a>exp1g[, group_mean <span class="sc">:</span><span class="er">=</span> <span class="fu">predict</span>(exp1g_m1)]</span>
<span id="cb93-10"><a href="#cb93-10" aria-hidden="true" tabindex="-1"></a>exp1g_m1_emm2 <span class="ot">&lt;-</span> exp1g[, .(<span class="at">group_mean =</span> <span class="fu">mean</span>(group_mean)),</span>
<span id="cb93-11"><a href="#cb93-11" aria-hidden="true" tabindex="-1"></a>                       by <span class="ot">=</span> .(treatment, experiment_id)]</span>
<span id="cb93-12"><a href="#cb93-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb93-13"><a href="#cb93-13" aria-hidden="true" tabindex="-1"></a><span class="co"># add group1 and group2 columns to exp1g_m1_planned</span></span>
<span id="cb93-14"><a href="#cb93-14" aria-hidden="true" tabindex="-1"></a>exp1g_m1_planned_dt <span class="ot">&lt;-</span> <span class="fu">data.table</span>(exp1g_m1_planned)</span>
<span id="cb93-15"><a href="#cb93-15" aria-hidden="true" tabindex="-1"></a>exp1g_m1_planned_dt[, pretty_p <span class="sc">:</span><span class="er">=</span> <span class="fu">pvalString</span>(p.value)]</span>
<span id="cb93-16"><a href="#cb93-16" aria-hidden="true" tabindex="-1"></a>exp1g_m1_planned_dt[, group1 <span class="sc">:</span><span class="er">=</span> <span class="fu">c</span>(<span class="st">"Gpr174- M"</span>, <span class="st">"Gpr174- F"</span>, <span class="st">""</span>)]</span>
<span id="cb93-17"><a href="#cb93-17" aria-hidden="true" tabindex="-1"></a>exp1g_m1_planned_dt[, group2 <span class="sc">:</span><span class="er">=</span> <span class="fu">c</span>(<span class="st">"Gpr174+ M"</span>, <span class="st">"Gpr174+ F"</span>, <span class="st">""</span>)]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb94"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a><span class="co"># get coefficients of model</span></span>
<span id="cb94-2"><a href="#cb94-2" aria-hidden="true" tabindex="-1"></a>b <span class="ot">&lt;-</span> exp1g_m1_coef[, <span class="st">"Estimate"</span>]</span>
<span id="cb94-3"><a href="#cb94-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb94-4"><a href="#cb94-4" aria-hidden="true" tabindex="-1"></a><span class="co"># get interaction p</span></span>
<span id="cb94-5"><a href="#cb94-5" aria-hidden="true" tabindex="-1"></a>p_ixn <span class="ot">&lt;-</span> exp1g_m1_planned_dt[contrast <span class="sc">==</span> <span class="st">"Interaction"</span>, pretty_p]</span>
<span id="cb94-6"><a href="#cb94-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb94-7"><a href="#cb94-7" aria-hidden="true" tabindex="-1"></a>exp1g_plot_a <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> exp1g,</span>
<span id="cb94-8"><a href="#cb94-8" aria-hidden="true" tabindex="-1"></a>              <span class="fu">aes</span>(<span class="at">x =</span> treatment,</span>
<span id="cb94-9"><a href="#cb94-9" aria-hidden="true" tabindex="-1"></a>                  <span class="at">y =</span> gc,</span>
<span id="cb94-10"><a href="#cb94-10" aria-hidden="true" tabindex="-1"></a>                  <span class="at">color =</span> experiment_id)) <span class="sc">+</span></span>
<span id="cb94-11"><a href="#cb94-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># modeled experiment by treatment means</span></span>
<span id="cb94-12"><a href="#cb94-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> exp1g_m1_emm2,</span>
<span id="cb94-13"><a href="#cb94-13" aria-hidden="true" tabindex="-1"></a>            <span class="fu">aes</span>(<span class="at">y =</span> group_mean,</span>
<span id="cb94-14"><a href="#cb94-14" aria-hidden="true" tabindex="-1"></a>                <span class="at">color =</span> experiment_id),</span>
<span id="cb94-15"><a href="#cb94-15" aria-hidden="true" tabindex="-1"></a>            <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.4</span>),</span>
<span id="cb94-16"><a href="#cb94-16" aria-hidden="true" tabindex="-1"></a>            <span class="at">alpha =</span> <span class="dv">1</span>,</span>
<span id="cb94-17"><a href="#cb94-17" aria-hidden="true" tabindex="-1"></a>            <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb94-18"><a href="#cb94-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">data =</span> exp1g_m1_emm2,</span>
<span id="cb94-19"><a href="#cb94-19" aria-hidden="true" tabindex="-1"></a>            <span class="fu">aes</span>(<span class="at">y =</span> group_mean,</span>
<span id="cb94-20"><a href="#cb94-20" aria-hidden="true" tabindex="-1"></a>                <span class="at">group =</span> experiment_id,</span>
<span id="cb94-21"><a href="#cb94-21" aria-hidden="true" tabindex="-1"></a>                <span class="at">color =</span> experiment_id),</span>
<span id="cb94-22"><a href="#cb94-22" aria-hidden="true" tabindex="-1"></a>            <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.4</span>)) <span class="sc">+</span></span>
<span id="cb94-23"><a href="#cb94-23" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb94-24"><a href="#cb94-24" aria-hidden="true" tabindex="-1"></a>  <span class="co"># raw data</span></span>
<span id="cb94-25"><a href="#cb94-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.4</span>),</span>
<span id="cb94-26"><a href="#cb94-26" aria-hidden="true" tabindex="-1"></a>             <span class="at">alpha =</span> <span class="fl">0.3</span></span>
<span id="cb94-27"><a href="#cb94-27" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb94-28"><a href="#cb94-28" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb94-29"><a href="#cb94-29" aria-hidden="true" tabindex="-1"></a>  <span class="co"># modeled treatment means</span></span>
<span id="cb94-30"><a href="#cb94-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> exp1g_m1_emm_dt,</span>
<span id="cb94-31"><a href="#cb94-31" aria-hidden="true" tabindex="-1"></a>             <span class="fu">aes</span>(<span class="at">x =</span> treatment,</span>
<span id="cb94-32"><a href="#cb94-32" aria-hidden="true" tabindex="-1"></a>                 <span class="at">y =</span> emmean),</span>
<span id="cb94-33"><a href="#cb94-33" aria-hidden="true" tabindex="-1"></a>             <span class="at">color =</span> <span class="st">"black"</span>,</span>
<span id="cb94-34"><a href="#cb94-34" aria-hidden="true" tabindex="-1"></a>             <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb94-35"><a href="#cb94-35" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_errorbar(data = exp1g_m1_emm_dt,</span></span>
<span id="cb94-36"><a href="#cb94-36" aria-hidden="true" tabindex="-1"></a>  <span class="co">#            aes(y = emmean,</span></span>
<span id="cb94-37"><a href="#cb94-37" aria-hidden="true" tabindex="-1"></a>  <span class="co">#                ymin = lower.CL,</span></span>
<span id="cb94-38"><a href="#cb94-38" aria-hidden="true" tabindex="-1"></a>  <span class="co">#                ymax = upper.CL),</span></span>
<span id="cb94-39"><a href="#cb94-39" aria-hidden="true" tabindex="-1"></a>  <span class="co">#             color = "black",</span></span>
<span id="cb94-40"><a href="#cb94-40" aria-hidden="true" tabindex="-1"></a>  <span class="co">#           width = .05) +</span></span>
<span id="cb94-41"><a href="#cb94-41" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb94-42"><a href="#cb94-42" aria-hidden="true" tabindex="-1"></a>  <span class="co"># pvalue brackets</span></span>
<span id="cb94-43"><a href="#cb94-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_pvalue_manual</span>(exp1g_m1_planned_dt[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>],</span>
<span id="cb94-44"><a href="#cb94-44" aria-hidden="true" tabindex="-1"></a>                     <span class="at">label =</span> <span class="st">"pretty_p"</span>,</span>
<span id="cb94-45"><a href="#cb94-45" aria-hidden="true" tabindex="-1"></a>                     <span class="at">y.position =</span> <span class="fu">c</span>(<span class="dv">20</span>,<span class="dv">20</span>),</span>
<span id="cb94-46"><a href="#cb94-46" aria-hidden="true" tabindex="-1"></a>                     <span class="at">size =</span> <span class="fl">2.5</span>,</span>
<span id="cb94-47"><a href="#cb94-47" aria-hidden="true" tabindex="-1"></a>                     <span class="at">tip.length =</span> <span class="fl">0.01</span>) <span class="sc">+</span></span>
<span id="cb94-48"><a href="#cb94-48" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb94-49"><a href="#cb94-49" aria-hidden="true" tabindex="-1"></a>  <span class="co"># additive line + interaction p bracket</span></span>
<span id="cb94-50"><a href="#cb94-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_segment</span>(<span class="at">x =</span> <span class="fl">3.85</span>,</span>
<span id="cb94-51"><a href="#cb94-51" aria-hidden="true" tabindex="-1"></a>               <span class="at">y =</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>] <span class="sc">+</span> b[<span class="dv">3</span>],</span>
<span id="cb94-52"><a href="#cb94-52" aria-hidden="true" tabindex="-1"></a>               <span class="at">xend =</span> <span class="fl">4.15</span>,</span>
<span id="cb94-53"><a href="#cb94-53" aria-hidden="true" tabindex="-1"></a>               <span class="at">yend =</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>] <span class="sc">+</span> b[<span class="dv">3</span>],</span>
<span id="cb94-54"><a href="#cb94-54" aria-hidden="true" tabindex="-1"></a>               <span class="at">linetype =</span> <span class="st">"dashed"</span>,</span>
<span id="cb94-55"><a href="#cb94-55" aria-hidden="true" tabindex="-1"></a>               <span class="at">color =</span> <span class="st">"gray"</span>) <span class="sc">+</span></span>
<span id="cb94-56"><a href="#cb94-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bracket</span>(</span>
<span id="cb94-57"><a href="#cb94-57" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="fl">4.2</span>,</span>
<span id="cb94-58"><a href="#cb94-58" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>] <span class="sc">+</span> b[<span class="dv">3</span>],</span>
<span id="cb94-59"><a href="#cb94-59" aria-hidden="true" tabindex="-1"></a>    <span class="at">yend =</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>] <span class="sc">+</span> b[<span class="dv">3</span>] <span class="sc">+</span> b[<span class="dv">4</span>],</span>
<span id="cb94-60"><a href="#cb94-60" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="fu">paste0</span>(<span class="st">"interaction</span><span class="sc">\n</span><span class="st">p = "</span>, p_ixn),</span>
<span id="cb94-61"><a href="#cb94-61" aria-hidden="true" tabindex="-1"></a>    <span class="at">text.size =</span> <span class="dv">3</span>,</span>
<span id="cb94-62"><a href="#cb94-62" aria-hidden="true" tabindex="-1"></a>    <span class="at">text.hjust =</span> <span class="dv">0</span>,</span>
<span id="cb94-63"><a href="#cb94-63" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb94-64"><a href="#cb94-64" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb94-65"><a href="#cb94-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"GC (%)"</span>) <span class="sc">+</span></span>
<span id="cb94-66"><a href="#cb94-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> pal_okabe_ito_blue) <span class="sc">+</span></span>
<span id="cb94-67"><a href="#cb94-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_pubr</span>() <span class="sc">+</span></span>
<span id="cb94-68"><a href="#cb94-68" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">xlim =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="fl">4.1</span>)) <span class="sc">+</span></span>
<span id="cb94-69"><a href="#cb94-69" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb94-70"><a href="#cb94-70" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(), <span class="co"># no x-axis title</span></span>
<span id="cb94-71"><a href="#cb94-71" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">"none"</span></span>
<span id="cb94-72"><a href="#cb94-72" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span> </span>
<span id="cb94-73"><a href="#cb94-73" aria-hidden="true" tabindex="-1"></a>  <span class="cn">NULL</span></span>
<span id="cb94-74"><a href="#cb94-74" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb94-75"><a href="#cb94-75" aria-hidden="true" tabindex="-1"></a>exp1g_plot_a <span class="ot">&lt;-</span> <span class="fu">factor_wrap</span>(exp1g_plot_a)</span>
<span id="cb94-76"><a href="#cb94-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb94-77"><a href="#cb94-77" aria-hidden="true" tabindex="-1"></a><span class="co">#exp1g_plot_a</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb95"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb95-1"><a href="#cb95-1" aria-hidden="true" tabindex="-1"></a><span class="co"># get coefficients of model</span></span>
<span id="cb95-2"><a href="#cb95-2" aria-hidden="true" tabindex="-1"></a>b <span class="ot">&lt;-</span> exp1g_m1_coef[, <span class="st">"Estimate"</span>]</span>
<span id="cb95-3"><a href="#cb95-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-4"><a href="#cb95-4" aria-hidden="true" tabindex="-1"></a><span class="co"># get interaction p</span></span>
<span id="cb95-5"><a href="#cb95-5" aria-hidden="true" tabindex="-1"></a>p_ixn <span class="ot">&lt;-</span> exp1g_m1_planned_dt[contrast <span class="sc">==</span> <span class="st">"Interaction"</span>, pretty_p]</span>
<span id="cb95-6"><a href="#cb95-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-7"><a href="#cb95-7" aria-hidden="true" tabindex="-1"></a>dodge_width <span class="ot">=</span> <span class="fl">0.6</span></span>
<span id="cb95-8"><a href="#cb95-8" aria-hidden="true" tabindex="-1"></a>exp1g_plot_b <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> exp1g,</span>
<span id="cb95-9"><a href="#cb95-9" aria-hidden="true" tabindex="-1"></a>              <span class="fu">aes</span>(<span class="at">x =</span> treatment,</span>
<span id="cb95-10"><a href="#cb95-10" aria-hidden="true" tabindex="-1"></a>                  <span class="at">y =</span> gc,</span>
<span id="cb95-11"><a href="#cb95-11" aria-hidden="true" tabindex="-1"></a>                  <span class="at">color =</span> experiment_id)) <span class="sc">+</span></span>
<span id="cb95-12"><a href="#cb95-12" aria-hidden="true" tabindex="-1"></a>  <span class="co"># modeled experiment by treatment means</span></span>
<span id="cb95-13"><a href="#cb95-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> exp1g_m1_emm2,</span>
<span id="cb95-14"><a href="#cb95-14" aria-hidden="true" tabindex="-1"></a>            <span class="fu">aes</span>(<span class="at">y =</span> group_mean,</span>
<span id="cb95-15"><a href="#cb95-15" aria-hidden="true" tabindex="-1"></a>                <span class="at">color =</span> experiment_id),</span>
<span id="cb95-16"><a href="#cb95-16" aria-hidden="true" tabindex="-1"></a>            <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> dodge_width),</span>
<span id="cb95-17"><a href="#cb95-17" aria-hidden="true" tabindex="-1"></a>            <span class="at">alpha =</span> <span class="dv">1</span>,</span>
<span id="cb95-18"><a href="#cb95-18" aria-hidden="true" tabindex="-1"></a>            <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb95-19"><a href="#cb95-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">data =</span> exp1g_m1_emm2,</span>
<span id="cb95-20"><a href="#cb95-20" aria-hidden="true" tabindex="-1"></a>            <span class="fu">aes</span>(<span class="at">y =</span> group_mean,</span>
<span id="cb95-21"><a href="#cb95-21" aria-hidden="true" tabindex="-1"></a>                <span class="at">group =</span> experiment_id,</span>
<span id="cb95-22"><a href="#cb95-22" aria-hidden="true" tabindex="-1"></a>                <span class="at">color =</span> experiment_id),</span>
<span id="cb95-23"><a href="#cb95-23" aria-hidden="true" tabindex="-1"></a>            <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> dodge_width)) <span class="sc">+</span></span>
<span id="cb95-24"><a href="#cb95-24" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb95-25"><a href="#cb95-25" aria-hidden="true" tabindex="-1"></a>  <span class="co"># raw data</span></span>
<span id="cb95-26"><a href="#cb95-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> dodge_width),</span>
<span id="cb95-27"><a href="#cb95-27" aria-hidden="true" tabindex="-1"></a>             <span class="at">color =</span> <span class="st">"gray80"</span></span>
<span id="cb95-28"><a href="#cb95-28" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb95-29"><a href="#cb95-29" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb95-30"><a href="#cb95-30" aria-hidden="true" tabindex="-1"></a>  <span class="co"># modeled treatment means</span></span>
<span id="cb95-31"><a href="#cb95-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> exp1g_m1_emm_dt,</span>
<span id="cb95-32"><a href="#cb95-32" aria-hidden="true" tabindex="-1"></a>             <span class="fu">aes</span>(<span class="at">x =</span> treatment,</span>
<span id="cb95-33"><a href="#cb95-33" aria-hidden="true" tabindex="-1"></a>                 <span class="at">y =</span> emmean),</span>
<span id="cb95-34"><a href="#cb95-34" aria-hidden="true" tabindex="-1"></a>             <span class="at">color =</span> <span class="st">"black"</span>,</span>
<span id="cb95-35"><a href="#cb95-35" aria-hidden="true" tabindex="-1"></a>             <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb95-36"><a href="#cb95-36" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_errorbar(data = exp1g_m1_emm_dt,</span></span>
<span id="cb95-37"><a href="#cb95-37" aria-hidden="true" tabindex="-1"></a>  <span class="co">#            aes(y = emmean,</span></span>
<span id="cb95-38"><a href="#cb95-38" aria-hidden="true" tabindex="-1"></a>  <span class="co">#                ymin = lower.CL,</span></span>
<span id="cb95-39"><a href="#cb95-39" aria-hidden="true" tabindex="-1"></a>  <span class="co">#                ymax = upper.CL),</span></span>
<span id="cb95-40"><a href="#cb95-40" aria-hidden="true" tabindex="-1"></a>  <span class="co">#             color = "black",</span></span>
<span id="cb95-41"><a href="#cb95-41" aria-hidden="true" tabindex="-1"></a>  <span class="co">#           width = .05) +</span></span>
<span id="cb95-42"><a href="#cb95-42" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb95-43"><a href="#cb95-43" aria-hidden="true" tabindex="-1"></a>  <span class="co"># pvalue brackets</span></span>
<span id="cb95-44"><a href="#cb95-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_pvalue_manual</span>(exp1g_m1_planned_dt[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>],</span>
<span id="cb95-45"><a href="#cb95-45" aria-hidden="true" tabindex="-1"></a>                     <span class="at">label =</span> <span class="st">"pretty_p"</span>,</span>
<span id="cb95-46"><a href="#cb95-46" aria-hidden="true" tabindex="-1"></a>                     <span class="at">y.position =</span> <span class="fu">c</span>(<span class="dv">20</span>,<span class="dv">20</span>),</span>
<span id="cb95-47"><a href="#cb95-47" aria-hidden="true" tabindex="-1"></a>                     <span class="at">size =</span> <span class="fl">2.5</span>,</span>
<span id="cb95-48"><a href="#cb95-48" aria-hidden="true" tabindex="-1"></a>                     <span class="at">tip.length =</span> <span class="fl">0.01</span>) <span class="sc">+</span></span>
<span id="cb95-49"><a href="#cb95-49" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb95-50"><a href="#cb95-50" aria-hidden="true" tabindex="-1"></a>  <span class="co"># additive line + interaction p bracket</span></span>
<span id="cb95-51"><a href="#cb95-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_segment</span>(<span class="at">x =</span> <span class="dv">4</span> <span class="sc">-</span> dodge_width<span class="sc">/</span><span class="dv">2</span>,</span>
<span id="cb95-52"><a href="#cb95-52" aria-hidden="true" tabindex="-1"></a>               <span class="at">y =</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>] <span class="sc">+</span> b[<span class="dv">3</span>],</span>
<span id="cb95-53"><a href="#cb95-53" aria-hidden="true" tabindex="-1"></a>               <span class="at">xend =</span> <span class="dv">4</span> <span class="sc">+</span> dodge_width<span class="sc">/</span><span class="dv">2</span>,</span>
<span id="cb95-54"><a href="#cb95-54" aria-hidden="true" tabindex="-1"></a>               <span class="at">yend =</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>] <span class="sc">+</span> b[<span class="dv">3</span>],</span>
<span id="cb95-55"><a href="#cb95-55" aria-hidden="true" tabindex="-1"></a>               <span class="at">linetype =</span> <span class="st">"dashed"</span>,</span>
<span id="cb95-56"><a href="#cb95-56" aria-hidden="true" tabindex="-1"></a>               <span class="at">color =</span> <span class="st">"gray"</span>) <span class="sc">+</span></span>
<span id="cb95-57"><a href="#cb95-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bracket</span>(</span>
<span id="cb95-58"><a href="#cb95-58" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="dv">4</span> <span class="sc">+</span> dodge_width<span class="sc">/</span><span class="fl">1.9</span>,</span>
<span id="cb95-59"><a href="#cb95-59" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>] <span class="sc">+</span> b[<span class="dv">3</span>],</span>
<span id="cb95-60"><a href="#cb95-60" aria-hidden="true" tabindex="-1"></a>    <span class="at">yend =</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>] <span class="sc">+</span> b[<span class="dv">3</span>] <span class="sc">+</span> b[<span class="dv">4</span>],</span>
<span id="cb95-61"><a href="#cb95-61" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="fu">paste0</span>(<span class="st">"interaction</span><span class="sc">\n</span><span class="st">p = "</span>, p_ixn),</span>
<span id="cb95-62"><a href="#cb95-62" aria-hidden="true" tabindex="-1"></a>    <span class="at">text.size =</span> <span class="dv">3</span>,</span>
<span id="cb95-63"><a href="#cb95-63" aria-hidden="true" tabindex="-1"></a>    <span class="at">text.hjust =</span> <span class="dv">0</span>,</span>
<span id="cb95-64"><a href="#cb95-64" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb95-65"><a href="#cb95-65" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb95-66"><a href="#cb95-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"GC (%)"</span>) <span class="sc">+</span></span>
<span id="cb95-67"><a href="#cb95-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> pal_okabe_ito_blue) <span class="sc">+</span></span>
<span id="cb95-68"><a href="#cb95-68" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_pubr</span>() <span class="sc">+</span></span>
<span id="cb95-69"><a href="#cb95-69" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">xlim =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="fl">4.2</span>)) <span class="sc">+</span></span>
<span id="cb95-70"><a href="#cb95-70" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb95-71"><a href="#cb95-71" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(), <span class="co"># no x-axis title</span></span>
<span id="cb95-72"><a href="#cb95-72" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">"none"</span></span>
<span id="cb95-73"><a href="#cb95-73" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span> </span>
<span id="cb95-74"><a href="#cb95-74" aria-hidden="true" tabindex="-1"></a>  <span class="cn">NULL</span></span>
<span id="cb95-75"><a href="#cb95-75" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb95-76"><a href="#cb95-76" aria-hidden="true" tabindex="-1"></a>exp1g_plot_b <span class="ot">&lt;-</span> <span class="fu">factor_wrap</span>(exp1g_plot_b)</span>
<span id="cb95-77"><a href="#cb95-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-78"><a href="#cb95-78" aria-hidden="true" tabindex="-1"></a><span class="co"># exp1g_plot_b</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="lmm_files/figure-html/unnamed-chunk-53-1.png" class="img-fluid figure-img" width="1066"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="repeated-measures-anova-randomized-complete-block-with-no-subsampling" class="level3" data-number="14.13.2">
<h3 data-number="14.13.2" class="anchored" data-anchor-id="repeated-measures-anova-randomized-complete-block-with-no-subsampling"><span class="header-section-number">14.13.2</span> Repeated measures ANOVA (randomized complete block with no subsampling)</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb96"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb96-1"><a href="#cb96-1" aria-hidden="true" tabindex="-1"></a><span class="co"># this is the rm-ANOVA</span></span>
<span id="cb96-2"><a href="#cb96-2" aria-hidden="true" tabindex="-1"></a>m1 <span class="ot">&lt;-</span> <span class="fu">aov_4</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb96-3"><a href="#cb96-3" aria-hidden="true" tabindex="-1"></a>              (treatment <span class="sc">*</span> activity <span class="sc">|</span> donor),</span>
<span id="cb96-4"><a href="#cb96-4" aria-hidden="true" tabindex="-1"></a>            <span class="at">data =</span> exp5c)</span>
<span id="cb96-5"><a href="#cb96-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-6"><a href="#cb96-6" aria-hidden="true" tabindex="-1"></a><span class="co"># lmm equivalent</span></span>
<span id="cb96-7"><a href="#cb96-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-8"><a href="#cb96-8" aria-hidden="true" tabindex="-1"></a>m2 <span class="ot">&lt;-</span> <span class="fu">lmer</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb96-9"><a href="#cb96-9" aria-hidden="true" tabindex="-1"></a>         (<span class="dv">1</span> <span class="sc">|</span> donor) <span class="sc">+</span></span>
<span id="cb96-10"><a href="#cb96-10" aria-hidden="true" tabindex="-1"></a>         (<span class="dv">1</span> <span class="sc">|</span> donor<span class="sc">:</span>treatment) <span class="sc">+</span></span>
<span id="cb96-11"><a href="#cb96-11" aria-hidden="true" tabindex="-1"></a>         (<span class="dv">1</span> <span class="sc">|</span> donor<span class="sc">:</span>activity),</span>
<span id="cb96-12"><a href="#cb96-12" aria-hidden="true" tabindex="-1"></a>       <span class="at">data =</span> exp5c)</span>
<span id="cb96-13"><a href="#cb96-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-14"><a href="#cb96-14" aria-hidden="true" tabindex="-1"></a><span class="co"># random intercept and slope model that is *not* equivalent</span></span>
<span id="cb96-15"><a href="#cb96-15" aria-hidden="true" tabindex="-1"></a><span class="co"># this isn't solvable because there is no subsampling</span></span>
<span id="cb96-16"><a href="#cb96-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-17"><a href="#cb96-17" aria-hidden="true" tabindex="-1"></a>m3 <span class="ot">&lt;-</span> <span class="fu">lmer</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb96-18"><a href="#cb96-18" aria-hidden="true" tabindex="-1"></a>              (treatment <span class="sc">*</span> activity <span class="sc">|</span> donor),</span>
<span id="cb96-19"><a href="#cb96-19" aria-hidden="true" tabindex="-1"></a>            <span class="at">data =</span> exp5c)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Notes</p>
<ol type="1">
<li><code>afex</code> computes the repeated measures anova model using both <code>aov</code> (classical univariate repeated measures ANOVA) and using <code>lm</code> with multiple response variables (the multivariate repeated measures ANOVA). As of this writing, the output from <code>aov</code> is the default.</li>
<li>Given only one measure for each donor within a <span class="math inline">\(\texttt{treatment} \times \texttt{activity}\)</span> combination, the linear mixed model <code>m2</code> is equivalent to the univariate repeated measures model m1.</li>
<li>The model formula in <code>m1</code> looks like that in the linear mixed model <code>m3</code> but the two models are not equivalent.</li>
</ol>
<section id="univariate-vs.-multivariate-repeated-measures-anova" class="level4" data-number="14.13.2.1">
<h4 data-number="14.13.2.1" class="anchored" data-anchor-id="univariate-vs.-multivariate-repeated-measures-anova"><span class="header-section-number">14.13.2.1</span> univariate vs.&nbsp;multivariate repeated measures ANOVA</h4>
<p>Use the multivariate model unless you want to replicate the result of someone who used a univariate model. By default, <code>aov_4</code> computes only the multivariate model (prior to version xxx, the default was to compute both models).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb97"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1" aria-hidden="true" tabindex="-1"></a><span class="co"># default mode -- should be multivariate</span></span>
<span id="cb97-2"><a href="#cb97-2" aria-hidden="true" tabindex="-1"></a>exp5c_aov <span class="ot">&lt;-</span> <span class="fu">aov_4</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb97-3"><a href="#cb97-3" aria-hidden="true" tabindex="-1"></a>                    (treatment <span class="sc">*</span> activity <span class="sc">|</span> donor),</span>
<span id="cb97-4"><a href="#cb97-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> exp5c)</span>
<span id="cb97-5"><a href="#cb97-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-6"><a href="#cb97-6" aria-hidden="true" tabindex="-1"></a><span class="co"># force aov_4 to compute univariate model</span></span>
<span id="cb97-7"><a href="#cb97-7" aria-hidden="true" tabindex="-1"></a>exp5c_aov2 <span class="ot">&lt;-</span> <span class="fu">aov_4</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb97-8"><a href="#cb97-8" aria-hidden="true" tabindex="-1"></a>                    (treatment <span class="sc">*</span> activity <span class="sc">|</span> donor),</span>
<span id="cb97-9"><a href="#cb97-9" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> exp5c,</span>
<span id="cb97-10"><a href="#cb97-10" aria-hidden="true" tabindex="-1"></a>                  <span class="at">include_aov =</span> <span class="cn">TRUE</span>)</span>
<span id="cb97-11"><a href="#cb97-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-12"><a href="#cb97-12" aria-hidden="true" tabindex="-1"></a><span class="co"># explicitly exclude the univariate model</span></span>
<span id="cb97-13"><a href="#cb97-13" aria-hidden="true" tabindex="-1"></a>exp5c_aov1 <span class="ot">&lt;-</span> <span class="fu">aov_4</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb97-14"><a href="#cb97-14" aria-hidden="true" tabindex="-1"></a>                    (treatment <span class="sc">*</span> activity <span class="sc">|</span> donor),</span>
<span id="cb97-15"><a href="#cb97-15" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> exp5c,</span>
<span id="cb97-16"><a href="#cb97-16" aria-hidden="true" tabindex="-1"></a>                  <span class="at">include_aov =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Notes</p>
<ol type="1">
<li>The <code>include_aov = TRUE</code> argument forces output from <code>aov_4</code> to include the univariate model.</li>
</ol>
<p><strong>contrasts from multivariate model</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb98"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb98-1"><a href="#cb98-1" aria-hidden="true" tabindex="-1"></a><span class="co"># These three should give equivalent results</span></span>
<span id="cb98-2"><a href="#cb98-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb98-3"><a href="#cb98-3" aria-hidden="true" tabindex="-1"></a><span class="co"># exp5c_aov was fit with the default -- multivariate model only</span></span>
<span id="cb98-4"><a href="#cb98-4" aria-hidden="true" tabindex="-1"></a><span class="fu">emmeans</span>(exp5c_aov,</span>
<span id="cb98-5"><a href="#cb98-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">"treatment"</span>, <span class="st">"activity"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb98-6"><a href="#cb98-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">contrast</span>(<span class="at">method =</span> <span class="st">"revpairwise"</span>,</span>
<span id="cb98-7"><a href="#cb98-7" aria-hidden="true" tabindex="-1"></a>           <span class="at">simple =</span> <span class="st">"each"</span>,</span>
<span id="cb98-8"><a href="#cb98-8" aria-hidden="true" tabindex="-1"></a>           <span class="at">combine =</span> <span class="cn">TRUE</span>,</span>
<span id="cb98-9"><a href="#cb98-9" aria-hidden="true" tabindex="-1"></a>           <span class="at">adjust =</span> <span class="st">"none"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> activity treatment contrast      estimate     SE df t.ratio p.value
 Basal    .         siNR4A3 - Scr   0.0813 0.1012  5   0.804  0.4581
 EPS      .         siNR4A3 - Scr  -0.1858 0.0359  5  -5.180  0.0035
 .        Scr       EPS - Basal     0.1510 0.0395  5   3.821  0.0124
 .        siNR4A3   EPS - Basal    -0.1161 0.0626  5  -1.855  0.1228</code></pre>
</div>
<div class="sourceCode cell-code" id="cb100"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb100-1"><a href="#cb100-1" aria-hidden="true" tabindex="-1"></a><span class="co"># exp5c_aov included the univariate model but the default is still the multivariate model</span></span>
<span id="cb100-2"><a href="#cb100-2" aria-hidden="true" tabindex="-1"></a><span class="fu">emmeans</span>(exp5c_aov2,</span>
<span id="cb100-3"><a href="#cb100-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">"treatment"</span>, <span class="st">"activity"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb100-4"><a href="#cb100-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">contrast</span>(<span class="at">method =</span> <span class="st">"revpairwise"</span>,</span>
<span id="cb100-5"><a href="#cb100-5" aria-hidden="true" tabindex="-1"></a>           <span class="at">simple =</span> <span class="st">"each"</span>,</span>
<span id="cb100-6"><a href="#cb100-6" aria-hidden="true" tabindex="-1"></a>           <span class="at">combine =</span> <span class="cn">TRUE</span>,</span>
<span id="cb100-7"><a href="#cb100-7" aria-hidden="true" tabindex="-1"></a>           <span class="at">adjust =</span> <span class="st">"none"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> activity treatment contrast      estimate     SE df t.ratio p.value
 Basal    .         siNR4A3 - Scr   0.0813 0.1012  5   0.804  0.4581
 EPS      .         siNR4A3 - Scr  -0.1858 0.0359  5  -5.180  0.0035
 .        Scr       EPS - Basal     0.1510 0.0395  5   3.821  0.0124
 .        siNR4A3   EPS - Basal    -0.1161 0.0626  5  -1.855  0.1228</code></pre>
</div>
<div class="sourceCode cell-code" id="cb102"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb102-1"><a href="#cb102-1" aria-hidden="true" tabindex="-1"></a><span class="co"># passing `model = "multivariate"` makes the model choice transparent</span></span>
<span id="cb102-2"><a href="#cb102-2" aria-hidden="true" tabindex="-1"></a><span class="fu">emmeans</span>(exp5c_aov2,</span>
<span id="cb102-3"><a href="#cb102-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">"treatment"</span>, <span class="st">"activity"</span>),</span>
<span id="cb102-4"><a href="#cb102-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">model =</span> <span class="st">"multivariate"</span>) <span class="sc">|&gt;</span></span>
<span id="cb102-5"><a href="#cb102-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">contrast</span>(<span class="at">method =</span> <span class="st">"revpairwise"</span>,</span>
<span id="cb102-6"><a href="#cb102-6" aria-hidden="true" tabindex="-1"></a>           <span class="at">simple =</span> <span class="st">"each"</span>,</span>
<span id="cb102-7"><a href="#cb102-7" aria-hidden="true" tabindex="-1"></a>           <span class="at">combine =</span> <span class="cn">TRUE</span>,</span>
<span id="cb102-8"><a href="#cb102-8" aria-hidden="true" tabindex="-1"></a>           <span class="at">adjust =</span> <span class="st">"none"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> activity treatment contrast      estimate     SE df t.ratio p.value
 Basal    .         siNR4A3 - Scr   0.0813 0.1012  5   0.804  0.4581
 EPS      .         siNR4A3 - Scr  -0.1858 0.0359  5  -5.180  0.0035
 .        Scr       EPS - Basal     0.1510 0.0395  5   3.821  0.0124
 .        siNR4A3   EPS - Basal    -0.1161 0.0626  5  -1.855  0.1228</code></pre>
</div>
</div>
<p>Notes</p>
<ol type="1">
<li>If our rmANOVA model is fit with the default specification (<code>exp5c_aov2</code>), we can get the multivariate output using the <code>model = "multivariate"</code> argument within <code>emmeans</code> (not the <code>contrast</code> function!). Or, if our fit excluded the univariate model (<code>exp5c_aov1</code>), then we don’t need the <code>model</code> argument.</li>
</ol>
<p><strong>contrasts from univariate model</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb104"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb104-1"><a href="#cb104-1" aria-hidden="true" tabindex="-1"></a><span class="co"># get the univariate model results using $aov</span></span>
<span id="cb104-2"><a href="#cb104-2" aria-hidden="true" tabindex="-1"></a><span class="fu">emmeans</span>(exp5c_aov2<span class="sc">$</span>aov, <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">"treatment"</span>, <span class="st">"activity"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb104-3"><a href="#cb104-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">contrast</span>(<span class="at">method =</span> <span class="st">"revpairwise"</span>,</span>
<span id="cb104-4"><a href="#cb104-4" aria-hidden="true" tabindex="-1"></a>           <span class="at">simple =</span> <span class="st">"each"</span>,</span>
<span id="cb104-5"><a href="#cb104-5" aria-hidden="true" tabindex="-1"></a>           <span class="at">combine =</span> <span class="cn">TRUE</span>,</span>
<span id="cb104-6"><a href="#cb104-6" aria-hidden="true" tabindex="-1"></a>           <span class="at">adjust =</span> <span class="st">"none"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> activity treatment contrast      estimate     SE   df t.ratio p.value
 Basal    .         siNR4A3 - Scr   0.0813 0.0759 8.60   1.071  0.3132
 EPS      .         siNR4A3 - Scr  -0.1858 0.0759 8.60  -2.448  0.0380
 .        Scr       EPS - Basal     0.1510 0.0524 9.39   2.884  0.0173
 .        siNR4A3   EPS - Basal    -0.1161 0.0524 9.39  -2.218  0.0525</code></pre>
</div>
</div>
<p>Notes</p>
<ol type="1">
<li>Passing <code>exp5c_aov2</code> to <code>emmeans</code> will return the contrasts from the multivariate. change this to the univariate model by passing <code>exp5c_aov2$aov</code>.</li>
</ol>
</section>
<section id="the-anova-table" class="level4" data-number="14.13.2.2">
<h4 data-number="14.13.2.2" class="anchored" data-anchor-id="the-anova-table"><span class="header-section-number">14.13.2.2</span> The ANOVA table</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb106"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb106-1"><a href="#cb106-1" aria-hidden="true" tabindex="-1"></a><span class="fu">nice</span>(exp5c_aov2, <span class="at">correction =</span> <span class="st">"GG"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Anova Table (Type 3 tests)

Response: glucose_uptake
              Effect   df  MSE       F   ges p.value
1          treatment 1, 5 0.02    0.68  .006    .448
2           activity 1, 5 0.01    0.30 &lt;.001    .609
3 treatment:activity 1, 5 0.01 10.37 *  .037    .023
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '+' 0.1 ' ' 1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb108"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb108-1"><a href="#cb108-1" aria-hidden="true" tabindex="-1"></a><span class="fu">nice</span>(exp5c_aov2, <span class="at">correction =</span> <span class="st">"none"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Anova Table (Type 3 tests)

Response: glucose_uptake
              Effect   df  MSE       F   ges p.value
1          treatment 1, 5 0.02    0.68  .006    .448
2           activity 1, 5 0.01    0.30 &lt;.001    .609
3 treatment:activity 1, 5 0.01 10.37 *  .037    .023
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '+' 0.1 ' ' 1</code></pre>
</div>
</div>
<p>Notes</p>
<ol type="1">
<li>The Greenhouse-Geiger (“GG”) correction is the default. While the <code>correction = "GG"</code> argument is not needed, it makes the script more transparent.</li>
<li>For <em>these data</em> the Greenhouse-Geiger correction doesn’t make a difference in the table.</li>
</ol>
</section>
</section>
</section>
<section id="hidden-code" class="level2" data-number="14.14">
<h2 data-number="14.14" class="anchored" data-anchor-id="hidden-code"><span class="header-section-number">14.14</span> Hidden code</h2>
<section id="import-exp5c" class="level3" data-number="14.14.1">
<h3 data-number="14.14.1" class="anchored" data-anchor-id="import-exp5c"><span class="header-section-number">14.14.1</span> Import exp5c</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb110"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb110-1"><a href="#cb110-1" aria-hidden="true" tabindex="-1"></a>data_from <span class="ot">&lt;-</span> <span class="st">"Transcriptomic profiling of skeletal muscle adaptations to exercise and inactivity"</span></span>
<span id="cb110-2"><a href="#cb110-2" aria-hidden="true" tabindex="-1"></a>file_name <span class="ot">&lt;-</span> <span class="st">"41467_2019_13869_MOESM6_ESM.xlsx"</span></span>
<span id="cb110-3"><a href="#cb110-3" aria-hidden="true" tabindex="-1"></a>file_path <span class="ot">&lt;-</span> <span class="fu">here</span>(data_folder, data_from, file_name)</span>
<span id="cb110-4"><a href="#cb110-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb110-5"><a href="#cb110-5" aria-hidden="true" tabindex="-1"></a>exp5c_wide <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(file_path,</span>
<span id="cb110-6"><a href="#cb110-6" aria-hidden="true" tabindex="-1"></a>                         <span class="at">sheet =</span> <span class="st">"Fig5c"</span>,</span>
<span id="cb110-7"><a href="#cb110-7" aria-hidden="true" tabindex="-1"></a>                         <span class="at">range =</span> <span class="st">"A2:M3"</span>,</span>
<span id="cb110-8"><a href="#cb110-8" aria-hidden="true" tabindex="-1"></a>                         <span class="at">col_names =</span> <span class="cn">FALSE</span>) <span class="sc">|&gt;</span></span>
<span id="cb110-9"><a href="#cb110-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.table</span>() <span class="sc">|&gt;</span></span>
<span id="cb110-10"><a href="#cb110-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">transpose</span>(<span class="at">make.names =</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>New names:
• `` -&gt; `...1`
• `` -&gt; `...2`
• `` -&gt; `...3`
• `` -&gt; `...4`
• `` -&gt; `...5`
• `` -&gt; `...6`
• `` -&gt; `...7`
• `` -&gt; `...8`
• `` -&gt; `...9`
• `` -&gt; `...10`
• `` -&gt; `...11`
• `` -&gt; `...12`
• `` -&gt; `...13`</code></pre>
</div>
<div class="sourceCode cell-code" id="cb112"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb112-1"><a href="#cb112-1" aria-hidden="true" tabindex="-1"></a>activity_levels <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Basal"</span>, <span class="st">"EPS"</span>)</span>
<span id="cb112-2"><a href="#cb112-2" aria-hidden="true" tabindex="-1"></a>treatment_levels <span class="ot">&lt;-</span> <span class="fu">names</span>(exp5c_wide)</span>
<span id="cb112-3"><a href="#cb112-3" aria-hidden="true" tabindex="-1"></a>exp5c_wide[, activity <span class="sc">:</span><span class="er">=</span> <span class="fu">rep</span>(activity_levels, <span class="at">each =</span> <span class="dv">6</span>)]</span>
<span id="cb112-4"><a href="#cb112-4" aria-hidden="true" tabindex="-1"></a>exp5c_wide[, activity <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(activity, <span class="at">levels =</span> activity_levels)]</span>
<span id="cb112-5"><a href="#cb112-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb112-6"><a href="#cb112-6" aria-hidden="true" tabindex="-1"></a>exp5c <span class="ot">&lt;-</span> <span class="fu">melt</span>(exp5c_wide,</span>
<span id="cb112-7"><a href="#cb112-7" aria-hidden="true" tabindex="-1"></a>              <span class="at">id.vars =</span> <span class="st">"activity"</span>,</span>
<span id="cb112-8"><a href="#cb112-8" aria-hidden="true" tabindex="-1"></a>              <span class="at">variable.name =</span> <span class="st">"treatment"</span>,</span>
<span id="cb112-9"><a href="#cb112-9" aria-hidden="true" tabindex="-1"></a>              <span class="at">value.name =</span> <span class="st">"glucose_uptake"</span>)</span>
<span id="cb112-10"><a href="#cb112-10" aria-hidden="true" tabindex="-1"></a>exp5c[, treatment <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(treatment, <span class="at">levels =</span> treatment_levels)]</span>
<span id="cb112-11"><a href="#cb112-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb112-12"><a href="#cb112-12" aria-hidden="true" tabindex="-1"></a>exp5c[, donor <span class="sc">:</span><span class="er">=</span> <span class="fu">rep</span>(<span class="fu">paste0</span>(<span class="st">"donor_"</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>), <span class="dv">4</span>)]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="import-exp1g" class="level3" data-number="14.14.2">
<h3 data-number="14.14.2" class="anchored" data-anchor-id="import-exp1g"><span class="header-section-number">14.14.2</span> Import exp1g</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb113"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb113-1"><a href="#cb113-1" aria-hidden="true" tabindex="-1"></a>data_from <span class="ot">&lt;-</span> <span class="st">"A GPR174–CCL21 module imparts sexual dimorphism to humoral immunity"</span></span>
<span id="cb113-2"><a href="#cb113-2" aria-hidden="true" tabindex="-1"></a>file_name <span class="ot">&lt;-</span> <span class="st">"41586_2019_1873_MOESM3_ESM.xlsx"</span></span>
<span id="cb113-3"><a href="#cb113-3" aria-hidden="true" tabindex="-1"></a>file_path <span class="ot">&lt;-</span> <span class="fu">here</span>(data_folder, data_from, file_name)</span>
<span id="cb113-4"><a href="#cb113-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-5"><a href="#cb113-5" aria-hidden="true" tabindex="-1"></a>exp1g_wide <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(file_path,</span>
<span id="cb113-6"><a href="#cb113-6" aria-hidden="true" tabindex="-1"></a>                         <span class="at">sheet =</span> <span class="st">"Fig 1g"</span>,</span>
<span id="cb113-7"><a href="#cb113-7" aria-hidden="true" tabindex="-1"></a>                         <span class="at">range =</span> <span class="st">"B4:E25"</span>,</span>
<span id="cb113-8"><a href="#cb113-8" aria-hidden="true" tabindex="-1"></a>                         <span class="at">col_types =</span> <span class="fu">c</span>(<span class="st">"numeric"</span>),</span>
<span id="cb113-9"><a href="#cb113-9" aria-hidden="true" tabindex="-1"></a>                         <span class="at">col_names =</span> <span class="cn">FALSE</span>) <span class="sc">|&gt;</span></span>
<span id="cb113-10"><a href="#cb113-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.table</span>()</span>
<span id="cb113-11"><a href="#cb113-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-12"><a href="#cb113-12" aria-hidden="true" tabindex="-1"></a>genotype_levels <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Gpr174+"</span>, <span class="st">"Gpr174-"</span>)</span>
<span id="cb113-13"><a href="#cb113-13" aria-hidden="true" tabindex="-1"></a>sex_levels <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"M"</span>, <span class="st">"F"</span>)</span>
<span id="cb113-14"><a href="#cb113-14" aria-hidden="true" tabindex="-1"></a>g.by.s_levels <span class="ot">&lt;-</span> <span class="fu">do.call</span>(paste, <span class="fu">expand.grid</span>(genotype_levels, sex_levels))</span>
<span id="cb113-15"><a href="#cb113-15" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(exp1g_wide) <span class="ot">&lt;-</span> g.by.s_levels</span>
<span id="cb113-16"><a href="#cb113-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-17"><a href="#cb113-17" aria-hidden="true" tabindex="-1"></a>exp_levels <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"exp_"</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>)</span>
<span id="cb113-18"><a href="#cb113-18" aria-hidden="true" tabindex="-1"></a>exp1g_wide[, experiment_id <span class="sc">:</span><span class="er">=</span> <span class="fu">rep</span>(exp_levels, <span class="fu">c</span>(<span class="dv">5</span>,<span class="dv">6</span>,<span class="dv">6</span>,<span class="dv">5</span>))] <span class="co">#check!</span></span>
<span id="cb113-19"><a href="#cb113-19" aria-hidden="true" tabindex="-1"></a>exp1g_wide[, experiment_id <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(experiment_id)] <span class="co">#check!</span></span>
<span id="cb113-20"><a href="#cb113-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-21"><a href="#cb113-21" aria-hidden="true" tabindex="-1"></a>exp1g <span class="ot">&lt;-</span> <span class="fu">melt</span>(exp1g_wide,</span>
<span id="cb113-22"><a href="#cb113-22" aria-hidden="true" tabindex="-1"></a>              <span class="at">id.vars =</span> <span class="st">"experiment_id"</span>,</span>
<span id="cb113-23"><a href="#cb113-23" aria-hidden="true" tabindex="-1"></a>              <span class="at">measure.vars =</span> g.by.s_levels,</span>
<span id="cb113-24"><a href="#cb113-24" aria-hidden="true" tabindex="-1"></a>              <span class="at">variable.name =</span> <span class="st">"treatment"</span>,</span>
<span id="cb113-25"><a href="#cb113-25" aria-hidden="true" tabindex="-1"></a>              <span class="at">value.name =</span> <span class="st">"gc"</span>) <span class="sc">|&gt;</span> <span class="co"># cell count</span></span>
<span id="cb113-26"><a href="#cb113-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">na.omit</span>()</span>
<span id="cb113-27"><a href="#cb113-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-28"><a href="#cb113-28" aria-hidden="true" tabindex="-1"></a>exp1g[, <span class="fu">c</span>(<span class="st">"genotype"</span>, <span class="st">"sex"</span>)<span class="sc">:</span><span class="er">=</span> <span class="fu">tstrsplit</span>(treatment,</span>
<span id="cb113-29"><a href="#cb113-29" aria-hidden="true" tabindex="-1"></a>                                             <span class="st">" "</span>,</span>
<span id="cb113-30"><a href="#cb113-30" aria-hidden="true" tabindex="-1"></a>                                             <span class="at">fixed =</span> <span class="cn">TRUE</span>)]</span>
<span id="cb113-31"><a href="#cb113-31" aria-hidden="true" tabindex="-1"></a>exp1g[, genotype <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(genotype,</span>
<span id="cb113-32"><a href="#cb113-32" aria-hidden="true" tabindex="-1"></a>                           <span class="at">levels =</span> genotype_levels)]</span>
<span id="cb113-33"><a href="#cb113-33" aria-hidden="true" tabindex="-1"></a>exp1g[, sex <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(sex,</span>
<span id="cb113-34"><a href="#cb113-34" aria-hidden="true" tabindex="-1"></a>                           <span class="at">levels =</span> sex_levels)]</span>
<span id="cb113-35"><a href="#cb113-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-36"><a href="#cb113-36" aria-hidden="true" tabindex="-1"></a>exp1g_means <span class="ot">&lt;-</span> exp1g[, .(<span class="at">gc =</span> <span class="fu">mean</span>(gc)),</span>
<span id="cb113-37"><a href="#cb113-37" aria-hidden="true" tabindex="-1"></a>                     by <span class="ot">=</span> .(treatment, genotype, sex, experiment_id)]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="../chapters/two-factors.html" class="pagination-link" aria-label="Linear models with two categorical $X$ -- Factorial linear models (&quot;two-way ANOVA&quot;)">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">13</span>&nbsp; <span class="chapter-title">Linear models with two categorical <span class="math inline">\(X\)</span> – Factorial linear models (“two-way ANOVA”)</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
  </div>
</nav>
</div> <!-- /content -->




</body></html>