<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.52">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>14&nbsp; Models for non-independence – linear mixed models – Statistics for the Experimental Bench Biologist</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../chapters/two-factors.html" rel="prev">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script src="../site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="../site_libs/lightable-0.0.1/lightable.css" rel="stylesheet">

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../chapters/lmm.html"><span class="chapter-number">14</span>&nbsp; <span class="chapter-title">Models for non-independence – linear mixed models</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../">Statistics for the Experimental Bench Biologist</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Preface</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../intro.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Analyzing experimental data with a linear model</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/setup.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Getting Started – R Projects and R Markdown</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/reading-writing-wrangling.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Data – Reading, Wrangling, and Writing</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/plotting-models.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Plotting models</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/uncertainty-sd-and-se.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Uncertainty – Standard Deviations, Standard Errors, and Confidence Intervals</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/p-values.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">P-values</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/inference-errors.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Errors in inference</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/intro-stat-modeling.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">An introduction to linear models</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/continuous_x.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">Linear models with a single, continuous <em>X</em> (“linear regression”)</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/one-factor.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">Linear models with a single, categorical <em>X</em> (“t-tests” and “ANOVA”)</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/model-checking.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">11</span>&nbsp; <span class="chapter-title">Model checking</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/multiple-testing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">12</span>&nbsp; <span class="chapter-title">Multiple tests – why and when to adjust <em>p</em>-values</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/two-factors.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">13</span>&nbsp; <span class="chapter-title">Linear models with two categorical <span class="math inline">\(X\)</span> – Factorial linear models (“two-way ANOVA”)</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/lmm.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">14</span>&nbsp; <span class="chapter-title">Models for non-independence – linear mixed models</span></span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#liberal-inference-from-pseudoreplication" id="toc-liberal-inference-from-pseudoreplication" class="nav-link active" data-scroll-target="#liberal-inference-from-pseudoreplication"><span class="header-section-number">14.1</span> Liberal inference from pseudoreplication</a></li>
  <li><a href="#conservative-inference-from-failure-to-identify-blocks" id="toc-conservative-inference-from-failure-to-identify-blocks" class="nav-link" data-scroll-target="#conservative-inference-from-failure-to-identify-blocks"><span class="header-section-number">14.2</span> Conservative inference from failure to identify blocks</a></li>
  <li><a href="#introduction-to-models-for-non-independent-data-linear-mixed-models" id="toc-introduction-to-models-for-non-independent-data-linear-mixed-models" class="nav-link" data-scroll-target="#introduction-to-models-for-non-independent-data-linear-mixed-models"><span class="header-section-number">14.3</span> Introduction to models for non-independent data (linear mixed models)</a></li>
  <li><a href="#experimental-designs-in-experimental-bench-biology" id="toc-experimental-designs-in-experimental-bench-biology" class="nav-link" data-scroll-target="#experimental-designs-in-experimental-bench-biology"><span class="header-section-number">14.4</span> Experimental designs in experimental bench biology</a>
  <ul>
  <li><a href="#notation-for-models" id="toc-notation-for-models" class="nav-link" data-scroll-target="#notation-for-models"><span class="header-section-number">14.4.1</span> Notation for models</a></li>
  <li><a href="#completely-randomized-design-crd" id="toc-completely-randomized-design-crd" class="nav-link" data-scroll-target="#completely-randomized-design-crd"><span class="header-section-number">14.4.2</span> Completely Randomized Design (CRD)</a></li>
  <li><a href="#completely-randomized-design-with-subsampling-crds-or-nested-design" id="toc-completely-randomized-design-with-subsampling-crds-or-nested-design" class="nav-link" data-scroll-target="#completely-randomized-design-with-subsampling-crds-or-nested-design"><span class="header-section-number">14.4.3</span> Completely Randomized Design with Subsampling (CRDS) or “Nested Design”</a></li>
  <li><a href="#randomized-complete-block-design-rcbd" id="toc-randomized-complete-block-design-rcbd" class="nav-link" data-scroll-target="#randomized-complete-block-design-rcbd"><span class="header-section-number">14.4.4</span> Randomized Complete Block Design (RCBD)</a></li>
  <li><a href="#randomized-split-plot-design-rspd" id="toc-randomized-split-plot-design-rspd" class="nav-link" data-scroll-target="#randomized-split-plot-design-rspd"><span class="header-section-number">14.4.5</span> Randomized Split Plot Design (RSPD)</a></li>
  <li><a href="#generalized-randomized-complete-block-design-grcbd" id="toc-generalized-randomized-complete-block-design-grcbd" class="nav-link" data-scroll-target="#generalized-randomized-complete-block-design-grcbd"><span class="header-section-number">14.4.6</span> Generalized Randomized Complete Block Design (GRCBD)</a></li>
  <li><a href="#nested-randomized-complete-block-design-nrcbd" id="toc-nested-randomized-complete-block-design-nrcbd" class="nav-link" data-scroll-target="#nested-randomized-complete-block-design-nrcbd"><span class="header-section-number">14.4.7</span> Nested Randomized Complete Block Design (NRCBD)</a></li>
  <li><a href="#longitudinal-randomized-complete-block-design-lrcbd" id="toc-longitudinal-randomized-complete-block-design-lrcbd" class="nav-link" data-scroll-target="#longitudinal-randomized-complete-block-design-lrcbd"><span class="header-section-number">14.4.8</span> Longitudinal Randomized Complete Block Design (LRCBD)</a></li>
  <li><a href="#variations-due-to-multiple-measures-of-the-response-variable" id="toc-variations-due-to-multiple-measures-of-the-response-variable" class="nav-link" data-scroll-target="#variations-due-to-multiple-measures-of-the-response-variable"><span class="header-section-number">14.4.9</span> Variations due to multiple measures of the response variable</a></li>
  </ul></li>
  <li><a href="#building-the-linear-mixed-model-for-clustered-data" id="toc-building-the-linear-mixed-model-for-clustered-data" class="nav-link" data-scroll-target="#building-the-linear-mixed-model-for-clustered-data"><span class="header-section-number">14.5</span> Building the linear (mixed) model for clustered data</a></li>
  <li><a href="#lmm-demo" id="toc-lmm-demo" class="nav-link" data-scroll-target="#lmm-demo"><span class="header-section-number">14.6</span> Example 1 – A random intercepts and slopes explainer (demo1)</a>
  <ul>
  <li><a href="#batched-measurements-result-in-clustered-residuals" id="toc-batched-measurements-result-in-clustered-residuals" class="nav-link" data-scroll-target="#batched-measurements-result-in-clustered-residuals"><span class="header-section-number">14.6.1</span> Batched measurements result in clustered residuals</a></li>
  <li><a href="#lmm-corerr" id="toc-lmm-corerr" class="nav-link" data-scroll-target="#lmm-corerr"><span class="header-section-number">14.6.2</span> Clustered residuals result in correlated error</a></li>
  <li><a href="#in-blocked-designs-clustered-residuals-adds-a-variance-component-that-masks-treatment-effects" id="toc-in-blocked-designs-clustered-residuals-adds-a-variance-component-that-masks-treatment-effects" class="nav-link" data-scroll-target="#in-blocked-designs-clustered-residuals-adds-a-variance-component-that-masks-treatment-effects"><span class="header-section-number">14.6.3</span> In blocked designs, clustered residuals adds a variance component that masks treatment effects</a></li>
  <li><a href="#linear-mixed-models-are-linear-models-with-added-random-factors" id="toc-linear-mixed-models-are-linear-models-with-added-random-factors" class="nav-link" data-scroll-target="#linear-mixed-models-are-linear-models-with-added-random-factors"><span class="header-section-number">14.6.4</span> Linear mixed models are linear models with added random factors</a></li>
  <li><a href="#what-the-random-effects-are" id="toc-what-the-random-effects-are" class="nav-link" data-scroll-target="#what-the-random-effects-are"><span class="header-section-number">14.6.5</span> What the random effects are</a></li>
  <li><a href="#in-a-blocked-design-a-linear-model-with-added-random-effects-increases-precision-of-treatment-effects" id="toc-in-a-blocked-design-a-linear-model-with-added-random-effects-increases-precision-of-treatment-effects" class="nav-link" data-scroll-target="#in-a-blocked-design-a-linear-model-with-added-random-effects-increases-precision-of-treatment-effects"><span class="header-section-number">14.6.6</span> In a blocked design, a linear model with added random effects increases precision of treatment effects</a></li>
  <li><a href="#lmm-varcorr" id="toc-lmm-varcorr" class="nav-link" data-scroll-target="#lmm-varcorr"><span class="header-section-number">14.6.7</span> The correlation among random intercepts and slopes</a></li>
  <li><a href="#clustered-residuals-create-heterogeneity-among-treatments" id="toc-clustered-residuals-create-heterogeneity-among-treatments" class="nav-link" data-scroll-target="#clustered-residuals-create-heterogeneity-among-treatments"><span class="header-section-number">14.6.8</span> Clustered residuals create heterogeneity among treatments</a></li>
  <li><a href="#linear-mixed-models-are-flexible" id="toc-linear-mixed-models-are-flexible" class="nav-link" data-scroll-target="#linear-mixed-models-are-flexible"><span class="header-section-number">14.6.9</span> Linear mixed models are flexible</a></li>
  <li><a href="#a-random-intercept-only-model" id="toc-a-random-intercept-only-model" class="nav-link" data-scroll-target="#a-random-intercept-only-model"><span class="header-section-number">14.6.10</span> A random intercept only model</a></li>
  <li><a href="#a-model-including-an-interaction-intercept" id="toc-a-model-including-an-interaction-intercept" class="nav-link" data-scroll-target="#a-model-including-an-interaction-intercept"><span class="header-section-number">14.6.11</span> A model including an interaction intercept</a></li>
  <li><a href="#lmm-demo-aic" id="toc-lmm-demo-aic" class="nav-link" data-scroll-target="#lmm-demo-aic"><span class="header-section-number">14.6.12</span> AIC and model selection – which model to report?</a></li>
  <li><a href="#the-specification-of-random-effects-matters" id="toc-the-specification-of-random-effects-matters" class="nav-link" data-scroll-target="#the-specification-of-random-effects-matters"><span class="header-section-number">14.6.13</span> The specification of random effects matters</a></li>
  <li><a href="#mixed-effect-and-repeated-measures-anova" id="toc-mixed-effect-and-repeated-measures-anova" class="nav-link" data-scroll-target="#mixed-effect-and-repeated-measures-anova"><span class="header-section-number">14.6.14</span> Mixed Effect and Repeated Measures ANOVA</a></li>
  <li><a href="#pseudoreplication" id="toc-pseudoreplication" class="nav-link" data-scroll-target="#pseudoreplication"><span class="header-section-number">14.6.15</span> Pseudoreplication</a></li>
  </ul></li>
  <li><a href="#lmm-example2" id="toc-lmm-example2" class="nav-link" data-scroll-target="#lmm-example2"><span class="header-section-number">14.7</span> Example 2 – experiments without subsampling replication (exp6g)</a>
  <ul>
  <li><a href="#understand-the-data" id="toc-understand-the-data" class="nav-link" data-scroll-target="#understand-the-data"><span class="header-section-number">14.7.1</span> Understand the data</a></li>
  <li><a href="#model-fit-and-inference" id="toc-model-fit-and-inference" class="nav-link" data-scroll-target="#model-fit-and-inference"><span class="header-section-number">14.7.2</span> Model fit and inference</a>
  <ul class="collapse">
  <li><a href="#fit-the-model" id="toc-fit-the-model" class="nav-link" data-scroll-target="#fit-the-model"><span class="header-section-number">14.7.2.1</span> Fit the model</a></li>
  <li><a href="#inference-from-the-model" id="toc-inference-from-the-model" class="nav-link" data-scroll-target="#inference-from-the-model"><span class="header-section-number">14.7.2.2</span> Inference from the model</a></li>
  <li><a href="#lmm-exp6g-plotthemodel" id="toc-lmm-exp6g-plotthemodel" class="nav-link" data-scroll-target="#lmm-exp6g-plotthemodel"><span class="header-section-number">14.7.2.3</span> Plot the model</a></li>
  <li><a href="#alternaplot-the-model" id="toc-alternaplot-the-model" class="nav-link" data-scroll-target="#alternaplot-the-model"><span class="header-section-number">14.7.2.4</span> Alternaplot the model</a></li>
  </ul></li>
  <li><a href="#the-model-exp6g_m1-adds-a-random-intercept-but-not-a-random-slope" id="toc-the-model-exp6g_m1-adds-a-random-intercept-but-not-a-random-slope" class="nav-link" data-scroll-target="#the-model-exp6g_m1-adds-a-random-intercept-but-not-a-random-slope"><span class="header-section-number">14.7.3</span> The model exp6g_m1 adds a random intercept but not a random slope</a></li>
  <li><a href="#the-fixed-effect-coefficients-of-model-exp6g_m1" id="toc-the-fixed-effect-coefficients-of-model-exp6g_m1" class="nav-link" data-scroll-target="#the-fixed-effect-coefficients-of-model-exp6g_m1"><span class="header-section-number">14.7.4</span> The fixed effect coefficients of model exp6g_m1</a></li>
  <li><a href="#the-random-intercept-coefficients-of-exp6g_m1" id="toc-the-random-intercept-coefficients-of-exp6g_m1" class="nav-link" data-scroll-target="#the-random-intercept-coefficients-of-exp6g_m1"><span class="header-section-number">14.7.5</span> The random intercept coefficients of exp6g_m1</a></li>
  <li><a href="#the-random-and-residual-variance-and-the-intraclass-correlation-of-model-exp6g_m1" id="toc-the-random-and-residual-variance-and-the-intraclass-correlation-of-model-exp6g_m1" class="nav-link" data-scroll-target="#the-random-and-residual-variance-and-the-intraclass-correlation-of-model-exp6g_m1"><span class="header-section-number">14.7.6</span> The random and residual variance and the intraclass correlation of model exp6g_m1</a></li>
  <li><a href="#the-linear-mixed-model-exp6g_m1-increases-precision-of-treatment-effects-relative-to-a-fixed-effects-model" id="toc-the-linear-mixed-model-exp6g_m1-increases-precision-of-treatment-effects-relative-to-a-fixed-effects-model" class="nav-link" data-scroll-target="#the-linear-mixed-model-exp6g_m1-increases-precision-of-treatment-effects-relative-to-a-fixed-effects-model"><span class="header-section-number">14.7.7</span> The linear mixed model exp6g_m1 increases precision of treatment effects, relative to a fixed effects model</a></li>
  <li><a href="#lmm-exp6g-alt" id="toc-lmm-exp6g-alt" class="nav-link" data-scroll-target="#lmm-exp6g-alt"><span class="header-section-number">14.7.8</span> Alternative models for exp6g</a></li>
  <li><a href="#paired-t-tests-and-repeated-measures-anova-are-special-cases-of-linear-mixed-models" id="toc-paired-t-tests-and-repeated-measures-anova-are-special-cases-of-linear-mixed-models" class="nav-link" data-scroll-target="#paired-t-tests-and-repeated-measures-anova-are-special-cases-of-linear-mixed-models"><span class="header-section-number">14.7.9</span> Paired t-tests and repeated measures ANOVA are special cases of linear mixed models</a></li>
  <li><a href="#lmm-exp6g-rmanova" id="toc-lmm-exp6g-rmanova" class="nav-link" data-scroll-target="#lmm-exp6g-rmanova"><span class="header-section-number">14.7.10</span> Classical (“univariate model”) repeated measures ANOVA of exp6g</a></li>
  <li><a href="#lmm-exp6g-rmanova-multi" id="toc-lmm-exp6g-rmanova-multi" class="nav-link" data-scroll-target="#lmm-exp6g-rmanova-multi"><span class="header-section-number">14.7.11</span> “Multivariate model” repeated measures ANOVA</a></li>
  <li><a href="#linear-mixed-models-vs-repeated-measures-anova" id="toc-linear-mixed-models-vs-repeated-measures-anova" class="nav-link" data-scroll-target="#linear-mixed-models-vs-repeated-measures-anova"><span class="header-section-number">14.7.12</span> Linear mixed models vs repeated measures ANOVA</a></li>
  <li><a href="#modeling-textttmouse_id-as-a-fixed-effect" id="toc-modeling-textttmouse_id-as-a-fixed-effect" class="nav-link" data-scroll-target="#modeling-textttmouse_id-as-a-fixed-effect"><span class="header-section-number">14.7.13</span> Modeling <span class="math inline">\(\texttt{mouse_id}\)</span> as a fixed effect</a></li>
  </ul></li>
  <li><a href="#example-3-factorial-experiments-and-no-subsampling-replicates-exp5c" id="toc-example-3-factorial-experiments-and-no-subsampling-replicates-exp5c" class="nav-link" data-scroll-target="#example-3-factorial-experiments-and-no-subsampling-replicates-exp5c"><span class="header-section-number">14.8</span> Example 3 – Factorial experiments and no subsampling replicates (exp5c)</a>
  <ul>
  <li><a href="#understand-the-data-1" id="toc-understand-the-data-1" class="nav-link" data-scroll-target="#understand-the-data-1"><span class="header-section-number">14.8.1</span> Understand the data</a></li>
  <li><a href="#examine-the-data" id="toc-examine-the-data" class="nav-link" data-scroll-target="#examine-the-data"><span class="header-section-number">14.8.2</span> Examine the data</a></li>
  <li><a href="#lmm-exp5c-m1" id="toc-lmm-exp5c-m1" class="nav-link" data-scroll-target="#lmm-exp5c-m1"><span class="header-section-number">14.8.3</span> Model fit and inference</a>
  <ul class="collapse">
  <li><a href="#check-the-model" id="toc-check-the-model" class="nav-link" data-scroll-target="#check-the-model"><span class="header-section-number">14.8.3.1</span> Check the model</a></li>
  <li><a href="#inference-from-the-model-1" id="toc-inference-from-the-model-1" class="nav-link" data-scroll-target="#inference-from-the-model-1"><span class="header-section-number">14.8.3.2</span> Inference from the model</a></li>
  <li><a href="#lmm-exp5c-plotthemodel" id="toc-lmm-exp5c-plotthemodel" class="nav-link" data-scroll-target="#lmm-exp5c-plotthemodel"><span class="header-section-number">14.8.3.3</span> Plot the model</a></li>
  <li><a href="#alternaplot-the-model-1" id="toc-alternaplot-the-model-1" class="nav-link" data-scroll-target="#alternaplot-the-model-1"><span class="header-section-number">14.8.3.4</span> Alternaplot the model</a></li>
  </ul></li>
  <li><a href="#why-we-care-about-modeling-batch-in-exp5c" id="toc-why-we-care-about-modeling-batch-in-exp5c" class="nav-link" data-scroll-target="#why-we-care-about-modeling-batch-in-exp5c"><span class="header-section-number">14.8.4</span> Why we care about modeling batch in exp5c</a></li>
  <li><a href="#the-linear-mixed-model-exp5c_m1-adds-two-random-intercepts" id="toc-the-linear-mixed-model-exp5c_m1-adds-two-random-intercepts" class="nav-link" data-scroll-target="#the-linear-mixed-model-exp5c_m1-adds-two-random-intercepts"><span class="header-section-number">14.8.5</span> The linear mixed model exp5c_m1 adds two random intercepts</a></li>
  <li><a href="#the-fixed-effect-coefficients-of-model-exp5c_m1" id="toc-the-fixed-effect-coefficients-of-model-exp5c_m1" class="nav-link" data-scroll-target="#the-fixed-effect-coefficients-of-model-exp5c_m1"><span class="header-section-number">14.8.6</span> The fixed effect coefficients of model exp5c_m1</a></li>
  <li><a href="#the-random-effect-coefficients-of-model-exp5c_m1" id="toc-the-random-effect-coefficients-of-model-exp5c_m1" class="nav-link" data-scroll-target="#the-random-effect-coefficients-of-model-exp5c_m1"><span class="header-section-number">14.8.7</span> The random effect coefficients of model exp5c_m1</a></li>
  <li><a href="#lmm-exp5c-alt" id="toc-lmm-exp5c-alt" class="nav-link" data-scroll-target="#lmm-exp5c-alt"><span class="header-section-number">14.8.8</span> Alternative models for exp5c</a></li>
  <li><a href="#lmm-rmanova-5c" id="toc-lmm-rmanova-5c" class="nav-link" data-scroll-target="#lmm-rmanova-5c"><span class="header-section-number">14.8.9</span> Classical (“univariate model”) repeated measures ANOVA</a></li>
  <li><a href="#lmm-rmanova-5c-multi" id="toc-lmm-rmanova-5c-multi" class="nav-link" data-scroll-target="#lmm-rmanova-5c-multi"><span class="header-section-number">14.8.10</span> “Multivariate model” repeated measures ANOVA of exp5c</a></li>
  <li><a href="#modeling-textttdonor-as-a-fixed-effect" id="toc-modeling-textttdonor-as-a-fixed-effect" class="nav-link" data-scroll-target="#modeling-textttdonor-as-a-fixed-effect"><span class="header-section-number">14.8.11</span> Modeling <span class="math inline">\(\texttt{donor}\)</span> as a fixed effect</a></li>
  </ul></li>
  <li><a href="#lmm-example4" id="toc-lmm-example4" class="nav-link" data-scroll-target="#lmm-example4"><span class="header-section-number">14.9</span> Example 4 – Experiments with subsampling replication (exp1g)</a>
  <ul>
  <li><a href="#understand-the-data-2" id="toc-understand-the-data-2" class="nav-link" data-scroll-target="#understand-the-data-2"><span class="header-section-number">14.9.1</span> Understand the data</a></li>
  <li><a href="#examine-the-data-1" id="toc-examine-the-data-1" class="nav-link" data-scroll-target="#examine-the-data-1"><span class="header-section-number">14.9.2</span> Examine the data</a></li>
  <li><a href="#fit-the-model-1" id="toc-fit-the-model-1" class="nav-link" data-scroll-target="#fit-the-model-1"><span class="header-section-number">14.9.3</span> Fit the model</a></li>
  <li><a href="#inference-from-the-model-2" id="toc-inference-from-the-model-2" class="nav-link" data-scroll-target="#inference-from-the-model-2"><span class="header-section-number">14.9.4</span> Inference from the model</a></li>
  <li><a href="#plot-the-model" id="toc-plot-the-model" class="nav-link" data-scroll-target="#plot-the-model"><span class="header-section-number">14.9.5</span> Plot the model</a></li>
  <li><a href="#alternaplot-the-model-2" id="toc-alternaplot-the-model-2" class="nav-link" data-scroll-target="#alternaplot-the-model-2"><span class="header-section-number">14.9.6</span> Alternaplot the model</a></li>
  <li><a href="#understanding-the-alternative-models" id="toc-understanding-the-alternative-models" class="nav-link" data-scroll-target="#understanding-the-alternative-models"><span class="header-section-number">14.9.7</span> Understanding the alternative models</a></li>
  <li><a href="#the-varcorr-matrix-of-models-exp1g_m1a-and-exp1g_m1b" id="toc-the-varcorr-matrix-of-models-exp1g_m1a-and-exp1g_m1b" class="nav-link" data-scroll-target="#the-varcorr-matrix-of-models-exp1g_m1a-and-exp1g_m1b"><span class="header-section-number">14.9.8</span> The VarCorr matrix of models exp1g_m1a and exp1g_m1b</a></li>
  <li><a href="#the-linear-mixed-model-has-more-precision-and-power-than-the-fixed-effect-model-of-batch-means" id="toc-the-linear-mixed-model-has-more-precision-and-power-than-the-fixed-effect-model-of-batch-means" class="nav-link" data-scroll-target="#the-linear-mixed-model-has-more-precision-and-power-than-the-fixed-effect-model-of-batch-means"><span class="header-section-number">14.9.9</span> The linear mixed model has more precision and power than the fixed effect model of batch means</a></li>
  <li><a href="#fixed-effect-models-and-pseudoreplication" id="toc-fixed-effect-models-and-pseudoreplication" class="nav-link" data-scroll-target="#fixed-effect-models-and-pseudoreplication"><span class="header-section-number">14.9.10</span> Fixed effect models and pseudoreplication</a></li>
  <li><a href="#mixed-effect-anova" id="toc-mixed-effect-anova" class="nav-link" data-scroll-target="#mixed-effect-anova"><span class="header-section-number">14.9.11</span> Mixed-effect ANOVA</a></li>
  </ul></li>
  <li><a href="#statistical-models-for-experimental-designs" id="toc-statistical-models-for-experimental-designs" class="nav-link" data-scroll-target="#statistical-models-for-experimental-designs"><span class="header-section-number">14.10</span> Statistical models for experimental designs</a>
  <ul>
  <li><a href="#models-for-completely-randomized-designs-crd" id="toc-models-for-completely-randomized-designs-crd" class="nav-link" data-scroll-target="#models-for-completely-randomized-designs-crd"><span class="header-section-number">14.10.1</span> Models for Completely Randomized Designs (CRD)</a></li>
  <li><a href="#models-for-batched-data-crds-rcbd-rspd-grcbd-nrcbd" id="toc-models-for-batched-data-crds-rcbd-rspd-grcbd-nrcbd" class="nav-link" data-scroll-target="#models-for-batched-data-crds-rcbd-rspd-grcbd-nrcbd"><span class="header-section-number">14.10.2</span> Models for batched data (CRDS, RCBD, RSPD, GRCBD, NRCBD)</a>
  <ul class="collapse">
  <li><a href="#linear-models" id="toc-linear-models" class="nav-link" data-scroll-target="#linear-models"><span class="header-section-number">14.10.2.1</span> Linear models</a></li>
  <li><a href="#linear-mixed-models-random-effects-models" id="toc-linear-mixed-models-random-effects-models" class="nav-link" data-scroll-target="#linear-mixed-models-random-effects-models"><span class="header-section-number">14.10.2.2</span> linear mixed models (“random effects” models)</a></li>
  <li><a href="#anova-models-mixed-models-repeated-measures-anova" id="toc-anova-models-mixed-models-repeated-measures-anova" class="nav-link" data-scroll-target="#anova-models-mixed-models-repeated-measures-anova"><span class="header-section-number">14.10.2.3</span> ANOVA models (“mixed models”, “repeated measures ANOVA”)</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#which-model-and-why" id="toc-which-model-and-why" class="nav-link" data-scroll-target="#which-model-and-why"><span class="header-section-number">14.11</span> Which model and why?</a>
  <ul>
  <li><a href="#crd" id="toc-crd" class="nav-link" data-scroll-target="#crd"><span class="header-section-number">14.11.1</span> CRD</a></li>
  <li><a href="#crds" id="toc-crds" class="nav-link" data-scroll-target="#crds"><span class="header-section-number">14.11.2</span> CRDS</a></li>
  <li><a href="#rcbd" id="toc-rcbd" class="nav-link" data-scroll-target="#rcbd"><span class="header-section-number">14.11.3</span> RCBD</a></li>
  <li><a href="#rspd" id="toc-rspd" class="nav-link" data-scroll-target="#rspd"><span class="header-section-number">14.11.4</span> RSPD</a></li>
  <li><a href="#rcbds" id="toc-rcbds" class="nav-link" data-scroll-target="#rcbds"><span class="header-section-number">14.11.5</span> RCBDS</a></li>
  <li><a href="#grcbd" id="toc-grcbd" class="nav-link" data-scroll-target="#grcbd"><span class="header-section-number">14.11.6</span> GRCBD</a></li>
  <li><a href="#grcbds" id="toc-grcbds" class="nav-link" data-scroll-target="#grcbds"><span class="header-section-number">14.11.7</span> GRCBDS</a></li>
  </ul></li>
  <li><a href="#working-in-r" id="toc-working-in-r" class="nav-link" data-scroll-target="#working-in-r"><span class="header-section-number">14.12</span> Working in R</a>
  <ul>
  <li><a href="#plotting-models-fit-to-batched-data" id="toc-plotting-models-fit-to-batched-data" class="nav-link" data-scroll-target="#plotting-models-fit-to-batched-data"><span class="header-section-number">14.12.1</span> Plotting models fit to batched data</a>
  <ul class="collapse">
  <li><a href="#models-without-subsampling---experiment-6g" id="toc-models-without-subsampling---experiment-6g" class="nav-link" data-scroll-target="#models-without-subsampling---experiment-6g"><span class="header-section-number">14.12.1.1</span> Models without subsampling - Experiment 6g</a></li>
  <li><a href="#models-with-subsampling---experiment-1g" id="toc-models-with-subsampling---experiment-1g" class="nav-link" data-scroll-target="#models-with-subsampling---experiment-1g"><span class="header-section-number">14.12.1.2</span> Models with subsampling - Experiment 1g</a></li>
  </ul></li>
  <li><a href="#repeated-measures-anova-randomized-complete-block-with-no-subsampling" id="toc-repeated-measures-anova-randomized-complete-block-with-no-subsampling" class="nav-link" data-scroll-target="#repeated-measures-anova-randomized-complete-block-with-no-subsampling"><span class="header-section-number">14.12.2</span> Repeated measures ANOVA (randomized complete block with no subsampling)</a>
  <ul class="collapse">
  <li><a href="#univariate-vs.-multivariate-repeated-measures-anova" id="toc-univariate-vs.-multivariate-repeated-measures-anova" class="nav-link" data-scroll-target="#univariate-vs.-multivariate-repeated-measures-anova"><span class="header-section-number">14.12.2.1</span> univariate vs.&nbsp;multivariate repeated measures ANOVA</a></li>
  <li><a href="#the-anova-table" id="toc-the-anova-table" class="nav-link" data-scroll-target="#the-anova-table"><span class="header-section-number">14.12.2.2</span> The ANOVA table</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#hidden-code" id="toc-hidden-code" class="nav-link" data-scroll-target="#hidden-code"><span class="header-section-number">14.13</span> Hidden code</a>
  <ul>
  <li><a href="#import-exp5c" id="toc-import-exp5c" class="nav-link" data-scroll-target="#import-exp5c"><span class="header-section-number">14.13.1</span> Import exp5c</a></li>
  <li><a href="#import-exp1g" id="toc-import-exp1g" class="nav-link" data-scroll-target="#import-exp1g"><span class="header-section-number">14.13.2</span> Import exp1g</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title"><span id="lmm" class="quarto-section-identifier"><span class="chapter-number">14</span>&nbsp; <span class="chapter-title">Models for non-independence – linear mixed models</span></span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<p>Probably no chapter in this book is more important for the best-practice analysis of experimental data than this chapter. Why? Because many if not most experimental data violates the assumption of independence and any analysis using standard <em>t</em>-tests and ANOVA will always lead to quantitative error in inference (confidence intervals and <em>p</em>-values) and often lead to qualitative errors in inference (statements about “significance”). Standard analysis of non-independent data can lead to absurdly liberal inference (the <em>p</em>-values are far lower than the data support), but can also lead to moderately conservative inference (the <em>p</em>-values are higher than the data supports). Liberal inference generates false discovery and lures researchers down dead-end research pathways. Conservative inference steers researchers away from true discovery.</p>
<p><strong>Linear Mixed Models</strong> are an extension of linear models that appropriately adjust inferential statistics for non-independent data. <strong>Paired t-tests</strong> and <strong>Repeated Measures ANOVA</strong> are classical tests that are special cases of linear mixed models. Linear mixed models are more flexible than these classical tests because the models can include added covariates or more complex models generally. And, linear mixed models can be extended to <strong>Generalized Linear Mixed Models</strong> for counts, binary responses, skewed responses, and ratios.</p>
<p>Before introducing experimental designs that generate non-independent data and the models used to analyze these, let’s explore two ubiquitous examples, one that leads to liberal inference and one that leads to conservative inference.</p>
<section id="liberal-inference-from-pseudoreplication" class="level2" data-number="14.1">
<h2 data-number="14.1" class="anchored" data-anchor-id="liberal-inference-from-pseudoreplication"><span class="header-section-number">14.1</span> Liberal inference from pseudoreplication</h2>
<p>Researchers are interested in regulation and repair of DNA double-stranded breaks and use a proximity ligation assay (PLA) of HeLa cells to investigate the number of damage response events (“foci”) per cell with and without an inhibitor of transcription elongation (DRB). The number of foci in each of fifty cells per treatment is measured. The experiment is replicated three times. The researchers use a <em>t</em>-test to compare the effect of DRB on foci count and naively include all measures in the analysis.</p>
<p>What is naive about the analysis? The fifty measures per cell are <strong>technical replicates</strong> and the values within a cell are not independent of each other because they share aspects of cell environment not shared by values in other cells. Including technical replicates in an analysis without accounting for this non-independence is a kind of <strong>pseudoreplication</strong></p>
<p>To show how this naive analysis results in extremely liberal inference and an increase in false discovery, I simulate this experiment using a case in which there is no effect of DRB treatment, so a low <em>p</em>-value indicates a false-discovery. The simulation is simplified with the two following conditions: 1) the model pretends that count data are normally distributed (this is because we want to focus on pseudoreplication and not a misspecified distribution) and 2) the model pretends that values from each treatment within an experiment are independent (this is because we want to focus on pseudoreplication).</p>
<p>In this naive analysis of the experiment, the researcher finds an effect of treatment with a <em>p</em>-value of 0.000073 and uses this small <em>p</em>-value to justify a decision to move forward with follow-up experiments. But this low <em>p</em>-value is not supported by the data – this discovery is false. This very small <em>p</em>-value is not an example of a “rare event”. In fact, if the researcher repeats the experiment 1000 times, then the median <em>p</em>-value is 0.000195 and 72.3% of the 1000 <em>p</em>-values lead to the same false discovery if 0.05 is used to make the decision to move forward.</p>
<p>While this example is fake, I see this naive analysis a lot: tumor area of multiple tumors per mouse, islet area of multiple islets per mouse, number of vesicles docked to a membrane in multiple cells per mouse, the number of mitochondria in multiple cells per mouse, the number of neurites in multiple neurons in multiple cells per mouse, etc. etc. Indeed, when I’m looking for examples of pseudoreplication to teach, I just look for figures with a bunch of points per treatment – something similar to this plot of the fake data experiment. Regardless, this is a huge source of false discovery that could disappear overnight.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-lmm-pseudoreplication-fake-example" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-lmm-pseudoreplication-fake-example-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="lmm_files/figure-html/fig-lmm-pseudoreplication-fake-example-1.png" class="img-fluid figure-img" width="672">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-lmm-pseudoreplication-fake-example-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;14.1: A fake example of pseudoreplication motivated by a real experiment.
</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="conservative-inference-from-failure-to-identify-blocks" class="level2" data-number="14.2">
<h2 data-number="14.2" class="anchored" data-anchor-id="conservative-inference-from-failure-to-identify-blocks"><span class="header-section-number">14.2</span> Conservative inference from failure to identify blocks</h2>
<p>Researchers are investigating the regulation of the differentiation of stem blood cells into osteoclasts, which can cause osteoporosis if overactivated. The researchers randomly sample three mice from 6 litters and use a <strong>littermate control</strong> design: within a litter, one sib is assigned to control (CN), one to glucortacoid (GC) treatment, and one to buthionine sulphoximine (BSO) treatment. The researchers are investigating the mechanism of GC-induced osteoporosis and if the researcher’s model is correct, then BSO should block this mechanism. Following an ANOVA, the researchers report the unadjusted <em>p</em>-value for each pairwise comparison with the expectation that BSO will reverse the effect of GC.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-lmm-failed-discovery-example" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-lmm-failed-discovery-example-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="lmm_files/figure-html/fig-lmm-failed-discovery-example-1.png" class="img-fluid figure-img" width="672">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-lmm-failed-discovery-example-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;14.2: Results of the fake experiment. A) results using ANOVA using on initial six litters, B) published results with four additional litters, C) results using LMM on initial six litters
</figcaption>
</figure>
</div>
</div>
</div>
<p>What is naive about the analysis? Any value measured from sibling mice within a litter are not independent of each other because the sibling mice share aspects of genetics and maternal environment not shared by mice in other litters. This shared variance adds correlated noise to data and failure to account for the shared variance will almost always lead to more conservative inference.</p>
<p>In the experiment, the results “looked” like the expected results if the model GC-induced osteoporosis were correct, but the <em>p</em>-value of the focal contrast (BSO - GC) was not &lt; 0.05 (<a href="#fig-lmm-failed-discovery-example" class="quarto-xref">Figure&nbsp;<span>14.2</span></a> A). The researchers expanded the experiment, adding data from four more litters (<a href="#fig-lmm-failed-discovery-example" class="quarto-xref">Figure&nbsp;<span>14.2</span></a> B). Had the researchers analyzed the initial set of data using a statistical model that accounts for the shared variance within a litter (<a href="#fig-lmm-failed-discovery-example" class="quarto-xref">Figure&nbsp;<span>14.2</span></a> C), the researchers would have been satisfied and fewer mice would have been killed, fewer resources used, and more time to pursue continued probing of the mechanism. The statistical model used to take advantage of the littermate control design is a <strong>linear-mixed model</strong>. An experiment with littermate control is known as a <strong>blocked design</strong>.</p>
<div class="callout callout-style-default callout-warning callout-titled" title="NHST Blues">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
NHST Blues
</div>
</div>
<div class="callout-body-container callout-body">
<p>NHST encourages peeking at the data to see if p &lt; 0.05 for a focal contrast, and collecting more data if this isn’t the case. Don’t do this. Peeking makes the <em>p</em>-value non-valid and will increase the false discovery rate. Peeking is a variation of the multiple test problem. In clinical trials there are statistically rigorous methods for peeking, which allows a trial to stop early.</p>
</div>
</div>
<p>The small <em>p</em>-value in the BSO - GC contrast using a linear mixed model is not an example of a “rare event”. In fact, if the researchers repeated the experiment 1000 times, 89.1% of the BSO - GC <em>p</em>-values using the linear mixed model are less than 0.05 while only 29.9% of the <em>p</em>-values following classical ANOVA are less than 0.05.</p>
<p>While this example is fake, I see this naive analysis a lot – littermate controls is very, very common but other examples are too, including replicated experiments (each experiment is a block). There are certain instances where researchers do recognize non-independence and do use a paired <em>t</em>-test but the vast majority of blocked designs (occurring in almost all experimental biology papers) go unrecognized. This is a huge source of failed discovery that could disappear overnight.</p>
</section>
<section id="introduction-to-models-for-non-independent-data-linear-mixed-models" class="level2" data-number="14.3">
<h2 data-number="14.3" class="anchored" data-anchor-id="introduction-to-models-for-non-independent-data-linear-mixed-models"><span class="header-section-number">14.3</span> Introduction to models for non-independent data (linear mixed models)</h2>
<p>This chapter is about models for <strong>correlated error</strong>, including linear models with added <strong>random factors</strong>, which are known as <strong>linear mixed models</strong>. In classical hypothesis testing, a <strong>paired t-test</strong>, <strong>repeated measures ANOVA</strong>, and <strong>mixed-effect ANOVA</strong> are equivalent to specific cases of linear mixed models. Linear mixed models are used for analyzing data composed of subsets – or <strong>batches</strong> – of data that were measured from the “same thing”, such as multiple measures within a mouse, or multiple mice within a litter. Batched data results in <strong>correlated error</strong>, which violates a key assumption of linear models (and their “which test” equivalents) and muddles statistical inference unless the correlated error is modeled, explicitly or implicitly. In some experimental designs (<strong>blocked designs</strong>), failure to model the correlated error reduces precision and power, contributing to reduced rates of discovery or confirmation. In other designs (<strong>nested designs</strong>), failure to model the correlated error results in falsely high precision and low <em>p</em>-values, leading to increased rates of false discovery. The falsely high precision is due to <strong>pseudoreplication</strong>. I think it’s fair to infer from the experimental biology literature, that experimental biologists don’t recognize the ubiquitousness of batched data and correlated error. This is probably the biggest issue in inference in the field (far more of an issue than say, a <em>t</em>-test on non-normal data).</p>
<p>What do I mean by “batch” and how can correlated error both increase and decrease false discovery? Consider an experiment to measure pancreatic islet area in response to two experimental factors: <span class="math inline">\(\texttt{genotype}\)</span> (WT, KO) and <span class="math inline">\(\texttt{treatment}\)</span> (presence/absence of some drug believed to be an agonist of the knocked out protein). While it may seem like the data from this experiment should be analyzed using the ANOVA option in GraphPad Prism (or, as advocated in this book, a general linear model that is equivalent to the ANOVA), the best practice statistical model actually depends on the <strong>experimental design</strong>. Experimental design matters because different designs introduce different patterns of <strong>correlated error</strong> due to shared genetics and environment. Recall that inference from a linear model (including <em>t</em>-tests and ANOVA) assumes <em>independence</em> (Chapter xxx) – that is, each response value has no relationship to any other value, other than that due to treatment. Lack of independence results in patterns of correlation among the residuals, or correlated error.</p>
<p>Something like the first experiment below (Design 1) is the necessary design to use the statistics that have been covered in this book to this point, without extreme violation of the independence assumption. But many (most?) experiments in experimental bench biology do not look like the design in Design 1 below. Instead, many (most?) experiments are variants of Designs 2-4, all of which have extreme violations of the independence assumption. Interestingly, some of these violations result in conservative statistics and reduced, true discovery rate while others result in liberal statistics and increased, false discovery rate.</p>
<p>Design 1. The design in Figure @ref(fig:lmm-biological-replicates_1) is a factorial design with two factors, <span class="math inline">\(\texttt{genotype}\)</span> and <span class="math inline">\(\texttt{treatment}\)</span>, each with two levels. Twenty mice of the same sex, each from a different litter from a unique dam and sire mating, are randomly sampled and assigned to one <span class="math inline">\(genotype \times treatment\)</span> combination (five mice per combination). All mice are housed individually (20 cages). The pancreatic tissue from all mice is prepared in a single batch and the area of a single islet is measured from each mouse. The entire experiment is carried out at the same time and each component (tissue preparation, measuring) is carried out by the same person (these could be different people for each component). This is a <strong>Completely Randomized Design</strong> (CRD). The five replicate mice per treatment are <strong>treatment replicates</strong> (often called <strong>biological replicates</strong> in experimental biology. A CRD does not have batched data.</p>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="../images/experimental_designs/lmm-biological-replicates 500 W.png" class="img-fluid figure-img" style="width:63.0%"></p>
</figure>
</div>
</div>
</div>
<p>Design 2. In the design in Figure @ref(fig:lmm-biological-replicates), four littermates are randomly sampled from five litters, each with a different dam and sire. Within each litter, mice are randomly assigned to each of the four treatment combinations (one per combination). Each litter is randomly assigned to cage with only a single litter per cage. All other aspects of this design are as in Design 1. This is a <strong>Randomized Complete Block Design</strong>. The five replicate mice per treatment are the treatment replicates. Each litter/cage combination is a type of batch called a <strong>block</strong>. A blocked design typically functions to reduce noise in the model fit (this increases power) and to reduce the number of litters and cages needed for an experiment. The four measures of Islet Area within a litter/cage (one per mouse) are not independent of each other. Each cage has four mice from the same litter and these mice share genetic and maternal factors that contribute to mouse anatomy and physiology that are not shared by mice in other litters. Additionally, each cage has a unique set of environmental factors that contribute to the error variance of the measure of the response. Each cage shares a cage-specific history of temperature, humidity, food, light, interactions with animal facilities staff, and behavioral interactions among the mice. All response measures within a litter/cage share the component of the error variance unique to that litter/cage and, as a consequence, the error (residuals) within a litter/cage are more similar to each other than they are to the residuals among litters/cages.</p>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="../images/experimental_designs/lmm-technical-replicates 500 W.png" class="img-fluid figure-img" style="width:63.0%"></p>
</figure>
</div>
</div>
</div>
<p>Design 3. The design in Figure @ref(fig:lmm-technical-replicates) is exactly like that in Design 2, except that the researchers take three measures of iselet area per mouse. The three measures are <strong>subsampled replicates</strong>. Experimental biologists often call these <strong>technical replicates</strong>, especially when the multiple measures are taken from the same preparation. Subsampling is a kind of <strong>nested design</strong> in which one variable is nested within (as opposed to crossed with) another variable. Here, the subsampled variable (subsample_id) is nested within the mouse_id variable. In addition to each litter/cage being a batch, each mouse is a batch. Each mouse has a unique set of factors that contribute to the error variance of the measures of the response in that mouse. All response measures within a mouse share the component of the error variance unique to that mouse and, as a consequence, the error (residuals) within a mouse are more similar to each other than they are to the residuals between mice</p>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="../images/lmm-pseudoreplication.png" class="img-fluid figure-img" style="width:63.0%"></p>
</figure>
</div>
</div>
</div>
<p>Design 4. The design in Figure @ref(fig:lmm-segregated) is a variation of Design 2, but the five treatment replicates of each combination are housed together in the same cage. In this design, each litter is a batch and each cage is a batch but these are different batches, unlike Design 2.</p>
<p>In each of these experiments, there is systematic variation at multiple levels: among treatments due to treatment effects and among batches due to <strong>batch effects</strong>. Batches come in lots of flavors, including experiment, litter, cage, flask, plate, slide, donor, and individual. The among-treatment differences in means are the <strong>fixed effects</strong>. The among-batch differences are the <strong>random effect</strong>. An assumption of modeling random effects is that the batches are a random sample of the batches that could have been sampled. This is often not strictly true as batches are often <strong>convenience samples</strong> (example: the human donors of the Type 2 diabetes beta cells are those that were in the hospital).</p>
<p>The variation among batches/lack of independence within batches has different consequences on the uncertainty of the estimate of a treatment effect. The batches in Experiment 1 contain all treatment combinations. The researcher is interested in the treatment effect but not the variation due to differences among the batches. The batches are nuissance factors that add additional variance to the response, with the consequence that estimates of treatment effects are less precise, unless the variance due to the batches is explicitly modeled. <strong>Modeling a batch that contains some or all treatment combinations will increase precision and power</strong>.</p>
<p>Batches that contain at least two treatment combinations are known as <strong>blocks</strong>. A block that contains all treatment combinations is a <strong>complete block</strong>. A block that contains fewer than all combinations is an <strong>incomplete block</strong>. Including block structure in the design is known as <strong>blocking</strong>. Blocks are non-experimental factors. Adding a blocking factor to a statistical model is used to increase the precision of an estimated treatment effect. Design 2 is an example of a <strong>randomized complete block design</strong>.</p>
<p>In Design 3, there are multiple measures per mouse and the design is a <strong>randomized complete block design with subsampling</strong>. The subsampling is not the kind of replication that can be used to infer the among treatment effect because the treatment assignment was not at the level of the subsamples. The <strong>treatment replicates</strong> are the litters/cages, because <em>it was at this level that treatment assignment was randomized</em>. A statistical analysis of all measures from a subsampled design without modeling the correlated error due to the subsampling is a kind of <a href="https://en.wikipedia.org/wiki/Pseudoreplication" target="_blank">pseudoreplication</a>. Pseudoreplication results in falsely precise standard errors and false small <em>p</em>-values and, consequently, increased rates of false discovery.</p>
<p>In Design 4, the treatment is randomized <em>to</em> batch, so each batch contains only a single treatment level. In these <strong>segregated</strong> experimental designs, the variation among batches that arises from non-treatment related differences among batches <strong>confounds</strong> the variation among batches due to a true treatment effect. Design 4 is an extreme example of this – there is only a single cage with a specific treatment combination. Imagine 1) the true effect of a treatment combination is zero and 2) an aggressive mouse in the control cage stimulates the stress response in the other mice and this stress response has a large effect on the value of the response variable measured by the researchers. The researcher is fooled into thinking that the treatment caused the difference in the response.</p>
<p>In all of these designs, it is important for the researcher to identify the <strong>experimental unit</strong> and the <strong>measurement</strong> unit. The experimental unit is the entity that was randomly assigned the treatment. In designs 1 – 3, the experimental unit is the mouse. In experiment 4, the experimental unit is the cage. The measurement unit is the entity that was measured. In designs 1, 2, and 4, the measurement unit is the mouse. In design 3, the measurement unit is specific islet that was measured.</p>
<div class="callout callout-style-default callout-note callout-titled" title="Pseudoreplication">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Pseudoreplication
</div>
</div>
<div class="callout-body-container callout-body">
<p>In pseudoreplication, the degrees of freedom used to compute the test statistic and the <em>p</em>-value are inflated given the experimental design and research question. An example: A researcher wants to investigate the effect of some protein on mitochondrial biogenesis and designs an experiment with a wildtype (WT) and a conditional knockout (KO) mouse. Mitochondrial counts from twenty cells in one WT mouse and one KO mouse are measured and the researcher uses a <em>t</em>-test to compare counts. The sample size used to compute the standard error in the denominator of the <em>t</em>-value is 20. The <em>t</em>-distribution used to compute the <em>p</em>-value uses 38 df (20 measures times two groups minus two estimated parameters). This is wrong. The df are inflated and the estimate of the standard error of the difference (the denominator of the t-value) is falsely small. The correct sample size for this design is 1 and the correct df is zero. The sample size and df are inflated for this design because the treatment was randomized to mouse and not to cell. Mouse is the experimental unit – the number of experimental units is what gives the degrees of freedom. The df are correct for inference about the two individuals (how compatible are the data and a model of sampling from the same individual?), but not for inference about the effect of genotype. We cannot infer anything about genotype with a sample size of 1, even with 20 measures per mouse, because any effect of treatment is completely confounded with other differences between the two mice.</p>
</div>
</div>
</section>
<section id="experimental-designs-in-experimental-bench-biology" class="level2" data-number="14.4">
<h2 data-number="14.4" class="anchored" data-anchor-id="experimental-designs-in-experimental-bench-biology"><span class="header-section-number">14.4</span> Experimental designs in experimental bench biology</h2>
<p>Given the basic principles above, let’s consider the kinds of experimental designs seen in experimental bench biology (Fig. @ref(lmm-designs)).</p>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../images/experimental_designs/designs_1.png" class="img-fluid figure-img"></p>
<figcaption>Experimental designs in experimental bench biology. Images created with BioRender.com</figcaption>
</figure>
</div>
</div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../images/experimental_designs/designs_2.png" class="img-fluid figure-img"></p>
<figcaption>Experimental designs in experimental bench biology. Images created with BioRender.com</figcaption>
</figure>
</div>
</div>
</div>
<section id="notation-for-models" class="level3" data-number="14.4.1">
<h3 data-number="14.4.1" class="anchored" data-anchor-id="notation-for-models"><span class="header-section-number">14.4.1</span> Notation for models</h3>
<p>i = 1..t (treatments)</p>
<p>j = 1..b (blocks)</p>
<p>k = 1..r (experimental replications within a block or within a CRD with no block structure)</p>
<p>m = 1..s (subsamples or technical replicates)</p>
</section>
<section id="completely-randomized-design-crd" class="level3" data-number="14.4.2">
<h3 data-number="14.4.2" class="anchored" data-anchor-id="completely-randomized-design-crd"><span class="header-section-number">14.4.2</span> Completely Randomized Design (CRD)</h3>
<p>The <strong>Completely Randomized Design</strong> experiment in Figure @ref(fig:lmm-designs)A has a single factor, <span class="math inline">\(\texttt{treatment}\)</span> with two levels (“Cn” and “Tr”). Five mice are randomly assigned to each treatment level. Each mouse is bred from a different litter and housed in a separate cage. The researchers measure a single value of the response variable from each mouse. The five replicate mice per treatment are the <strong>treatment (biological) replicates</strong>. The design is completely randomized because there is no subgrouping due to batches. What kinds of subgrouping does this design avoid?</p>
<p>By using a single mouse per litter, there are no litter batches and subsets of mice don’t share <strong>litter effects</strong> – common litter responses to the Cn or Tr treatments. Each litter has a unique set of factors that contribute to the error variance of the measure of the response. Siblings from the same dam and sire share more genetic variation than non-siblings and this shared genetic variation contributes to phenotypes (including the response to treatment) that are more likely to be similar to each other than to non-siblings. Siblings from the same litter share the same history of maternal factors (<strong>maternal effects</strong>, including <strong>epigenetic effects</strong>) specific to the pregnancy and even the history of events leading up to the pregnancy. This shared non-genetic and epigenetic variation contributes to phenotypes (including the response to treatment) that are more likely to be similar to each other than to non-siblings. All response measures within a litter share the genetic, maternal environmental, and epigenetic components of the error variance unique to that litter and, as a consequence, the error (residuals) within a litter are more similar to each other than they are to the residuals between litters.</p>
<p>By housing each mouse in it’s own cage, there is no cage batch and subsets of mice don’t share <strong>cage effects</strong> – common cage responses to the Cn or Tr treatments. As stated earlier, each cage has a unique set of factors that contribute to the error variance of the measure of the response. Each cage shares a cage-specific history of temperature, humidity, food, light, interactions with animal facilities staff and behavioral interactions among the mice. All response measures within a cage share the component of the error variance unique to that cage and, as a consequence, the error (residuals) within a cage are more similar to each other than they are to the residuals between cages.</p>
<p>Examples:</p>
<ol type="1">
<li>Ten mice from separate litters are sampled. Five mice are randomly assigned to control. Five mice are randomly assigned to treatment. A single measure per mouse is taken. Mouse is the experimental unit. <span class="math inline">\(t=2\)</span>, <span class="math inline">\(b=0\)</span>, <span class="math inline">\(r=5\)</span>, and <span class="math inline">\(s=1\)</span>.</li>
<li>Ten cell cultures are created. Five cultures are randomly assigned to control and five to treatment. A single measure per culture is taken. Culture is the experimental unit. <span class="math inline">\(t=2\)</span>, <span class="math inline">\(b=0\)</span>, <span class="math inline">\(r=5\)</span>, and <span class="math inline">\(s=1\)</span>.</li>
</ol>
<div class="callout callout-style-default callout-note callout-titled" title="Known Unknowns">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Known Unknowns
</div>
</div>
<div class="callout-body-container callout-body">
<p>While most data from experiments in bench biology are analyzed as if the experimental design is a CRD, a good question is, what fraction of these actually are CRD? We know that many (most?) mouse experiments will have batched measures and correlated responses because most experiments are conducted with multiple mice per litter and/or cage (or equivalents in other model systems). Many cell culture data come from multiple replicates of the whole experiment – the experiment functions as a block. And time series experiments include multiple measures on the same experimental unit over time. Except for time series experiments, researchers in experimental bench biology are using almost exclusively statistical tests that assume independence of errors (the tests appropriate for CRDs). How this mismatch between experimental design and statistical practice affects the rate of false and true discovery in cell and molecular biology is entirely unknown.</p>
</div>
</div>
</section>
<section id="completely-randomized-design-with-subsampling-crds-or-nested-design" class="level3" data-number="14.4.3">
<h3 data-number="14.4.3" class="anchored" data-anchor-id="completely-randomized-design-with-subsampling-crds-or-nested-design"><span class="header-section-number">14.4.3</span> Completely Randomized Design with Subsampling (CRDS) or “Nested Design”</h3>
<p>The <strong>Completely Randomized Design with Subsampling</strong> experiment in Fig. @ref(fig:lmm-designs)B is exactly like the CRD except that the researchers measure multiple values of the response variable from each mouse <em>under the same condition</em> (that is, not different in treatment or time). The multiple measures are <strong>subsampled (technical) replicates</strong>.</p>
<ol type="1">
<li>Do not confuse subsampled replicates with measures of the response under different conditions in the same mouse, for example a measure from one brain slice under the control treatment and a measure from a second brain slice under the drug treatment. This example is a kind of Randomized Complete Block Design, which is outlined next and the core design in this chapter.</li>
<li>Do not confuse subsampled replicates with a measures of the response at different times in the same mouse, for example, the plasma glucose levels at baseline and at five post-baseline time points. This example is a kind of longitudinal design, which is outlined below and more thoroughly in the chapter <a href="#pre-post">Linear models for longitudinal experiments</a>.</li>
<li>Do not confuse subsampled replicates with measures of different response variables from the same mouse, for example measures of the weights of five different skeletal muscles. This example is a kind of <strong>multiple response</strong> which is addressed xxx.</li>
</ol>
<p>The technical replicates are a kind of <strong>pseudoreplication</strong>. The general linear model <code>y ~ treatment</code> fit to these data, including t-tests and traditional ANOVA, will have falsely high precision and falsely low <em>p</em>-values.</p>
<p>Examples:</p>
<ol type="1">
<li>Ten mice from separate litters are sampled. Five mice are randomly assigned to control. Five mice are randomly assigned to treatment. Multiple measures per mouse are taken. Example: five measures of Islet Area are measured in each pancreas. Mouse is the experimental unit. Each of the measures is a technical replicate because the treatment is not randomly assigned to each islet but to the whole mouse. <span class="math inline">\(t=2\)</span>, <span class="math inline">\(b=0\)</span>, <span class="math inline">\(r=5\)</span>, and <span class="math inline">\(s=5\)</span>.</li>
<li>Ten cell cultures are created. Five cultures are randomly assigned to control and five to treatment. Multiple measures per culture are taken. Example: Mitochondrial counts are measured from five cells in each culture. Culture is the experimental unit. Each of the counts is a technical replicate because the treatment is not randomly assigned to each cell but to the whole culture. <span class="math inline">\(t=2\)</span>, <span class="math inline">\(b=0\)</span>, <span class="math inline">\(r=5\)</span>, and <span class="math inline">\(s=5\)</span>.</li>
</ol>
<p>Notes:</p>
<ol type="1">
<li>Subsampling can occur at multiple levels. Example: Ten mice from separate litters are sampled. Five mice are randomly assigned to control. Five mice are randomly assigned to treatment. Five neurons in a slice of brain tissue are identified in each mouse. From each neuron, the length of five dendrite spines are measured. The five measures of spine length are “nested within” neuron and the five neurons are “nested within” mouse. Nested subsampling can quickly lead to massive pseudoreplication and false discovery.</li>
</ol>
</section>
<section id="randomized-complete-block-design-rcbd" class="level3" data-number="14.4.4">
<h3 data-number="14.4.4" class="anchored" data-anchor-id="randomized-complete-block-design-rcbd"><span class="header-section-number">14.4.4</span> Randomized Complete Block Design (RCBD)</h3>
<p>The <strong>Randomized Complete Block Design</strong> experiment in Figure @ref(fig:lmm-designs)C is similar to the CRD except that all treatment combinations (two here) are randomly assigned to sibling mice within a litter. Here, two mice from each litter are randomly selected and one is randomly assigned to “Cn” and the other to “Tr”. Each litter is randomly assigned to a unique cage. The researchers measure a single value of the response variable from each mouse. The five replicate mice per treatment are the treatment replicates. The litters (or cage) are the blocks. In this design, litter and cage effects are confounded but this has no consequence on the statistical model and inference unless the researchers want to explicitly estimate these effects separately. Compared to the CRD, this design requires fewer resources (five litters instead of ten, five cages instead of ten). Compared to the general linear model fit to data from the CRD (<code>y ~ treatment</code>), including t-tests and traditional ANOVA, the linear mixed model fit to the RCBD has increased precision and power. While many researchers seem to be designing experiments similar to this (“littermate controls”), most are failing to fit a statistical model that accounts for the batching and taking advantage of the increased precision and power.</p>
<p>Examples:</p>
<ol type="1">
<li>Ten mice are sampled. In each mouse, one forelimb is assigned to control and the other forelimb is assigned to treatment. Only a single measure on each side is taken. Limb is the experimental unit. Mouse is a block. <span class="math inline">\(t=2\)</span>, <span class="math inline">\(b=10\)</span>, <span class="math inline">\(r=1\)</span>, and <span class="math inline">\(s=1\)</span>.</li>
<li>Ten litters are sampled. In each litter, one sib is assigned to control and the other sib is assigned to treatment. Only a single measure on each sib is taken. Mouse is the experimental unit. Litter is a block. <span class="math inline">\(t=2\)</span>, <span class="math inline">\(b=10\)</span>, <span class="math inline">\(r=1\)</span>, and <span class="math inline">\(s=1\)</span>.</li>
<li>Two mice, each from a separate litter are sampled. One is randomly assigned to control and the other to treatment. Only a single measure of the response variable is taken per mouse. The experiment is replicated five times (five different days, each with a newly made set of reagents and machine calibrations). Mouse is the experimental unit. Experiment is a block. <span class="math inline">\(t=2\)</span>, <span class="math inline">\(b=5\)</span>, <span class="math inline">\(r=1\)</span>, and <span class="math inline">\(s=1\)</span>.</li>
</ol>
</section>
<section id="randomized-split-plot-design-rspd" class="level3" data-number="14.4.5">
<h3 data-number="14.4.5" class="anchored" data-anchor-id="randomized-split-plot-design-rspd"><span class="header-section-number">14.4.5</span> Randomized Split Plot Design (RSPD)</h3>
<p>The <strong>Randomized Split Plot Design</strong> experiment in Figure @ref(fig:lmm-designs)D is similar to the RCBD except that there is now a second experimental factor that is crossed with the first experimental factor. An individual mouse acts as single experimental unit for one factor (here, <span class="math inline">\(\texttt{genotype}\)</span> with levels “WT” and “KO”) but acts as a block for the second experimental factor (here, <span class="math inline">\(\texttt{treatment}\)</span> with levels “Cn” and “Tr”). The first factor (<span class="math inline">\(\texttt{genotype}\)</span>) is the <strong>main plot</strong> – the levels of the factor are randomly assigned to the main plots. The second factor (<span class="math inline">\(\texttt{treatment}\)</span>) is the <strong>subplot</strong> – the levels of this factor are randomly assigned to the subplots. <span class="math inline">\(\texttt{Litter}\)</span> is a replicated block. Also in Figure @ref(fig:lmm-designs)D, a 2 x 2 RCBD with the same two experimental factors is shown for comparison. In the 2 x 2 RCBD, four mice per block (litter) are each randomly assigned one of the 2 x 2 combinations of <span class="math inline">\(\texttt{genotype}\)</span> and <span class="math inline">\(\texttt{treatment}\)</span>).</p>
</section>
<section id="generalized-randomized-complete-block-design-grcbd" class="level3" data-number="14.4.6">
<h3 data-number="14.4.6" class="anchored" data-anchor-id="generalized-randomized-complete-block-design-grcbd"><span class="header-section-number">14.4.6</span> Generalized Randomized Complete Block Design (GRCBD)</h3>
<p>The <strong>Generalized Randomized Complete Block Design</strong> experiment in Figure @ref(fig:lmm-designs)E is similar to the RCBD except that two treatment replicates per block (litter/cage) are assigned.</p>
<p><strong>Important and somewhat not intuitive</strong> Because the treatment replicates within a litter share common error variance, these do not act like independent replicates. One consequence of this is, the sample size (<span class="math inline">\(n\)</span>) is five and not ten (<span class="math inline">\(litters \times treatment replicates\)</span>). The general linear model <code>y ~ treatment</code> fit to these data, including t-tests and traditional ANOVA, will generally have falsely high precision and falsely low <em>p</em>-values.</p>
</section>
<section id="nested-randomized-complete-block-design-nrcbd" class="level3" data-number="14.4.7">
<h3 data-number="14.4.7" class="anchored" data-anchor-id="nested-randomized-complete-block-design-nrcbd"><span class="header-section-number">14.4.7</span> Nested Randomized Complete Block Design (NRCBD)</h3>
<p>The <strong>Nested Randomized Complete Block Design</strong> experiment in Figure @ref(fig:lmm-designs)F is similar to the RCBD except that there are now replicated experiments. This is a nested block design with litter (a block) nested within experiment (a block).</p>
</section>
<section id="longitudinal-randomized-complete-block-design-lrcbd" class="level3" data-number="14.4.8">
<h3 data-number="14.4.8" class="anchored" data-anchor-id="longitudinal-randomized-complete-block-design-lrcbd"><span class="header-section-number">14.4.8</span> Longitudinal Randomized Complete Block Design (LRCBD)</h3>
<p>The <strong>Longitudinal Randomized Complete Block Design</strong> experiment in Figure @ref(fig:lmm-designs)G is similar to the RCBD except that there are multiple measures of the response variable taken, each taken at a different time point, including <strong>baseline</strong> (time zero).</p>
</section>
<section id="variations-due-to-multiple-measures-of-the-response-variable" class="level3" data-number="14.4.9">
<h3 data-number="14.4.9" class="anchored" data-anchor-id="variations-due-to-multiple-measures-of-the-response-variable"><span class="header-section-number">14.4.9</span> Variations due to multiple measures of the response variable</h3>
<p>Similar to the CRDS above, the other basic designs can include subsampling, resulting in, for example, RCBDS or RSPDS. If there is subsampling within subsampled units, then we can designated these with “SS”, for example RCBDSS.</p>
<p>Similar to the LRCBD above, the other basic designs can include longitudinal sampling, resulting in, for example, LCRD or LRSPD.</p>
</section>
</section>
<section id="building-the-linear-mixed-model-for-clustered-data" class="level2" data-number="14.5">
<h2 data-number="14.5" class="anchored" data-anchor-id="building-the-linear-mixed-model-for-clustered-data"><span class="header-section-number">14.5</span> Building the linear (mixed) model for clustered data</h2>
<p>Notation</p>
<ol type="1">
<li>i = 1..t (treatments)</li>
<li>j = 1..b (blocks)</li>
<li>k = 1..r (experimental replications within a block)</li>
<li>m = 1..s (subsamples or technical replicates)</li>
</ol>
<p><span class="math display">\[y_i = \beta_0 + \beta_i treatment_i + (\gamma_j block_j) + (\gamma_{ij} block_j treatment_i) + \varepsilon\]</span></p>
</section>
<section id="lmm-demo" class="level2" data-number="14.6">
<h2 data-number="14.6" class="anchored" data-anchor-id="lmm-demo"><span class="header-section-number">14.6</span> Example 1 – A random intercepts and slopes explainer (demo1)</h2>
<p>To introduce linear mixed models, I’m using data from <a href="#lmm-example4">Experiment 1g</a> below. The design is <span class="math inline">\(2 \times 2\)</span> factorial with 4-5 mice per treatment combination. To simplify the explanation of <strong>random intercepts</strong> and <strong>random slopes</strong> in linear models with added random factors (<strong>linear mixed models</strong>), I flatten the analysis to a single treatment factor (<span class="math inline">\(\texttt{treatment}\)</span>) with four levels (“Control”, “Tr1”, “Tr2”, “Tr3). The response is percent germinal centers (<span class="math inline">\(\texttt{gc}\)</span>) in <a href="https://en.wikipedia.org/wiki/Lymphatic_system#Secondary_lymphoid_organs" target="_blank">secondary lymphoid tissue</a>. The experiment was replicated 4 times. Each replication is a batch. This batch information is in the variable <span class="math inline">\(\texttt{experiment_id}\)</span>. Further detail isn’t necessary at this point.</p>
<p>Figure @ref(fig:lmm-explainer1a)A is a response plot of the linear model <code>lm(gc ~ treatment)</code> fit to the whole data set, ignoring the fact that the data were collected in batches. This is the <strong>complete pooling</strong> fit. Figure @ref(fig:lmm-explainer1a)B is a response plot of the linear model <code>lm(gc ~ treatment)</code> fit to the means of each treatment combination from each experiment. This is the <strong>means pooling</strong> fit.</p>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="lmm_files/figure-html/lmm-explainer1a-1.png" class="img-fluid figure-img" width="1066"></p>
<figcaption>A. Response plot of the linear model gc ~ treatment fit to all exp1g data. B. Response plot of the linear model gc ~ treatment fit to the experiment means of the exp1g data.</figcaption>
</figure>
</div>
</div>
</div>
<section id="batched-measurements-result-in-clustered-residuals" class="level3" data-number="14.6.1">
<h3 data-number="14.6.1" class="anchored" data-anchor-id="batched-measurements-result-in-clustered-residuals"><span class="header-section-number">14.6.1</span> Batched measurements result in clustered residuals</h3>
<p>Figure @ref(fig:lmm-explainer1b)A is a plot of the residuals of the complete-pooling fit against <span class="math inline">\(\texttt{experiment_id}\)</span>. The residuals are <strong>clustered</strong> by experiment. All residuals from experiment 1 are positive. All residuals from experiment 2 are negative. Residuals from experiment 3 are generally positive. Residuals from experiment 4 seem pretty random. This clustering by experiment is the same in the plot of the residuals of the means-pooling fit against <span class="math inline">\(\texttt{experiment_id}\)</span> (Figure @ref(fig:lmm-explainer1b)B). The residuals are not independent in either fit. If you asked me to guess the sign of a residual and gave me the information that the measure was from experiment 1, I’d be correct 100% of the time. If the residuals were independent, I’d be correct, on average, 50% of the time. Independent residuals are randomly scattered about zero for within each experiment (Figure @ref(fig:lmm-explainer1b)C).</p>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="lmm_files/figure-html/lmm-explainer1b-1.png" class="img-fluid figure-img" width="1066"></p>
<figcaption>A. Residuals of the model fit to all demo1 data. B. Residuals of the model fit to the mean demo1 data.</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="lmm-corerr" class="level3" data-number="14.6.2">
<h3 data-number="14.6.2" class="anchored" data-anchor-id="lmm-corerr"><span class="header-section-number">14.6.2</span> Clustered residuals result in correlated error</h3>
<p>An assumption of inference from a linear model is independence – each response is independently drawn from a distribution of random values. In Experiment 1g, the experiments are batches and the batched data results in correlated error unless modeled. One way to see this correlated error is to use the residuals from the means-pooled fit.</p>
<ol type="1">
<li>aggregate the data by computing the means for each treatment level within each experiment.</li>
<li>fit the fixed effect model (the model without added random factors) to the aggregated data</li>
<li>compute the residuals from the model</li>
<li>cast (or spread) the residuals for each treatment into its own column. This creates a 4 rows (experiment) <span class="math inline">\(\times\)</span> 4 columns (treatments) matrix of residuals.</li>
<li>Compute the correlations among the four treatment combination columns. This is the correlated error due to the batch effect of <span class="math inline">\(\texttt{experiment_id}\)</span>.</li>
</ol>
<div class="cell">
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<caption>Residuals of fixed affect model fit to aggregated data. The residuals are split into each treatment.</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">experiment_id</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Control</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Tr1</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Tr2</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Tr3</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">exp_1</td>
<td style="text-align: right;">2.578125</td>
<td style="text-align: right;">1.64075</td>
<td style="text-align: right;">2.2425</td>
<td style="text-align: right;">3.027125</td>
</tr>
<tr class="even">
<td style="text-align: left;">exp_2</td>
<td style="text-align: right;">-2.229375</td>
<td style="text-align: right;">-2.91625</td>
<td style="text-align: right;">-5.7475</td>
<td style="text-align: right;">-4.435375</td>
</tr>
<tr class="odd">
<td style="text-align: left;">exp_3</td>
<td style="text-align: right;">-0.615375</td>
<td style="text-align: right;">1.09975</td>
<td style="text-align: right;">3.4625</td>
<td style="text-align: right;">3.697125</td>
</tr>
<tr class="even">
<td style="text-align: left;">exp_4</td>
<td style="text-align: right;">0.266625</td>
<td style="text-align: right;">0.17575</td>
<td style="text-align: right;">0.0425</td>
<td style="text-align: right;">-2.288875</td>
</tr>
</tbody>
</table>


</div>
</div>
<div class="cell">
<div class="cell-output cell-output-stderr">
<pre><code>Registered S3 method overwritten by 'GGally':
  method from   
  +.gg   ggplot2</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="lmm_files/figure-html/lmm-demo1-cor-error-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>I’ve used <code>GGally::ggpairs</code> to compute and display the correlations as a matrix. The lower triangle of matrix elements contains the scatterplot of the residuals for the treatment combination defined by the row and column headers. The upper triangle of elements contains the Pearson correlation. With only four experiment residuals per treatment combination, large correlations will be common. But all correlations are large, positive values. The asterisks indicate values that would be an unexpected surprise under a null model of no correlation.</p>
<p>We could explicitly model correlated error with a <strong>linear model for correlated error</strong> using the <code>nlme::gls</code> function, using a model for the correlated error that matches our knowledge of how the data were generated (from experiment batches). In this chapter, we implicitly model the correlated error using a linear model with added random factors – a <strong>linear mixed model</strong>. What we explicitly model in a linear mixed model is hierarchical levels of variance.</p>
</section>
<section id="in-blocked-designs-clustered-residuals-adds-a-variance-component-that-masks-treatment-effects" class="level3" data-number="14.6.3">
<h3 data-number="14.6.3" class="anchored" data-anchor-id="in-blocked-designs-clustered-residuals-adds-a-variance-component-that-masks-treatment-effects"><span class="header-section-number">14.6.3</span> In blocked designs, clustered residuals adds a variance component that masks treatment effects</h3>
<p>The variance among the experiments within a treatment is much greater than the variance among the treatment means. A consequence of this is, the experiment effect masks the effect of treatment. We can manually unmask this by</p>
<ol type="1">
<li>compute the experiment means across all treatment combinations.</li>
<li>create a gc variable without variation among experiment means (“adjusted for experiment_id”).</li>
</ol>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="lmm_files/figure-html/lmm-demo1-manual-lmm-plot-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption>Adjusting for variance among experiments. The black, dashed line is the grand-mean response. In the left panel, the colored, dashed lines are the mean gc for each experiment, ignoring treatment. In the right panel, the individual values have been shifted (adjusted) by centering the experiment means. This has the effect of reducing the error variance – the spread of the values around the treatment means (large black dots).</figcaption>
</figure>
</div>
</div>
</div>
<p>In Figure @ref(fig:lmm-demo1-manual-lmm-plot), the black dots are the modeled means of each treatment combination. The small colored dots are the measured values of the response for each <span class="math inline">\(\texttt{experiment_id}\)</span> in the left panel and the experiment-adjusted values in the right panel. The black, dashed line is the grand-mean response. The colored, dashed lines are the means of all responses in each experiment. These means are equal in the right panel (and covered by the black line) because the variation among the means has been adjusted away. What is left is error variation uncontaminated by <span class="math inline">\(\texttt{experiment_id}\)</span>.</p>
<p>In Experiment 1g, <span class="math inline">\(\texttt{experiment_id}\)</span> is a nuissance variable – it adds to the noise. In the exercise above, the effects of the treatment variables are adjusted for the elevation of batch effects on the overal batch mean. Linear mixed models are more sophisticated than this. In a linear mixed model, the effects of the treatment variables are adjusted for the elevations of batch effects on the intercept and batch effects on the slopes (or some combination of these). These are the <strong>random intercepts</strong> and <strong>random slopes</strong>.</p>
</section>
<section id="linear-mixed-models-are-linear-models-with-added-random-factors" class="level3" data-number="14.6.4">
<h3 data-number="14.6.4" class="anchored" data-anchor-id="linear-mixed-models-are-linear-models-with-added-random-factors"><span class="header-section-number">14.6.4</span> Linear mixed models are linear models with added random factors</h3>
<p>A linear model adds some combination of <strong>random intercepts</strong> and <strong>random slopes</strong> to a linear model.</p>
<p><span class="math display">\[
\begin{equation}
\texttt{gc}_{jk} = (\beta_{0} + \gamma_{0j}) + (\beta_{k} + \gamma_{kj}) \texttt{treatment}_{k} + \varepsilon
(\#eq:lmm-demo1-m1)
\end{equation}
\]</span></p>
<p>A <strong>random intercept</strong> for experiment <em>j</em> is the sum of the fixed intercept (<span class="math inline">\(\beta_0\)</span>) and a <strong>random intercept effect</strong> (<span class="math inline">\(\gamma_{0j}\)</span>). I’ve embedded these within parentheses to show how these combine into the random intercept. A <strong>random slope</strong> for batch <em>j</em> is the sum of the fixed slope (<span class="math inline">\(\beta_k\)</span>) for the non-reference level <span class="math inline">\(k\)</span> and a <strong>random slope effect</strong> (<span class="math inline">\(\gamma_{kj}\)</span>). I’ve embedded these within parentheses to show how these combine into the random slopes.</p>
<p>There is a different <span class="math inline">\(\gamma_{0j}\)</span> for each experiment. There is a different <span class="math inline">\(\gamma_{kj}\)</span> for each combination of non-reference level and experiment. The <span class="math inline">\(\gamma_0j\)</span> and <span class="math inline">\(\gamma_kj\)</span> are modeled as if the values for each experiment is a random draw from an infinite number of experiments. This is why <span class="math inline">\(\gamma_{0j}\)</span> and <span class="math inline">\(\gamma_{kj}\)</span> are <strong>random effects</strong>. In contrast, <span class="math inline">\(\beta_0\)</span> and the three <span class="math inline">\(\beta_k\)</span> for the non-reference treatment levels are the same for all experiments – this is why <span class="math inline">\(\beta_0\)</span> is known as <strong>fixed effects</strong> (technically, <span class="math inline">\(\beta_0\)</span> is a mean and not an effect).</p>
<p>Model @ref(eq:lmm-demo1-m1) is fit to the Example 1 data using <code>lme4::lmer()</code></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>demo1_m1 <span class="ot">&lt;-</span> <span class="fu">lmer</span>(gc <span class="sc">~</span> treatment <span class="sc">+</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>                   (treatment <span class="sc">|</span> experiment_id),</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> demo1)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Notes</p>
<ol type="1">
<li><code>(treatment | experiment_id)</code> specifies a random intercept for all levels of <span class="math inline">\(\texttt{experiment_id}\)</span> and a random slope for all combinations the levels of <span class="math inline">\(\texttt{experiment_id}\)</span> and the non-reference levels of <span class="math inline">\(\texttt{treatment}\)</span></li>
</ol>
</section>
<section id="what-the-random-effects-are" class="level3" data-number="14.6.5">
<h3 data-number="14.6.5" class="anchored" data-anchor-id="what-the-random-effects-are"><span class="header-section-number">14.6.5</span> What the random effects are</h3>
<p><strong>Random intercepts model batch effects in the reference treatment level</strong>. @ref(fig:lmm-demo1-explainer-2c)A illustrates random intercepts and random intercept effects. The large, colored dots are the modeled means of each experiment for each treatment combination. For the reference treatment level (“Control”), each mean is the sum of the estimated fixed intercept (<span class="math inline">\(b_0\)</span>), shown by the dashed gray line, and the estimated random intercept effect (<span class="math inline">\(g_{0j}\)</span>) for experiment <em>j</em>. The random intercept effects are the vertical, colored lines.</p>
<p><strong>Random slopes model the effect of treatment on batch effects in the non-reference treatment levels</strong>. @ref(fig:lmm-demo1-explainer-2c)B illustrates random slopes and random slope effects, focusing on the slopes for the 2nd non-reference treatment level (“Tr2”). The angled black line is the estimated fixed slope <span class="math inline">\(b_2\)</span> for this level. The colored lines are the random slopes for each experiment. The pale, gray dots are where the modeled means at the Tr2 level would be if there were no random slope effect – as if we took the large colored dots at “Control” and rigidly shifted them up the black line to “Tr2”. The estimated random slope effects <span class="math inline">\(\mathrm{g}_{2j}\)</span> are the difference between these large, gray dots and the modeled means.</p>
<div class="cell">
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.
ℹ Please use `linewidth` instead.</code></pre>
</div>
</div>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="lmm_files/figure-html/lmm-demo1-explainer-2c-1.png" class="img-fluid figure-img" width="1066"></p>
<figcaption>What random intercepts and slopes are. (A) A random intercept for batch <span class="math inline">\(j\)</span> is the difference between the fixed intercept and the modeled mean for batch <span class="math inline">\(j\)</span> in the reference treatment level. The random intercepts for experiments 1 (<span class="math inline">\(\mathrm{g}_{0.1}\)</span>) and 2 (<span class="math inline">\(\mathrm{g}_{0.2}\)</span>) are shown with brackets. (B) The fixed slope for the “Tr2” is illustrated with a bracket. The large, grey dots are the expected values for each batch (experiment_id) in the “Tr2” treatment if the random slope effects are zero. A random slope for batch <span class="math inline">\(j\)</span> is the difference between the expected value for batch j and the modeled mean for batch <span class="math inline">\(j\)</span>. The random slope for experiment 2 in “Tr2” (<span class="math inline">\(\mathrm{g}_{2.2}\)</span>) is shown with a bracket.</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="in-a-blocked-design-a-linear-model-with-added-random-effects-increases-precision-of-treatment-effects" class="level3" data-number="14.6.6">
<h3 data-number="14.6.6" class="anchored" data-anchor-id="in-a-blocked-design-a-linear-model-with-added-random-effects-increases-precision-of-treatment-effects"><span class="header-section-number">14.6.6</span> In a blocked design, a linear model with added random effects increases precision of treatment effects</h3>
<p><span class="math display">\[
\begin{equation}
\texttt{gc}_{jk} = \beta_{0} + \beta_{k} \texttt{treatment}_{k} +  (\gamma_{0j} + \gamma_{kj}\texttt{treatment}_{k} + \varepsilon)
(\#eq:lmm-demo1-m0)
\end{equation}
\]</span></p>
<p>If the random intercepts and random slopes aren’t modeled, this among-experiment variance is shifted to the error variance because the intercept effects and slope effects aren’t estimated but absorbed by the error – everything in Model @ref(eq:lmm-demo1-m0) will be estimated by the residuals. As a consequence, the estimate of <span class="math inline">\(\sigma\)</span> (the square root of the error variance) for the linear mixed model is smaller than that for the linear model with only fixed effects.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># sigma for the lmm</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>m1 <span class="ot">&lt;-</span> <span class="fu">lmer</span>(gc <span class="sc">~</span> treatment <span class="sc">+</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>                   (treatment <span class="sc">|</span> experiment_id),</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> demo1)</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(m1)<span class="sc">$</span>sigma</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 1.931356</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># sigma for the fixed lm</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>m2 <span class="ot">&lt;-</span> <span class="fu">lm</span>(gc <span class="sc">~</span> treatment,</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> demo1)</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(m2)<span class="sc">$</span>sigma</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 3.376402</code></pre>
</div>
</div>
<p>The consequence of the smaller estimate of <span class="math inline">\(\sigma\)</span> in the linear mixed model on inference (confidence intervals and <em>p</em>-values) depends on the number of subsamples, the <strong>variance of the random effects</strong> relative to the variance of the residual error, and the <strong>correlation among the random effects</strong>.</p>
</section>
<section id="lmm-varcorr" class="level3" data-number="14.6.7">
<h3 data-number="14.6.7" class="anchored" data-anchor-id="lmm-varcorr"><span class="header-section-number">14.6.7</span> The correlation among random intercepts and slopes</h3>
<p>Again, here is the linear mixed model fit to the experiment 1g data.</p>
<p><span class="math display">\[
\begin{equation}
\texttt{gc}_{jk} = (\beta_{0} + \gamma_{0j}) + (\beta_{k} + \gamma_{kj}) \texttt{treatment}_{k} + \varepsilon
\end{equation}
\]</span></p>
<p>Think about how this model <strong>generates data</strong>. We have four experiments, so we randomly draw four <span class="math inline">\(\gamma_{0j}\)</span> from a normal distribution with some variance <span class="math inline">\(\sigma_{0}^2\)</span>. And, for each non-reference treatment, we randomly draw four <span class="math inline">\(\gamma_{0k}\)</span> from a normal distribution with some variance <span class="math inline">\(\sigma_{k}^2\)</span>. This gives us a matrix of four columns (one random intercept and three random slopes) and four rows (four experiments).</p>
<div class="cell">
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">intercept</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">slope 1</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">slope 2</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">slope 3</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">$\gamma_{0.1}$</td>
<td style="text-align: left;">$\gamma_{1.1}$</td>
<td style="text-align: left;">$\gamma_{2.1}$</td>
<td style="text-align: left;">$\gamma_{3.1}$</td>
</tr>
<tr class="even">
<td style="text-align: left;">$\gamma_{0.2}$</td>
<td style="text-align: left;">$\gamma_{1.2}$</td>
<td style="text-align: left;">$\gamma_{2.2}$</td>
<td style="text-align: left;">$\gamma_{3.2}$</td>
</tr>
<tr class="odd">
<td style="text-align: left;">$\gamma_{0.3}$</td>
<td style="text-align: left;">$\gamma_{1.3}$</td>
<td style="text-align: left;">$\gamma_{2.3}$</td>
<td style="text-align: left;">$\gamma_{3.3}$</td>
</tr>
<tr class="even">
<td style="text-align: left;">$\gamma_{0.4}$</td>
<td style="text-align: left;">$\gamma_{1.4}$</td>
<td style="text-align: left;">$\gamma_{2.4}$</td>
<td style="text-align: left;">$\gamma_{3.4}$</td>
</tr>
</tbody>
</table>


</div>
</div>
<p>To randomly sample these values, the model needs not only the variances (<span class="math inline">\(\sigma_k^2\)</span>) for each column (random effect) <em>but also</em> a correlation for each pair of columns. These correlations are the off-diagonal elements of the correlation matrix of random effects.</p>
<div class="cell">
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">1</td>
<td style="text-align: left;">COR($\gamma_{0}$, $\gamma_{1}$)</td>
<td style="text-align: left;">COR($\gamma_{0}$, $\gamma_{2}$)</td>
<td style="text-align: left;">COR($\gamma_{0}$, $\gamma_{3}$)</td>
</tr>
<tr class="even">
<td style="text-align: left;">COR($\gamma_{1}$, $\gamma_{0}$)</td>
<td style="text-align: left;">1</td>
<td style="text-align: left;">COR($\gamma_{1}$, $\gamma_{2}$)</td>
<td style="text-align: left;">COR($\gamma_{1}$, $\gamma_{3}$)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">COR($\gamma_{2}$, $\gamma_{0}$)</td>
<td style="text-align: left;">COR($\gamma_{2}$, $\gamma_{1}$)</td>
<td style="text-align: left;">1</td>
<td style="text-align: left;">COR($\gamma_{2}$, $\gamma_{3}$)</td>
</tr>
<tr class="even">
<td style="text-align: left;">COR($\gamma_{3}$, $\gamma_{0}$)</td>
<td style="text-align: left;">COR($\gamma_{3}$, $\gamma_{1}$)</td>
<td style="text-align: left;">COR($\gamma_{3}$, $\gamma_{2}$)</td>
<td style="text-align: left;">1</td>
</tr>
</tbody>
</table>


</div>
</div>
<p>In the models fit in this text, a researcher doesn’t specify these variances and correlations. Instead, these are parameters estimated by the model. Here is a summary of the estimates of the variances of the random effects and of the correlations among the random effects for the linear mixed model fit to the experiment 1g data.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code> Groups        Name         Std.Dev. Corr                
 experiment_id (Intercept)  1.77054                      
               treatmentTr1 0.96417  -0.025              
               treatmentTr2 2.90269   0.459  0.876       
               treatmentTr3 2.99109   0.300  0.879  0.938
 Residual                   1.93136                      </code></pre>
</div>
</div>
<p>The first four values in the column “Std.Dev.” are the square roots of the estimated variances for the random effects given in the column “Name”. The last value in column “Std.Dev.” is the square roots of the estimate of <span class="math inline">\(\sigma^2\)</span> (the error variance). The (lower) triangular matrix of values under “Corr” are the estimates of the correlations among the random effects. The variance of the random effects and the correlation among the random effects creates the <a href="#lmm-corerr">correlated error described above</a> but do not confuse these different correlations (this is easy to confuse, you are not alone).</p>
<p>A compact way to view these variances and correlations is a matrix with the random effect standard deviations on the diagonal and the correlations on the off-diagonal. I’ll refer to this as the <em>VarCorr</em> matrix after the <code>lme4</code> function used to get the values.</p>
<div class="cell">
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<caption>The Varcorr matrix. Standard deviations of random effects on the diagonal. Correlations of random effects on the off-diagonal.</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: right;" data-quarto-table-cell-role="th"></th>
<th style="text-align: right;" data-quarto-table-cell-role="th"></th>
<th style="text-align: right;" data-quarto-table-cell-role="th"></th>
<th style="text-align: right;" data-quarto-table-cell-role="th"></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">(Intercept)</td>
<td style="text-align: right;">1.77</td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
</tr>
<tr class="even">
<td style="text-align: left;">treatmentTr1</td>
<td style="text-align: right;">-0.02</td>
<td style="text-align: right;">0.96</td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;">treatmentTr2</td>
<td style="text-align: right;">0.46</td>
<td style="text-align: right;">0.88</td>
<td style="text-align: right;">2.90</td>
<td style="text-align: right;"></td>
</tr>
<tr class="even">
<td style="text-align: left;">treatmentTr3</td>
<td style="text-align: right;">0.30</td>
<td style="text-align: right;">0.88</td>
<td style="text-align: right;">0.94</td>
<td style="text-align: right;">2.99</td>
</tr>
</tbody>
</table>


</div>
</div>
<p>It’s probably not worth trying to understand the experimental reason underneath the correlations among the random effects. But, researchers might want to sleuth out why a lab is getting high random intercept and slope variances, relative to the error variances, as these could indicate potential sources of improvement in lab protocols.</p>
</section>
<section id="clustered-residuals-create-heterogeneity-among-treatments" class="level3" data-number="14.6.8">
<h3 data-number="14.6.8" class="anchored" data-anchor-id="clustered-residuals-create-heterogeneity-among-treatments"><span class="header-section-number">14.6.8</span> Clustered residuals create heterogeneity among treatments</h3>
<p>The variances of the four treatments are</p>
<div class="cell">
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">treatment</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Var</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Control</td>
<td style="text-align: right;">7.8</td>
</tr>
<tr class="even">
<td style="text-align: left;">Tr1</td>
<td style="text-align: right;">5.2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Tr2</td>
<td style="text-align: right;">16.2</td>
</tr>
<tr class="even">
<td style="text-align: left;">Tr3</td>
<td style="text-align: right;">16.7</td>
</tr>
</tbody>
</table>


</div>
</div>
<p>In chapter xxx, I stated that heterogeneity of variances can arise because of clustered data. Why does clustered data generate heterogeneity? Let’s keep peeking at the linear mixed model fit to the experiment 1g data.</p>
<p><span class="math display">\[
\begin{equation}
\texttt{gc}_{jk} = (\beta_{0} + \gamma_{0j}) + (\beta_{k} + \gamma_{kj}) \texttt{treatment}_{k} + \varepsilon
\end{equation}
\]</span></p>
<p>In a linear model with fixed effects only, the expected variance for any treatment for any treatment is <span class="math inline">\(\sigma^2\)</span>. But if the data are batched, the expected variances include components due to the batch and these batch components depend on the treatment. This creates heterogeneity.</p>
<p>To understand this, first some rules of expected variance. The random variable <span class="math inline">\(\texttt{C}\)</span> is the sum of two random variables <span class="math inline">\(\texttt{A}\)</span> and <span class="math inline">\(\texttt{B}\)</span>. The variances of these variables are <span class="math inline">\(\sigma_{C}^2\)</span>, <span class="math inline">\(\sigma_{A}^2\)</span>, and <span class="math inline">\(\sigma_{B}^2\)</span>.</p>
<ol type="1">
<li>The expected variance of <span class="math inline">\(\texttt{C}\)</span> if <span class="math inline">\(\texttt{A}\)</span> and <span class="math inline">\(\texttt{B}\)</span> are independent (uncorrelated) is <span class="math inline">\(\sigma_{C}^2 = \sigma_{A}^2 + \sigma_{B}^2\)</span> (<a href="https://en.wikipedia.org/wiki/Pythagorean_theorem" target="_blank">this equation should look familiar</a>).</li>
<li>The expected variance of <span class="math inline">\(\texttt{C}\)</span> if <span class="math inline">\(\texttt{A}\)</span> and <span class="math inline">\(\texttt{B}\)</span> are not independent (correlated) is <span class="math inline">\(\sigma_{C}^2 = \sigma_{A}^2 + \sigma_{B}^2 + 2\sigma_{A}\ \sigma_{B}\ \rho_{A,B}\)</span> where <span class="math inline">\(\rho_{A,B}\)</span> is the expected correlation between <span class="math inline">\(\texttt{A}\)</span> and <span class="math inline">\(\texttt{B}\)</span> (<a href="https://en.wikipedia.org/wiki/Law_of_cosines" target="_blank">this equation might also look familiar</a>).</li>
</ol>
<p>Here is some code to better know expected variances of the sum of two correlated random variables.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># copy, paste, and explore</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>n <span class="ot">&lt;-</span> <span class="dv">10</span><span class="sc">^</span><span class="dv">4</span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>rho <span class="ot">&lt;-</span> <span class="fl">0.6</span> <span class="co"># change this to any value between -1 and 1</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>b <span class="ot">&lt;-</span> <span class="fu">sqrt</span>(<span class="fu">abs</span>(rho))</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>z <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(n)</span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>A <span class="ot">&lt;-</span> b<span class="sc">*</span>z <span class="sc">+</span> <span class="fu">sqrt</span>(<span class="dv">1</span><span class="sc">-</span>b<span class="sc">^</span><span class="dv">2</span>)<span class="sc">*</span><span class="fu">rnorm</span>(n)</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>B <span class="ot">&lt;-</span> <span class="fu">sign</span>(rho)<span class="sc">*</span>b<span class="sc">*</span>z <span class="sc">+</span> <span class="fu">sqrt</span>(<span class="dv">1</span><span class="sc">-</span>b<span class="sc">^</span><span class="dv">2</span>)<span class="sc">*</span><span class="fu">rnorm</span>(n)</span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a><span class="fu">cor</span>(A,B) <span class="co"># should be close to rho</span></span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>C <span class="ot">&lt;-</span> A <span class="sc">+</span> B</span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a><span class="fu">sd</span>(A)<span class="sc">^</span><span class="dv">2</span> <span class="co"># should be close to 1</span></span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a><span class="fu">sd</span>(B)<span class="sc">^</span><span class="dv">2</span> <span class="co"># should be close to 1</span></span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a><span class="fu">sd</span>(C)<span class="sc">^</span><span class="dv">2</span> <span class="co"># should be close to 1^2 + 1^2 + 2*1*1*rho</span></span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a><span class="fu">sd</span>(A)<span class="sc">^</span><span class="dv">2</span> <span class="sc">+</span> <span class="fu">sd</span>(B)<span class="sc">^</span><span class="dv">2</span> <span class="sc">+</span> <span class="dv">2</span><span class="sc">*</span><span class="fu">sd</span>(A)<span class="sc">*</span><span class="fu">sd</span>(B)<span class="sc">*</span><span class="fu">cor</span>(A,B) <span class="co"># should equal previous line</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Using these rules and the standard deviations of the random effects given above we can compute the expected variances of the treatment groups given the fit model.</p>
<ol type="1">
<li>For the variance of the reference (“Control”) group, we need to add to <span class="math inline">\(\sigma^2\)</span> the variance of the random intercept using rule #1 (the residuals are not correlated with random intercepts or slopes). The modeled variance is a less than the actual variance of the Control group.</li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(m1)<span class="sc">$</span>sigma<span class="sc">^</span><span class="dv">2</span> <span class="sc">+</span> <span class="fl">1.7710</span><span class="sc">^</span><span class="dv">2</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 6.866577</code></pre>
</div>
</div>
<ol start="2" type="1">
<li>For the variance of a non-reference group, we need to add to <span class="math inline">\(\sigma^2\)</span> the variance of the random intercept and the variance of the random slope for the treatment and the component due to the correlation between the random slope for the treatment and the random intercept. For Tr2, this is</li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co"># error + intercept + slope + cor(intercept, slope)</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>(<span class="fu">summary</span>(m1)<span class="sc">$</span>sigma<span class="sc">^</span><span class="dv">2</span>) <span class="sc">+</span> (<span class="fl">1.7710</span><span class="sc">^</span><span class="dv">2</span>) <span class="sc">+</span> (<span class="fl">2.9026</span><span class="sc">^</span><span class="dv">2</span>) <span class="sc">+</span> (<span class="dv">2</span> <span class="sc">*</span> <span class="fl">1.7710</span> <span class="sc">*</span> <span class="fl">2.9026</span> <span class="sc">*</span> <span class="fl">0.458</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 20.00037</code></pre>
</div>
</div>
<p>which is a bit higher than the measured variance.</p>
</section>
<section id="linear-mixed-models-are-flexible" class="level3" data-number="14.6.9">
<h3 data-number="14.6.9" class="anchored" data-anchor-id="linear-mixed-models-are-flexible"><span class="header-section-number">14.6.9</span> Linear mixed models are flexible</h3>
<p>One more look at the linear mixed model fit to the experiment 1g data.</p>
<p><span class="math display">\[
\begin{equation}
\texttt{gc}_{jk} = (\beta_{0} + \gamma_{0j}) + (\beta_{k} + \gamma_{kj}) \texttt{treatment}_{k} + \varepsilon
\end{equation}
\]</span></p>
<p>The linear mixed model specifies both a random intercept and a random slope but a researcher might limit the model to the random intercept only, or less commonly, the random slope only. Or a researcher might replace the random slope with a second random intercept that captures variance in the batch by treatment combinations like a random slope. Or a researcher might model the structure (correlated error and heterogeneity of variances) in the residuals <em>in addition to</em> adding random factors to the model.</p>
</section>
<section id="a-random-intercept-only-model" class="level3" data-number="14.6.10">
<h3 data-number="14.6.10" class="anchored" data-anchor-id="a-random-intercept-only-model"><span class="header-section-number">14.6.10</span> A random intercept only model</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>demo1_m2 <span class="ot">&lt;-</span> <span class="fu">lmer</span>(gc <span class="sc">~</span> treatment <span class="sc">+</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> experiment_id),</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> demo1)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Notes</p>
<ol type="1">
<li>Model <code>demo1_m2</code> specifies only a random intercept for each of the levels of <span class="math inline">\(\texttt{experiment_id}\)</span>. The exclusion of the random slopes ia a kind of <strong>model simplification</strong>.</li>
<li>In experiments without subsampling, random slopes cannot be added to the model because there is no variation with a treatment by batch combination.</li>
<li>In experiments with subsampling, a researcher might exclude a random slope term for several reasons, including</li>
</ol>
<ul>
<li>it is the culture in many subfields to only include a random intercept (no, this is not a good reason)</li>
<li>the computation of the model fit returned a convergence warning</li>
<li>model comparison suggested that a model with the random slope was too complex given the data. A useful statistic for comparing models with different random effects specifications is the <strong>AIC</strong>, which is introduced in Section @ref(lmm-demo-aic) below.</li>
</ul>
</section>
<section id="a-model-including-an-interaction-intercept" class="level3" data-number="14.6.11">
<h3 data-number="14.6.11" class="anchored" data-anchor-id="a-model-including-an-interaction-intercept"><span class="header-section-number">14.6.11</span> A model including an interaction intercept</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>demo1_m3 <span class="ot">&lt;-</span> <span class="fu">lmer</span>(gc <span class="sc">~</span> treatment <span class="sc">+</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> experiment_id) <span class="sc">+</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> experiment_id<span class="sc">:</span>treatment),</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> demo1)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Notes</p>
<ol type="1">
<li><code>(1 | experiment_id:treatment)</code> models a random intercept for all combinations of the levels of <span class="math inline">\(\texttt{experiment_id}\)</span> and <span class="math inline">\(\texttt{treatment}\)</span>. This <strong>interaction intercept</strong> is an alternative to a random slope for modeling treatment-specific batch effects.</li>
</ol>
</section>
<section id="lmm-demo-aic" class="level3" data-number="14.6.12">
<h3 data-number="14.6.12" class="anchored" data-anchor-id="lmm-demo-aic"><span class="header-section-number">14.6.12</span> AIC and model selection – which model to report?</h3>
<p>Three different linear mixed models were fit to the Example 1 data: <code>demo1_m1</code>, <code>demo1_m2</code>, and <code>demo1_m3</code>. Which model do we report? A useful stastistic for this decision is a statistic known as the AIC.</p>
<div class="cell">
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<caption>AIC of the two alternative models fit to Example 1 data</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">Model</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">AIC</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">demo1_m1</td>
<td style="text-align: right;">344.1529</td>
</tr>
<tr class="even">
<td style="text-align: left;">demo1_m2</td>
<td style="text-align: right;">344.0765</td>
</tr>
<tr class="odd">
<td style="text-align: left;">demo1_m3</td>
<td style="text-align: right;">337.5332</td>
</tr>
</tbody>
</table>


</div>
</div>
<p>Notes</p>
<ol type="1">
<li>AIC (<a href="https://en.wikipedia.org/wiki/Akaike_information_criterion" target="_blank">Akaike Information Criterion</a>) is a relative measure of model quality. Compare this to <span class="math inline">\(R^2\)</span>, which is an absolute measure of goodness of fit. The AIC formula has two parts, one is a kind of goodness of fit (like <span class="math inline">\(R^2\)</span>) and the other is a penalty based on the number of parameters in the model. As the goodness of fit increases, the AIC goes down. As the number of parameters increases, the AIC goes up. The model with the lowest AIC is the highest quality model. The actual value of AIC, unlike <span class="math inline">\(R^2\)</span>, does not have any absolute meaning; it is only meaningful relative to the AICs computed from fits to different models to the same data.</li>
<li>The AIC of the three models suggests that we report Model <code>demo1_m3</code>.</li>
</ol>
</section>
<section id="the-specification-of-random-effects-matters" class="level3" data-number="14.6.13">
<h3 data-number="14.6.13" class="anchored" data-anchor-id="the-specification-of-random-effects-matters"><span class="header-section-number">14.6.13</span> The specification of random effects matters</h3>
<p>Inference will often be very different between these two models as they make very different</p>
<div class="cell">
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<caption>Contrasts from the three linear mixed models fit to the Example 1 data.</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">contrast</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">estimate</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">SE</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">df</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">lower.CL</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">upper.CL</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">t.ratio</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">p.value</th>
</tr>
</thead>
<tbody>
<tr class="odd" data-grouplength="3">
<td colspan="8" style="border-bottom: 1px solid"><strong>intercepts + slopes (m1)</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Tr1 - Control</td>
<td style="text-align: right;">2.54</td>
<td style="text-align: right;">0.804</td>
<td style="text-align: right;">2.978</td>
<td style="text-align: right;">-0.03</td>
<td style="text-align: right;">5.11</td>
<td style="text-align: right;">3.16</td>
<td style="text-align: right;">0.05150</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Tr2 - Control</td>
<td style="text-align: right;">5.61</td>
<td style="text-align: right;">1.591</td>
<td style="text-align: right;">2.997</td>
<td style="text-align: right;">0.54</td>
<td style="text-align: right;">10.67</td>
<td style="text-align: right;">3.52</td>
<td style="text-align: right;">0.03886</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Tr3 - Control</td>
<td style="text-align: right;">2.85</td>
<td style="text-align: right;">1.631</td>
<td style="text-align: right;">2.997</td>
<td style="text-align: right;">-2.35</td>
<td style="text-align: right;">8.04</td>
<td style="text-align: right;">1.74</td>
<td style="text-align: right;">0.17950</td>
</tr>
<tr class="odd" data-grouplength="3">
<td colspan="8" style="border-bottom: 1px solid"><strong>intercepts only (m2)</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Tr1 - Control</td>
<td style="text-align: right;">2.64</td>
<td style="text-align: right;">0.755</td>
<td style="text-align: right;">66.005</td>
<td style="text-align: right;">1.13</td>
<td style="text-align: right;">4.15</td>
<td style="text-align: right;">3.49</td>
<td style="text-align: right;">0.00086</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Tr2 - Control</td>
<td style="text-align: right;">5.65</td>
<td style="text-align: right;">0.766</td>
<td style="text-align: right;">66.014</td>
<td style="text-align: right;">4.12</td>
<td style="text-align: right;">7.18</td>
<td style="text-align: right;">7.38</td>
<td style="text-align: right;">0.00000</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Tr3 - Control</td>
<td style="text-align: right;">2.94</td>
<td style="text-align: right;">0.765</td>
<td style="text-align: right;">66.000</td>
<td style="text-align: right;">1.41</td>
<td style="text-align: right;">4.47</td>
<td style="text-align: right;">3.85</td>
<td style="text-align: right;">0.00027</td>
</tr>
<tr class="odd" data-grouplength="3">
<td colspan="8" style="border-bottom: 1px solid"><strong>intercept + interaction intercept (m3)</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Tr1 - Control</td>
<td style="text-align: right;">2.54</td>
<td style="text-align: right;">1.214</td>
<td style="text-align: right;">8.912</td>
<td style="text-align: right;">-0.21</td>
<td style="text-align: right;">5.29</td>
<td style="text-align: right;">2.09</td>
<td style="text-align: right;">0.06616</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Tr2 - Control</td>
<td style="text-align: right;">5.60</td>
<td style="text-align: right;">1.219</td>
<td style="text-align: right;">9.061</td>
<td style="text-align: right;">2.84</td>
<td style="text-align: right;">8.35</td>
<td style="text-align: right;">4.59</td>
<td style="text-align: right;">0.00128</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Tr3 - Control</td>
<td style="text-align: right;">2.87</td>
<td style="text-align: right;">1.219</td>
<td style="text-align: right;">9.052</td>
<td style="text-align: right;">0.12</td>
<td style="text-align: right;">5.63</td>
<td style="text-align: right;">2.36</td>
<td style="text-align: right;">0.04269</td>
</tr>
</tbody>
</table>


</div>
</div>
<p>Notes</p>
<ol type="1">
<li>Inference from the intercept only model (<code>demo1_m2</code>) is almost certainly too optimistic based on simulations that show that intercept only models can be highly anti-conservative (too narrow confidence intervals and too small <em>p</em>-values).</li>
<li>Humans have evolved to make up rational explanations – do not convince yourself that the model with the smallest <em>p</em>-values is the scientifically most rational model.</li>
</ol>
</section>
<section id="mixed-effect-and-repeated-measures-anova" class="level3" data-number="14.6.14">
<h3 data-number="14.6.14" class="anchored" data-anchor-id="mixed-effect-and-repeated-measures-anova"><span class="header-section-number">14.6.14</span> Mixed Effect and Repeated Measures ANOVA</h3>
<p>Two-way mixed-effect ANOVA (some fields would call this a repeated measures ANOVA) is equivalent to Model <code>demo1_m3</code> <em>in this case</em>. More generally, the two are equal in balanced designs – the same number of subsamples in all treatment x batch combinations.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>demo1_m4 <span class="ot">&lt;-</span> <span class="fu">aov_4</span>(gc <span class="sc">~</span> treatment <span class="sc">+</span> </span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>                   (treatment <span class="sc">|</span> experiment_id),</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> demo1)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Notes 1. The function <code>afex::aov4</code> is used for specifying ANOVA models that are special cases of linear mixed models. 2. The model formula in Model <code>demo1_m2</code> looks exactly like that the random intercepts and slopes model (Model <code>demo1_m1</code>) <em>but these are not the same</em>. 3. One can use either a univariate or multivariate model for mixed or repeated measures ANOVA – see section @ref(lmm-exp6g-rmanova-multi).</p>
<div class="cell">
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<caption>Contrasts from the two-way mixed effect ANOVA model demo1_m4.</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">contrast</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">estimate</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">SE</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">df</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">lower.CL</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">upper.CL</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">t.ratio</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">p.value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Tr1 - Control</td>
<td style="text-align: right;">2.50</td>
<td style="text-align: right;">0.599</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">0.60</td>
<td style="text-align: right;">4.41</td>
<td style="text-align: right;">4.18</td>
<td style="text-align: right;">0.02488</td>
</tr>
<tr class="even">
<td style="text-align: left;">Tr2 - Control</td>
<td style="text-align: right;">5.58</td>
<td style="text-align: right;">1.559</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">0.62</td>
<td style="text-align: right;">10.54</td>
<td style="text-align: right;">3.58</td>
<td style="text-align: right;">0.03734</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Tr3 - Control</td>
<td style="text-align: right;">2.84</td>
<td style="text-align: right;">1.586</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">-2.20</td>
<td style="text-align: right;">7.89</td>
<td style="text-align: right;">1.79</td>
<td style="text-align: right;">0.17095</td>
</tr>
</tbody>
</table>


</div>
</div>
</section>
<section id="pseudoreplication" class="level3" data-number="14.6.15">
<h3 data-number="14.6.15" class="anchored" data-anchor-id="pseudoreplication"><span class="header-section-number">14.6.15</span> Pseudoreplication</h3>
</section>
</section>
<section id="lmm-example2" class="level2" data-number="14.7">
<h2 data-number="14.7" class="anchored" data-anchor-id="lmm-example2"><span class="header-section-number">14.7</span> Example 2 – experiments without subsampling replication (exp6g)</h2>
<p>This example introduces linear mixed models for batches that contain all treatment levels of a single factor but no subsampling replication. In this example, the batch is the individual mouse (<span class="math inline">\(\texttt{mouse_id}\)</span>). There are four measures of the response variable on each mouse, one measure per treatment level. When there is no subsampling replication, we cannot add a random slope to the model because there is only a single observation at each treatment level and a slope would fit the point at the reference level and the point at the non-reference level perfectly. However, we can explicitly model variation in the correlated error and heterogeneity in the variances among treatments as an alternative to modeling a random slope.</p>
<p><a href="https://www.nature.com/articles/s41586-020-2388-4" target="_blank">Reversing a model of Parkinson’s disease with in situ converted nigral neurons</a></p>
<p><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/7521455/" target="_blank">Public source</a></p>
<p>Source figure: <a href="https://www.nature.com/articles/s41586-020-2388-4#Fig6" target="_blank">Fig. 6g</a></p>
<p>Source data: <a href="https://www.nature.com/articles/s41586-020-2388-4#MOESM11" target="_blank">Source Data Fig. 6</a></p>
<section id="understand-the-data" class="level3" data-number="14.7.1">
<h3 data-number="14.7.1" class="anchored" data-anchor-id="understand-the-data"><span class="header-section-number">14.7.1</span> Understand the data</h3>
<p>In this study, the researchers investigated the effectiveness of knocking down the protein <a href="https://en.wikipedia.org/wiki/PTBP1" target="_blank">PTBP1</a> to induce astrocytes to convert to neurons in a motor processing region of the brain. Experimental lesions of this region of the brain is a model of <a href="https://en.wikipedia.org/wiki/Parkinson%27s_disease" target="_blank">Parkinson’s disease</a>. In Experiment 6g, the researchers</p>
<ol type="1">
<li>Generated a lesion in the motor processing region using 6-hydroxydopamine (6-OHDA). The lesion disrupts the ability to control the <strong>contralateral</strong> (opposite side) forelimb.</li>
<li>One month after the lesion, measured the percent of ipsilateral (same side) forepaw touches (the forelimb extending out and touching the surface) in a test of exploration in a new environment (the “cylinder test”). The expected percent in an intact mouse is 50%. In a lesioned mouse, the percent should be much greater than 50% since there is less control of the contralateral limb. The measure at this point is in the treatment “Lesion”. This is the positive control.</li>
<li>Converted astrocytes in the lesion to functional neurons by knocking down PTBP1.</li>
<li>Two months after knockdown, gave the mouse saline and remeasured percent ipsilateral touches in a cylinder test. If the knockdown worked as expected, there should be closer to 50% ipsilateral touches. The measure at this point is in the treatment “Saline”. The comparison with Lesion is a focal test.</li>
<li>Inhibited neuron action in the converted neurons using clozapine-N-oxide (CNO), which suppresses neuron electrical activity. Then, remeasured percent ipsilateral touches in a cylinder test. If the CNO worked as expected, there should be much greater than 50% ipsilateral touches since there should be re-loss of control of the contralateral limb. The measure at this point is in the treatment “CNO”. The comparison with Saline is a focal test.</li>
<li>Allowed three days for the CNO to degrade, then, remeasured percent ipsilateral touches in a cylinder test. If the CNO degraded as expected, the converted neurons should be functional and there should be closer to 50% ipsilateral touches. The measure at this point is in the treatment “Post_CNO”. The comparison with CNO is a focal test.</li>
</ol>
<p>The design is <span class="math inline">\(4 \times 1\)</span> – a single treatment with four levels (“Lesion”, “Saline”, “CNO”, “Post_CNO”)</p>
<p>The planned contrasts are</p>
<ol type="1">
<li>Saline - Lesion. This measures the effect of the knockdown and conversion of astrocytes to functional neurons.</li>
<li>CNO - Saline. This measures the effect of inhibiting the converted neurons to test if it was these and not some other neurons that account for the effect in contrast 1.</li>
<li>Post_CNO - CNO. This is probing the same expectation as contrast 2.</li>
</ol>
</section>
<section id="model-fit-and-inference" class="level3" data-number="14.7.2">
<h3 data-number="14.7.2" class="anchored" data-anchor-id="model-fit-and-inference"><span class="header-section-number">14.7.2</span> Model fit and inference</h3>
<section id="fit-the-model" class="level4" data-number="14.7.2.1">
<h4 data-number="14.7.2.1" class="anchored" data-anchor-id="fit-the-model"><span class="header-section-number">14.7.2.1</span> Fit the model</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>exp6g_m1a <span class="ot">&lt;-</span> <span class="fu">lmer</span>(touch <span class="sc">~</span> treatment <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>mouse_id), <span class="at">data =</span> exp6g)</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a><span class="co"># alt model</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>exp6g_m1b <span class="ot">&lt;-</span> <span class="fu">lme</span>(touch <span class="sc">~</span> treatment,</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>                <span class="at">random =</span> <span class="sc">~</span><span class="dv">1</span><span class="sc">|</span>mouse_id,</span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>                <span class="at">correlation =</span> <span class="fu">corSymm</span>(<span class="at">form =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> mouse_id),</span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>                <span class="at">weights =</span> <span class="fu">varIdent</span>(<span class="at">form =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> treatment),</span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>                <span class="at">data =</span> exp6g)</span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a><span class="fu">AIC</span>(exp6g_m1a, exp6g_m1b)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in AIC.default(exp6g_m1a, exp6g_m1b): models are not all fitted to the
same number of observations</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>          df      AIC
exp6g_m1a  6 191.3367
exp6g_m1b 15 198.3888</code></pre>
</div>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co"># report model a</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>exp6g_m1 <span class="ot">&lt;-</span> exp6g_m1a</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>exp6g_m1b overparameterizes, report exp6g_m1a (see <a href="#lmm-exp6g-alt">Alternative models for exp6g</a> below)</p>
</section>
<section id="inference-from-the-model" class="level4" data-number="14.7.2.2">
<h4 data-number="14.7.2.2" class="anchored" data-anchor-id="inference-from-the-model"><span class="header-section-number">14.7.2.2</span> Inference from the model</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>exp6g_m1_coef <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">coef</span>(<span class="fu">summary</span>(exp6g_m1)),</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">confint</span>(exp6g_m1)[<span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>),])</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a><span class="co"># exp5c_m1_coef %&gt;%</span></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a><span class="co">#   kable(digits = c(2,3,1,1,4,2,2)) %&gt;%</span></span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a><span class="co">#   kable_styling()</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>exp6g_m1_emm <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(exp6g_m1, <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">"treatment"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">treatment</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">emmean</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">SE</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">df</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">lower.CL</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">upper.CL</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Lesion</td>
<td style="text-align: right;">83.3</td>
<td style="text-align: right;">3.44</td>
<td style="text-align: right;">18.9</td>
<td style="text-align: right;">76.1</td>
<td style="text-align: right;">90.47</td>
</tr>
<tr class="even">
<td style="text-align: left;">Saline</td>
<td style="text-align: right;">62.8</td>
<td style="text-align: right;">3.44</td>
<td style="text-align: right;">18.9</td>
<td style="text-align: right;">55.5</td>
<td style="text-align: right;">69.96</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CNO</td>
<td style="text-align: right;">81.4</td>
<td style="text-align: right;">3.44</td>
<td style="text-align: right;">18.9</td>
<td style="text-align: right;">74.2</td>
<td style="text-align: right;">88.63</td>
</tr>
<tr class="even">
<td style="text-align: left;">Post_CNO</td>
<td style="text-align: right;">58.8</td>
<td style="text-align: right;">3.44</td>
<td style="text-align: right;">18.9</td>
<td style="text-align: right;">51.6</td>
<td style="text-align: right;">66.00</td>
</tr>
</tbody>
</table>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="co"># exp6g_m1_emm # print in console to get row numbers</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a><span class="co"># set the mean as the row number from the emmeans table</span></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>lesion <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>)</span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>saline <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">0</span>,<span class="dv">0</span>)</span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>cno <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>post_cno <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>)</span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a>exp6g_m1_planned <span class="ot">&lt;-</span> <span class="fu">contrast</span>(exp6g_m1_emm,</span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a>                       <span class="at">method =</span> <span class="fu">list</span>(</span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a>                         <span class="st">"Saline - Lesion"</span> <span class="ot">=</span> <span class="fu">c</span>(saline <span class="sc">-</span> lesion),</span>
<span id="cb23-11"><a href="#cb23-11" aria-hidden="true" tabindex="-1"></a>                         <span class="st">"CNO - Saline"</span> <span class="ot">=</span> <span class="fu">c</span>(cno <span class="sc">-</span> saline),</span>
<span id="cb23-12"><a href="#cb23-12" aria-hidden="true" tabindex="-1"></a>                         <span class="st">"Post_CNO - CNO"</span> <span class="ot">=</span> <span class="fu">c</span>(post_cno <span class="sc">-</span> cno)</span>
<span id="cb23-13"><a href="#cb23-13" aria-hidden="true" tabindex="-1"></a>                       ),</span>
<span id="cb23-14"><a href="#cb23-14" aria-hidden="true" tabindex="-1"></a>                       <span class="at">adjust =</span> <span class="st">"none"</span></span>
<span id="cb23-15"><a href="#cb23-15" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb23-16"><a href="#cb23-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">contrast</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">estimate</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">SE</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">df</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">lower.CL</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">upper.CL</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">t.ratio</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">p.value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Saline - Lesion</td>
<td style="text-align: right;">-20.52</td>
<td style="text-align: right;">4.071</td>
<td style="text-align: right;">18</td>
<td style="text-align: right;">-29.07</td>
<td style="text-align: right;">-11.96</td>
<td style="text-align: right;">-5.04</td>
<td style="text-align: right;">0.00009</td>
</tr>
<tr class="even">
<td style="text-align: left;">CNO - Saline</td>
<td style="text-align: right;">18.67</td>
<td style="text-align: right;">4.071</td>
<td style="text-align: right;">18</td>
<td style="text-align: right;">10.12</td>
<td style="text-align: right;">27.23</td>
<td style="text-align: right;">4.59</td>
<td style="text-align: right;">0.00023</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Post_CNO - CNO</td>
<td style="text-align: right;">-22.63</td>
<td style="text-align: right;">4.071</td>
<td style="text-align: right;">18</td>
<td style="text-align: right;">-31.18</td>
<td style="text-align: right;">-14.08</td>
<td style="text-align: right;">-5.56</td>
<td style="text-align: right;">0.00003</td>
</tr>
</tbody>
</table>


</div>
</div>
</section>
<section id="lmm-exp6g-plotthemodel" class="level4" data-number="14.7.2.3">
<h4 data-number="14.7.2.3" class="anchored" data-anchor-id="lmm-exp6g-plotthemodel"><span class="header-section-number">14.7.2.3</span> Plot the model</h4>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="lmm_files/figure-html/unnamed-chunk-19-1.png" class="img-fluid figure-img" width="576"></p>
<figcaption>Treatment effect on ipsilateral touch, as percent of all touches. Gray dots connected by lines are individual mice.</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="alternaplot-the-model" class="level4" data-number="14.7.2.4">
<h4 data-number="14.7.2.4" class="anchored" data-anchor-id="alternaplot-the-model"><span class="header-section-number">14.7.2.4</span> Alternaplot the model</h4>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="lmm_files/figure-html/unnamed-chunk-20-1.png" class="img-fluid figure-img" width="576"></p>
<figcaption>Ipsilateral touch (percent of all touches) response to different treatments. Gray dots connected by lines are individual mice.</figcaption>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="the-model-exp6g_m1-adds-a-random-intercept-but-not-a-random-slope" class="level3" data-number="14.7.3">
<h3 data-number="14.7.3" class="anchored" data-anchor-id="the-model-exp6g_m1-adds-a-random-intercept-but-not-a-random-slope"><span class="header-section-number">14.7.3</span> The model exp6g_m1 adds a random intercept but not a random slope</h3>
<p>The model fit to the exp6g data is</p>
<p><span class="math display">\[
\begin{equation}
\texttt{touch}_{jk} = (\beta_{0} + \gamma_{0j}) + (\beta_{k} + \gamma_{jk}) \texttt{treatment}_{k} + \varepsilon
\end{equation}
\]</span></p>
<p>Notes</p>
<ol type="1">
<li>Again, in experiments without subsampling, we cannot add a random slope to the model (for each mouse, there is a single observation at each treatment level so a slope would fit the two points perfectly).</li>
</ol>
</section>
<section id="the-fixed-effect-coefficients-of-model-exp6g_m1" class="level3" data-number="14.7.4">
<h3 data-number="14.7.4" class="anchored" data-anchor-id="the-fixed-effect-coefficients-of-model-exp6g_m1"><span class="header-section-number">14.7.4</span> The fixed effect coefficients of model exp6g_m1</h3>
<p>The fixed effect coefficients of model <code>exp6g_m1</code> are</p>
<div class="cell">
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Estimate</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Std. Error</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">df</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">t value</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Pr(&gt;&amp;#124;t&amp;#124;)</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">2.5 %</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">97.5 %</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">(Intercept)</td>
<td style="text-align: right;">83.27</td>
<td style="text-align: right;">3.441</td>
<td style="text-align: right;">18.9</td>
<td style="text-align: right;">24.20</td>
<td style="text-align: right;">0.0000</td>
<td style="text-align: right;">76.72</td>
<td style="text-align: right;">89.81</td>
</tr>
<tr class="even">
<td style="text-align: left;">treatmentSaline</td>
<td style="text-align: right;">-20.52</td>
<td style="text-align: right;">4.071</td>
<td style="text-align: right;">18.0</td>
<td style="text-align: right;">-5.04</td>
<td style="text-align: right;">0.0001</td>
<td style="text-align: right;">-28.25</td>
<td style="text-align: right;">-12.78</td>
</tr>
<tr class="odd">
<td style="text-align: left;">treatmentCNO</td>
<td style="text-align: right;">-1.84</td>
<td style="text-align: right;">4.071</td>
<td style="text-align: right;">18.0</td>
<td style="text-align: right;">-0.45</td>
<td style="text-align: right;">0.6565</td>
<td style="text-align: right;">-9.58</td>
<td style="text-align: right;">5.90</td>
</tr>
<tr class="even">
<td style="text-align: left;">treatmentPost_CNO</td>
<td style="text-align: right;">-24.47</td>
<td style="text-align: right;">4.071</td>
<td style="text-align: right;">18.0</td>
<td style="text-align: right;">-6.01</td>
<td style="text-align: right;">0.0000</td>
<td style="text-align: right;">-32.21</td>
<td style="text-align: right;">-16.73</td>
</tr>
</tbody>
</table>


</div>
</div>
<p>Notes</p>
<ol type="1">
<li>The interpretation of the fixed effectcs coefficients have the usual interpretation (see <a href="one-factor.html#oneway-what-coefs-are">The coefficients of a linear model using dummy coding have a useful interpretation</a>). Figure @ref(fig:lmm-exp6g-explainer-1) is a reminder.</li>
</ol>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="lmm_files/figure-html/lmm-exp6g-explainer-1-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption>Fixed effects estimated by exp6g_m1. <span class="math inline">\(b_0\)</span> is the modeled mean of the Lesian treatment. <span class="math inline">\(b_1\)</span> is the difference (Saline - Lesian). <span class="math inline">\(b_2\)</span> is the difference (CNO - Lesian). <span class="math inline">\(b_3\)</span> is the difference (Post_CNO - Lesian).</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="the-random-intercept-coefficients-of-exp6g_m1" class="level3" data-number="14.7.5">
<h3 data-number="14.7.5" class="anchored" data-anchor-id="the-random-intercept-coefficients-of-exp6g_m1"><span class="header-section-number">14.7.5</span> The random intercept coefficients of exp6g_m1</h3>
<div class="cell">
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<caption>Random intercept coefficients (g_0j) for exp6g_m1. For each mouse, the random intercept is the sum of the fixed intercept ($b_0$) and the random intercept effect $g_{0j}$.</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">mouse_id</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">random intercept</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">b_0</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">g_0j</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">mouse_1</td>
<td style="text-align: right;">82.874</td>
<td style="text-align: right;">83.268</td>
<td style="text-align: right;">-0.393</td>
</tr>
<tr class="even">
<td style="text-align: left;">mouse_2</td>
<td style="text-align: right;">83.640</td>
<td style="text-align: right;">83.268</td>
<td style="text-align: right;">0.372</td>
</tr>
<tr class="odd">
<td style="text-align: left;">mouse_3</td>
<td style="text-align: right;">77.499</td>
<td style="text-align: right;">83.268</td>
<td style="text-align: right;">-5.769</td>
</tr>
<tr class="even">
<td style="text-align: left;">mouse_4</td>
<td style="text-align: right;">79.943</td>
<td style="text-align: right;">83.268</td>
<td style="text-align: right;">-3.325</td>
</tr>
<tr class="odd">
<td style="text-align: left;">mouse_5</td>
<td style="text-align: right;">89.286</td>
<td style="text-align: right;">83.268</td>
<td style="text-align: right;">6.018</td>
</tr>
<tr class="even">
<td style="text-align: left;">mouse_6</td>
<td style="text-align: right;">82.732</td>
<td style="text-align: right;">83.268</td>
<td style="text-align: right;">-0.536</td>
</tr>
<tr class="odd">
<td style="text-align: left;">mouse_7</td>
<td style="text-align: right;">86.901</td>
<td style="text-align: right;">83.268</td>
<td style="text-align: right;">3.633</td>
</tr>
</tbody>
</table>


</div>
</div>
<p>Notes</p>
<ol type="1">
<li>The random intercept effect <span class="math inline">\(g_{0j}\)</span> is the difference between the modeled mean for mouse <em>j</em> and the mean of the reference treatment. Unlike the fixed intercept (<span class="math inline">\(b_0\)</span>), <span class="math inline">\(g_{0j}\)</span> is an effect.</li>
<li>Figure @ref(fig:lmm-exp1g-explainer-2) illustrates the random intercept coefficients <span class="math inline">\(g_{0j}\)</span> for model <code>exp6g_m1</code>.</li>
<li>The random intercept coefficients are often (but not always) treated as a source of nuisance variation – that is, the coefficients are not generally of interest and the values are not typically reported.</li>
</ol>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="lmm_files/figure-html/lmm-exp6g-explainer-2-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption>Random intercept effects for model exp6g_m1. The pale, colored dots are the measured percent ipsilateral touch values for each mouse for each treatment. The dashed, gray lines are the modeled means for each treatment. The dashed grey line for the reference level (“Lesion”) is the fixed intercept. The dark, colored dots at the reference level are the random intercepts. The value of each random intercept is the sum of the fixed intercept and the random intercept effect for that mouse. The vertical, colored line segments at the reference level are the random intercept effects <span class="math inline">\(g_{0j}\)</span>. The length of the segment is the residual from the dashed, gray line to the pale dot. The random intercept effects for mice 1 and 2 are too short to see.</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="the-random-and-residual-variance-and-the-intraclass-correlation-of-model-exp6g_m1" class="level3" data-number="14.7.6">
<h3 data-number="14.7.6" class="anchored" data-anchor-id="the-random-and-residual-variance-and-the-intraclass-correlation-of-model-exp6g_m1"><span class="header-section-number">14.7.6</span> The random and residual variance and the intraclass correlation of model exp6g_m1</h3>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code> Groups   Name        Std.Dev.
 mouse_id (Intercept) 4.9882  
 Residual             7.6155  </code></pre>
</div>
</div>
<p>Notes</p>
<ol type="1">
<li>The first element in “Std.Dev.” is the estimate of <span class="math inline">\(\sqrt{\sigma^2_{0j}}\)</span>, the standard deviation among the donors due to the random effect of <span class="math inline">\(\texttt{mouse_id}\)</span>.</li>
<li>The second element in “Std.Dev.” is <span class="math inline">\(\sqrt{\sigma^2_{0}}\)</span>, the estimate of the standard deviation of the error variance (<span class="math inline">\(\varepsilon^2\)</span>). Remember that the residuals of the model are the estimates of <span class="math inline">\(\varepsilon\)</span>.</li>
<li>The ratio <span class="math inline">\(\frac{\sigma^2_{0j}}{\sigma^2_{0j} + \sigma^2_{0}}\)</span> (the ratio of the among-block variance to total random variance) is known as the <a href="https://en.wikipedia.org/wiki/Intraclass_correlation" target="_blank">intraclass correlation</a>. This correlation is an estimate of the correlated error due to the by-mouse clustering if the model were fit without the added random intercept. For <code>exp6g_m1</code>, this correlation is 0.3.</li>
<li>The intraclass correlation ranges between 0 and 1 and makes a pretty good qualitative indicator of repeatability.</li>
</ol>
</section>
<section id="the-linear-mixed-model-exp6g_m1-increases-precision-of-treatment-effects-relative-to-a-fixed-effects-model" class="level3" data-number="14.7.7">
<h3 data-number="14.7.7" class="anchored" data-anchor-id="the-linear-mixed-model-exp6g_m1-increases-precision-of-treatment-effects-relative-to-a-fixed-effects-model"><span class="header-section-number">14.7.7</span> The linear mixed model exp6g_m1 increases precision of treatment effects, relative to a fixed effects model</h3>
<p>Let’s compare the effects estimated by the linear mixed model <code>exp6g_m1</code> with a linear model that ignores donor (a <strong>fixed effects</strong> model).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>exp6g_m2 <span class="ot">&lt;-</span> <span class="fu">lm</span>(touch <span class="sc">~</span> treatment, <span class="at">data =</span> exp6g)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="lmm_files/figure-html/lmm-exp6g-why2-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption>A. Inference from a linear mixed model with blocking factor (mouse_id) added as a random intercept. B. Inference from a fixed effects model.</figcaption>
</figure>
</div>
</div>
</div>
<p>Figure @ref(fig:lmm-exp6g-why2)A is a plot of the effects from the linear mixed model <code>exp6g_m1</code> that models the added variance due to mouse. Figure @ref(fig:lmm-exp6g-why2)B is a plot of the effects from the fixed effect model <code>exp6g_m2</code> that ignores the added variance due to mouse. The 95% confidence intervals of the treatment effects in model <code>exp6g_m1</code> are slightly smaller than those in model <code>exp6g_m2</code>. Adding <span class="math inline">\(\texttt{mouse_id}\)</span> as a random factor to the linear model increases the precision of the estimate of the treatment effects by eliminating the among-mouse component of variance from the error variance.</p>
<p>The error variance (the estimate of $^2) in the linear mixed model and the fixed effects model is</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(exp6g_m1)<span class="sc">$</span>sigma<span class="sc">^</span><span class="dv">2</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 57.99549</code></pre>
</div>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(exp6g_m2)<span class="sc">$</span>sigma<span class="sc">^</span><span class="dv">2</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 82.8777</code></pre>
</div>
</div>
<p>The error variance is 43% higher in the fixed effects model. In the linear mixed model, the variance lost from the error was shifted to the random intercept. We can track this shift with the table of the two variance components of the linear mixed model, shown above, and here again (the values are the square roots of the variances).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="fu">VarCorr</span>(exp6g_m1)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> Groups   Name        Std.Dev.
 mouse_id (Intercept) 4.9882  
 Residual             7.6155  </code></pre>
</div>
</div>
<p>The sum of the two variance components of the linear mixed model is equal to the error variance of the fixed effects model:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(<span class="fu">as.data.frame</span>(<span class="fu">VarCorr</span>(exp6g_m1))<span class="sc">$</span>vcov)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 82.8777</code></pre>
</div>
</div>
</section>
<section id="lmm-exp6g-alt" class="level3" data-number="14.7.8">
<h3 data-number="14.7.8" class="anchored" data-anchor-id="lmm-exp6g-alt"><span class="header-section-number">14.7.8</span> Alternative models for exp6g</h3>
<p>Linear mixed models are very flexible, a topic which is too advanced for this text. Here I want to focus on an alternative model because of its relevance to <strong>repeated measures ANOVA</strong>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a>exp6g_m1a <span class="ot">&lt;-</span> <span class="fu">lmer</span>(touch <span class="sc">~</span> treatment <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> mouse_id),</span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp6g)</span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>exp6g_m1b <span class="ot">&lt;-</span> <span class="fu">lme</span>(touch <span class="sc">~</span> treatment,</span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">random =</span> <span class="sc">~</span><span class="dv">1</span><span class="sc">|</span>mouse_id,</span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a>                <span class="at">correlation =</span> <span class="fu">corSymm</span>(<span class="at">form =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> mouse_id),</span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a>                <span class="at">weights =</span> <span class="fu">varIdent</span>(<span class="at">form =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> treatment),</span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a>                <span class="at">data =</span> exp6g)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Notes</p>
<ol type="1">
<li>Two models are fit. The two models have the same fixed effect but differ in how they model the random effect and the pattern of correlations in the residuals. The difference in specification determines the error variance and degrees of freedom for computing uncertainty. I’ll return to inference and the problem of “which model to choose” in a moment. First, how do these models differ?</li>
<li>Model <code>exp6g_m1a</code> is the same model analyzed and explained above. This model specifies a random intercept for each level of <span class="math inline">\(\texttt{mouse_id}\)</span>.</li>
<li>Model <code>exp6g_m1b</code> has the same random effects as <code>exp6g_m1a</code> but adds two additional arguments, a <code>correlation</code> argument that explicitly models correlated error in the residuals and a <code>weights</code> argument that models heterogeneity in the residuals.</li>
<li>Consider the error (the residuals) of the fixed effect model <code>touch ~ treatment</code> cast into a matrix with the residuals for each treatment in its own column. This matrix of residuals will be a <span class="math inline">\(7 \times 4\)</span> (7 mice, 4 treatments) matrix that looks like the table of residuals in Example 1 (Table @ref(tab:lmm-demo1-cor-error-table)).</li>
<li>Model <code>exp6g_m1a</code> implicitly models <strong>compound symmetric</strong> correlated error. This means</li>
</ol>
<ul>
<li>the correlation between every pair of columns of the residual matrix is the same.</li>
<li>the variances of each of the columns of the residual matrix is the same.</li>
<li>Model <code>exp6g_m1a</code> assumes zero correlation among the columns of a residual matrix <em>from Model exp6g_m1a</em>. This is because the source of the correlated residuals from the fixed model has been modeled by the random intercept.</li>
</ul>
<ol start="5" type="1">
<li>Model <code>exp6g_m1b</code> does not assume <strong>compound symmetric</strong> correlated error.</li>
</ol>
<ul>
<li>the argument <code>correlation = corSymm(form = ~ 1 | mouse_id)</code> in model <code>exp6g_m1b</code> explicitly models different correlations for all pairs of treatment combinations – that is, <strong>unstructured</strong> correlated error.</li>
<li>the argument <code>weights = varIdent(form = ~ 1 | treatment)</code> in model <code>exp6g_m1b</code> explicitly models heterogeneity in the residuals among the levels of $.</li>
</ul>
<ol start="6" type="1">
<li>Model <code>exp6g_m1b</code> makes fewer assumptions for inference. The trade-off is the estimation of more parameters and the potential of <a href="https://en.wikipedia.org/wiki/Overfitting" target="_blank">overfitting</a>.</li>
<li>A univariate model of a <strong>repeated measures ANOVA</strong> fit to the <code>exp6g</code> data is equivalent to model <code>exp6g_m1a</code>. See section @ref(lmm-exp6g-rmanova).</li>
<li>A multivariate model of a <strong>repeated measures ANOVA</strong> fit to the <code>exp6g</code> data is equivalent to model <code>exp6g_m1b</code>. See section @ref(lmm-exp6g-rmanova-multi).</li>
</ol>
<div class="cell">
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<caption>Planned contrasts from the two alternative models.</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">contrast</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">estimate</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">SE</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">df</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">lower.CL</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">upper.CL</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">t.ratio</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">p.value</th>
</tr>
</thead>
<tbody>
<tr class="odd" data-grouplength="3">
<td colspan="8" style="border-bottom: 1px solid"><strong>exp6g_m1a</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Saline - Lesion</td>
<td style="text-align: right;">-20.52</td>
<td style="text-align: right;">4.071</td>
<td style="text-align: right;">18</td>
<td style="text-align: right;">-29.07</td>
<td style="text-align: right;">-11.96</td>
<td style="text-align: right;">-5.04</td>
<td style="text-align: right;">0.000085</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">CNO - Saline</td>
<td style="text-align: right;">18.67</td>
<td style="text-align: right;">4.071</td>
<td style="text-align: right;">18</td>
<td style="text-align: right;">10.12</td>
<td style="text-align: right;">27.23</td>
<td style="text-align: right;">4.59</td>
<td style="text-align: right;">0.000228</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Post_CNO - CNO</td>
<td style="text-align: right;">-22.63</td>
<td style="text-align: right;">4.071</td>
<td style="text-align: right;">18</td>
<td style="text-align: right;">-31.18</td>
<td style="text-align: right;">-14.08</td>
<td style="text-align: right;">-5.56</td>
<td style="text-align: right;">0.000028</td>
</tr>
<tr class="odd" data-grouplength="3">
<td colspan="8" style="border-bottom: 1px solid"><strong>exp6g_m1b</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Saline - Lesion</td>
<td style="text-align: right;">-20.52</td>
<td style="text-align: right;">3.172</td>
<td style="text-align: right;">18</td>
<td style="text-align: right;">-27.18</td>
<td style="text-align: right;">-13.85</td>
<td style="text-align: right;">-6.47</td>
<td style="text-align: right;">0.000004</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">CNO - Saline</td>
<td style="text-align: right;">18.67</td>
<td style="text-align: right;">3.057</td>
<td style="text-align: right;">18</td>
<td style="text-align: right;">12.25</td>
<td style="text-align: right;">25.10</td>
<td style="text-align: right;">6.11</td>
<td style="text-align: right;">0.000009</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Post_CNO - CNO</td>
<td style="text-align: right;">-22.63</td>
<td style="text-align: right;">5.610</td>
<td style="text-align: right;">18</td>
<td style="text-align: right;">-34.42</td>
<td style="text-align: right;">-10.84</td>
<td style="text-align: right;">-4.03</td>
<td style="text-align: right;">0.000779</td>
</tr>
</tbody>
</table>


</div>
</div>
<ol start="9" type="1">
<li>Planned comparisons of the two models are given in the tables above. The effect estimates are the same but inference differs quantitatively among the models (but not qualitatively) because of how each model models the error. Which model do we report? One way to evaluate the models is a statistic known as the AIC.</li>
</ol>
<div class="cell">
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<caption>AIC of the two alternative models fit to exp6g data</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">Model</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">AIC</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">exp6g_m1a</td>
<td style="text-align: right;">191.3367</td>
</tr>
<tr class="even">
<td style="text-align: left;">exp6g_m1b</td>
<td style="text-align: right;">198.3888</td>
</tr>
</tbody>
</table>


</div>
</div>
<ol start="10" type="1">
<li>AIC (<a href="https://en.wikipedia.org/wiki/Akaike_information_criterion" target="_blank">Akaike Information Criterion</a>) is a relative measure of model quality. Compare this to <span class="math inline">\(R^2\)</span>, which is an absolute measure of goodness of fit. The AIC formula has two parts, one is a kind of goodness of fit (like <span class="math inline">\(R^2\)</span>) and the other is a penalty based on the number of parameters in the model. As the goodness of fit increases, the AIC goes down. As the number of parameters increases, the AIC goes up. The model with the lowest AIC is the highest quality model. The actual number, however does not have any absolute meaning; it is only meaningful relative to the AICs computed from fits to different models using the same data.</li>
<li>The AICs of models <code>exp6g_m1a</code> and models <code>exp6g_m1b</code> suggest that model <code>exp6g_m1b</code> is too complex given the data. This has relevance for the repeated measures ANOVA analysis in the next section.</li>
</ol>
</section>
<section id="paired-t-tests-and-repeated-measures-anova-are-special-cases-of-linear-mixed-models" class="level3" data-number="14.7.9">
<h3 data-number="14.7.9" class="anchored" data-anchor-id="paired-t-tests-and-repeated-measures-anova-are-special-cases-of-linear-mixed-models"><span class="header-section-number">14.7.9</span> Paired t-tests and repeated measures ANOVA are special cases of linear mixed models</h3>
<p>A paired <em>t</em>-test is a special case of linear mixed model fit to data from a randomized complete block design with no subsampling and only two treatment levels (see <a href="#oneway-paired-t">Lack of independence</a> in the Violations chapter). A repeated measures ANOVA is a special case of linear mixed model fit to data from a randomized complete block design with no subsampling and more than two treatment levels (see <a href="#oneway-paired-t">Lack of independence</a> in the Violations chapter).</p>
<p>Experiment 6g is a randomized complete block design with no subsampling and four treatments. A “which test” key in a traditional, experimental statistics textbook would guide a researcher to analyze these data using <a href="https://en.wikipedia.org/wiki/Repeated_measures_design#Repeated_measures_ANOVA" target="_blank">repeated measures ANOVA</a>. There are two methods of repeated measures anova, the univariate model and the multivariate model.</p>
</section>
<section id="lmm-exp6g-rmanova" class="level3" data-number="14.7.10">
<h3 data-number="14.7.10" class="anchored" data-anchor-id="lmm-exp6g-rmanova"><span class="header-section-number">14.7.10</span> Classical (“univariate model”) repeated measures ANOVA of exp6g</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>exp6g_aov1 <span class="ot">&lt;-</span> <span class="fu">aov_4</span>(touch <span class="sc">~</span> treatment <span class="sc">+</span></span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>                    (treatment <span class="sc">|</span> mouse_id),</span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> exp6g)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stderr">
<pre><code>aov object missing, substituting multivariate/lm model.
to get univariate tests, refit ANOVA with include_aov = TRUE</code></pre>
</div>
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<caption>Planned contrasts for the linear mixed model exp6g_m1a and the univariate model of the repeated measures ANOVA.</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">contrast</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">estimate</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">SE</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">df</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">lower.CL</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">upper.CL</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">t.ratio</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">p.value</th>
</tr>
</thead>
<tbody>
<tr class="odd" data-grouplength="3">
<td colspan="8" style="border-bottom: 1px solid"><strong>exp6g_m1a</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Saline - Lesion</td>
<td style="text-align: right;">-20.52</td>
<td style="text-align: right;">4.071</td>
<td style="text-align: right;">18</td>
<td style="text-align: right;">-29.07</td>
<td style="text-align: right;">-11.96</td>
<td style="text-align: right;">-5.04</td>
<td style="text-align: right;">0.000085</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">CNO - Saline</td>
<td style="text-align: right;">18.67</td>
<td style="text-align: right;">4.071</td>
<td style="text-align: right;">18</td>
<td style="text-align: right;">10.12</td>
<td style="text-align: right;">27.23</td>
<td style="text-align: right;">4.59</td>
<td style="text-align: right;">0.000228</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Post_CNO - CNO</td>
<td style="text-align: right;">-22.63</td>
<td style="text-align: right;">4.071</td>
<td style="text-align: right;">18</td>
<td style="text-align: right;">-31.18</td>
<td style="text-align: right;">-14.08</td>
<td style="text-align: right;">-5.56</td>
<td style="text-align: right;">0.000028</td>
</tr>
<tr class="odd" data-grouplength="3">
<td colspan="8" style="border-bottom: 1px solid"><strong>univariate RM-ANOVA</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Saline - Lesion</td>
<td style="text-align: right;">-20.52</td>
<td style="text-align: right;">3.172</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">-28.28</td>
<td style="text-align: right;">-12.75</td>
<td style="text-align: right;">-6.47</td>
<td style="text-align: right;">0.000649</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">CNO - Saline</td>
<td style="text-align: right;">18.67</td>
<td style="text-align: right;">3.057</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">11.19</td>
<td style="text-align: right;">26.15</td>
<td style="text-align: right;">6.11</td>
<td style="text-align: right;">0.000877</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Post_CNO - CNO</td>
<td style="text-align: right;">-22.63</td>
<td style="text-align: right;">5.610</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">-36.36</td>
<td style="text-align: right;">-8.90</td>
<td style="text-align: right;">-4.03</td>
<td style="text-align: right;">0.006853</td>
</tr>
</tbody>
</table>


</div>
</div>
<p>Notes</p>
<ol type="1">
<li>In addition to the assumptions of the linear model outlined in the Violations chapter, the classical (“univariate model”) repeated measures ANOVA assumes <a href="https://en.wikipedia.org/wiki/Mauchly%27s_sphericity_test#Sphericity" target="_blank">sphericity</a>, which is the equality of the variances of <em>all pairwise differences</em> among treatment combinations.</li>
<li>The univariate model of the classic RM ANOVA is equivalent to Model <code>exp6g_m1a</code>.</li>
</ol>
</section>
<section id="lmm-exp6g-rmanova-multi" class="level3" data-number="14.7.11">
<h3 data-number="14.7.11" class="anchored" data-anchor-id="lmm-exp6g-rmanova-multi"><span class="header-section-number">14.7.11</span> “Multivariate model” repeated measures ANOVA</h3>
<div class="cell">
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<caption>Planned contrasts for the linear mixed model exp6g_m1b and the multivariate model of the repeated measures ANOVA.</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">contrast</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">estimate</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">SE</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">df</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">lower.CL</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">upper.CL</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">t.ratio</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">p.value</th>
</tr>
</thead>
<tbody>
<tr class="odd" data-grouplength="3">
<td colspan="8" style="border-bottom: 1px solid"><strong>exp6g_m1b</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Saline - Lesion</td>
<td style="text-align: right;">-20.52</td>
<td style="text-align: right;">3.172</td>
<td style="text-align: right;">18</td>
<td style="text-align: right;">-27.18</td>
<td style="text-align: right;">-13.85</td>
<td style="text-align: right;">-6.47</td>
<td style="text-align: right;">0.000004</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">CNO - Saline</td>
<td style="text-align: right;">18.67</td>
<td style="text-align: right;">3.057</td>
<td style="text-align: right;">18</td>
<td style="text-align: right;">12.25</td>
<td style="text-align: right;">25.10</td>
<td style="text-align: right;">6.11</td>
<td style="text-align: right;">0.000009</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Post_CNO - CNO</td>
<td style="text-align: right;">-22.63</td>
<td style="text-align: right;">5.610</td>
<td style="text-align: right;">18</td>
<td style="text-align: right;">-34.42</td>
<td style="text-align: right;">-10.84</td>
<td style="text-align: right;">-4.03</td>
<td style="text-align: right;">0.000779</td>
</tr>
<tr class="odd" data-grouplength="3">
<td colspan="8" style="border-bottom: 1px solid"><strong>multivariate RM-ANOVA</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Saline - Lesion</td>
<td style="text-align: right;">-20.52</td>
<td style="text-align: right;">3.172</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">-28.28</td>
<td style="text-align: right;">-12.75</td>
<td style="text-align: right;">-6.47</td>
<td style="text-align: right;">0.000649</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">CNO - Saline</td>
<td style="text-align: right;">18.67</td>
<td style="text-align: right;">3.057</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">11.19</td>
<td style="text-align: right;">26.15</td>
<td style="text-align: right;">6.11</td>
<td style="text-align: right;">0.000877</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Post_CNO - CNO</td>
<td style="text-align: right;">-22.63</td>
<td style="text-align: right;">5.610</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">-36.36</td>
<td style="text-align: right;">-8.90</td>
<td style="text-align: right;">-4.03</td>
<td style="text-align: right;">0.006853</td>
</tr>
</tbody>
</table>


</div>
</div>
<p>Notes</p>
<ol type="1">
<li>The sphericity assumption of classical repeated measures ANOVA is relaxed if the model is fit using the “multivariate model”.</li>
<li>The multivariate model repeated measures ANOVA is a linear model with a multivariate response and not a linear mixed model. In the multivariate model, each treatment combination is a different response variable and there is a single row for each level of the random factor (<span class="math inline">\(\texttt{mouse_id}\)</span> in Experiment 6g). The multivariate model is a different way of handling the correlated error that occurs when conceiving of the design as univariate.</li>
<li>The multivariate repeated measures ANOVA is equivalent to the linear mixed model <code>exp6g_m1b</code>. The SEs of the contrasts are the same but the degrees of freedom differ. Because of the increased df of the linear mixed model, the CIs are narrower and the <em>p</em>-value is smaller – that is the linear mixed model is less conservative than the repeated measures ANOVA. There is no correct degrees of freedom for a linear mixed model like this and emmeans outputs one way to compute these (with options for others, none of which are equivalent to those from the multivariate model RM-ANOVA).</li>
</ol>
</section>
<section id="linear-mixed-models-vs-repeated-measures-anova" class="level3" data-number="14.7.12">
<h3 data-number="14.7.12" class="anchored" data-anchor-id="linear-mixed-models-vs-repeated-measures-anova"><span class="header-section-number">14.7.12</span> Linear mixed models vs repeated measures ANOVA</h3>
<p>Many modern textbooks encourage researchers to use linear mixed models instead of repeated measures ANOVA for randomized complete block designs (with or without subsampling) because</p>
<ol type="1">
<li>linear mixed models do not exclude random units (subject/mouse/donor/cage) with missing measures of one of the treatment combinations. <a href="#violations-rmanova">Example 2 (diHOME exp2a) – A repeated measures ANOVA is a special case of a linear mixed model</a> is an example of this kind of missing data.</li>
<li>if there is subsampling, linear mixed models do not <strong>aggregate</strong> the random-unit data, that is, linear mixed models do not simply compute the means and ignore the variance of the sample within each random unit.</li>
<li>linear mixed models allow for modeling additional sources of correlated error, that is, an experiment may have two or more random factor variables (for example, donor and experiment).</li>
<li>linear mixed models allow a researcher to model different patterns of correlated error. This is especially important in longitudinal experiments.</li>
<li>linear mixed models can be generalized to model sampling from non-normal distributions – these are generalized linear mixed models.</li>
</ol>
</section>
<section id="modeling-textttmouse_id-as-a-fixed-effect" class="level3" data-number="14.7.13">
<h3 data-number="14.7.13" class="anchored" data-anchor-id="modeling-textttmouse_id-as-a-fixed-effect"><span class="header-section-number">14.7.13</span> Modeling <span class="math inline">\(\texttt{mouse_id}\)</span> as a fixed effect</h3>
<p><span class="math display">\[
\begin{equation}
\texttt{touch} \sim \texttt{treatment + mouse_id}
\end{equation}
(\#eq:lmm-exp6g-m3)
\]</span></p>
<p>Model @ref(eq:lmm-exp6g-m3) (using R formula syntax) is a linear model with the block <span class="math inline">\(\texttt{mouse_id}\)</span> added as a fixed covariate instead of a random intercept. The coefficients of the fit model are</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>exp6g_m3 <span class="ot">&lt;-</span> <span class="fu">lm</span>(touch <span class="sc">~</span> treatment <span class="sc">+</span> mouse_id,</span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>               <span class="at">data =</span> exp6g)</span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a>exp6g_m3_coef <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">coef</span>(<span class="fu">summary</span>(exp6g_m3)),</span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">confint</span>(exp6g_m3))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Estimate</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Std. Error</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">t value</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Pr(&gt;&amp;#124;t&amp;#124;)</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">2.5 %</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">97.5 %</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">(Intercept)</td>
<td style="text-align: right;">82.65</td>
<td style="text-align: right;">4.551</td>
<td style="text-align: right;">18.2</td>
<td style="text-align: right;">0.0000</td>
<td style="text-align: right;">73.08</td>
<td style="text-align: right;">92.21</td>
</tr>
<tr class="even">
<td style="text-align: left;">treatmentSaline</td>
<td style="text-align: right;">-20.52</td>
<td style="text-align: right;">4.071</td>
<td style="text-align: right;">-5.0</td>
<td style="text-align: right;">0.0001</td>
<td style="text-align: right;">-29.07</td>
<td style="text-align: right;">-11.96</td>
</tr>
<tr class="odd">
<td style="text-align: left;">treatmentCNO</td>
<td style="text-align: right;">-1.84</td>
<td style="text-align: right;">4.071</td>
<td style="text-align: right;">-0.5</td>
<td style="text-align: right;">0.6565</td>
<td style="text-align: right;">-10.39</td>
<td style="text-align: right;">6.71</td>
</tr>
<tr class="even">
<td style="text-align: left;">treatmentPost_CNO</td>
<td style="text-align: right;">-24.47</td>
<td style="text-align: right;">4.071</td>
<td style="text-align: right;">-6.0</td>
<td style="text-align: right;">0.0000</td>
<td style="text-align: right;">-33.02</td>
<td style="text-align: right;">-15.92</td>
</tr>
<tr class="odd">
<td style="text-align: left;">mouse_idmouse_2</td>
<td style="text-align: right;">1.21</td>
<td style="text-align: right;">5.385</td>
<td style="text-align: right;">0.2</td>
<td style="text-align: right;">0.8245</td>
<td style="text-align: right;">-10.10</td>
<td style="text-align: right;">12.52</td>
</tr>
<tr class="even">
<td style="text-align: left;">mouse_idmouse_3</td>
<td style="text-align: right;">-8.51</td>
<td style="text-align: right;">5.385</td>
<td style="text-align: right;">-1.6</td>
<td style="text-align: right;">0.1315</td>
<td style="text-align: right;">-19.82</td>
<td style="text-align: right;">2.81</td>
</tr>
<tr class="odd">
<td style="text-align: left;">mouse_idmouse_4</td>
<td style="text-align: right;">-4.64</td>
<td style="text-align: right;">5.385</td>
<td style="text-align: right;">-0.9</td>
<td style="text-align: right;">0.4003</td>
<td style="text-align: right;">-15.95</td>
<td style="text-align: right;">6.67</td>
</tr>
<tr class="even">
<td style="text-align: left;">mouse_idmouse_5</td>
<td style="text-align: right;">10.15</td>
<td style="text-align: right;">5.385</td>
<td style="text-align: right;">1.9</td>
<td style="text-align: right;">0.0758</td>
<td style="text-align: right;">-1.17</td>
<td style="text-align: right;">21.46</td>
</tr>
<tr class="odd">
<td style="text-align: left;">mouse_idmouse_6</td>
<td style="text-align: right;">-0.23</td>
<td style="text-align: right;">5.385</td>
<td style="text-align: right;">0.0</td>
<td style="text-align: right;">0.9671</td>
<td style="text-align: right;">-11.54</td>
<td style="text-align: right;">11.09</td>
</tr>
<tr class="even">
<td style="text-align: left;">mouse_idmouse_7</td>
<td style="text-align: right;">6.37</td>
<td style="text-align: right;">5.385</td>
<td style="text-align: right;">1.2</td>
<td style="text-align: right;">0.2521</td>
<td style="text-align: right;">-4.94</td>
<td style="text-align: right;">17.69</td>
</tr>
</tbody>
</table>


</div>
</div>
<p>Notes</p>
<ol type="1">
<li>The coefficients include a slope for the seven non-reference levels of mouse_id We typically don’t care about these.</li>
<li>In a randomized complete block design with only 1 replicate of each treatment combination per block, like that in Experiment 6g, inference about treatment effects is exactly that same between this model and the random intercept model <code>exp6g_m1.</code> Compare the SE, CI and <em>p</em>-value for the coefficients of the treatment effects to those from the linear mixed model <code>exp6g_m1</code>.</li>
</ol>
<div class="cell">
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Estimate</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Std. Error</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">df</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">t value</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Pr(&gt;&amp;#124;t&amp;#124;)</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">2.5 %</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">97.5 %</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">(Intercept)</td>
<td style="text-align: right;">83.27</td>
<td style="text-align: right;">3.441</td>
<td style="text-align: right;">18.9</td>
<td style="text-align: right;">24.20</td>
<td style="text-align: right;">0.0000</td>
<td style="text-align: right;">76.72</td>
<td style="text-align: right;">89.81</td>
</tr>
<tr class="even">
<td style="text-align: left;">treatmentSaline</td>
<td style="text-align: right;">-20.52</td>
<td style="text-align: right;">4.071</td>
<td style="text-align: right;">18.0</td>
<td style="text-align: right;">-5.04</td>
<td style="text-align: right;">0.0001</td>
<td style="text-align: right;">-28.25</td>
<td style="text-align: right;">-12.78</td>
</tr>
<tr class="odd">
<td style="text-align: left;">treatmentCNO</td>
<td style="text-align: right;">-1.84</td>
<td style="text-align: right;">4.071</td>
<td style="text-align: right;">18.0</td>
<td style="text-align: right;">-0.45</td>
<td style="text-align: right;">0.6565</td>
<td style="text-align: right;">-9.58</td>
<td style="text-align: right;">5.90</td>
</tr>
<tr class="even">
<td style="text-align: left;">treatmentPost_CNO</td>
<td style="text-align: right;">-24.47</td>
<td style="text-align: right;">4.071</td>
<td style="text-align: right;">18.0</td>
<td style="text-align: right;">-6.01</td>
<td style="text-align: right;">0.0000</td>
<td style="text-align: right;">-32.21</td>
<td style="text-align: right;">-16.73</td>
</tr>
</tbody>
</table>


</div>
</div>
<ol start="3" type="1">
<li>In a randomized complete block design with only 1 replicate of each treatment combination per block, like that in Experiment 6g, inference about <em>treatment means</em> differs between this model and the random intercept model – the SE of the means of the fixed effect model are smaller than the SE of the means of the linear mixed model. Compare the SE and CI of the <code>(Intercept)</code> (the mean of the reference treatment combination) between the fixed effect and linear mixed model.</li>
<li>The equivalence of inference in the treatment effect between the fixed effect and linear mixed model holds only for balanced randomized complete block designs – where all blocks contain all treatment combinations and the subsampling replicate size is the same for all treatment combinations for all blocks. This means, outside of special cases like Experiment 6g, the choice between adding a blocking variable as a random or fixed factor depends on assumptions about the model.</li>
</ol>
</section>
</section>
<section id="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c" class="level2" data-number="14.8">
<h2 data-number="14.8" class="anchored" data-anchor-id="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c"><span class="header-section-number">14.8</span> Example 3 – Factorial experiments and no subsampling replicates (exp5c)</h2>
<p>Example 3 is similar to example 2 in that there is no subsampling replication and we cannot add random slopes to the linear mixed model. Example 3 differs in that the design is factorial – there are two, crossed fixed factors. Consequently, there are several alternative models with different sets of random intercepts. The reported model includes two random intercepts, one of which models differences in batch effects among treatment levels (treatment by batch interactions). This <strong>interaction intercept</strong> is an alternative to random slope for modeling treatment by batch interactions.</p>
<p><a href="https://www.nature.com/articles/s41467-019-13869-w" target="_blank">Transcriptomic profiling of skeletal muscle adaptations to exercise and inactivity</a></p>
<p>Source figure: <a href="https://www.nature.com/articles/s41467-019-13869-w#Fig5" target="_blank">Fig. 5c</a></p>
<p>Source data: <a href="https://www.nature.com/articles/s41467-019-13869-w#Sec27" target="_blank">Source Data Fig. 5</a></p>
<div class="cell">
<div class="cell-output cell-output-stderr">
<pre><code>New names:
• `` -&gt; `...1`
• `` -&gt; `...2`
• `` -&gt; `...3`
• `` -&gt; `...4`
• `` -&gt; `...5`
• `` -&gt; `...6`
• `` -&gt; `...7`
• `` -&gt; `...8`
• `` -&gt; `...9`
• `` -&gt; `...10`
• `` -&gt; `...11`
• `` -&gt; `...12`
• `` -&gt; `...13`</code></pre>
</div>
</div>
<section id="understand-the-data-1" class="level3" data-number="14.8.1">
<h3 data-number="14.8.1" class="anchored" data-anchor-id="understand-the-data-1"><span class="header-section-number">14.8.1</span> Understand the data</h3>
<p>The data for Example 1 are from Figure 5c. Six muscle source cells were used to start six independent cultures. Cells from each culture were treated with either a negative control (“Scr”) or a siRNA (“siNR4A3”) that “silences” expression of the NR4A3 gene product by <a href="https://en.wikipedia.org/wiki/Small_interfering_RNA" target="_blank">cleaving the mRNA</a>. Glucose uptake in the two cell types was measured at rest (“Basal”) and during electrical pulse stimulation (“EPS”).</p>
<p>The design is a <span class="math inline">\(2 \times 2\)</span> <a href="https://en.wikipedia.org/wiki/Blocking_(statistics)#Randomized_block_design" target="_blank">Randomized complete block with no subsampling</a>. There are two factors each with two levels: <span class="math inline">\(\texttt{treatment}\)</span> (“Scr”, “siNR4A3”) and <span class="math inline">\(\texttt{activity}\)</span> (“Basal”, “EPS”). Each source cell is a block. All four treatment combinations were measured once per block.</p>
</section>
<section id="examine-the-data" class="level3" data-number="14.8.2">
<h3 data-number="14.8.2" class="anchored" data-anchor-id="examine-the-data"><span class="header-section-number">14.8.2</span> Examine the data</h3>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="lmm_files/figure-html/lmm-exp5c-examine-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>The plot shows a strong donor effect.</p>
</section>
<section id="lmm-exp5c-m1" class="level3" data-number="14.8.3">
<h3 data-number="14.8.3" class="anchored" data-anchor-id="lmm-exp5c-m1"><span class="header-section-number">14.8.3</span> Model fit and inference</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>exp5c_m1a <span class="ot">&lt;-</span> <span class="fu">lmer</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor),</span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp5c)</span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a>exp5c_m1b <span class="ot">&lt;-</span> <span class="fu">lmer</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor) <span class="sc">+</span></span>
<span id="cb39-7"><a href="#cb39-7" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor<span class="sc">:</span>treatment) <span class="sc">+</span></span>
<span id="cb39-8"><a href="#cb39-8" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor<span class="sc">:</span>activity),</span>
<span id="cb39-9"><a href="#cb39-9" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp5c)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>boundary (singular) fit: see help('isSingular')</code></pre>
</div>
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a>exp5c_m1c <span class="ot">&lt;-</span> <span class="fu">lmer</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor) <span class="sc">+</span></span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor<span class="sc">:</span>treatment),</span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp5c)</span>
<span id="cb41-5"><a href="#cb41-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-6"><a href="#cb41-6" aria-hidden="true" tabindex="-1"></a>exp5c_m1d <span class="ot">&lt;-</span> <span class="fu">lme</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity,</span>
<span id="cb41-7"><a href="#cb41-7" aria-hidden="true" tabindex="-1"></a>                 <span class="at">random =</span>  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> donor,</span>
<span id="cb41-8"><a href="#cb41-8" aria-hidden="true" tabindex="-1"></a>                 <span class="at">correlation =</span> <span class="fu">corSymm</span>(<span class="at">form =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> donor),</span>
<span id="cb41-9"><a href="#cb41-9" aria-hidden="true" tabindex="-1"></a>                 <span class="at">weights =</span> <span class="fu">varIdent</span>(<span class="at">form =</span> <span class="sc">~</span> <span class="dv">1</span><span class="sc">|</span>t.by.a),</span>
<span id="cb41-10"><a href="#cb41-10" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp5c)</span>
<span id="cb41-11"><a href="#cb41-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-12"><a href="#cb41-12" aria-hidden="true" tabindex="-1"></a><span class="co"># check AIC</span></span>
<span id="cb41-13"><a href="#cb41-13" aria-hidden="true" tabindex="-1"></a><span class="fu">AIC</span>(exp5c_m1a, exp5c_m1b, exp5c_m1c, exp5c_m1d)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in AIC.default(exp5c_m1a, exp5c_m1b, exp5c_m1c, exp5c_m1d): models are
not all fitted to the same number of observations</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>          df       AIC
exp5c_m1a  6  8.153813
exp5c_m1b  8 10.052676
exp5c_m1c  7  8.052676
exp5c_m1d 15 14.763024</code></pre>
</div>
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="co"># check VarCorr model c</span></span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a><span class="fu">VarCorr</span>(exp5c_m1c) <span class="co"># fine</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> Groups          Name        Std.Dev.
 donor:treatment (Intercept) 0.089519
 donor           (Intercept) 0.352097
 Residual                    0.090685</code></pre>
</div>
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="co"># report 1c (based on AIC and VarCorr check)</span></span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>exp5c_m1 <span class="ot">&lt;-</span> exp5c_m1c</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>exp5c_m1b (equivalent to univariate repeated measures ANOVA) is singular fit. don’t use. Trivial difference in AIC between exp5c_m1a and exp5c_m1c. Nothing in VarCorr with exp5c_m1c raises red flags. Report exp5c_m1c.</p>
<section id="check-the-model" class="level4" data-number="14.8.3.1">
<h4 data-number="14.8.3.1" class="anchored" data-anchor-id="check-the-model"><span class="header-section-number">14.8.3.1</span> Check the model</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggcheck_the_model</span>(exp5c_m1)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>boundary (singular) fit: see help('isSingular')
boundary (singular) fit: see help('isSingular')
boundary (singular) fit: see help('isSingular')
boundary (singular) fit: see help('isSingular')
boundary (singular) fit: see help('isSingular')
boundary (singular) fit: see help('isSingular')
boundary (singular) fit: see help('isSingular')
boundary (singular) fit: see help('isSingular')</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in rlm.default(x, y, weights, method = method, wt.method = wt.method, :
'rlm' failed to converge in 20 steps</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>boundary (singular) fit: see help('isSingular')
boundary (singular) fit: see help('isSingular')
boundary (singular) fit: see help('isSingular')
boundary (singular) fit: see help('isSingular')
boundary (singular) fit: see help('isSingular')
boundary (singular) fit: see help('isSingular')
boundary (singular) fit: see help('isSingular')
boundary (singular) fit: see help('isSingular')</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in rlm.default(x, y, weights, method = method, wt.method = wt.method, :
'rlm' failed to converge in 20 steps</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>boundary (singular) fit: see help('isSingular')
boundary (singular) fit: see help('isSingular')
boundary (singular) fit: see help('isSingular')</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Model failed to converge with 1 negative eigenvalue: -8.3e-02</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>boundary (singular) fit: see help('isSingular')</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>`geom_smooth()` using formula = 'y ~ x'</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="lmm_files/figure-html/lmm-exp5c_m1-check-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>fine.</p>
</section>
<section id="inference-from-the-model-1" class="level4" data-number="14.8.3.2">
<h4 data-number="14.8.3.2" class="anchored" data-anchor-id="inference-from-the-model-1"><span class="header-section-number">14.8.3.2</span> Inference from the model</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a>exp5c_m1_coef <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">coef</span>(<span class="fu">summary</span>(exp5c_m1)),</span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">confint</span>(exp5c_m1)[<span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>),])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a>exp5c_m1_coef <span class="sc">%&gt;%</span></span>
<span id="cb57-2"><a href="#cb57-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>(<span class="at">digits =</span> <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">4</span>,<span class="dv">2</span>,<span class="dv">2</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb57-3"><a href="#cb57-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Estimate</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Std. Error</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">df</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">t value</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Pr(&gt;&amp;#124;t&amp;#124;)</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">2.5 %</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">97.5 %</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">(Intercept)</td>
<td style="text-align: right;">1.34</td>
<td style="text-align: right;">0.153</td>
<td style="text-align: right;">5.8</td>
<td style="text-align: right;">8.8</td>
<td style="text-align: right;">0.0001</td>
<td style="text-align: right;">1.02</td>
<td style="text-align: right;">1.66</td>
</tr>
<tr class="even">
<td style="text-align: left;">treatmentsiNR4A3</td>
<td style="text-align: right;">0.08</td>
<td style="text-align: right;">0.074</td>
<td style="text-align: right;">8.5</td>
<td style="text-align: right;">1.1</td>
<td style="text-align: right;">0.2994</td>
<td style="text-align: right;">-0.07</td>
<td style="text-align: right;">0.23</td>
</tr>
<tr class="odd">
<td style="text-align: left;">activityEPS</td>
<td style="text-align: right;">0.15</td>
<td style="text-align: right;">0.052</td>
<td style="text-align: right;">10.0</td>
<td style="text-align: right;">2.9</td>
<td style="text-align: right;">0.0163</td>
<td style="text-align: right;">0.05</td>
<td style="text-align: right;">0.25</td>
</tr>
<tr class="even">
<td style="text-align: left;">treatmentsiNR4A3:activityEPS</td>
<td style="text-align: right;">-0.27</td>
<td style="text-align: right;">0.074</td>
<td style="text-align: right;">10.0</td>
<td style="text-align: right;">-3.6</td>
<td style="text-align: right;">0.0048</td>
<td style="text-align: right;">-0.41</td>
<td style="text-align: right;">-0.12</td>
</tr>
</tbody>
</table>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a>exp5c_m1_emm <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(exp5c_m1, <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">"treatment"</span>, <span class="st">"activity"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">treatment</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">activity</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">emmean</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">SE</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">df</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">lower.CL</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">upper.CL</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Scr</td>
<td style="text-align: left;">Basal</td>
<td style="text-align: right;">1.34</td>
<td style="text-align: right;">0.153</td>
<td style="text-align: right;">5.8</td>
<td style="text-align: right;">0.96</td>
<td style="text-align: right;">1.72</td>
</tr>
<tr class="even">
<td style="text-align: left;">siNR4A3</td>
<td style="text-align: left;">Basal</td>
<td style="text-align: right;">1.42</td>
<td style="text-align: right;">0.153</td>
<td style="text-align: right;">5.8</td>
<td style="text-align: right;">1.04</td>
<td style="text-align: right;">1.80</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Scr</td>
<td style="text-align: left;">EPS</td>
<td style="text-align: right;">1.49</td>
<td style="text-align: right;">0.153</td>
<td style="text-align: right;">5.8</td>
<td style="text-align: right;">1.11</td>
<td style="text-align: right;">1.87</td>
</tr>
<tr class="even">
<td style="text-align: left;">siNR4A3</td>
<td style="text-align: left;">EPS</td>
<td style="text-align: right;">1.31</td>
<td style="text-align: right;">0.153</td>
<td style="text-align: right;">5.8</td>
<td style="text-align: right;">0.93</td>
<td style="text-align: right;">1.68</td>
</tr>
</tbody>
</table>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a><span class="co"># exp5c_emm # print in console to get row numbers</span></span>
<span id="cb59-2"><a href="#cb59-2" aria-hidden="true" tabindex="-1"></a><span class="co"># set the mean as the row number from the emmeans table</span></span>
<span id="cb59-3"><a href="#cb59-3" aria-hidden="true" tabindex="-1"></a>scr_basal <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>)</span>
<span id="cb59-4"><a href="#cb59-4" aria-hidden="true" tabindex="-1"></a>siNR4A3_basal <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">0</span>,<span class="dv">0</span>)</span>
<span id="cb59-5"><a href="#cb59-5" aria-hidden="true" tabindex="-1"></a>scr_eps <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb59-6"><a href="#cb59-6" aria-hidden="true" tabindex="-1"></a>siNR4A3_eps <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>)</span>
<span id="cb59-7"><a href="#cb59-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-8"><a href="#cb59-8" aria-hidden="true" tabindex="-1"></a>exp5c_m1_planned <span class="ot">&lt;-</span> <span class="fu">contrast</span>(exp5c_m1_emm,</span>
<span id="cb59-9"><a href="#cb59-9" aria-hidden="true" tabindex="-1"></a>                       <span class="at">method =</span> <span class="fu">list</span>(</span>
<span id="cb59-10"><a href="#cb59-10" aria-hidden="true" tabindex="-1"></a>                         <span class="st">"(Scr EPS) - (Scr Basal)"</span> <span class="ot">=</span> <span class="fu">c</span>(scr_eps <span class="sc">-</span> scr_basal),</span>
<span id="cb59-11"><a href="#cb59-11" aria-hidden="true" tabindex="-1"></a>                         <span class="st">"(siNR4A3 EPS) - (siNR4A3 Basal)"</span> <span class="ot">=</span> <span class="fu">c</span>(siNR4A3_eps <span class="sc">-</span> siNR4A3_basal),</span>
<span id="cb59-12"><a href="#cb59-12" aria-hidden="true" tabindex="-1"></a>                         <span class="st">"Interaction"</span> <span class="ot">=</span> <span class="fu">c</span>(siNR4A3_eps <span class="sc">-</span> siNR4A3_basal) <span class="sc">-</span></span>
<span id="cb59-13"><a href="#cb59-13" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">c</span>(scr_eps <span class="sc">-</span> scr_basal)</span>
<span id="cb59-14"><a href="#cb59-14" aria-hidden="true" tabindex="-1"></a>                           </span>
<span id="cb59-15"><a href="#cb59-15" aria-hidden="true" tabindex="-1"></a>                       ),</span>
<span id="cb59-16"><a href="#cb59-16" aria-hidden="true" tabindex="-1"></a>                       <span class="at">adjust =</span> <span class="st">"none"</span></span>
<span id="cb59-17"><a href="#cb59-17" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb59-18"><a href="#cb59-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">contrast</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">estimate</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">SE</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">df</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">lower.CL</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">upper.CL</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">t.ratio</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">p.value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">(Scr EPS) - (Scr Basal)</td>
<td style="text-align: right;">0.15</td>
<td style="text-align: right;">0.052</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">0.03</td>
<td style="text-align: right;">0.27</td>
<td style="text-align: right;">2.88</td>
<td style="text-align: right;">0.016</td>
</tr>
<tr class="even">
<td style="text-align: left;">(siNR4A3 EPS) - (siNR4A3 Basal)</td>
<td style="text-align: right;">-0.12</td>
<td style="text-align: right;">0.052</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">-0.23</td>
<td style="text-align: right;">0.00</td>
<td style="text-align: right;">-2.22</td>
<td style="text-align: right;">0.051</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Interaction</td>
<td style="text-align: right;">-0.27</td>
<td style="text-align: right;">0.074</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">-0.43</td>
<td style="text-align: right;">-0.10</td>
<td style="text-align: right;">-3.61</td>
<td style="text-align: right;">0.005</td>
</tr>
</tbody>
</table>


</div>
</div>
</section>
<section id="lmm-exp5c-plotthemodel" class="level4" data-number="14.8.3.3">
<h4 data-number="14.8.3.3" class="anchored" data-anchor-id="lmm-exp5c-plotthemodel"><span class="header-section-number">14.8.3.3</span> Plot the model</h4>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="lmm_files/figure-html/unnamed-chunk-30-1.png" class="img-fluid figure-img" width="576"></p>
<figcaption>Treatment effect on glucose uptake. Gray dots connected by lines are individual donors.</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="alternaplot-the-model-1" class="level4" data-number="14.8.3.4">
<h4 data-number="14.8.3.4" class="anchored" data-anchor-id="alternaplot-the-model-1"><span class="header-section-number">14.8.3.4</span> Alternaplot the model</h4>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="lmm_files/figure-html/lmm-exp5c-alternaplot-1.png" class="img-fluid figure-img" width="576"></p>
<figcaption>Glucose response to different treatments. Gray dots connected by lines are individual donors. Dashed gray line is expected additive mean of “siNR4A3 EPS”.</figcaption>
</figure>
</div>
</div>
</div>
<p>Notes</p>
<ol type="1">
<li>Many researchers might look at the wide confidence intervals relative to the short distance between the means and think “no effect”. The confidence intervals are correct, they simply are <em>not</em> meant to be tools for inferring anything about differences in means. This is one of many reasons why plots of means and error bars can be misleading for inference, despite the ubiquity of their use for communicating results. And, its why I prefer the <a href="#lmm-exp5c-plotthemodel">effects-and-response plots</a>, which explicitly communicate correct inference about effects.</li>
</ol>
</section>
</section>
<section id="why-we-care-about-modeling-batch-in-exp5c" class="level3" data-number="14.8.4">
<h3 data-number="14.8.4" class="anchored" data-anchor-id="why-we-care-about-modeling-batch-in-exp5c"><span class="header-section-number">14.8.4</span> Why we care about modeling batch in exp5c</h3>
<p>Figure @ref(fig:lmm-exp5c-why) shows the modeled means of the four treatment combinations and the individual values colored by donor. It is pretty easy to see that the glucose uptake values for donors 4 and 5 are well above the mean for all four treatments. And, the values for donors 1, 2, and 3 are well below the mean for all four treatments. The values for donor 6 are near the mean for all four treatments.</p>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="lmm_files/figure-html/lmm-exp5c-why-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption>Why we care about blocking. The black dots are the modeled means of each treatment combination. The colored dots are the measured values of the response for each donor. The position of a donor relative to the mean is easy to see with these data.</figcaption>
</figure>
</div>
</div>
</div>
<p>Let’s compare the effects estimated by the linear mixed model <code>exp5c_m1</code> with a linear model that ignores donor.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a>exp5c_m2 <span class="ot">&lt;-</span> <span class="fu">lm</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity,</span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a>           <span class="at">data =</span> exp5c)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="lmm_files/figure-html/lmm-exp5c-why2-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption>A. Inference from a linear mixed model with blocking factor (donor) added as a random intercept. B. Inference from a fixed effects model.</figcaption>
</figure>
</div>
</div>
</div>
<p>Figure @ref(fig:lmm-exp5c-why2)A is a plot of the effects from the linear mixed model that models the correlated error due to donor. Figure @ref(fig:lmm-exp5c-why2)B is a plot of the effects from the fixed effect model that ignores the correlated error due to donor. Adding <span class="math inline">\(\texttt{donor}\)</span> as a factor to the linear model increases the precision of the estimate of the treatment effects by eliminating the among-donor component of variance from the error variance.</p>
</section>
<section id="the-linear-mixed-model-exp5c_m1-adds-two-random-intercepts" class="level3" data-number="14.8.5">
<h3 data-number="14.8.5" class="anchored" data-anchor-id="the-linear-mixed-model-exp5c_m1-adds-two-random-intercepts"><span class="header-section-number">14.8.5</span> The linear mixed model exp5c_m1 adds two random intercepts</h3>
<p>There are <em>two</em> random intercepts in the linear mixed model <code>exp5c_m1</code> fit to the exp5c data (<code>exp5c_m1</code> is copied from <code>exp5c_m1c</code> in <a href="#lmm-exp5c-m1">Model fit and inference</a>).</p>
<p><span class="math display">\[
\begin{align}
\texttt{glucose_uptake}_j = \ &amp;(\beta_0 + \gamma_{0j} + \gamma_{0jk}) + \beta_1 (\texttt{treatment}_\texttt{siNR4A3}) + \beta_2 (\texttt{activity}_\texttt{EPS}) \ + \\
&amp;\beta_3 (\texttt{treatment}_\texttt{siNR4A3}:\texttt{activity}_\texttt{EPS}) + \varepsilon
\end{align}
(\#eq:lmm-exp5c-m1)
\]</span></p>
<ol type="1">
<li>The random intercept effect <span class="math inline">\(\gamma_{0j}\)</span> models donor variation in the reference level. <em>j</em> indexes donor <em>j</em>. A coefficient is estimated for each donor.</li>
<li>The random intercept effect <span class="math inline">\(\gamma_{0jk}\)</span> models the variation due to the <span class="math inline">\(\texttt{donor}\)</span> by <span class="math inline">\(\texttt{treatment}\)</span> combinations. A coefficient is estimated for each of the 6 (donors) <span class="math inline">\(\times\)</span> 2 (levels of <span class="math inline">\(\texttt{treatment}\)</span>). <em>k</em> indexes treatment level <em>k</em>. This <strong>interaction intercept</strong> is an alternative to random slope for modeling treatment by batch interactions. Unlike a random slope, there is a <span class="math inline">\(\gamma_{0jk}\)</span> coefficient for each donor in the reference level.</li>
</ol>
</section>
<section id="the-fixed-effect-coefficients-of-model-exp5c_m1" class="level3" data-number="14.8.6">
<h3 data-number="14.8.6" class="anchored" data-anchor-id="the-fixed-effect-coefficients-of-model-exp5c_m1"><span class="header-section-number">14.8.6</span> The fixed effect coefficients of model exp5c_m1</h3>
<p>The fixed effect coefficients of model <code>exp5c_m1</code> are</p>
<div class="cell">
<div class="cell-output cell-output-stderr">
<pre><code>Computing profile confidence intervals ...</code></pre>
</div>
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Estimate</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Std. Error</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">df</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">t value</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Pr(&gt;&amp;#124;t&amp;#124;)</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">2.5 %</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">97.5 %</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">(Intercept)</td>
<td style="text-align: right;">1.34</td>
<td style="text-align: right;">0.153</td>
<td style="text-align: right;">5.8</td>
<td style="text-align: right;">8.77</td>
<td style="text-align: right;">0.0001</td>
<td style="text-align: right;">1.02</td>
<td style="text-align: right;">1.66</td>
</tr>
<tr class="even">
<td style="text-align: left;">treatmentsiNR4A3</td>
<td style="text-align: right;">0.08</td>
<td style="text-align: right;">0.074</td>
<td style="text-align: right;">8.5</td>
<td style="text-align: right;">1.11</td>
<td style="text-align: right;">0.2994</td>
<td style="text-align: right;">-0.07</td>
<td style="text-align: right;">0.23</td>
</tr>
<tr class="odd">
<td style="text-align: left;">activityEPS</td>
<td style="text-align: right;">0.15</td>
<td style="text-align: right;">0.052</td>
<td style="text-align: right;">10.0</td>
<td style="text-align: right;">2.88</td>
<td style="text-align: right;">0.0163</td>
<td style="text-align: right;">0.05</td>
<td style="text-align: right;">0.25</td>
</tr>
<tr class="even">
<td style="text-align: left;">treatmentsiNR4A3:activityEPS</td>
<td style="text-align: right;">-0.27</td>
<td style="text-align: right;">0.074</td>
<td style="text-align: right;">10.0</td>
<td style="text-align: right;">-3.61</td>
<td style="text-align: right;">0.0048</td>
<td style="text-align: right;">-0.41</td>
<td style="text-align: right;">-0.12</td>
</tr>
</tbody>
</table>


</div>
</div>
<p>Notes</p>
<ol type="1">
<li>The interpretation of the fixed effectcs coefficients have the <a href="two-factors.html#twoway-what-coefs-are">usual interpretation for a factorial linear model</a>. Figure @ref(fig:lmm-exp5c-explainer-1) is a reminder.</li>
</ol>
<div class="cell">
<div class="cell-output cell-output-stderr">
<pre><code>Warning in geom_point(aes(x = 4, y = b[1] + b[2] + b[3]), color = "gray", : All aesthetics have length 1, but the data has 24 rows.
ℹ Please consider using `annotate()` or provide this layer with data containing
  a single row.</code></pre>
</div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="lmm_files/figure-html/lmm-exp5c-explainer-1-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption>Fixed effects estimated by exp5c_m1. The light gray point is the expected value of the GPR174- F treatment if genotype and sex were additive.</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="the-random-effect-coefficients-of-model-exp5c_m1" class="level3" data-number="14.8.7">
<h3 data-number="14.8.7" class="anchored" data-anchor-id="the-random-effect-coefficients-of-model-exp5c_m1"><span class="header-section-number">14.8.7</span> The random effect coefficients of model exp5c_m1</h3>
<div class="cell">
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<caption>Random intercept effects for each donor (g_0j) and for each donor:treatment combination (g_0jk) for exp5c_m1. There are two random intercepts. random_intercept_j is the sum of the fixed intercept (b_0) and g_0j. random_intercept_jk is the sum of the fixed intercept (b_0) and g_0jk.</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">donor:treatment</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">random_intercept_j</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">random_intercept_jk</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">b_0</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">g_0j</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">g_0jk</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">donor_1:Scr</td>
<td style="text-align: right;">1.018</td>
<td style="text-align: right;">1.349</td>
<td style="text-align: right;">1.341</td>
<td style="text-align: right;">-0.323</td>
<td style="text-align: right;">0.008</td>
</tr>
<tr class="even">
<td style="text-align: left;">donor_1:siNR4A3</td>
<td style="text-align: right;">1.018</td>
<td style="text-align: right;">1.312</td>
<td style="text-align: right;">1.341</td>
<td style="text-align: right;">-0.323</td>
<td style="text-align: right;">-0.029</td>
</tr>
<tr class="odd">
<td style="text-align: left;">donor_2:Scr</td>
<td style="text-align: right;">1.033</td>
<td style="text-align: right;">1.283</td>
<td style="text-align: right;">1.341</td>
<td style="text-align: right;">-0.307</td>
<td style="text-align: right;">-0.058</td>
</tr>
<tr class="even">
<td style="text-align: left;">donor_2:siNR4A3</td>
<td style="text-align: right;">1.033</td>
<td style="text-align: right;">1.379</td>
<td style="text-align: right;">1.341</td>
<td style="text-align: right;">-0.307</td>
<td style="text-align: right;">0.038</td>
</tr>
<tr class="odd">
<td style="text-align: left;">donor_3:Scr</td>
<td style="text-align: right;">1.168</td>
<td style="text-align: right;">1.310</td>
<td style="text-align: right;">1.341</td>
<td style="text-align: right;">-0.173</td>
<td style="text-align: right;">-0.031</td>
</tr>
<tr class="even">
<td style="text-align: left;">donor_3:siNR4A3</td>
<td style="text-align: right;">1.168</td>
<td style="text-align: right;">1.360</td>
<td style="text-align: right;">1.341</td>
<td style="text-align: right;">-0.173</td>
<td style="text-align: right;">0.020</td>
</tr>
<tr class="odd">
<td style="text-align: left;">donor_4:Scr</td>
<td style="text-align: right;">1.785</td>
<td style="text-align: right;">1.302</td>
<td style="text-align: right;">1.341</td>
<td style="text-align: right;">0.445</td>
<td style="text-align: right;">-0.039</td>
</tr>
<tr class="even">
<td style="text-align: left;">donor_4:siNR4A3</td>
<td style="text-align: right;">1.785</td>
<td style="text-align: right;">1.408</td>
<td style="text-align: right;">1.341</td>
<td style="text-align: right;">0.445</td>
<td style="text-align: right;">0.068</td>
</tr>
<tr class="odd">
<td style="text-align: left;">donor_5:Scr</td>
<td style="text-align: right;">1.744</td>
<td style="text-align: right;">1.433</td>
<td style="text-align: right;">1.341</td>
<td style="text-align: right;">0.403</td>
<td style="text-align: right;">0.092</td>
</tr>
<tr class="even">
<td style="text-align: left;">donor_5:siNR4A3</td>
<td style="text-align: right;">1.744</td>
<td style="text-align: right;">1.275</td>
<td style="text-align: right;">1.341</td>
<td style="text-align: right;">0.403</td>
<td style="text-align: right;">-0.066</td>
</tr>
<tr class="odd">
<td style="text-align: left;">donor_6:Scr</td>
<td style="text-align: right;">1.296</td>
<td style="text-align: right;">1.368</td>
<td style="text-align: right;">1.341</td>
<td style="text-align: right;">-0.045</td>
<td style="text-align: right;">0.028</td>
</tr>
<tr class="even">
<td style="text-align: left;">donor_6:siNR4A3</td>
<td style="text-align: right;">1.296</td>
<td style="text-align: right;">1.310</td>
<td style="text-align: right;">1.341</td>
<td style="text-align: right;">-0.045</td>
<td style="text-align: right;">-0.031</td>
</tr>
</tbody>
</table>


</div>
</div>
<p>Notes</p>
<ol type="1">
<li>The random intercept “random_intercept_j” is the sum of the fixed intercept <span class="math inline">\(b_0\)</span> and the random intercept effect <span class="math inline">\(g_{0j}\)</span> for donor <em>j</em>. The <span class="math inline">\(g_{0j}\)</span> estimate the <span class="math inline">\(\gamma_{0j}\)</span> in Model @ref(eq:lmm-exp5c-m1). Note that <span class="math inline">\(g_{0j}\)</span> is the same for both treatment levels within a donor. The illustration of these random effects is similar to that for Example 2 <code>exp6g_m1</code> (Figure @ref(fig:lmm-exp6g-explainer-2).</li>
<li>The random intercept “random_intercept_jk” is the sum of the fixed intercept <span class="math inline">\(b_0\)</span> and the random intercept effect <span class="math inline">\(g_{0jk}\)</span> for donor <em>j</em> in treatment level <em>k</em> (with the reference level equal to 1). The <span class="math inline">\(g_{0jk}\)</span> estimate the <span class="math inline">\(\gamma_{0jk}\)</span> in Model @ref(eq:lmm-exp5c-m1). The <span class="math inline">\(g_{0jk}\)</span> differ for each combination of <span class="math inline">\(\texttt{donor}\)</span> and <span class="math inline">\(\texttt{treatment}\)</span>.</li>
<li>The donor:treatment random intercepts (“random_intercept_jk”) are shown with the dark, colored dots in Figure @ref(fig:lmm-exp5c-explainer-2)A. The distance from the dark colored dot to the dashed, gray line (the modeled mean for the treatment:activity combination) is the random intercept effect estimate of <span class="math inline">\(\gamma_{0jk}\)</span>.</li>
<li>The set of random intercept effects for the donor:treatment combinations (<span class="math inline">\(g_{0jk}\)</span>) are the same between the two activity levels. This is seen in Figure @ref(fig:lmm-exp5c-explainer-2)A - compare the pattern of dark, colored dots in the two EPS treatments to the two Basal treatments.</li>
<li>The combined random intercept effects (<span class="math inline">\(g_{0j} + g_{0jk}\)</span>) are shown in Figure @ref(fig:lmm-exp5c-explainer-2)B as the distance between the dark, colored dots and the dashed, gray lines.</li>
<li>There is some communication decision-making in Figure @ref(fig:lmm-exp5c-explainer-2). The actual y-values of the dark, colored dots are the random intercept effect plus the treatment-combination modeled mean. Thus, in A, the dark colored dots are equal to “random_intercept_jk” only at the reference level (since the reference model mean is the fixed intercept).</li>
</ol>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="lmm_files/figure-html/lmm-exp5c-explainer-2-1.png" class="img-fluid figure-img" width="1066"></p>
<figcaption>Random intercepts for model exp5c_m1. The pale, colored dots are the measured glucose uptake values for each donor at each treatment combination. The dashed, gray lines are the modeled means for each treatment combination. (A) The distance from a dark colored dot to the dashed, gray line is the random intercept effect <span class="math inline">\(g_{0jk}\)</span>. (B) The distance from a dark colored dot to the dashed, gray line is the combined random intercept effect <span class="math inline">\(g_{0j} + g_{0jk}\)</span>.</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="lmm-exp5c-alt" class="level3" data-number="14.8.8">
<h3 data-number="14.8.8" class="anchored" data-anchor-id="lmm-exp5c-alt"><span class="header-section-number">14.8.8</span> Alternative models for exp5c</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb63"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a>exp5c_m1a <span class="ot">&lt;-</span> <span class="fu">lmer</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb63-2"><a href="#cb63-2" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor),</span>
<span id="cb63-3"><a href="#cb63-3" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp5c)</span>
<span id="cb63-4"><a href="#cb63-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-5"><a href="#cb63-5" aria-hidden="true" tabindex="-1"></a>exp5c_m1b <span class="ot">&lt;-</span> <span class="fu">lmer</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb63-6"><a href="#cb63-6" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor) <span class="sc">+</span></span>
<span id="cb63-7"><a href="#cb63-7" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor<span class="sc">:</span>treatment) <span class="sc">+</span></span>
<span id="cb63-8"><a href="#cb63-8" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor<span class="sc">:</span>activity),</span>
<span id="cb63-9"><a href="#cb63-9" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp5c)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>boundary (singular) fit: see help('isSingular')</code></pre>
</div>
<div class="sourceCode cell-code" id="cb65"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a>exp5c_m1c <span class="ot">&lt;-</span> <span class="fu">lmer</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb65-2"><a href="#cb65-2" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor) <span class="sc">+</span></span>
<span id="cb65-3"><a href="#cb65-3" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor<span class="sc">:</span>treatment),</span>
<span id="cb65-4"><a href="#cb65-4" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp5c)</span>
<span id="cb65-5"><a href="#cb65-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb65-6"><a href="#cb65-6" aria-hidden="true" tabindex="-1"></a>exp5c_m1d <span class="ot">&lt;-</span> <span class="fu">lme</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity,</span>
<span id="cb65-7"><a href="#cb65-7" aria-hidden="true" tabindex="-1"></a>                 <span class="at">random =</span>  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> donor,</span>
<span id="cb65-8"><a href="#cb65-8" aria-hidden="true" tabindex="-1"></a>                 <span class="at">correlation =</span> <span class="fu">corSymm</span>(<span class="at">form =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> donor),</span>
<span id="cb65-9"><a href="#cb65-9" aria-hidden="true" tabindex="-1"></a>                 <span class="at">weights =</span> <span class="fu">varIdent</span>(<span class="at">form =</span> <span class="sc">~</span> <span class="dv">1</span><span class="sc">|</span>t.by.a),</span>
<span id="cb65-10"><a href="#cb65-10" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp5c)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Notes</p>
<ol type="1">
<li>Four models are fit. The models specify the same fixed effects but different random effects or patterns of correlated error. Again, we care about the fixed effects – this is the point of the experiment – but the specification of the random effects determines the error variance and degrees of freedom for computing uncertainty.</li>
<li>Model <code>exp5c_m1a</code> specifies a random intercept for each level of <span class="math inline">\(\texttt{donor}\)</span>.</li>
<li>Model <code>exp5c_m1b</code> adds two additional random intercepts to Model <code>exp5c_m1a</code>. The code <code>(1 | donor:treatment)</code> adds an intercept for each combination of <span class="math inline">\(\texttt{donor}\)</span> and <span class="math inline">\(\texttt{treatment}\)</span>. The code <code>(1 | donor:activity)</code> adds an intercept for each combination of <span class="math inline">\(\texttt{donor}\)</span> and <span class="math inline">\(\texttt{activity}\)</span>. This model is the equivalent of the univariate model of a <strong>repeated measures ANOVA</strong> of these data (see <a href="#lmm-rmanova-5c">Classical (“univariate model”) repeated measures ANOVA</a> below).</li>
<li>Model <code>exp5c_m1c</code> adds only one additional random intercept (<code>(1 | donor:treatment)</code>). It is a simplification of Model <code>exp5c_m1c</code>. This is the model reported and explained above.</li>
<li>Model <code>exp5c_m1d</code> has the same random effects as <code>exp5c_m1a</code> but models unstructured correlated error and heterogeneity of the error, as described for Example 2: <a href="#lmm-exp6g-alt">Alternative models for exp6g</a>. This model is equivalent to the multivariate model of a <strong>repeated measures ANOVA</strong> of these data (see <a href="#lmm-rmanova-5c-multi">“Multivariate model” repeated measures ANOVA</a> below).</li>
<li><code>lmer</code> returns a message “boundary (singular) fit: see ?isSingular” for <code>exp5c_m1b</code>. The model <em>was</em> fit but the message means that we should be cautious about (or simply avoid) interpreting any inferential statistics (SEs, CIs, <em>p</em>-values). Looking at the estimates of the estimated parameters of the random effects in the table below (the variances of the effects and the covariances among the effects), Model <code>exp5c_m1b</code> estimates zero variance for the <span class="math inline">\(\texttt{donor:activity}\)</span> random intercept. This suggests that we could simplify <code>exp5c_m1b</code> by removing <code>(1 | donor:activity)</code> from the model – this is <code>exp5c_m1c</code>. Think about this warning. If we fit using repeated measures ANOVA (next section), we won’t get a warning message but we still may be <a href="https://en.wikipedia.org/wiki/Overfitting" target="_blank">overfitting</a>.</li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a><span class="fu">VarCorr</span>(exp5c_m1b)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> Groups          Name        Std.Dev.
 donor:activity  (Intercept) 0.000000
 donor:treatment (Intercept) 0.089519
 donor           (Intercept) 0.352092
 Residual                    0.090686</code></pre>
</div>
</div>
<ol start="7" type="1">
<li>Planned comparisons of the four alternative models are given in Table @ref(tab:lmm-exp5c-alternatives-planned). The effect estimates are the same but inference differs slightly among the models because of how each model partitions error variance to either random effects or the residuals.</li>
</ol>
<div class="cell">
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<caption>Planned contrasts from the four alternative models.</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">contrast</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">estimate</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">SE</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">df</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">lower.CL</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">upper.CL</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">t.ratio</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">p.value</th>
</tr>
</thead>
<tbody>
<tr class="odd" data-grouplength="3">
<td colspan="8" style="border-bottom: 1px solid"><strong>exp5c_m1a</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(Scr EPS) - (Scr Basal)</td>
<td style="text-align: right;">0.15</td>
<td style="text-align: right;">0.067</td>
<td style="text-align: right;">15</td>
<td style="text-align: right;">0.01</td>
<td style="text-align: right;">0.29</td>
<td style="text-align: right;">2.25</td>
<td style="text-align: right;">0.040</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(siNR4A3 EPS) - (siNR4A3 Basal)</td>
<td style="text-align: right;">-0.12</td>
<td style="text-align: right;">0.067</td>
<td style="text-align: right;">15</td>
<td style="text-align: right;">-0.26</td>
<td style="text-align: right;">0.03</td>
<td style="text-align: right;">-1.73</td>
<td style="text-align: right;">0.105</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Interaction</td>
<td style="text-align: right;">-0.27</td>
<td style="text-align: right;">0.095</td>
<td style="text-align: right;">15</td>
<td style="text-align: right;">-0.47</td>
<td style="text-align: right;">-0.06</td>
<td style="text-align: right;">-2.81</td>
<td style="text-align: right;">0.013</td>
</tr>
<tr class="odd" data-grouplength="3">
<td colspan="8" style="border-bottom: 1px solid"><strong>exp5c_m1b</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(Scr EPS) - (Scr Basal)</td>
<td style="text-align: right;">0.15</td>
<td style="text-align: right;">0.052</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">0.03</td>
<td style="text-align: right;">0.27</td>
<td style="text-align: right;">2.88</td>
<td style="text-align: right;">0.016</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(siNR4A3 EPS) - (siNR4A3 Basal)</td>
<td style="text-align: right;">-0.12</td>
<td style="text-align: right;">0.052</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">-0.23</td>
<td style="text-align: right;">0.00</td>
<td style="text-align: right;">-2.22</td>
<td style="text-align: right;">0.051</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Interaction</td>
<td style="text-align: right;">-0.27</td>
<td style="text-align: right;">0.074</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">-0.46</td>
<td style="text-align: right;">-0.08</td>
<td style="text-align: right;">-3.61</td>
<td style="text-align: right;">0.015</td>
</tr>
<tr class="odd" data-grouplength="3">
<td colspan="8" style="border-bottom: 1px solid"><strong>exp5c_m1c</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(Scr EPS) - (Scr Basal)</td>
<td style="text-align: right;">0.15</td>
<td style="text-align: right;">0.052</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">0.03</td>
<td style="text-align: right;">0.27</td>
<td style="text-align: right;">2.88</td>
<td style="text-align: right;">0.016</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(siNR4A3 EPS) - (siNR4A3 Basal)</td>
<td style="text-align: right;">-0.12</td>
<td style="text-align: right;">0.052</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">-0.23</td>
<td style="text-align: right;">0.00</td>
<td style="text-align: right;">-2.22</td>
<td style="text-align: right;">0.051</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Interaction</td>
<td style="text-align: right;">-0.27</td>
<td style="text-align: right;">0.074</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">-0.43</td>
<td style="text-align: right;">-0.10</td>
<td style="text-align: right;">-3.61</td>
<td style="text-align: right;">0.005</td>
</tr>
<tr class="odd" data-grouplength="3">
<td colspan="8" style="border-bottom: 1px solid"><strong>exp5c_m1d</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(Scr EPS) - (Scr Basal)</td>
<td style="text-align: right;">0.15</td>
<td style="text-align: right;">0.040</td>
<td style="text-align: right;">15</td>
<td style="text-align: right;">0.07</td>
<td style="text-align: right;">0.24</td>
<td style="text-align: right;">3.82</td>
<td style="text-align: right;">0.002</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(siNR4A3 EPS) - (siNR4A3 Basal)</td>
<td style="text-align: right;">-0.12</td>
<td style="text-align: right;">0.063</td>
<td style="text-align: right;">15</td>
<td style="text-align: right;">-0.25</td>
<td style="text-align: right;">0.02</td>
<td style="text-align: right;">-1.85</td>
<td style="text-align: right;">0.083</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Interaction</td>
<td style="text-align: right;">-0.27</td>
<td style="text-align: right;">0.083</td>
<td style="text-align: right;">15</td>
<td style="text-align: right;">-0.44</td>
<td style="text-align: right;">-0.09</td>
<td style="text-align: right;">-3.22</td>
<td style="text-align: right;">0.006</td>
</tr>
</tbody>
</table>


</div>
</div>
<div class="cell">
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<caption>AIC of the four alternative models firt to exp5c data</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">Model</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">AIC</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">exp5c_m1a</td>
<td style="text-align: right;">8.153813</td>
</tr>
<tr class="even">
<td style="text-align: left;">exp5c_m1b</td>
<td style="text-align: right;">10.052675</td>
</tr>
<tr class="odd">
<td style="text-align: left;">exp5c_m1c</td>
<td style="text-align: right;">8.052675</td>
</tr>
<tr class="even">
<td style="text-align: left;">exp5c_m1d</td>
<td style="text-align: right;">14.763024</td>
</tr>
</tbody>
</table>


</div>
</div>
<ol start="6" type="1">
<li>The use of AIC for selecting among models with different random effects was described above. The AICs of the models suggest that models <code>exp5c_m1b</code> and <code>exp5c_m1d</code> are too complex given the data. The AICs of models <code>exp5c_m1a</code> and <code>exp5c_m1c</code> are effectively the same. I reported <code>exp5c_m1c</code> simply to allow me to explain the <code>donor:treatment</code> random effect.</li>
</ol>
</section>
<section id="lmm-rmanova-5c" class="level3" data-number="14.8.9">
<h3 data-number="14.8.9" class="anchored" data-anchor-id="lmm-rmanova-5c"><span class="header-section-number">14.8.9</span> Classical (“univariate model”) repeated measures ANOVA</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb68"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a>exp5c_aov1 <span class="ot">&lt;-</span> <span class="fu">aov_4</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb68-2"><a href="#cb68-2" aria-hidden="true" tabindex="-1"></a>                    (treatment <span class="sc">*</span> activity <span class="sc">|</span> donor),</span>
<span id="cb68-3"><a href="#cb68-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> exp5c)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stderr">
<pre><code>aov object missing, substituting multivariate/lm model.
to get univariate tests, refit ANOVA with include_aov = TRUE</code></pre>
</div>
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<caption>Planned contrasts for the linear mixed model exp5c_m1c and the univariate model of the repeated measures ANOVA.</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">contrast</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">estimate</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">SE</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">df</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">lower.CL</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">upper.CL</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">t.ratio</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">p.value</th>
</tr>
</thead>
<tbody>
<tr class="odd" data-grouplength="3">
<td colspan="8" style="border-bottom: 1px solid"><strong>exp5c_m1c</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(Scr EPS) - (Scr Basal)</td>
<td style="text-align: right;">0.15</td>
<td style="text-align: right;">0.052</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">0.03</td>
<td style="text-align: right;">0.27</td>
<td style="text-align: right;">2.88</td>
<td style="text-align: right;">0.016</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(siNR4A3 EPS) - (siNR4A3 Basal)</td>
<td style="text-align: right;">-0.12</td>
<td style="text-align: right;">0.052</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">-0.23</td>
<td style="text-align: right;">0.00</td>
<td style="text-align: right;">-2.22</td>
<td style="text-align: right;">0.051</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Interaction</td>
<td style="text-align: right;">-0.27</td>
<td style="text-align: right;">0.074</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">-0.43</td>
<td style="text-align: right;">-0.10</td>
<td style="text-align: right;">-3.61</td>
<td style="text-align: right;">0.005</td>
</tr>
<tr class="odd" data-grouplength="3">
<td colspan="8" style="border-bottom: 1px solid"><strong>univariate RM-ANOVA</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(Scr EPS) - (Scr Basal)</td>
<td style="text-align: right;">0.15</td>
<td style="text-align: right;">0.040</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">0.05</td>
<td style="text-align: right;">0.25</td>
<td style="text-align: right;">3.82</td>
<td style="text-align: right;">0.012</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(siNR4A3 EPS) - (siNR4A3 Basal)</td>
<td style="text-align: right;">-0.12</td>
<td style="text-align: right;">0.063</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">-0.28</td>
<td style="text-align: right;">0.04</td>
<td style="text-align: right;">-1.85</td>
<td style="text-align: right;">0.123</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Interaction</td>
<td style="text-align: right;">-0.27</td>
<td style="text-align: right;">0.083</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">-0.48</td>
<td style="text-align: right;">-0.05</td>
<td style="text-align: right;">-3.22</td>
<td style="text-align: right;">0.023</td>
</tr>
</tbody>
</table>


</div>
</div>
<p>Notes</p>
<ol type="1">
<li>In addition to the assumptions of the linear model outlined in the Violations chapter, the classical (“univariate model”) repeated measures ANOVA assumes <a href="https://en.wikipedia.org/wiki/Mauchly%27s_sphericity_test#Sphericity" target="_blank">sphericity</a>, which is the equality of the variances of all pairwise differences among treatment combinations.</li>
<li>The univariate model of the repeated measures ANOVA is equivalent to Model <code>exp5c_m1c</code>. Inference (SEs, CIs, <em>p</em>-values) differs slightly in this here because the estimated variance for the <span class="math inline">\(\texttt{donor:activity}\)</span> random intercept of Model <code>exp5c_m1c</code> is zero. In cases where all variances are greater than zero <em>and the design is balanced</em> (no block is missing a treatment combination), the two tables would be identical.</li>
</ol>
</section>
<section id="lmm-rmanova-5c-multi" class="level3" data-number="14.8.10">
<h3 data-number="14.8.10" class="anchored" data-anchor-id="lmm-rmanova-5c-multi"><span class="header-section-number">14.8.10</span> “Multivariate model” repeated measures ANOVA of exp5c</h3>
<div class="cell">
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<caption>Planned contrasts for the linear mixed model exp5c_m1d and the multivariate model of the repeated measures ANOVA.</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">contrast</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">estimate</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">SE</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">df</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">lower.CL</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">upper.CL</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">t.ratio</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">p.value</th>
</tr>
</thead>
<tbody>
<tr class="odd" data-grouplength="3">
<td colspan="8" style="border-bottom: 1px solid"><strong>exp5c_m1d</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(Scr EPS) - (Scr Basal)</td>
<td style="text-align: right;">0.15</td>
<td style="text-align: right;">0.040</td>
<td style="text-align: right;">15</td>
<td style="text-align: right;">0.07</td>
<td style="text-align: right;">0.24</td>
<td style="text-align: right;">3.82</td>
<td style="text-align: right;">0.002</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(siNR4A3 EPS) - (siNR4A3 Basal)</td>
<td style="text-align: right;">-0.12</td>
<td style="text-align: right;">0.063</td>
<td style="text-align: right;">15</td>
<td style="text-align: right;">-0.25</td>
<td style="text-align: right;">0.02</td>
<td style="text-align: right;">-1.85</td>
<td style="text-align: right;">0.083</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Interaction</td>
<td style="text-align: right;">-0.27</td>
<td style="text-align: right;">0.083</td>
<td style="text-align: right;">15</td>
<td style="text-align: right;">-0.44</td>
<td style="text-align: right;">-0.09</td>
<td style="text-align: right;">-3.22</td>
<td style="text-align: right;">0.006</td>
</tr>
<tr class="odd" data-grouplength="3">
<td colspan="8" style="border-bottom: 1px solid"><strong>multivariate RM-ANOVA</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(Scr EPS) - (Scr Basal)</td>
<td style="text-align: right;">0.15</td>
<td style="text-align: right;">0.040</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">0.05</td>
<td style="text-align: right;">0.25</td>
<td style="text-align: right;">3.82</td>
<td style="text-align: right;">0.012</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(siNR4A3 EPS) - (siNR4A3 Basal)</td>
<td style="text-align: right;">-0.12</td>
<td style="text-align: right;">0.063</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">-0.28</td>
<td style="text-align: right;">0.04</td>
<td style="text-align: right;">-1.85</td>
<td style="text-align: right;">0.123</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Interaction</td>
<td style="text-align: right;">-0.27</td>
<td style="text-align: right;">0.083</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">-0.48</td>
<td style="text-align: right;">-0.05</td>
<td style="text-align: right;">-3.22</td>
<td style="text-align: right;">0.023</td>
</tr>
</tbody>
</table>


</div>
</div>
<p>Notes</p>
<ol type="1">
<li>The sphericity assumption of the repeated measures ANOVA is relaxed if the model is fit using the “multivariate model”.</li>
<li>The multivariate model repeated measures ANOVA is a linear model with a multivariate response and not a linear mixed model. In the multivariate model, each treatment combination is a different response variable and there is a single row for each level of the random factor (<span class="math inline">\(\texttt{donor}\)</span> in Experiment 5c). The multivariate model is a different way of handling the correlated error that occurs when conceiving of the design as univariate.</li>
<li>The multivariate repeated measures ANOVA can be specified using the linear mixed model <code>exp5c_m1d</code>. Compared to the contrasts from the multivariate model of the repeated measures ANOVA above, the SEs of the contrasts are the same but the degrees of freedom differ. Because of the increased df of the linear mixed model, the CIs are narrower and the <em>p</em>-value is smaller – that is the linear mixed model is less conservative than the repeated measures ANOVA. There is no correct degrees of freedom for a linear mixed model like this and emmeans outputs one way to compute these (with options for others, none of which are equivalent to those from the multivariate model RM-ANOVA).</li>
</ol>
</section>
<section id="modeling-textttdonor-as-a-fixed-effect" class="level3" data-number="14.8.11">
<h3 data-number="14.8.11" class="anchored" data-anchor-id="modeling-textttdonor-as-a-fixed-effect"><span class="header-section-number">14.8.11</span> Modeling <span class="math inline">\(\texttt{donor}\)</span> as a fixed effect</h3>
<p><span class="math display">\[
\begin{equation}
\texttt{glucose_uptake} \sim \texttt{treatment * activity + donor}
\end{equation}
(\#eq:lmm-exp5c-m3)
\]</span></p>
<p>Model @ref(eq:lmm-exp5c-m3) (using R formula syntax) is a linear model with the block <span class="math inline">\(\texttt{donor}\)</span> added as a fixed covariate instead of a random intercept.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a>exp5c_m3 <span class="ot">&lt;-</span> <span class="fu">lm</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span> donor,</span>
<span id="cb70-2"><a href="#cb70-2" aria-hidden="true" tabindex="-1"></a>               <span class="at">data =</span> exp5c)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>The coefficients of the fit model are</p>
<div class="cell">
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Estimate</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Std. Error</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">t value</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Pr(&gt;&amp;#124;t&amp;#124;)</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">2.5 %</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">97.5 %</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">(Intercept)</td>
<td style="text-align: right;">1.00</td>
<td style="text-align: right;">0.071</td>
<td style="text-align: right;">14.0</td>
<td style="text-align: right;">0.0000</td>
<td style="text-align: right;">0.85</td>
<td style="text-align: right;">1.15</td>
</tr>
<tr class="even">
<td style="text-align: left;">treatmentsiNR4A3</td>
<td style="text-align: right;">0.08</td>
<td style="text-align: right;">0.067</td>
<td style="text-align: right;">1.2</td>
<td style="text-align: right;">0.2453</td>
<td style="text-align: right;">-0.06</td>
<td style="text-align: right;">0.22</td>
</tr>
<tr class="odd">
<td style="text-align: left;">activityEPS</td>
<td style="text-align: right;">0.15</td>
<td style="text-align: right;">0.067</td>
<td style="text-align: right;">2.2</td>
<td style="text-align: right;">0.0402</td>
<td style="text-align: right;">0.01</td>
<td style="text-align: right;">0.29</td>
</tr>
<tr class="even">
<td style="text-align: left;">donordonor_2</td>
<td style="text-align: right;">0.02</td>
<td style="text-align: right;">0.082</td>
<td style="text-align: right;">0.2</td>
<td style="text-align: right;">0.8444</td>
<td style="text-align: right;">-0.16</td>
<td style="text-align: right;">0.19</td>
</tr>
<tr class="odd">
<td style="text-align: left;">donordonor_3</td>
<td style="text-align: right;">0.16</td>
<td style="text-align: right;">0.082</td>
<td style="text-align: right;">1.9</td>
<td style="text-align: right;">0.0757</td>
<td style="text-align: right;">-0.02</td>
<td style="text-align: right;">0.33</td>
</tr>
<tr class="even">
<td style="text-align: left;">donordonor_4</td>
<td style="text-align: right;">0.81</td>
<td style="text-align: right;">0.082</td>
<td style="text-align: right;">9.8</td>
<td style="text-align: right;">0.0000</td>
<td style="text-align: right;">0.63</td>
<td style="text-align: right;">0.98</td>
</tr>
<tr class="odd">
<td style="text-align: left;">donordonor_5</td>
<td style="text-align: right;">0.76</td>
<td style="text-align: right;">0.082</td>
<td style="text-align: right;">9.2</td>
<td style="text-align: right;">0.0000</td>
<td style="text-align: right;">0.59</td>
<td style="text-align: right;">0.94</td>
</tr>
<tr class="even">
<td style="text-align: left;">donordonor_6</td>
<td style="text-align: right;">0.29</td>
<td style="text-align: right;">0.082</td>
<td style="text-align: right;">3.5</td>
<td style="text-align: right;">0.0030</td>
<td style="text-align: right;">0.12</td>
<td style="text-align: right;">0.47</td>
</tr>
<tr class="odd">
<td style="text-align: left;">treatmentsiNR4A3:activityEPS</td>
<td style="text-align: right;">-0.27</td>
<td style="text-align: right;">0.095</td>
<td style="text-align: right;">-2.8</td>
<td style="text-align: right;">0.0132</td>
<td style="text-align: right;">-0.47</td>
<td style="text-align: right;">-0.06</td>
</tr>
</tbody>
</table>


</div>
</div>
<p>Notes</p>
<ol type="1">
<li>The coefficients include a slope for the five non-reference levels of donor. We typically don’t care about these.</li>
</ol>
<div class="cell">
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<caption>Planned contrasts for the linear mixed model exp5c_m1a and the fixed effect model exp5c_m3.</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">contrast</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">estimate</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">SE</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">df</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">lower.CL</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">upper.CL</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">t.ratio</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">p.value</th>
</tr>
</thead>
<tbody>
<tr class="odd" data-grouplength="3">
<td colspan="8" style="border-bottom: 1px solid"><strong>exp5c_m1a</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(Scr EPS) - (Scr Basal)</td>
<td style="text-align: right;">0.15</td>
<td style="text-align: right;">0.067</td>
<td style="text-align: right;">15</td>
<td style="text-align: right;">0.01</td>
<td style="text-align: right;">0.29</td>
<td style="text-align: right;">2.25</td>
<td style="text-align: right;">0.040</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(siNR4A3 EPS) - (siNR4A3 Basal)</td>
<td style="text-align: right;">-0.12</td>
<td style="text-align: right;">0.067</td>
<td style="text-align: right;">15</td>
<td style="text-align: right;">-0.26</td>
<td style="text-align: right;">0.03</td>
<td style="text-align: right;">-1.73</td>
<td style="text-align: right;">0.105</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Interaction</td>
<td style="text-align: right;">-0.27</td>
<td style="text-align: right;">0.095</td>
<td style="text-align: right;">15</td>
<td style="text-align: right;">-0.47</td>
<td style="text-align: right;">-0.06</td>
<td style="text-align: right;">-2.81</td>
<td style="text-align: right;">0.013</td>
</tr>
<tr class="odd" data-grouplength="3">
<td colspan="8" style="border-bottom: 1px solid"><strong>exp5c_m3</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(Scr EPS) - (Scr Basal)</td>
<td style="text-align: right;">0.15</td>
<td style="text-align: right;">0.067</td>
<td style="text-align: right;">15</td>
<td style="text-align: right;">0.01</td>
<td style="text-align: right;">0.29</td>
<td style="text-align: right;">2.25</td>
<td style="text-align: right;">0.040</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(siNR4A3 EPS) - (siNR4A3 Basal)</td>
<td style="text-align: right;">-0.12</td>
<td style="text-align: right;">0.067</td>
<td style="text-align: right;">15</td>
<td style="text-align: right;">-0.26</td>
<td style="text-align: right;">0.03</td>
<td style="text-align: right;">-1.73</td>
<td style="text-align: right;">0.105</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Interaction</td>
<td style="text-align: right;">-0.27</td>
<td style="text-align: right;">0.095</td>
<td style="text-align: right;">15</td>
<td style="text-align: right;">-0.47</td>
<td style="text-align: right;">-0.06</td>
<td style="text-align: right;">-2.81</td>
<td style="text-align: right;">0.013</td>
</tr>
</tbody>
</table>


</div>
</div>
<p>Notes</p>
<ol type="1">
<li>In a randomized complete block design with only 1 replicate of each treatment combination per block, like that in Experiment 5c, inference about treatment effects is exactly that same between this fixed effect model and the random intercept model <code>exp5c_m1a.</code></li>
<li>The equivalence of inference in the treatment effect between the fixed effect and linear mixed model holds only for balanced randomized complete block designs – where all blocks contain all treatment combinations and the subsampling replicate size is the same for all treatment combinations for all blocks. This means, outside of special cases like Experiment 5c, the choice between adding a blocking variable as a random or fixed factor depends on assumptions about the model.</li>
</ol>
</section>
</section>
<section id="lmm-example4" class="level2" data-number="14.9">
<h2 data-number="14.9" class="anchored" data-anchor-id="lmm-example4"><span class="header-section-number">14.9</span> Example 4 – Experiments with subsampling replication (exp1g)</h2>
<p>This example is from a design with batches (independent experiments) that contain all treatment levels of a single factor <em>and</em> subsampling replication. These data were used to introduce linear mixed models in Example 1. The design of the experiment is <span class="math inline">\(2 \times 2\)</span> factorial. Example 1 flattened the analysis to simplify explanation of random intercepts and random slopes. Here, the data are analyzed with a factorial model.</p>
<p><a href="https://www.nature.com/articles/s41586-019-1873-0" target="_blank">A GPR174–CCL21 module imparts sexual dimorphism to humoral immunity</a></p>
<p><a href="https://pubmed.ncbi.nlm.nih.gov/31875850/" target="_blank">Public source</a></p>
<p>Source figure: <a href="https://www.nature.com/articles/s41586-019-1873-0/figures/1" target="_blank">Fig. 1g</a></p>
<p>Source data: <a href="https://www.nature.com/articles/s41586-019-1873-0#MOESM3" target="_blank">Source Data Fig. 1</a></p>
<section id="understand-the-data-2" class="level3" data-number="14.9.1">
<h3 data-number="14.9.1" class="anchored" data-anchor-id="understand-the-data-2"><span class="header-section-number">14.9.1</span> Understand the data</h3>
<p>The researchers in this paper are interested in discovering mechanisms causing the lower <a href="https://en.wikipedia.org/wiki/Humoral_immunity" target="_blank">antibody-mediated immune response</a> in males relative to females. The data in Fig. 1 are from a set of experiments on mice to investigate how the G-protein coupled receptor protein GPR174 regulates formation of the <a href="https://en.wikipedia.org/wiki/Germinal_center" target="_blank">B-cell germinal center in secondary lymph tissue</a>. GPR174 is a X-linked gene.</p>
<p><strong>Response variable</strong> <span class="math inline">\(\texttt{gc}\)</span> – germinal center size (%). The units are the percent of cells expressing germinal center markers.</p>
<p><strong>Factor 1</strong> – <span class="math inline">\(\texttt{sex}\)</span> (“M”, “F”). Male (“M”) is the reference level.</p>
<p><strong>Factor 2</strong> – <span class="math inline">\(\texttt{chromosome}\)</span> (“Gpr174+”, “Gpr174-”). “Gpr174-” is a GPR174 knockout. The wildtype (“Gpr174+”) condition is the reference level.</p>
<p><strong>Design</strong> – <span class="math inline">\(2 \times 2\)</span>, that is, two crossed factors each with two levels. This results in four groups, each with a unique combination of the levels from each factor. “M Gpr174+” is the control. “M Gpr174+” is the knockout genotype in males (“knockout added”). “F Gpr174+” is the wildtype female (“X chromosome added”). “F Gpr174-” is the knockout female (“knockout and X chromosome added”.</p>
</section>
<section id="examine-the-data-1" class="level3" data-number="14.9.2">
<h3 data-number="14.9.2" class="anchored" data-anchor-id="examine-the-data-1"><span class="header-section-number">14.9.2</span> Examine the data</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb71"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> exp1g,</span>
<span id="cb71-2"><a href="#cb71-2" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x =</span> treatment,</span>
<span id="cb71-3"><a href="#cb71-3" aria-hidden="true" tabindex="-1"></a>           <span class="at">y =</span> gc,</span>
<span id="cb71-4"><a href="#cb71-4" aria-hidden="true" tabindex="-1"></a>           <span class="at">color =</span> experiment_id)) <span class="sc">+</span></span>
<span id="cb71-5"><a href="#cb71-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="fl">0.4</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="lmm_files/figure-html/lmm-exp1g-examine-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="fit-the-model-1" class="level3" data-number="14.9.3">
<h3 data-number="14.9.3" class="anchored" data-anchor-id="fit-the-model-1"><span class="header-section-number">14.9.3</span> Fit the model</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb72"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="co"># three slope parameters</span></span>
<span id="cb72-2"><a href="#cb72-2" aria-hidden="true" tabindex="-1"></a>exp1g_m1a <span class="ot">&lt;-</span> <span class="fu">lmer</span>(gc <span class="sc">~</span> genotype <span class="sc">*</span> sex <span class="sc">+</span></span>
<span id="cb72-3"><a href="#cb72-3" aria-hidden="true" tabindex="-1"></a>                   (genotype <span class="sc">*</span> sex <span class="sc">|</span> experiment_id),</span>
<span id="cb72-4"><a href="#cb72-4" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp1g)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>boundary (singular) fit: see help('isSingular')</code></pre>
</div>
<div class="sourceCode cell-code" id="cb74"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="fu">VarCorr</span>(exp1g_m1a) <span class="co"># looks fine</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> Groups        Name                 Std.Dev. Corr                
 experiment_id (Intercept)          1.77088                      
               genotypeGpr174-      0.96193  -0.023              
               sexF                 2.90154   0.459  0.878       
               genotypeGpr174-:sexF 1.33170  -0.314 -0.625 -0.706
 Residual                           1.93155                      </code></pre>
</div>
<div class="sourceCode cell-code" id="cb76"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a><span class="co"># one slope parameter but capturing all treatment combinations</span></span>
<span id="cb76-2"><a href="#cb76-2" aria-hidden="true" tabindex="-1"></a>exp1g_m1b <span class="ot">&lt;-</span> <span class="fu">lmer</span>(gc <span class="sc">~</span> genotype <span class="sc">*</span> sex <span class="sc">+</span></span>
<span id="cb76-3"><a href="#cb76-3" aria-hidden="true" tabindex="-1"></a>                   (treatment <span class="sc">|</span> experiment_id),</span>
<span id="cb76-4"><a href="#cb76-4" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp1g)</span>
<span id="cb76-5"><a href="#cb76-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-6"><a href="#cb76-6" aria-hidden="true" tabindex="-1"></a><span class="co"># intercept interactions</span></span>
<span id="cb76-7"><a href="#cb76-7" aria-hidden="true" tabindex="-1"></a>exp1g_m1c <span class="ot">&lt;-</span> <span class="fu">lmer</span>(gc <span class="sc">~</span> genotype <span class="sc">*</span> sex <span class="sc">+</span></span>
<span id="cb76-8"><a href="#cb76-8" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> experiment_id) <span class="sc">+</span></span>
<span id="cb76-9"><a href="#cb76-9" aria-hidden="true" tabindex="-1"></a>                    (<span class="dv">1</span> <span class="sc">|</span> experiment_id<span class="sc">:</span>genotype) <span class="sc">+</span></span>
<span id="cb76-10"><a href="#cb76-10" aria-hidden="true" tabindex="-1"></a>                    (<span class="dv">1</span> <span class="sc">|</span> experiment_id<span class="sc">:</span>sex) <span class="sc">+</span></span>
<span id="cb76-11"><a href="#cb76-11" aria-hidden="true" tabindex="-1"></a>                    (<span class="dv">1</span> <span class="sc">|</span> experiment_id<span class="sc">:</span>genotype<span class="sc">:</span>sex),</span>
<span id="cb76-12"><a href="#cb76-12" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp1g)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>boundary (singular) fit: see help('isSingular')</code></pre>
</div>
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="fu">VarCorr</span>(exp1g_m1c) <span class="co"># id:genotype is low</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> Groups                     Name        Std.Dev. 
 experiment_id:genotype:sex (Intercept) 0.4046670
 experiment_id:sex          (Intercept) 1.6927499
 experiment_id:genotype     (Intercept) 0.0001216
 experiment_id              (Intercept) 2.5487069
 Residual                               1.9792428</code></pre>
</div>
<div class="sourceCode cell-code" id="cb80"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a><span class="co"># drop id:genotype which has low variance</span></span>
<span id="cb80-2"><a href="#cb80-2" aria-hidden="true" tabindex="-1"></a>exp1g_m1d <span class="ot">&lt;-</span> <span class="fu">lmer</span>(gc <span class="sc">~</span> genotype <span class="sc">*</span> sex <span class="sc">+</span></span>
<span id="cb80-3"><a href="#cb80-3" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> experiment_id) <span class="sc">+</span></span>
<span id="cb80-4"><a href="#cb80-4" aria-hidden="true" tabindex="-1"></a>                    (<span class="dv">1</span> <span class="sc">|</span> experiment_id<span class="sc">:</span>sex) <span class="sc">+</span></span>
<span id="cb80-5"><a href="#cb80-5" aria-hidden="true" tabindex="-1"></a>                    (<span class="dv">1</span> <span class="sc">|</span> experiment_id<span class="sc">:</span>sex<span class="sc">:</span>genotype),</span>
<span id="cb80-6"><a href="#cb80-6" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp1g)</span>
<span id="cb80-7"><a href="#cb80-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-8"><a href="#cb80-8" aria-hidden="true" tabindex="-1"></a><span class="fu">AIC</span>(exp1g_m1a, exp1g_m1b, exp1g_m1c, exp1g_m1d)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>          df      AIC
exp1g_m1a 15 344.1579
exp1g_m1b 15 344.1529
exp1g_m1c  9 337.7602
exp1g_m1d  8 335.7602</code></pre>
</div>
<div class="sourceCode cell-code" id="cb82"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a><span class="co"># go with exp1g_m1d.</span></span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a>exp1g_m1 <span class="ot">&lt;-</span> exp1g_m1d</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="inference-from-the-model-2" class="level3" data-number="14.9.4">
<h3 data-number="14.9.4" class="anchored" data-anchor-id="inference-from-the-model-2"><span class="header-section-number">14.9.4</span> Inference from the model</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb83"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb83-1"><a href="#cb83-1" aria-hidden="true" tabindex="-1"></a>exp1g_m1_coef <span class="ot">&lt;-</span> <span class="fu">coef</span>(<span class="fu">summary</span>(exp1g_m1))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb84"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a>exp1g_m1_coef</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                      Estimate Std. Error       df   t value    Pr(&gt;|t|)
(Intercept)           8.435718   1.612602 4.638595  5.231123 0.004202015
genotypeGpr174-       2.568365   0.712122 5.760582  3.606637 0.012094327
sexF                  5.583860   1.397388 4.024010  3.995927 0.015995349
genotypeGpr174-:sexF -5.304752   1.013552 5.918159 -5.233823 0.002033657</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb86"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a><span class="co"># order of factors reversed in specs because I want sex to be</span></span>
<span id="cb86-2"><a href="#cb86-2" aria-hidden="true" tabindex="-1"></a><span class="co"># main x-axis variable in plot</span></span>
<span id="cb86-3"><a href="#cb86-3" aria-hidden="true" tabindex="-1"></a>exp1g_m1_emm <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(exp1g_m1, <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">"sex"</span>, <span class="st">"genotype"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb87"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb87-1"><a href="#cb87-1" aria-hidden="true" tabindex="-1"></a><span class="co"># exp1g_m1_emm # print in console to get row numbers</span></span>
<span id="cb87-2"><a href="#cb87-2" aria-hidden="true" tabindex="-1"></a><span class="co"># set the mean as the row number from the emmeans table</span></span>
<span id="cb87-3"><a href="#cb87-3" aria-hidden="true" tabindex="-1"></a>wt_m <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>)</span>
<span id="cb87-4"><a href="#cb87-4" aria-hidden="true" tabindex="-1"></a>wt_f <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">0</span>,<span class="dv">0</span>)</span>
<span id="cb87-5"><a href="#cb87-5" aria-hidden="true" tabindex="-1"></a>ko_m <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb87-6"><a href="#cb87-6" aria-hidden="true" tabindex="-1"></a>ko_f <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>)</span>
<span id="cb87-7"><a href="#cb87-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb87-8"><a href="#cb87-8" aria-hidden="true" tabindex="-1"></a><span class="co"># simple effects within males and females + interaction </span></span>
<span id="cb87-9"><a href="#cb87-9" aria-hidden="true" tabindex="-1"></a><span class="co"># 1. (ko_m - wt_m) </span></span>
<span id="cb87-10"><a href="#cb87-10" aria-hidden="true" tabindex="-1"></a><span class="co"># 2. (ko_f - wt_f)</span></span>
<span id="cb87-11"><a href="#cb87-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb87-12"><a href="#cb87-12" aria-hidden="true" tabindex="-1"></a>exp1g_contrasts <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb87-13"><a href="#cb87-13" aria-hidden="true" tabindex="-1"></a>  <span class="st">"(Gpr174- M) - (Gpr174+ M)"</span> <span class="ot">=</span> <span class="fu">c</span>(ko_m <span class="sc">-</span> wt_m),</span>
<span id="cb87-14"><a href="#cb87-14" aria-hidden="true" tabindex="-1"></a>  <span class="st">"(Gpr174- F) - (Gpr174+ F)"</span> <span class="ot">=</span> <span class="fu">c</span>(ko_f <span class="sc">-</span> wt_f),</span>
<span id="cb87-15"><a href="#cb87-15" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Interaction"</span> <span class="ot">=</span> <span class="fu">c</span>(ko_f <span class="sc">-</span> wt_f) <span class="sc">-</span></span>
<span id="cb87-16"><a href="#cb87-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">c</span>(ko_m <span class="sc">-</span> wt_m)</span>
<span id="cb87-17"><a href="#cb87-17" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb87-18"><a href="#cb87-18" aria-hidden="true" tabindex="-1"></a>exp1g_m1_planned <span class="ot">&lt;-</span> <span class="fu">contrast</span>(exp1g_m1_emm,</span>
<span id="cb87-19"><a href="#cb87-19" aria-hidden="true" tabindex="-1"></a>                       <span class="at">method =</span> exp1g_contrasts,</span>
<span id="cb87-20"><a href="#cb87-20" aria-hidden="true" tabindex="-1"></a>                       <span class="at">adjust =</span> <span class="st">"none"</span></span>
<span id="cb87-21"><a href="#cb87-21" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb87-22"><a href="#cb87-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">contrast</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">estimate</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">SE</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">df</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">lower.CL</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">upper.CL</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">t.ratio</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">p.value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">(Gpr174- M) - (Gpr174+ M)</td>
<td style="text-align: right;">2.57</td>
<td style="text-align: right;">0.714</td>
<td style="text-align: right;">5.784</td>
<td style="text-align: right;">0.81</td>
<td style="text-align: right;">4.33</td>
<td style="text-align: right;">3.60</td>
<td style="text-align: right;">0.012</td>
</tr>
<tr class="even">
<td style="text-align: left;">(Gpr174- F) - (Gpr174+ F)</td>
<td style="text-align: right;">-2.74</td>
<td style="text-align: right;">0.722</td>
<td style="text-align: right;">6.100</td>
<td style="text-align: right;">-4.50</td>
<td style="text-align: right;">-0.98</td>
<td style="text-align: right;">-3.79</td>
<td style="text-align: right;">0.009</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Interaction</td>
<td style="text-align: right;">-5.30</td>
<td style="text-align: right;">1.016</td>
<td style="text-align: right;">5.942</td>
<td style="text-align: right;">-7.80</td>
<td style="text-align: right;">-2.81</td>
<td style="text-align: right;">-5.22</td>
<td style="text-align: right;">0.002</td>
</tr>
</tbody>
</table>


</div>
</div>
<p>Notes</p>
<ol type="1">
<li>The direction of the estimated effect is opposite in males and females</li>
</ol>
</section>
<section id="plot-the-model" class="level3" data-number="14.9.5">
<h3 data-number="14.9.5" class="anchored" data-anchor-id="plot-the-model"><span class="header-section-number">14.9.5</span> Plot the model</h3>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="lmm_files/figure-html/unnamed-chunk-38-1.png" class="img-fluid figure-img" width="576"></p>
<figcaption>Treatment effect on germinal center (GC) formation. Small, pale, colored dots are independent experiments. Intermediate size colored dots are experiment means.</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="alternaplot-the-model-2" class="level3" data-number="14.9.6">
<h3 data-number="14.9.6" class="anchored" data-anchor-id="alternaplot-the-model-2"><span class="header-section-number">14.9.6</span> Alternaplot the model</h3>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="lmm_files/figure-html/lmm-exp1g-alternaplot-1.png" class="img-fluid figure-img" width="576"></p>
<figcaption>Germinal center (GC) formation in response to treatment. Small, pale, colored dots are independent experiments. Intermediate size colored dots are experiment means. Dashed gray line is expected additive mean of Gpr174- F</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="understanding-the-alternative-models" class="level3" data-number="14.9.7">
<h3 data-number="14.9.7" class="anchored" data-anchor-id="understanding-the-alternative-models"><span class="header-section-number">14.9.7</span> Understanding the alternative models</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb88"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a>exp1g_m1a <span class="ot">&lt;-</span> <span class="fu">lmer</span>(gc <span class="sc">~</span> genotype <span class="sc">*</span> sex <span class="sc">+</span></span>
<span id="cb88-2"><a href="#cb88-2" aria-hidden="true" tabindex="-1"></a>                   (genotype <span class="sc">*</span> sex <span class="sc">|</span> experiment_id),</span>
<span id="cb88-3"><a href="#cb88-3" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp1g)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>boundary (singular) fit: see help('isSingular')</code></pre>
</div>
<div class="sourceCode cell-code" id="cb90"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a>exp1g_m1b <span class="ot">&lt;-</span> <span class="fu">lmer</span>(gc <span class="sc">~</span> genotype <span class="sc">*</span> sex <span class="sc">+</span></span>
<span id="cb90-2"><a href="#cb90-2" aria-hidden="true" tabindex="-1"></a>                   (treatment <span class="sc">|</span> experiment_id),</span>
<span id="cb90-3"><a href="#cb90-3" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp1g)</span>
<span id="cb90-4"><a href="#cb90-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-5"><a href="#cb90-5" aria-hidden="true" tabindex="-1"></a>exp1g_m1c <span class="ot">&lt;-</span> <span class="fu">lmer</span>(gc <span class="sc">~</span> genotype <span class="sc">*</span> sex <span class="sc">+</span></span>
<span id="cb90-6"><a href="#cb90-6" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> experiment_id) <span class="sc">+</span></span>
<span id="cb90-7"><a href="#cb90-7" aria-hidden="true" tabindex="-1"></a>                    (<span class="dv">1</span> <span class="sc">|</span> experiment_id<span class="sc">:</span>genotype) <span class="sc">+</span></span>
<span id="cb90-8"><a href="#cb90-8" aria-hidden="true" tabindex="-1"></a>                    (<span class="dv">1</span> <span class="sc">|</span> experiment_id<span class="sc">:</span>sex) <span class="sc">+</span></span>
<span id="cb90-9"><a href="#cb90-9" aria-hidden="true" tabindex="-1"></a>                    (<span class="dv">1</span> <span class="sc">|</span> experiment_id<span class="sc">:</span>genotype<span class="sc">:</span>sex),</span>
<span id="cb90-10"><a href="#cb90-10" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp1g)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>boundary (singular) fit: see help('isSingular')</code></pre>
</div>
<div class="sourceCode cell-code" id="cb92"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1" aria-hidden="true" tabindex="-1"></a>exp1g_m1d <span class="ot">&lt;-</span> <span class="fu">lmer</span>(gc <span class="sc">~</span> genotype <span class="sc">*</span> sex <span class="sc">+</span></span>
<span id="cb92-2"><a href="#cb92-2" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> experiment_id) <span class="sc">+</span></span>
<span id="cb92-3"><a href="#cb92-3" aria-hidden="true" tabindex="-1"></a>                    (<span class="dv">1</span> <span class="sc">|</span> experiment_id<span class="sc">:</span>sex) <span class="sc">+</span></span>
<span id="cb92-4"><a href="#cb92-4" aria-hidden="true" tabindex="-1"></a>                    (<span class="dv">1</span> <span class="sc">|</span> experiment_id<span class="sc">:</span>genotype<span class="sc">:</span>sex),</span>
<span id="cb92-5"><a href="#cb92-5" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp1g)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Notes</p>
<ol type="1">
<li>All four models model the same fixed effects. The models differ only in how they model the random effects.</li>
<li>Model <code>exp1g_m1a</code> fits one random intercept, two random slopes, and one random interaction.</li>
</ol>
<ul>
<li><span class="math inline">\(\gamma_{0j}\)</span> – a random intercept modeling batch effects of <span class="math inline">\(\texttt{experiment_id}\)</span> on the intercept</li>
<li><span class="math inline">\(\gamma_{1j}\)</span> – a random slope modeling the effect of the non-reference level of “Gpr174-” on the batch effect of <span class="math inline">\(\texttt{experiment_id}\)</span>. This is a <span class="math inline">\(\texttt{experiment_id} \times \texttt{genotype}\)</span> interaction.</li>
<li><span class="math inline">\(\gamma_{2j}\)</span> – a random slope modeling the effect of “F” (female) on the batch effect of <span class="math inline">\(\texttt{experiment_id}\)</span>. This is a <span class="math inline">\(\texttt{experiment_id} \times \texttt{sex}\)</span> interaction.</li>
<li><span class="math inline">\(\gamma_{3j}\)</span> – a random slope modeling the effect of the interaction effect of “Gpr174-” and “F” on the batch effect of <span class="math inline">\(\texttt{experiment_id}\)</span>. This is a <span class="math inline">\(\texttt{experiment_id} \times \texttt{sex}  \times \texttt{genotype}\)</span> interaction.</li>
</ul>
<ol start="3" type="1">
<li>Model <code>exp1g_m1b</code> fits one random intercept and three random slopes</li>
</ol>
<ul>
<li><span class="math inline">\(\gamma_{0j}\)</span> – a random intercept modeling batch effects of <span class="math inline">\(\texttt{experiment_id}\)</span> on the intercept. This is modeling the same thing as <span class="math inline">\(\gamma_{0j}\)</span> in Model <code>exp1g_m1a</code>.</li>
<li><span class="math inline">\(\gamma_{1j}\)</span> – a random slope modeling the effect of “Gpr174- M” on the batch effect of <span class="math inline">\(\texttt{experiment_id}\)</span>. This is modeling the same thing as <span class="math inline">\(\gamma_{1j}\)</span> in Model <code>exp1g_m1a</code>.</li>
<li><span class="math inline">\(\gamma_{2j}\)</span> – a random slope modeling the effect of “Gpr174+ F” on the batch effect of <span class="math inline">\(\texttt{experiment_id}\)</span>. This is modeling the same thing as <span class="math inline">\(\gamma_{2j}\)</span> in Model <code>exp1g_m1a</code>.</li>
<li><span class="math inline">\(\gamma_{3j}\)</span> – a random slope modeling the effect of “M Gpr174- F” on the batch effect of <span class="math inline">\(\texttt{experiment_id}\)</span>. This is modeling the added variance accounted for by the random interaction <span class="math inline">\(\gamma_{2j}\)</span> in Model <code>exp1g_m1a</code> but in a different way.</li>
</ul>
<ol start="4" type="1">
<li>Model <code>exp1g_m1c</code> fits four random intercepts</li>
</ol>
<ul>
<li><span class="math inline">\(\gamma_{0j}\)</span> – a random intercept modeling batch effects of <span class="math inline">\(\texttt{experiment_id}\)</span> on the intercept. This is modeling the same thing as <span class="math inline">\(\gamma_{0j}\)</span> in Model <code>exp1g_m1a</code>.</li>
<li><span class="math inline">\(\gamma_{0jk}\)</span> – a random intercept modeling the effects of the combination of <span class="math inline">\(\texttt{experiment_id}\)</span> and <span class="math inline">\(\texttt{genotype}\)</span>. This is very similar to the variance modeled by <span class="math inline">\(\gamma_{1j}\)</span> in Model <code>exp1g_m1a</code> except the draws from <span class="math inline">\(\gamma_{0jk}\)</span> are independent (uncorrelated) of draws from the other random intercepts.</li>
<li><span class="math inline">\(\gamma_{0jl}\)</span> – a random intercept modeling the effects of the combination of <span class="math inline">\(\texttt{experiment_id}\)</span> and <span class="math inline">\(\texttt{sex}\)</span>. This is very similar to the variance modeled by <span class="math inline">\(\gamma_{2j}\)</span> in Model <code>exp1g_m1a</code> except the draws from <span class="math inline">\(\gamma_{0jl}\)</span> are independent (uncorrelated) of draws from the other random intercepts (review <a href="#lmm-varcorr">The correlation among random intercepts and slopes</a> if this doesn’t make sense).</li>
<li><span class="math inline">\(\gamma_{0jkl}\)</span> – a random intercept modeling the effects of the combination of <span class="math inline">\(\texttt{experiment_id}\)</span>, <span class="math inline">\(\texttt{genotype}\)</span> and <span class="math inline">\(\texttt{sex}\)</span>. This is very similar to the variance modeled by <span class="math inline">\(\gamma_{3j}\)</span> in Model <code>exp1g_m1a</code> except the draws from <span class="math inline">\(\gamma_{0jkl}\)</span> are independent (uncorrelated) of draws from the other random intercepts.</li>
</ul>
<ol start="5" type="1">
<li>Model <code>exp1g_m1d</code> fits the same random intercepts as Model <code>exp1g_m1c</code> but excludes <span class="math inline">\(\gamma_{0jl}\)</span> (the random intercept for the experiment_id by genotyp combination. This was excluded because of the low variance of this component in the fit model.</li>
</ol>
</section>
<section id="the-varcorr-matrix-of-models-exp1g_m1a-and-exp1g_m1b" class="level3" data-number="14.9.8">
<h3 data-number="14.9.8" class="anchored" data-anchor-id="the-varcorr-matrix-of-models-exp1g_m1a-and-exp1g_m1b"><span class="header-section-number">14.9.8</span> The VarCorr matrix of models exp1g_m1a and exp1g_m1b</h3>
<p>The random effect similarity of models <code>exp1g_m1a</code> and <code>exp1g_m1b</code> can be seen in the estimated variance components and correlations among the random effects.</p>
<div class="cell">
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<caption>The Varcorr matrix. Standard deviations of random effects on the diagonal. Correlations of random effects on the off-diagonal.</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: right;" data-quarto-table-cell-role="th"></th>
<th style="text-align: right;" data-quarto-table-cell-role="th"></th>
<th style="text-align: right;" data-quarto-table-cell-role="th"></th>
<th style="text-align: right;" data-quarto-table-cell-role="th"></th>
</tr>
</thead>
<tbody>
<tr class="odd" data-grouplength="4">
<td colspan="5" style="border-bottom: 1px solid"><strong>exp1g_m1a</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(Intercept)</td>
<td style="text-align: right;">1.7709</td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">genotypeGpr174-</td>
<td style="text-align: right;">-0.0232</td>
<td style="text-align: right;">0.9619</td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">sexF</td>
<td style="text-align: right;">0.4589</td>
<td style="text-align: right;">0.8776</td>
<td style="text-align: right;">2.9015</td>
<td style="text-align: right;"></td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">genotypeGpr174-:sexF</td>
<td style="text-align: right;">-0.3137</td>
<td style="text-align: right;">-0.6247</td>
<td style="text-align: right;">-0.7056</td>
<td style="text-align: right;">1.3317</td>
</tr>
<tr class="even" data-grouplength="4">
<td colspan="5" style="border-bottom: 1px solid"><strong>exp1g_m1b</strong></td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(Intercept)</td>
<td style="text-align: right;">1.7711</td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">treatmentGpr174- M</td>
<td style="text-align: right;">-0.0250</td>
<td style="text-align: right;">0.9641</td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">treatmentGpr174+ F</td>
<td style="text-align: right;">0.4585</td>
<td style="text-align: right;">0.8761</td>
<td style="text-align: right;">2.9024</td>
<td style="text-align: right;"></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">treatmentGpr174- F</td>
<td style="text-align: right;">0.2999</td>
<td style="text-align: right;">0.8793</td>
<td style="text-align: right;">0.9384</td>
<td style="text-align: right;">2.9909</td>
</tr>
</tbody>
</table>


</div>
</div>
</section>
<section id="the-linear-mixed-model-has-more-precision-and-power-than-the-fixed-effect-model-of-batch-means" class="level3" data-number="14.9.9">
<h3 data-number="14.9.9" class="anchored" data-anchor-id="the-linear-mixed-model-has-more-precision-and-power-than-the-fixed-effect-model-of-batch-means"><span class="header-section-number">14.9.9</span> The linear mixed model has more precision and power than the fixed effect model of batch means</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb93"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb93-1"><a href="#cb93-1" aria-hidden="true" tabindex="-1"></a><span class="co"># means pooling model</span></span>
<span id="cb93-2"><a href="#cb93-2" aria-hidden="true" tabindex="-1"></a>exp1g_m2 <span class="ot">&lt;-</span> <span class="fu">lm</span>(gc <span class="sc">~</span> sex <span class="sc">*</span> genotype,</span>
<span id="cb93-3"><a href="#cb93-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">data =</span> exp1g_means)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<caption>Planned contrasts for the linear mixed model exp1g_m1 and the fixed effects model of experiment means exp1g_m2.</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">contrast</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">estimate</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">SE</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">df</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">lower.CL</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">upper.CL</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">t.ratio</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">p.value</th>
</tr>
</thead>
<tbody>
<tr class="odd" data-grouplength="3">
<td colspan="8" style="border-bottom: 1px solid"><strong>exp1g_m1 (lmm)</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(Gpr174- M) - (Gpr174+ M)</td>
<td style="text-align: right;">2.57</td>
<td style="text-align: right;">0.714</td>
<td style="text-align: right;">5.8</td>
<td style="text-align: right;">0.81</td>
<td style="text-align: right;">4.33</td>
<td style="text-align: right;">3.60</td>
<td style="text-align: right;">0.012</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(Gpr174- F) - (Gpr174+ F)</td>
<td style="text-align: right;">-2.74</td>
<td style="text-align: right;">0.722</td>
<td style="text-align: right;">6.1</td>
<td style="text-align: right;">-4.50</td>
<td style="text-align: right;">-0.98</td>
<td style="text-align: right;">-3.79</td>
<td style="text-align: right;">0.009</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Interaction</td>
<td style="text-align: right;">-5.30</td>
<td style="text-align: right;">1.016</td>
<td style="text-align: right;">5.9</td>
<td style="text-align: right;">-7.80</td>
<td style="text-align: right;">-2.81</td>
<td style="text-align: right;">-5.22</td>
<td style="text-align: right;">0.002</td>
</tr>
<tr class="odd" data-grouplength="3">
<td colspan="8" style="border-bottom: 1px solid"><strong>exp1g_m2 (lm means pooling)</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(Gpr174- M) - (Gpr174+ M)</td>
<td style="text-align: right;">2.50</td>
<td style="text-align: right;">2.257</td>
<td style="text-align: right;">12.0</td>
<td style="text-align: right;">-2.41</td>
<td style="text-align: right;">7.42</td>
<td style="text-align: right;">1.11</td>
<td style="text-align: right;">0.289</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(Gpr174- F) - (Gpr174+ F)</td>
<td style="text-align: right;">-2.73</td>
<td style="text-align: right;">2.257</td>
<td style="text-align: right;">12.0</td>
<td style="text-align: right;">-7.65</td>
<td style="text-align: right;">2.18</td>
<td style="text-align: right;">-1.21</td>
<td style="text-align: right;">0.249</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Interaction</td>
<td style="text-align: right;">-5.24</td>
<td style="text-align: right;">3.192</td>
<td style="text-align: right;">12.0</td>
<td style="text-align: right;">-12.20</td>
<td style="text-align: right;">1.72</td>
<td style="text-align: right;">-1.64</td>
<td style="text-align: right;">0.127</td>
</tr>
</tbody>
</table>


</div>
</div>
<p>Notes</p>
<ol type="1">
<li>A fixed effects model fit to batch-means pooled data is strongly conservative and will result in less discovery.</li>
<li>Means pooling does not make the data independent in a randomized complete block design. A linear mixed model of batch-means pooled data is a <strong>mixed-effect ANOVA</strong> (Next section. Also see Section @ref(issues-exp4d) in the Issues chapter).</li>
</ol>
</section>
<section id="fixed-effect-models-and-pseudoreplication" class="level3" data-number="14.9.10">
<h3 data-number="14.9.10" class="anchored" data-anchor-id="fixed-effect-models-and-pseudoreplication"><span class="header-section-number">14.9.10</span> Fixed effect models and pseudoreplication</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb94"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a><span class="co"># complete pooling model</span></span>
<span id="cb94-2"><a href="#cb94-2" aria-hidden="true" tabindex="-1"></a>exp1g_m3 <span class="ot">&lt;-</span> <span class="fu">lm</span>(gc <span class="sc">~</span> sex <span class="sc">*</span> genotype,</span>
<span id="cb94-3"><a href="#cb94-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">data =</span> exp1g)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<caption>Planned contrasts for the linear mixed model exp1g_m1 and the fixed effects model exp1g_m3 with complete pooling.</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">contrast</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">estimate</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">SE</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">df</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">lower.CL</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">upper.CL</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">t.ratio</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">p.value</th>
</tr>
</thead>
<tbody>
<tr class="odd" data-grouplength="3">
<td colspan="8" style="border-bottom: 1px solid"><strong>exp1g_m1 (lmm)</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(Gpr174- M) - (Gpr174+ M)</td>
<td style="text-align: right;">2.57</td>
<td style="text-align: right;">0.714</td>
<td style="text-align: right;">5.8</td>
<td style="text-align: right;">0.81</td>
<td style="text-align: right;">4.33</td>
<td style="text-align: right;">3.60</td>
<td style="text-align: right;">0.012</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(Gpr174- F) - (Gpr174+ F)</td>
<td style="text-align: right;">-2.74</td>
<td style="text-align: right;">0.722</td>
<td style="text-align: right;">6.1</td>
<td style="text-align: right;">-4.50</td>
<td style="text-align: right;">-0.98</td>
<td style="text-align: right;">-3.79</td>
<td style="text-align: right;">0.009</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Interaction</td>
<td style="text-align: right;">-5.30</td>
<td style="text-align: right;">1.016</td>
<td style="text-align: right;">5.9</td>
<td style="text-align: right;">-7.80</td>
<td style="text-align: right;">-2.81</td>
<td style="text-align: right;">-5.22</td>
<td style="text-align: right;">0.002</td>
</tr>
<tr class="odd" data-grouplength="3">
<td colspan="8" style="border-bottom: 1px solid"><strong>exp1g_m3 (lm complete pooling)</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(Gpr174- M) - (Gpr174+ M)</td>
<td style="text-align: right;">2.44</td>
<td style="text-align: right;">1.111</td>
<td style="text-align: right;">69.0</td>
<td style="text-align: right;">0.22</td>
<td style="text-align: right;">4.65</td>
<td style="text-align: right;">2.20</td>
<td style="text-align: right;">0.032</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(Gpr174- F) - (Gpr174+ F)</td>
<td style="text-align: right;">-2.53</td>
<td style="text-align: right;">1.125</td>
<td style="text-align: right;">69.0</td>
<td style="text-align: right;">-4.77</td>
<td style="text-align: right;">-0.28</td>
<td style="text-align: right;">-2.25</td>
<td style="text-align: right;">0.028</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Interaction</td>
<td style="text-align: right;">-4.97</td>
<td style="text-align: right;">1.581</td>
<td style="text-align: right;">69.0</td>
<td style="text-align: right;">-8.12</td>
<td style="text-align: right;">-1.81</td>
<td style="text-align: right;">-3.14</td>
<td style="text-align: right;">0.002</td>
</tr>
</tbody>
</table>


</div>
</div>
<p>Notes</p>
<ol type="1">
<li>Complete pooling is strongly anti-conservative and will result in increased false discovery. Complete pooling is a type of pseudoreplication – the subsamples have been analyzed as if they are independent replicates. Subsamples are not independent.</li>
<li>For the Experiment 1g data, the 95% confidence intervals are wider and the <em>p</em>-values are larger in the complete pool model <code>exp1g_m3</code> compared to the linear mixed model <code>exp1g_m1</code>. This is an unusual result.</li>
</ol>
</section>
<section id="mixed-effect-anova" class="level3" data-number="14.9.11">
<h3 data-number="14.9.11" class="anchored" data-anchor-id="mixed-effect-anova"><span class="header-section-number">14.9.11</span> Mixed-effect ANOVA</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb95"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb95-1"><a href="#cb95-1" aria-hidden="true" tabindex="-1"></a>exp1g_m1_aov <span class="ot">&lt;-</span> <span class="fu">aov_4</span>(gc <span class="sc">~</span> sex <span class="sc">*</span> genotype <span class="sc">+</span></span>
<span id="cb95-2"><a href="#cb95-2" aria-hidden="true" tabindex="-1"></a>                  (sex <span class="sc">*</span> genotype <span class="sc">|</span> experiment_id),</span>
<span id="cb95-3"><a href="#cb95-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">data =</span> exp1g)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: More than one observation per design cell, aggregating data using `fun_aggregate = mean`.
To turn off this warning, pass `fun_aggregate = mean` explicitly.</code></pre>
</div>
</div>
<p>Notes</p>
<ol type="1">
<li>The formula has the same format as that in Example 3 for repeated measures ANOVA on RCB designs with no subsampling. In biology, this is often called a <strong>mixed effect ANOVA</strong> with two fixed factors and one random factor.</li>
<li>The data are aggregated prior to fitting the model – this means the subsamples are averaged within each batch by treatment combination.</li>
<li>The mixed-effect ANOVA is equivalent to the linear mixed model <code>exp1g_m1c</code> if there are the same number of replicates in each treatment combination and the same number of subsamples in all treatment by <span class="math inline">\(\texttt{experiment_id}\)</span> combinations. The design is not balanced in this example.</li>
</ol>
<div class="cell">
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<caption>Planned contrasts from mixed ANOVA compared to lmm equivalent of mixed ANOVA and lowest AIC lmm.</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">contrast</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">estimate</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">SE</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">df</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">lower.CL</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">upper.CL</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">t.ratio</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">p.value</th>
</tr>
</thead>
<tbody>
<tr class="odd" data-grouplength="3">
<td colspan="8" style="border-bottom: 1px solid"><strong>exp1g_m1_aov (mixed ANOVA)</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(Gpr174- M) - (Gpr174+ M)</td>
<td style="text-align: right;">2.50</td>
<td style="text-align: right;">0.599</td>
<td style="text-align: right;">3.000</td>
<td style="text-align: right;">0.60</td>
<td style="text-align: right;">4.41</td>
<td style="text-align: right;">4.18</td>
<td style="text-align: right;">0.025</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(Gpr174- F) - (Gpr174+ F)</td>
<td style="text-align: right;">-2.73</td>
<td style="text-align: right;">0.808</td>
<td style="text-align: right;">3.000</td>
<td style="text-align: right;">-5.30</td>
<td style="text-align: right;">-0.16</td>
<td style="text-align: right;">-3.39</td>
<td style="text-align: right;">0.043</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Interaction</td>
<td style="text-align: right;">-5.24</td>
<td style="text-align: right;">1.087</td>
<td style="text-align: right;">3.000</td>
<td style="text-align: right;">-8.70</td>
<td style="text-align: right;">-1.78</td>
<td style="text-align: right;">-4.82</td>
<td style="text-align: right;">0.017</td>
</tr>
<tr class="odd" data-grouplength="3">
<td colspan="8" style="border-bottom: 1px solid"><strong>exp1g_m1_c (lmm equivalent of mixed ANOVA)</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(Gpr174- M) - (Gpr174+ M)</td>
<td style="text-align: right;">2.57</td>
<td style="text-align: right;">0.715</td>
<td style="text-align: right;">5.784</td>
<td style="text-align: right;">0.80</td>
<td style="text-align: right;">4.33</td>
<td style="text-align: right;">3.59</td>
<td style="text-align: right;">0.012</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(Gpr174- F) - (Gpr174+ F)</td>
<td style="text-align: right;">-2.74</td>
<td style="text-align: right;">0.724</td>
<td style="text-align: right;">6.100</td>
<td style="text-align: right;">-4.50</td>
<td style="text-align: right;">-0.97</td>
<td style="text-align: right;">-3.78</td>
<td style="text-align: right;">0.009</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Interaction</td>
<td style="text-align: right;">-5.30</td>
<td style="text-align: right;">1.017</td>
<td style="text-align: right;">2.972</td>
<td style="text-align: right;">-8.56</td>
<td style="text-align: right;">-2.05</td>
<td style="text-align: right;">-5.21</td>
<td style="text-align: right;">0.014</td>
</tr>
<tr class="odd" data-grouplength="3">
<td colspan="8" style="border-bottom: 1px solid"><strong>exp1g_md (lmm with min AIC)</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(Gpr174- M) - (Gpr174+ M)</td>
<td style="text-align: right;">2.57</td>
<td style="text-align: right;">0.714</td>
<td style="text-align: right;">5.784</td>
<td style="text-align: right;">0.81</td>
<td style="text-align: right;">4.33</td>
<td style="text-align: right;">3.60</td>
<td style="text-align: right;">0.012</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">(Gpr174- F) - (Gpr174+ F)</td>
<td style="text-align: right;">-2.74</td>
<td style="text-align: right;">0.722</td>
<td style="text-align: right;">6.100</td>
<td style="text-align: right;">-4.50</td>
<td style="text-align: right;">-0.98</td>
<td style="text-align: right;">-3.79</td>
<td style="text-align: right;">0.009</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Interaction</td>
<td style="text-align: right;">-5.30</td>
<td style="text-align: right;">1.016</td>
<td style="text-align: right;">5.942</td>
<td style="text-align: right;">-7.80</td>
<td style="text-align: right;">-2.81</td>
<td style="text-align: right;">-5.22</td>
<td style="text-align: right;">0.002</td>
</tr>
</tbody>
</table>


</div>
</div>
</section>
</section>
<section id="statistical-models-for-experimental-designs" class="level2" data-number="14.10">
<h2 data-number="14.10" class="anchored" data-anchor-id="statistical-models-for-experimental-designs"><span class="header-section-number">14.10</span> Statistical models for experimental designs</h2>
<section id="models-for-completely-randomized-designs-crd" class="level3" data-number="14.10.1">
<h3 data-number="14.10.1" class="anchored" data-anchor-id="models-for-completely-randomized-designs-crd"><span class="header-section-number">14.10.1</span> Models for Completely Randomized Designs (CRD)</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb97"><pre class="sourceCode r fold-show code-with-copy"><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1" aria-hidden="true" tabindex="-1"></a>lm0 <span class="ot">&lt;-</span> <span class="fu">lm</span>(y <span class="sc">~</span> treatment,</span>
<span id="cb97-2"><a href="#cb97-2" aria-hidden="true" tabindex="-1"></a>             <span class="at">data =</span> figx)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="models-for-batched-data-crds-rcbd-rspd-grcbd-nrcbd" class="level3" data-number="14.10.2">
<h3 data-number="14.10.2" class="anchored" data-anchor-id="models-for-batched-data-crds-rcbd-rspd-grcbd-nrcbd"><span class="header-section-number">14.10.2</span> Models for batched data (CRDS, RCBD, RSPD, GRCBD, NRCBD)</h3>
<section id="linear-models" class="level4" data-number="14.10.2.1">
<h4 data-number="14.10.2.1" class="anchored" data-anchor-id="linear-models"><span class="header-section-number">14.10.2.1</span> Linear models</h4>
<section id="fixed-effect-model" class="level5" data-number="14.10.2.1.1">
<h5 data-number="14.10.2.1.1" class="anchored" data-anchor-id="fixed-effect-model"><span class="header-section-number">14.10.2.1.1</span> fixed effect model</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb98"><pre class="sourceCode r fold-show code-with-copy"><code class="sourceCode r"><span id="cb98-1"><a href="#cb98-1" aria-hidden="true" tabindex="-1"></a>lm1 <span class="ot">&lt;-</span> <span class="fu">lm</span>(y <span class="sc">~</span> treatment <span class="sc">+</span> block,</span>
<span id="cb98-2"><a href="#cb98-2" aria-hidden="true" tabindex="-1"></a>             <span class="at">data =</span> figx)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="linear-model-of-batch-means" class="level5" data-number="14.10.2.1.2">
<h5 data-number="14.10.2.1.2" class="anchored" data-anchor-id="linear-model-of-batch-means"><span class="header-section-number">14.10.2.1.2</span> linear model of batch means</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb99"><pre class="sourceCode r fold-show code-with-copy"><code class="sourceCode r"><span id="cb99-1"><a href="#cb99-1" aria-hidden="true" tabindex="-1"></a>figx_means <span class="ot">&lt;-</span> figx[, .(<span class="at">y =</span> <span class="fu">mean</span>(y)),</span>
<span id="cb99-2"><a href="#cb99-2" aria-hidden="true" tabindex="-1"></a>               by <span class="ot">=</span> .(treatment, batch)]</span>
<span id="cb99-3"><a href="#cb99-3" aria-hidden="true" tabindex="-1"></a>lm2 <span class="ot">&lt;-</span> <span class="fu">lm</span>(y <span class="sc">~</span> treatment,</span>
<span id="cb99-4"><a href="#cb99-4" aria-hidden="true" tabindex="-1"></a>             <span class="at">data =</span> figx_means)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="linear-mixed-models-random-effects-models" class="level4" data-number="14.10.2.2">
<h4 data-number="14.10.2.2" class="anchored" data-anchor-id="linear-mixed-models-random-effects-models"><span class="header-section-number">14.10.2.2</span> linear mixed models (“random effects” models)</h4>
<section id="random-intercept-model" class="level5" data-number="14.10.2.2.1">
<h5 data-number="14.10.2.2.1" class="anchored" data-anchor-id="random-intercept-model"><span class="header-section-number">14.10.2.2.1</span> Random intercept model</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb100"><pre class="sourceCode r fold-show code-with-copy"><code class="sourceCode r"><span id="cb100-1"><a href="#cb100-1" aria-hidden="true" tabindex="-1"></a>lmm1 <span class="ot">&lt;-</span> <span class="fu">lmer</span>(y <span class="sc">~</span> treatment <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> block),</span>
<span id="cb100-2"><a href="#cb100-2" aria-hidden="true" tabindex="-1"></a>              <span class="at">data =</span> figx)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="lmm-for-correlated-error" class="level5" data-number="14.10.2.2.2">
<h5 data-number="14.10.2.2.2" class="anchored" data-anchor-id="lmm-for-correlated-error"><span class="header-section-number">14.10.2.2.2</span> lmm for correlated error</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb101"><pre class="sourceCode r fold-show code-with-copy"><code class="sourceCode r"><span id="cb101-1"><a href="#cb101-1" aria-hidden="true" tabindex="-1"></a>lmm2 <span class="ot">&lt;-</span> <span class="fu">lme</span>(y <span class="sc">~</span> treatment,</span>
<span id="cb101-2"><a href="#cb101-2" aria-hidden="true" tabindex="-1"></a>             <span class="at">random =</span> <span class="sc">~</span><span class="dv">1</span> <span class="sc">|</span> block,</span>
<span id="cb101-3"><a href="#cb101-3" aria-hidden="true" tabindex="-1"></a>             <span class="at">correlation =</span> <span class="fu">corSymm</span>(<span class="at">form =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> block),</span>
<span id="cb101-4"><a href="#cb101-4" aria-hidden="true" tabindex="-1"></a>             <span class="at">weights =</span> <span class="fu">varIdent</span>(<span class="at">form =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> treatment),</span>
<span id="cb101-5"><a href="#cb101-5" aria-hidden="true" tabindex="-1"></a>             <span class="at">data =</span> figx)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="random-intercept-and-slope-model" class="level5" data-number="14.10.2.2.3">
<h5 data-number="14.10.2.2.3" class="anchored" data-anchor-id="random-intercept-and-slope-model"><span class="header-section-number">14.10.2.2.3</span> random intercept and slope model</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb102"><pre class="sourceCode r fold-show code-with-copy"><code class="sourceCode r"><span id="cb102-1"><a href="#cb102-1" aria-hidden="true" tabindex="-1"></a>lmm3 <span class="ot">&lt;-</span>  <span class="fu">lmer</span>(y <span class="sc">~</span> treatment <span class="sc">+</span> (treatment <span class="sc">|</span> block),</span>
<span id="cb102-2"><a href="#cb102-2" aria-hidden="true" tabindex="-1"></a>               <span class="at">data =</span> figx)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="random-interaction-intercept-model" class="level5" data-number="14.10.2.2.4">
<h5 data-number="14.10.2.2.4" class="anchored" data-anchor-id="random-interaction-intercept-model"><span class="header-section-number">14.10.2.2.4</span> random interaction intercept model</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb103"><pre class="sourceCode r fold-show code-with-copy"><code class="sourceCode r"><span id="cb103-1"><a href="#cb103-1" aria-hidden="true" tabindex="-1"></a>lmm4 <span class="ot">&lt;-</span>  <span class="fu">lmer</span>(y <span class="sc">~</span> treatment <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> block) <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> block<span class="sc">:</span>treatment),</span>
<span id="cb103-2"><a href="#cb103-2" aria-hidden="true" tabindex="-1"></a>               <span class="at">data =</span> figx)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="random-interaction-model-with-subsampling" class="level5" data-number="14.10.2.2.5">
<h5 data-number="14.10.2.2.5" class="anchored" data-anchor-id="random-interaction-model-with-subsampling"><span class="header-section-number">14.10.2.2.5</span> random interaction model with subsampling</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb104"><pre class="sourceCode r fold-show code-with-copy"><code class="sourceCode r"><span id="cb104-1"><a href="#cb104-1" aria-hidden="true" tabindex="-1"></a>lmm5 <span class="ot">&lt;-</span>  <span class="fu">lmer</span>(y <span class="sc">~</span> treatment <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> block) <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> block<span class="sc">:</span>treatment) <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> block<span class="sc">:</span>treatment<span class="sc">:</span>replicate),</span>
<span id="cb104-2"><a href="#cb104-2" aria-hidden="true" tabindex="-1"></a>               <span class="at">data =</span> figx)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="crd-split-plot-model" class="level5" data-number="14.10.2.2.6">
<h5 data-number="14.10.2.2.6" class="anchored" data-anchor-id="crd-split-plot-model"><span class="header-section-number">14.10.2.2.6</span> CRD split plot model</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb105"><pre class="sourceCode r fold-show code-with-copy"><code class="sourceCode r"><span id="cb105-1"><a href="#cb105-1" aria-hidden="true" tabindex="-1"></a><span class="co"># tr1 is the main plot</span></span>
<span id="cb105-2"><a href="#cb105-2" aria-hidden="true" tabindex="-1"></a><span class="co"># tr2 is the subplot</span></span>
<span id="cb105-3"><a href="#cb105-3" aria-hidden="true" tabindex="-1"></a><span class="co"># main_plot is tr1:rep, where rep is the rep id</span></span>
<span id="cb105-4"><a href="#cb105-4" aria-hidden="true" tabindex="-1"></a>lmm6 <span class="ot">&lt;-</span>  <span class="fu">lmer</span>(y <span class="sc">~</span> tr1 <span class="sc">*</span> tr2 <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> main_plot),</span>
<span id="cb105-5"><a href="#cb105-5" aria-hidden="true" tabindex="-1"></a>               <span class="at">data =</span> figx)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="rcbd-split-plot-model" class="level5" data-number="14.10.2.2.7">
<h5 data-number="14.10.2.2.7" class="anchored" data-anchor-id="rcbd-split-plot-model"><span class="header-section-number">14.10.2.2.7</span> RCBD split plot model</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb106"><pre class="sourceCode r fold-show code-with-copy"><code class="sourceCode r"><span id="cb106-1"><a href="#cb106-1" aria-hidden="true" tabindex="-1"></a><span class="co"># tr1 is the main plot</span></span>
<span id="cb106-2"><a href="#cb106-2" aria-hidden="true" tabindex="-1"></a><span class="co"># tr2 is the subplot</span></span>
<span id="cb106-3"><a href="#cb106-3" aria-hidden="true" tabindex="-1"></a><span class="co"># main_plot is tr1:block</span></span>
<span id="cb106-4"><a href="#cb106-4" aria-hidden="true" tabindex="-1"></a>lmm7 <span class="ot">&lt;-</span>  <span class="fu">lmer</span>(y <span class="sc">~</span> tr1 <span class="sc">*</span> tr2 <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> block) <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> block<span class="sc">:</span>tr1),</span>
<span id="cb106-5"><a href="#cb106-5" aria-hidden="true" tabindex="-1"></a>               <span class="at">data =</span> figx)</span>
<span id="cb106-6"><a href="#cb106-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-7"><a href="#cb106-7" aria-hidden="true" tabindex="-1"></a>lmm7 <span class="ot">&lt;-</span>  <span class="fu">lmer</span>(y <span class="sc">~</span> tr1 <span class="sc">*</span> tr2 <span class="sc">+</span> block <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> block<span class="sc">:</span>tr1),</span>
<span id="cb106-8"><a href="#cb106-8" aria-hidden="true" tabindex="-1"></a>               <span class="at">data =</span> figx)</span>
<span id="cb106-9"><a href="#cb106-9" aria-hidden="true" tabindex="-1"></a><span class="co"># the two versions are numerically equivalent</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="anova-models-mixed-models-repeated-measures-anova" class="level4" data-number="14.10.2.3">
<h4 data-number="14.10.2.3" class="anchored" data-anchor-id="anova-models-mixed-models-repeated-measures-anova"><span class="header-section-number">14.10.2.3</span> ANOVA models (“mixed models”, “repeated measures ANOVA”)</h4>
<section id="multivariate-repeated-measures-anova" class="level5" data-number="14.10.2.3.1">
<h5 data-number="14.10.2.3.1" class="anchored" data-anchor-id="multivariate-repeated-measures-anova"><span class="header-section-number">14.10.2.3.1</span> Multivariate repeated measures ANOVA</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb107"><pre class="sourceCode r fold-show code-with-copy"><code class="sourceCode r"><span id="cb107-1"><a href="#cb107-1" aria-hidden="true" tabindex="-1"></a>aov1 <span class="ot">&lt;-</span> <span class="fu">aov_4</span>(y <span class="sc">~</span> treatment <span class="sc">+</span> (treatment <span class="sc">|</span> block),</span>
<span id="cb107-2"><a href="#cb107-2" aria-hidden="true" tabindex="-1"></a>             <span class="at">data =</span> figx)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="univariate-repeated-measures-anova" class="level5" data-number="14.10.2.3.2">
<h5 data-number="14.10.2.3.2" class="anchored" data-anchor-id="univariate-repeated-measures-anova"><span class="header-section-number">14.10.2.3.2</span> Univariate repeated measures ANOVA</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb108"><pre class="sourceCode r fold-show code-with-copy"><code class="sourceCode r"><span id="cb108-1"><a href="#cb108-1" aria-hidden="true" tabindex="-1"></a>aov2 <span class="ot">&lt;-</span> <span class="fu">aov_4</span>(y <span class="sc">~</span> treatment <span class="sc">+</span> (treatment <span class="sc">|</span> block),</span>
<span id="cb108-2"><a href="#cb108-2" aria-hidden="true" tabindex="-1"></a>           <span class="at">include_aov =</span> <span class="cn">TRUE</span>,</span>
<span id="cb108-3"><a href="#cb108-3" aria-hidden="true" tabindex="-1"></a>             <span class="at">data =</span> figx)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="pairwise-paired-t-tests" class="level5" data-number="14.10.2.3.3">
<h5 data-number="14.10.2.3.3" class="anchored" data-anchor-id="pairwise-paired-t-tests"><span class="header-section-number">14.10.2.3.3</span> Pairwise paired, t-tests</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb109"><pre class="sourceCode r fold-show code-with-copy"><code class="sourceCode r"><span id="cb109-1"><a href="#cb109-1" aria-hidden="true" tabindex="-1"></a>pptt <span class="ot">&lt;-</span> <span class="fu">pairwise_t_tests</span>(<span class="at">y_col =</span> <span class="st">"y"</span>,</span>
<span id="cb109-2"><a href="#cb109-2" aria-hidden="true" tabindex="-1"></a>                         <span class="at">g_col =</span> <span class="st">"treatment"</span>,</span>
<span id="cb109-3"><a href="#cb109-3" aria-hidden="true" tabindex="-1"></a>                         <span class="at">id_col =</span> <span class="st">"block"</span>,</span>
<span id="cb109-4"><a href="#cb109-4" aria-hidden="true" tabindex="-1"></a>                         <span class="at">data =</span> figx)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
</section>
</section>
<section id="which-model-and-why" class="level2" data-number="14.11">
<h2 data-number="14.11" class="anchored" data-anchor-id="which-model-and-why"><span class="header-section-number">14.11</span> Which model and why?</h2>
<section id="crd" class="level3" data-number="14.11.1">
<h3 data-number="14.11.1" class="anchored" data-anchor-id="crd"><span class="header-section-number">14.11.1</span> CRD</h3>
<ol type="1">
<li>No batch to model!</li>
</ol>
</section>
<section id="crds" class="level3" data-number="14.11.2">
<h3 data-number="14.11.2" class="anchored" data-anchor-id="crds"><span class="header-section-number">14.11.2</span> CRDS</h3>
<ol type="1">
<li>lmm1 – random intercept model</li>
<li>lm2 – linear model of batch means</li>
</ol>
<p>Notes</p>
<ol type="1">
<li>These are numerically equivalent and should result in same estimates, SE, CIs, and p-values</li>
<li>These are equivalent to a <strong>Nested t-test</strong> or <strong>Nested ANOVA</strong></li>
</ol>
</section>
<section id="rcbd" class="level3" data-number="14.11.3">
<h3 data-number="14.11.3" class="anchored" data-anchor-id="rcbd"><span class="header-section-number">14.11.3</span> RCBD</h3>
<p>Reasonable models:</p>
<ol type="1">
<li>lmm1 – random intercept model</li>
<li>lmm2 – lmm for correlated error</li>
<li>lm1 – fixed effect model</li>
<li>aov1 – multivariate RM-ANOVA</li>
<li>aov2 – univariate RM-ANOVA</li>
<li>pptt – pairwise, paired t-test</li>
</ol>
<p>(Cannot use lmm3 or lmm4 because there is no replication of each block:treatment combination.)</p>
<p>Assumptions:</p>
<ol type="1">
<li>lmm1, lm1, and aov2 assume
<ul>
<li><strong>compound symmetry</strong>.</li>
<li><strong>sphericity</strong>.</li>
</ul></li>
<li>lmm2, aov1, and pptt do not assume compound symmetry and sphericity.</li>
</ol>
<p>Notes:</p>
<ol type="1">
<li>lmm1 is the standard. If the number of treatments = 2, then lmm1 is eqivalent with a paired t-test.</li>
<li>If the design is balanced (all blocks have a single value for both treatments) AND the number of treatments = 2 then <em>all six methods are numerically equivalent</em>.</li>
<li>If the design is balanced AND the number of treatments &gt; 2, then
<ul>
<li>lmm1, lm1, and aov2 are numerically equivalent</li>
<li>lmm2, aov1, pptt result in the same estimates and SE but lmm2 has more df, so the CIs, and p-values of lmm2 are less conservative.</li>
</ul></li>
<li>If the number of treatments = 2 AND the design is not balanced (at least one block is missing value for one treatment level) then the linear models (lm1, aov1, aov2, pptt) are equivalent but the linear mixed models (lmm1, lmm2) differ from each other and from the linear models.</li>
<li>If a treatment within a block is missing, the whole block is deleted in the RM-ANOVA models. This reduces power (the loss of power depends partly on the number of missing values).</li>
<li>If a treatment within a block is missing, the block is deleted only in the comparisons including the missing treatment in the pairwise, paired t-tests. This makes pptt more powerful than aov1.</li>
<li>the lmm models are very flexible – covariates can be added or these can be used as generalized lmms for non-normal distributions.</li>
<li>the lmm models and especially lmm2 sometimes (often?) fail to converge, especially with small samples (small number of blocks) or if the among-block component of variance is small.</li>
</ol>
<p>Performance:</p>
<div class="cell">
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<caption>Type I error rate the RCBD statistical models under different levels of correlation structure among the observations. The Cor Error column is the average correlated error of the residuals fit by a simple linear model y ~ treatment. Model abbreviations as in text.</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">Contrast</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Cor Error</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">lm0</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">lm1</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">aov1</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">aov2</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">pptt</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">lmm2</th>
</tr>
</thead>
<tbody>
<tr class="odd" data-grouplength="3">
<td colspan="8" style="border-bottom: 1px solid"><strong>sim 1 - high correlated error</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Cn, Tr1</td>
<td style="text-align: right;">0.299</td>
<td style="text-align: right;">0.047</td>
<td style="text-align: right;">0.110</td>
<td style="text-align: right;">0.058</td>
<td style="text-align: right;">0.110</td>
<td style="text-align: right;">0.058</td>
<td style="text-align: right;">0.068</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Cn, Tr2</td>
<td style="text-align: right;">0.412</td>
<td style="text-align: right;">0.027</td>
<td style="text-align: right;">0.090</td>
<td style="text-align: right;">0.052</td>
<td style="text-align: right;">0.090</td>
<td style="text-align: right;">0.052</td>
<td style="text-align: right;">0.070</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Tr1, Tr2</td>
<td style="text-align: right;">0.662</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: right;">0.005</td>
<td style="text-align: right;">0.063</td>
<td style="text-align: right;">0.005</td>
<td style="text-align: right;">0.063</td>
<td style="text-align: right;">0.070</td>
</tr>
<tr class="odd" data-grouplength="3">
<td colspan="8" style="border-bottom: 1px solid"><strong>sim 2 - low correlated error</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Cn, Tr1</td>
<td style="text-align: right;">0.165</td>
<td style="text-align: right;">0.044</td>
<td style="text-align: right;">0.072</td>
<td style="text-align: right;">0.055</td>
<td style="text-align: right;">0.072</td>
<td style="text-align: right;">0.055</td>
<td style="text-align: right;">0.063</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Cn, Tr2</td>
<td style="text-align: right;">0.164</td>
<td style="text-align: right;">0.051</td>
<td style="text-align: right;">0.077</td>
<td style="text-align: right;">0.060</td>
<td style="text-align: right;">0.077</td>
<td style="text-align: right;">0.060</td>
<td style="text-align: right;">0.076</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Tr1, Tr2</td>
<td style="text-align: right;">0.238</td>
<td style="text-align: right;">0.016</td>
<td style="text-align: right;">0.028</td>
<td style="text-align: right;">0.062</td>
<td style="text-align: right;">0.028</td>
<td style="text-align: right;">0.062</td>
<td style="text-align: right;">0.079</td>
</tr>
<tr class="odd" data-grouplength="3">
<td colspan="8" style="border-bottom: 1px solid"><strong>sim 3 - equal correlated error</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Cn, Tr1</td>
<td style="text-align: right;">0.771</td>
<td style="text-align: right;">0.001</td>
<td style="text-align: right;">0.057</td>
<td style="text-align: right;">0.058</td>
<td style="text-align: right;">0.057</td>
<td style="text-align: right;">0.058</td>
<td style="text-align: right;">0.069</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Cn, Tr2</td>
<td style="text-align: right;">0.773</td>
<td style="text-align: right;">0.001</td>
<td style="text-align: right;">0.049</td>
<td style="text-align: right;">0.056</td>
<td style="text-align: right;">0.049</td>
<td style="text-align: right;">0.056</td>
<td style="text-align: right;">0.067</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Tr1, Tr2</td>
<td style="text-align: right;">0.769</td>
<td style="text-align: right;">0.002</td>
<td style="text-align: right;">0.056</td>
<td style="text-align: right;">0.056</td>
<td style="text-align: right;">0.056</td>
<td style="text-align: right;">0.056</td>
<td style="text-align: right;">0.074</td>
</tr>
<tr class="odd" data-grouplength="3">
<td colspan="8" style="border-bottom: 1px solid"><strong>sim 4 - zero correlated error</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Cn, Tr1</td>
<td style="text-align: right;">0.025</td>
<td style="text-align: right;">0.050</td>
<td style="text-align: right;">0.047</td>
<td style="text-align: right;">0.042</td>
<td style="text-align: right;">0.047</td>
<td style="text-align: right;">0.042</td>
<td style="text-align: right;">0.061</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Cn, Tr2</td>
<td style="text-align: right;">0.010</td>
<td style="text-align: right;">0.054</td>
<td style="text-align: right;">0.054</td>
<td style="text-align: right;">0.046</td>
<td style="text-align: right;">0.054</td>
<td style="text-align: right;">0.046</td>
<td style="text-align: right;">0.061</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em;" data-indentlevel="1">Tr1, Tr2</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: right;">0.056</td>
<td style="text-align: right;">0.060</td>
<td style="text-align: right;">0.053</td>
<td style="text-align: right;">0.060</td>
<td style="text-align: right;">0.053</td>
<td style="text-align: right;">0.076</td>
</tr>
</tbody>
</table>


</div>
</div>
<ol type="1">
<li>aov1, pptt, and lmm2 have well-behaved type I error rates (lmm2 is a little anti-conservative) regardless of correlation structure – False discoveries should be well controlled.</li>
<li>lm1, lmm1, and aov2 have poorly-behaved type I error rates when there is a block:treatment interaction (sim 1 and 2), resulting in heterogenous correlation structure (remember that this is moot when there are only two treatment levels) – this consequence is small when the block and block:treatment variance is small (small correlations in residuals). When there is block:treatment interaction, these models are conservative in contrasts associated with large residual correlation, resulting in low power, and anti-conservative in contrasts associated with small residual correlations. resulting in high false discovery rates.</li>
</ol>
<p>Best Practices:</p>
<ol type="1">
<li>If the number of treatments = 2 and
<ul>
<li>the design is balanced: it doesn’t matter which method you use.</li>
<li>the design is not balanced: use lmm1 or lmm2</li>
</ul></li>
<li>If the number of treatments &gt; 2 and
<ul>
<li>the design is balanced: use aov1/pptt (these are numerically equivalent). lmm1 is the standard but it assumes equal correlated error among treatment levels and equal standard errors among contrasts. lmm2 explicitly models heterogeneity of correlations and variances but 1) the model often fails and 2) has slightly anti-conservative Type I error.</li>
<li>the design is not balanced: lmm2 is attractive if the model runs. pptt is attractive if there are few missing values.</li>
</ul></li>
<li>If there are covariates, then start with lmm2. Use lmeControl if lmm2 fails. If this fails, use pairwise lmm1, which is the same as pairwise t-tests but allows covariates to be added to the model. Note that the covariate will be modeled independently in each pair, which effectively models the consequence of a treatment:covariate interaction.</li>
</ol>
</section>
<section id="rspd" class="level3" data-number="14.11.4">
<h3 data-number="14.11.4" class="anchored" data-anchor-id="rspd"><span class="header-section-number">14.11.4</span> RSPD</h3>
<p>Reasonable models:</p>
<ol type="1">
<li>lmm4 – random interaction intercept model</li>
</ol>
<p>Assumptions:</p>
<ol type="1">
<li></li>
</ol>
<p>Notes:</p>
<ol type="1">
<li></li>
</ol>
<p>Performance:</p>
<div class="cell">
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<caption>Type I error rate for the RSPD statistical models under different models of random variance. The design is 2 (main plot: WT, KO) x 3 (subplot: Cn, Tr1, Tr2) with ten blocks. All four simulations have a component of variance due to block. Sim 1 includes a block:main plot component, Sim 2 includes a block:sub plot component, Sim 3 includes both block:main and block:sub components. Type I error rates for the 9 simple effect contrasts were averaged within the two sets: contrast = main is the aggregate of the single main plot contrast (KO - WT) at each level of the subplot factor. contrast = sub is the aggregate of the three subplot contrasts (Tr1 - Cn, Tr2 - Cn, Tr2 - Tr1) at each level of the main plot factor. The Cor Error column is the average correlated error of the residuals fit by a simple linear model y ~ treatment. Model abbreviations as in text.</caption>
<thead>
<tr class="header">
<th style="text-align: right;" data-quarto-table-cell-role="th">sim_id</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">sig_block</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">sig_ss</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">sig_main</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">sig_sub</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">contrast</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">r</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">lm0</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">lm1</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">lmm1</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">lmm3</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">lmm4</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">lmm5</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">lmm6</th>
</tr>
</thead>
<tbody>
<tr class="odd" data-grouplength="2">
<td colspan="14" style="border-bottom: 1px solid"><strong>sim 1 - block:main plot</strong></td>
</tr>
<tr class="even">
<td style="text-align: right; padding-left: 2em;" data-indentlevel="1">1</td>
<td style="text-align: right;">0.8</td>
<td style="text-align: right;">0.5</td>
<td style="text-align: right;">0.6</td>
<td style="text-align: right;">0.0</td>
<td style="text-align: left;">main</td>
<td style="text-align: right;">0.494</td>
<td style="text-align: right;">0.010</td>
<td style="text-align: right;">0.199</td>
<td style="text-align: right;">0.087</td>
<td style="text-align: right;">0.051</td>
<td style="text-align: right;">0.052</td>
<td style="text-align: right;">0.083</td>
<td style="text-align: right;">0.054</td>
</tr>
<tr class="odd">
<td style="text-align: right; padding-left: 2em;" data-indentlevel="1">1</td>
<td style="text-align: right;">0.8</td>
<td style="text-align: right;">0.5</td>
<td style="text-align: right;">0.6</td>
<td style="text-align: right;">0.0</td>
<td style="text-align: left;">sub</td>
<td style="text-align: right;">0.783</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: right;">0.052</td>
<td style="text-align: right;">0.011</td>
<td style="text-align: right;">0.052</td>
<td style="text-align: right;">0.052</td>
<td style="text-align: right;">0.011</td>
<td style="text-align: right;">0.052</td>
</tr>
<tr class="even" data-grouplength="2">
<td colspan="14" style="border-bottom: 1px solid"><strong>sim 2 - block:sub plot</strong></td>
</tr>
<tr class="odd">
<td style="text-align: right; padding-left: 2em;" data-indentlevel="1">2</td>
<td style="text-align: right;">0.8</td>
<td style="text-align: right;">0.5</td>
<td style="text-align: right;">0.0</td>
<td style="text-align: right;">0.6</td>
<td style="text-align: left;">main</td>
<td style="text-align: right;">0.782</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: right;">0.002</td>
<td style="text-align: right;">0.003</td>
<td style="text-align: right;">0.002</td>
<td style="text-align: right;">0.003</td>
<td style="text-align: right;">0.042</td>
<td style="text-align: right;">0.041</td>
</tr>
<tr class="even">
<td style="text-align: right; padding-left: 2em;" data-indentlevel="1">2</td>
<td style="text-align: right;">0.8</td>
<td style="text-align: right;">0.5</td>
<td style="text-align: right;">0.0</td>
<td style="text-align: right;">0.6</td>
<td style="text-align: left;">sub</td>
<td style="text-align: right;">0.497</td>
<td style="text-align: right;">0.008</td>
<td style="text-align: right;">0.053</td>
<td style="text-align: right;">0.066</td>
<td style="text-align: right;">0.068</td>
<td style="text-align: right;">0.065</td>
<td style="text-align: right;">0.050</td>
<td style="text-align: right;">0.053</td>
</tr>
<tr class="odd" data-grouplength="2">
<td colspan="14" style="border-bottom: 1px solid"><strong>sim 3 - block:main + block:sub</strong></td>
</tr>
<tr class="even">
<td style="text-align: right; padding-left: 2em;" data-indentlevel="1">3</td>
<td style="text-align: right;">0.8</td>
<td style="text-align: right;">0.5</td>
<td style="text-align: right;">0.6</td>
<td style="text-align: right;">0.4</td>
<td style="text-align: left;">main</td>
<td style="text-align: right;">0.549</td>
<td style="text-align: right;">0.007</td>
<td style="text-align: right;">0.111</td>
<td style="text-align: right;">0.062</td>
<td style="text-align: right;">0.038</td>
<td style="text-align: right;">0.039</td>
<td style="text-align: right;">0.068</td>
<td style="text-align: right;">0.054</td>
</tr>
<tr class="odd">
<td style="text-align: right; padding-left: 2em;" data-indentlevel="1">3</td>
<td style="text-align: right;">0.8</td>
<td style="text-align: right;">0.5</td>
<td style="text-align: right;">0.6</td>
<td style="text-align: right;">0.4</td>
<td style="text-align: left;">sub</td>
<td style="text-align: right;">0.689</td>
<td style="text-align: right;">0.001</td>
<td style="text-align: right;">0.056</td>
<td style="text-align: right;">0.026</td>
<td style="text-align: right;">0.057</td>
<td style="text-align: right;">0.056</td>
<td style="text-align: right;">0.024</td>
<td style="text-align: right;">0.054</td>
</tr>
<tr class="even" data-grouplength="2">
<td colspan="14" style="border-bottom: 1px solid"><strong>sim 4 - block only</strong></td>
</tr>
<tr class="odd">
<td style="text-align: right; padding-left: 2em;" data-indentlevel="1">4</td>
<td style="text-align: right;">0.8</td>
<td style="text-align: right;">0.5</td>
<td style="text-align: right;">0.0</td>
<td style="text-align: right;">0.0</td>
<td style="text-align: left;">main</td>
<td style="text-align: right;">0.707</td>
<td style="text-align: right;">0.002</td>
<td style="text-align: right;">0.049</td>
<td style="text-align: right;">0.049</td>
<td style="text-align: right;">0.044</td>
<td style="text-align: right;">0.045</td>
<td style="text-align: right;">0.056</td>
<td style="text-align: right;">0.054</td>
</tr>
<tr class="even">
<td style="text-align: right; padding-left: 2em;" data-indentlevel="1">4</td>
<td style="text-align: right;">0.8</td>
<td style="text-align: right;">0.5</td>
<td style="text-align: right;">0.0</td>
<td style="text-align: right;">0.0</td>
<td style="text-align: left;">sub</td>
<td style="text-align: right;">0.704</td>
<td style="text-align: right;">0.002</td>
<td style="text-align: right;">0.057</td>
<td style="text-align: right;">0.055</td>
<td style="text-align: right;">0.061</td>
<td style="text-align: right;">0.060</td>
<td style="text-align: right;">0.052</td>
<td style="text-align: right;">0.059</td>
</tr>
</tbody>
</table>


</div>
</div>
</section>
<section id="rcbds" class="level3" data-number="14.11.5">
<h3 data-number="14.11.5" class="anchored" data-anchor-id="rcbds"><span class="header-section-number">14.11.5</span> RCBDS</h3>
<ol type="1">
<li>lmm3 – random intercept and slopes model</li>
<li>lmm4 – random interaction intercept model</li>
<li>lmm1 – random intercept model on batch means</li>
<li>lmm2 – lmm for correlated error on batch means</li>
<li>aov1 – multivariate RM-ANOVA</li>
<li>aov2 – univariate RM-ANOVA</li>
<li>pptt – pairwise, paired t-test</li>
</ol>
<p>Notes</p>
<ol type="1">
<li>lmm3 is the “maximal model” – it fits the most parameters and has the fewest assumptions. lmm1 on the full data is the minimal model. I’m not recommending this at all as this is pseudoreplication and will result in very anti-conservative inference. lmm4 is less conservative than lmm3 (and makes more assumptions).</li>
<li>lmm1 and lmm2 <em>here</em> are the models for RCBD designs but using the aggregated data - that is the subsampled replicates within a batch (block:treatment combination) are averaged.</li>
<li>If design is balanced
<ul>
<li>lmm3 and aov1 result in same estimates, SE, CIs, and p-values</li>
<li>lmm4, lmm1, and aov2 result in same estimates, SE, CIs, and p-values</li>
<li>lmm2 has same estimates and SE as lmm3/aov1 but more df so less conservative</li>
</ul></li>
<li>If a treatment within a block is missing, the whole block is deleted in the RM-ANOVA models. This reduces power</li>
<li>the lmm models are very flexible – covariates can be added or these can be used as generalized lmms for non-normal distributions.</li>
<li>the lmm models, and especially lmm2 and lmm3, sometimes (often?) fail to converge, especially with small samples (small number of blocks) or if within block correlation is small</li>
<li>lmm1, lmm4, and aov2 will be less conservative (more power but at higher type I error/false positive rate)</li>
</ol>
<p>Best practices</p>
<ol type="1">
<li>If balanced and no covariates then use lmm3/lmm2/aov1 if discovery is expensive (want to avoid false positives) or lmm4/aov2/lmm1 if discovery is cheap (can afford false positives). lmm2 is less conservative than lmm3/aov1. Use lmm3 if interested in the variance at different levels and/or covariance structure.</li>
<li>If unbalanced, or if there are covariates, then start with lmm3 or lmm2. Use lmeControl if lmm2 fails. If these fail, go to 1.</li>
</ol>
</section>
<section id="grcbd" class="level3" data-number="14.11.6">
<h3 data-number="14.11.6" class="anchored" data-anchor-id="grcbd"><span class="header-section-number">14.11.6</span> GRCBD</h3>
<ol type="1">
<li>lmm3 – random intercept and slopes model</li>
<li>lmm4 – random interaction intercept model</li>
<li>lmm1 – random intercept model on block:treatment means</li>
<li>lmm2 – lmm for correlated error on block:treatment means</li>
<li>aov1 – multivariate RM-ANOVA</li>
<li>aov2 – univariate RM-ANOVA</li>
<li>lm1 – fixed effect model on all data</li>
<li>pptt – pairwise, paired t-test on block:treatment means</li>
</ol>
</section>
<section id="grcbds" class="level3" data-number="14.11.7">
<h3 data-number="14.11.7" class="anchored" data-anchor-id="grcbds"><span class="header-section-number">14.11.7</span> GRCBDS</h3>
<ol type="1">
<li>lmm3 – random intercept and slopes model</li>
<li>lmm4 – random interaction intercept model</li>
<li>lmm1 – random intercept model on block:treatment means</li>
<li>lmm2 – lmm for correlated error on block:treatment means</li>
<li>aov1 – multivariate RM-ANOVA</li>
<li>aov2 – univariate RM-ANOVA</li>
<li>lm1 – fixed effect model on all block:treatment:rep means</li>
<li>pptt – pairwise, paired t-test on block:treatment:rep means</li>
</ol>
</section>
</section>
<section id="working-in-r" class="level2" data-number="14.12">
<h2 data-number="14.12" class="anchored" data-anchor-id="working-in-r"><span class="header-section-number">14.12</span> Working in R</h2>
<section id="plotting-models-fit-to-batched-data" class="level3" data-number="14.12.1">
<h3 data-number="14.12.1" class="anchored" data-anchor-id="plotting-models-fit-to-batched-data"><span class="header-section-number">14.12.1</span> Plotting models fit to batched data</h3>
<section id="models-without-subsampling---experiment-6g" class="level4" data-number="14.12.1.1">
<h4 data-number="14.12.1.1" class="anchored" data-anchor-id="models-without-subsampling---experiment-6g"><span class="header-section-number">14.12.1.1</span> Models without subsampling - Experiment 6g</h4>
<p>Data wrangling necessary for plot:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb110"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb110-1"><a href="#cb110-1" aria-hidden="true" tabindex="-1"></a><span class="co"># convert contrast table to a data.table</span></span>
<span id="cb110-2"><a href="#cb110-2" aria-hidden="true" tabindex="-1"></a>exp6g_m1_planned_dt <span class="ot">&lt;-</span> <span class="fu">data.table</span>(exp6g_m1_planned)</span>
<span id="cb110-3"><a href="#cb110-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb110-4"><a href="#cb110-4" aria-hidden="true" tabindex="-1"></a><span class="co"># create a pretty p-value column</span></span>
<span id="cb110-5"><a href="#cb110-5" aria-hidden="true" tabindex="-1"></a>exp6g_m1_planned_dt[, pretty_p <span class="sc">:</span><span class="er">=</span> <span class="fu">pvalString</span>(p.value)]</span>
<span id="cb110-6"><a href="#cb110-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb110-7"><a href="#cb110-7" aria-hidden="true" tabindex="-1"></a><span class="co"># add group1 and group2 columns to exp1g_m1_planned</span></span>
<span id="cb110-8"><a href="#cb110-8" aria-hidden="true" tabindex="-1"></a>exp6g_m1_planned_dt[, group1 <span class="sc">:</span><span class="er">=</span> <span class="fu">c</span>(<span class="st">"Saline"</span>, <span class="st">"CNO"</span>, <span class="st">"Post_CNO"</span>)]</span>
<span id="cb110-9"><a href="#cb110-9" aria-hidden="true" tabindex="-1"></a>exp6g_m1_planned_dt[, group2 <span class="sc">:</span><span class="er">=</span> <span class="fu">c</span>(<span class="st">"Lesion"</span>, <span class="st">"Saline"</span>, <span class="st">"CNO"</span>)]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Experiments are colored:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb111"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb111-1"><a href="#cb111-1" aria-hidden="true" tabindex="-1"></a>gg1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> exp6g,</span>
<span id="cb111-2"><a href="#cb111-2" aria-hidden="true" tabindex="-1"></a>              <span class="fu">aes</span>(<span class="at">x =</span> treatment,</span>
<span id="cb111-3"><a href="#cb111-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">y =</span> touch,</span>
<span id="cb111-4"><a href="#cb111-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">color =</span> mouse_id)) <span class="sc">+</span></span>
<span id="cb111-5"><a href="#cb111-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.2</span>)) <span class="sc">+</span></span>
<span id="cb111-6"><a href="#cb111-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group =</span> mouse_id),</span>
<span id="cb111-7"><a href="#cb111-7" aria-hidden="true" tabindex="-1"></a>            <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.2</span>),</span>
<span id="cb111-8"><a href="#cb111-8" aria-hidden="true" tabindex="-1"></a>            <span class="at">color =</span> <span class="st">"gray80"</span>) <span class="sc">+</span></span>
<span id="cb111-9"><a href="#cb111-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> <span class="fu">summary</span>(exp6g_m1_emm),</span>
<span id="cb111-10"><a href="#cb111-10" aria-hidden="true" tabindex="-1"></a>             <span class="fu">aes</span>(<span class="at">y =</span> emmean),</span>
<span id="cb111-11"><a href="#cb111-11" aria-hidden="true" tabindex="-1"></a>             <span class="at">color =</span> <span class="st">"black"</span>,</span>
<span id="cb111-12"><a href="#cb111-12" aria-hidden="true" tabindex="-1"></a>             <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb111-13"><a href="#cb111-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_errorbar</span>(<span class="at">data =</span> <span class="fu">summary</span>(exp6g_m1_emm),</span>
<span id="cb111-14"><a href="#cb111-14" aria-hidden="true" tabindex="-1"></a>             <span class="fu">aes</span>(<span class="at">y =</span> emmean,</span>
<span id="cb111-15"><a href="#cb111-15" aria-hidden="true" tabindex="-1"></a>                 <span class="at">ymin =</span> lower.CL,</span>
<span id="cb111-16"><a href="#cb111-16" aria-hidden="true" tabindex="-1"></a>                 <span class="at">ymax =</span> upper.CL),</span>
<span id="cb111-17"><a href="#cb111-17" aria-hidden="true" tabindex="-1"></a>             <span class="at">color =</span> <span class="st">"black"</span>,</span>
<span id="cb111-18"><a href="#cb111-18" aria-hidden="true" tabindex="-1"></a>             <span class="at">width =</span> .<span class="dv">05</span>) <span class="sc">+</span></span>
<span id="cb111-19"><a href="#cb111-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-20"><a href="#cb111-20" aria-hidden="true" tabindex="-1"></a>  <span class="co"># pvalue brackets</span></span>
<span id="cb111-21"><a href="#cb111-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_pvalue_manual</span>(exp6g_m1_planned_dt,</span>
<span id="cb111-22"><a href="#cb111-22" aria-hidden="true" tabindex="-1"></a>                     <span class="at">label =</span> <span class="st">"pretty_p"</span>,</span>
<span id="cb111-23"><a href="#cb111-23" aria-hidden="true" tabindex="-1"></a>                     <span class="at">y.position =</span> <span class="fu">c</span>(<span class="dv">99</span>,<span class="dv">97</span>,<span class="dv">95</span>),</span>
<span id="cb111-24"><a href="#cb111-24" aria-hidden="true" tabindex="-1"></a>                     <span class="at">size =</span> <span class="fl">2.5</span>,</span>
<span id="cb111-25"><a href="#cb111-25" aria-hidden="true" tabindex="-1"></a>                     <span class="at">tip.length =</span> <span class="fl">0.01</span>) <span class="sc">+</span></span>
<span id="cb111-26"><a href="#cb111-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-27"><a href="#cb111-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Percent ipsilateral touch"</span>) <span class="sc">+</span></span>
<span id="cb111-28"><a href="#cb111-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> pal_okabe_ito_blue) <span class="sc">+</span></span>
<span id="cb111-29"><a href="#cb111-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_pubr</span>() <span class="sc">+</span></span>
<span id="cb111-30"><a href="#cb111-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb111-31"><a href="#cb111-31" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(), <span class="co"># no x-axis title</span></span>
<span id="cb111-32"><a href="#cb111-32" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">"none"</span></span>
<span id="cb111-33"><a href="#cb111-33" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span> </span>
<span id="cb111-34"><a href="#cb111-34" aria-hidden="true" tabindex="-1"></a>  <span class="cn">NULL</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Treatments are colored:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb112"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb112-1"><a href="#cb112-1" aria-hidden="true" tabindex="-1"></a>gg2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> exp6g,</span>
<span id="cb112-2"><a href="#cb112-2" aria-hidden="true" tabindex="-1"></a>              <span class="fu">aes</span>(<span class="at">x =</span> treatment,</span>
<span id="cb112-3"><a href="#cb112-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">y =</span> touch)) <span class="sc">+</span></span>
<span id="cb112-4"><a href="#cb112-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">group =</span> mouse_id),</span>
<span id="cb112-5"><a href="#cb112-5" aria-hidden="true" tabindex="-1"></a>             <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.2</span>),</span>
<span id="cb112-6"><a href="#cb112-6" aria-hidden="true" tabindex="-1"></a>             <span class="at">color =</span> <span class="st">"gray"</span>) <span class="sc">+</span></span>
<span id="cb112-7"><a href="#cb112-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group =</span> mouse_id),</span>
<span id="cb112-8"><a href="#cb112-8" aria-hidden="true" tabindex="-1"></a>            <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.2</span>),</span>
<span id="cb112-9"><a href="#cb112-9" aria-hidden="true" tabindex="-1"></a>            <span class="at">color =</span> <span class="st">"gray80"</span>) <span class="sc">+</span></span>
<span id="cb112-10"><a href="#cb112-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> <span class="fu">summary</span>(exp6g_m1_emm),</span>
<span id="cb112-11"><a href="#cb112-11" aria-hidden="true" tabindex="-1"></a>             <span class="fu">aes</span>(<span class="at">y =</span> emmean,</span>
<span id="cb112-12"><a href="#cb112-12" aria-hidden="true" tabindex="-1"></a>             <span class="at">color =</span> treatment),</span>
<span id="cb112-13"><a href="#cb112-13" aria-hidden="true" tabindex="-1"></a>             <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb112-14"><a href="#cb112-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_errorbar</span>(<span class="at">data =</span> <span class="fu">summary</span>(exp6g_m1_emm),</span>
<span id="cb112-15"><a href="#cb112-15" aria-hidden="true" tabindex="-1"></a>             <span class="fu">aes</span>(<span class="at">y =</span> emmean,</span>
<span id="cb112-16"><a href="#cb112-16" aria-hidden="true" tabindex="-1"></a>                 <span class="at">ymin =</span> lower.CL,</span>
<span id="cb112-17"><a href="#cb112-17" aria-hidden="true" tabindex="-1"></a>                 <span class="at">ymax =</span> upper.CL,</span>
<span id="cb112-18"><a href="#cb112-18" aria-hidden="true" tabindex="-1"></a>             <span class="at">color =</span> treatment),</span>
<span id="cb112-19"><a href="#cb112-19" aria-hidden="true" tabindex="-1"></a>             <span class="at">width =</span> .<span class="dv">05</span>) <span class="sc">+</span></span>
<span id="cb112-20"><a href="#cb112-20" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb112-21"><a href="#cb112-21" aria-hidden="true" tabindex="-1"></a>  <span class="co"># pvalue brackets</span></span>
<span id="cb112-22"><a href="#cb112-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_pvalue_manual</span>(exp6g_m1_planned_dt,</span>
<span id="cb112-23"><a href="#cb112-23" aria-hidden="true" tabindex="-1"></a>                     <span class="at">label =</span> <span class="st">"pretty_p"</span>,</span>
<span id="cb112-24"><a href="#cb112-24" aria-hidden="true" tabindex="-1"></a>                     <span class="at">y.position =</span> <span class="fu">c</span>(<span class="dv">99</span>,<span class="dv">97</span>,<span class="dv">95</span>),</span>
<span id="cb112-25"><a href="#cb112-25" aria-hidden="true" tabindex="-1"></a>                     <span class="at">size =</span> <span class="fl">2.5</span>,</span>
<span id="cb112-26"><a href="#cb112-26" aria-hidden="true" tabindex="-1"></a>                     <span class="at">tip.length =</span> <span class="fl">0.01</span>) <span class="sc">+</span></span>
<span id="cb112-27"><a href="#cb112-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb112-28"><a href="#cb112-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Percent ipsilateral touch"</span>) <span class="sc">+</span></span>
<span id="cb112-29"><a href="#cb112-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> pal_okabe_ito_blue) <span class="sc">+</span></span>
<span id="cb112-30"><a href="#cb112-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_pubr</span>() <span class="sc">+</span></span>
<span id="cb112-31"><a href="#cb112-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb112-32"><a href="#cb112-32" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(), <span class="co"># no x-axis title</span></span>
<span id="cb112-33"><a href="#cb112-33" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">"none"</span></span>
<span id="cb112-34"><a href="#cb112-34" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span> </span>
<span id="cb112-35"><a href="#cb112-35" aria-hidden="true" tabindex="-1"></a>  <span class="cn">NULL</span></span>
<span id="cb112-36"><a href="#cb112-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb112-37"><a href="#cb112-37" aria-hidden="true" tabindex="-1"></a><span class="co">#gg2</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="lmm_files/figure-html/unnamed-chunk-57-1.png" class="img-fluid figure-img" width="1066"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="models-with-subsampling---experiment-1g" class="level4" data-number="14.12.1.2">
<h4 data-number="14.12.1.2" class="anchored" data-anchor-id="models-with-subsampling---experiment-1g"><span class="header-section-number">14.12.1.2</span> Models with subsampling - Experiment 1g</h4>
<p>Data wrangling necessary for plot:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb113"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb113-1"><a href="#cb113-1" aria-hidden="true" tabindex="-1"></a><span class="co"># add treatment column to emmeans table</span></span>
<span id="cb113-2"><a href="#cb113-2" aria-hidden="true" tabindex="-1"></a>exp1g_m1_emm_dt <span class="ot">&lt;-</span> <span class="fu">summary</span>(exp1g_m1_emm) <span class="sc">%&gt;%</span></span>
<span id="cb113-3"><a href="#cb113-3" aria-hidden="true" tabindex="-1"></a>  data.table</span>
<span id="cb113-4"><a href="#cb113-4" aria-hidden="true" tabindex="-1"></a>exp1g_m1_emm_dt[, treatment <span class="sc">:</span><span class="er">=</span> <span class="fu">paste</span>(genotype, sex)]</span>
<span id="cb113-5"><a href="#cb113-5" aria-hidden="true" tabindex="-1"></a>exp1g_m1_emm_dt[, treatment <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(treatment,</span>
<span id="cb113-6"><a href="#cb113-6" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">levels =</span> <span class="fu">levels</span>(exp1g<span class="sc">$</span>treatment))]</span>
<span id="cb113-7"><a href="#cb113-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-8"><a href="#cb113-8" aria-hidden="true" tabindex="-1"></a><span class="co"># create table of means for each treatment * experiment_id combination</span></span>
<span id="cb113-9"><a href="#cb113-9" aria-hidden="true" tabindex="-1"></a>exp1g[, group_mean <span class="sc">:</span><span class="er">=</span> <span class="fu">predict</span>(exp1g_m1)]</span>
<span id="cb113-10"><a href="#cb113-10" aria-hidden="true" tabindex="-1"></a>exp1g_m1_emm2 <span class="ot">&lt;-</span> exp1g[, .(<span class="at">group_mean =</span> <span class="fu">mean</span>(group_mean)),</span>
<span id="cb113-11"><a href="#cb113-11" aria-hidden="true" tabindex="-1"></a>                       by <span class="ot">=</span> .(treatment, experiment_id)]</span>
<span id="cb113-12"><a href="#cb113-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-13"><a href="#cb113-13" aria-hidden="true" tabindex="-1"></a><span class="co"># add group1 and group2 columns to exp1g_m1_planned</span></span>
<span id="cb113-14"><a href="#cb113-14" aria-hidden="true" tabindex="-1"></a>exp1g_m1_planned_dt <span class="ot">&lt;-</span> <span class="fu">data.table</span>(exp1g_m1_planned)</span>
<span id="cb113-15"><a href="#cb113-15" aria-hidden="true" tabindex="-1"></a>exp1g_m1_planned_dt[, pretty_p <span class="sc">:</span><span class="er">=</span> <span class="fu">pvalString</span>(p.value)]</span>
<span id="cb113-16"><a href="#cb113-16" aria-hidden="true" tabindex="-1"></a>exp1g_m1_planned_dt[, group1 <span class="sc">:</span><span class="er">=</span> <span class="fu">c</span>(<span class="st">"Gpr174- M"</span>, <span class="st">"Gpr174- F"</span>, <span class="st">""</span>)]</span>
<span id="cb113-17"><a href="#cb113-17" aria-hidden="true" tabindex="-1"></a>exp1g_m1_planned_dt[, group2 <span class="sc">:</span><span class="er">=</span> <span class="fu">c</span>(<span class="st">"Gpr174+ M"</span>, <span class="st">"Gpr174+ F"</span>, <span class="st">""</span>)]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb114"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb114-1"><a href="#cb114-1" aria-hidden="true" tabindex="-1"></a><span class="co"># get coefficients of model</span></span>
<span id="cb114-2"><a href="#cb114-2" aria-hidden="true" tabindex="-1"></a>b <span class="ot">&lt;-</span> exp1g_m1_coef[, <span class="st">"Estimate"</span>]</span>
<span id="cb114-3"><a href="#cb114-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-4"><a href="#cb114-4" aria-hidden="true" tabindex="-1"></a><span class="co"># get interaction p</span></span>
<span id="cb114-5"><a href="#cb114-5" aria-hidden="true" tabindex="-1"></a>p_ixn <span class="ot">&lt;-</span> exp1g_m1_planned_dt[contrast <span class="sc">==</span> <span class="st">"Interaction"</span>, pretty_p]</span>
<span id="cb114-6"><a href="#cb114-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-7"><a href="#cb114-7" aria-hidden="true" tabindex="-1"></a>exp1g_plot_a <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> exp1g,</span>
<span id="cb114-8"><a href="#cb114-8" aria-hidden="true" tabindex="-1"></a>              <span class="fu">aes</span>(<span class="at">x =</span> treatment,</span>
<span id="cb114-9"><a href="#cb114-9" aria-hidden="true" tabindex="-1"></a>                  <span class="at">y =</span> gc,</span>
<span id="cb114-10"><a href="#cb114-10" aria-hidden="true" tabindex="-1"></a>                  <span class="at">color =</span> experiment_id)) <span class="sc">+</span></span>
<span id="cb114-11"><a href="#cb114-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># modeled experiment by treatment means</span></span>
<span id="cb114-12"><a href="#cb114-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> exp1g_m1_emm2,</span>
<span id="cb114-13"><a href="#cb114-13" aria-hidden="true" tabindex="-1"></a>            <span class="fu">aes</span>(<span class="at">y =</span> group_mean,</span>
<span id="cb114-14"><a href="#cb114-14" aria-hidden="true" tabindex="-1"></a>                <span class="at">color =</span> experiment_id),</span>
<span id="cb114-15"><a href="#cb114-15" aria-hidden="true" tabindex="-1"></a>            <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.4</span>),</span>
<span id="cb114-16"><a href="#cb114-16" aria-hidden="true" tabindex="-1"></a>            <span class="at">alpha =</span> <span class="dv">1</span>,</span>
<span id="cb114-17"><a href="#cb114-17" aria-hidden="true" tabindex="-1"></a>            <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb114-18"><a href="#cb114-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">data =</span> exp1g_m1_emm2,</span>
<span id="cb114-19"><a href="#cb114-19" aria-hidden="true" tabindex="-1"></a>            <span class="fu">aes</span>(<span class="at">y =</span> group_mean,</span>
<span id="cb114-20"><a href="#cb114-20" aria-hidden="true" tabindex="-1"></a>                <span class="at">group =</span> experiment_id,</span>
<span id="cb114-21"><a href="#cb114-21" aria-hidden="true" tabindex="-1"></a>                <span class="at">color =</span> experiment_id),</span>
<span id="cb114-22"><a href="#cb114-22" aria-hidden="true" tabindex="-1"></a>            <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.4</span>)) <span class="sc">+</span></span>
<span id="cb114-23"><a href="#cb114-23" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb114-24"><a href="#cb114-24" aria-hidden="true" tabindex="-1"></a>  <span class="co"># raw data</span></span>
<span id="cb114-25"><a href="#cb114-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.4</span>),</span>
<span id="cb114-26"><a href="#cb114-26" aria-hidden="true" tabindex="-1"></a>             <span class="at">alpha =</span> <span class="fl">0.3</span></span>
<span id="cb114-27"><a href="#cb114-27" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb114-28"><a href="#cb114-28" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb114-29"><a href="#cb114-29" aria-hidden="true" tabindex="-1"></a>  <span class="co"># modeled treatment means</span></span>
<span id="cb114-30"><a href="#cb114-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> exp1g_m1_emm_dt,</span>
<span id="cb114-31"><a href="#cb114-31" aria-hidden="true" tabindex="-1"></a>             <span class="fu">aes</span>(<span class="at">y =</span> emmean),</span>
<span id="cb114-32"><a href="#cb114-32" aria-hidden="true" tabindex="-1"></a>             <span class="at">color =</span> <span class="st">"black"</span>,</span>
<span id="cb114-33"><a href="#cb114-33" aria-hidden="true" tabindex="-1"></a>             <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb114-34"><a href="#cb114-34" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_errorbar(data = exp1g_m1_emm_dt,</span></span>
<span id="cb114-35"><a href="#cb114-35" aria-hidden="true" tabindex="-1"></a>  <span class="co">#            aes(y = emmean,</span></span>
<span id="cb114-36"><a href="#cb114-36" aria-hidden="true" tabindex="-1"></a>  <span class="co">#                ymin = lower.CL,</span></span>
<span id="cb114-37"><a href="#cb114-37" aria-hidden="true" tabindex="-1"></a>  <span class="co">#                ymax = upper.CL),</span></span>
<span id="cb114-38"><a href="#cb114-38" aria-hidden="true" tabindex="-1"></a>  <span class="co">#             color = "black",</span></span>
<span id="cb114-39"><a href="#cb114-39" aria-hidden="true" tabindex="-1"></a>  <span class="co">#           width = .05) +</span></span>
<span id="cb114-40"><a href="#cb114-40" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb114-41"><a href="#cb114-41" aria-hidden="true" tabindex="-1"></a>  <span class="co"># pvalue brackets</span></span>
<span id="cb114-42"><a href="#cb114-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_pvalue_manual</span>(exp1g_m1_planned_dt[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>],</span>
<span id="cb114-43"><a href="#cb114-43" aria-hidden="true" tabindex="-1"></a>                     <span class="at">label =</span> <span class="st">"pretty_p"</span>,</span>
<span id="cb114-44"><a href="#cb114-44" aria-hidden="true" tabindex="-1"></a>                     <span class="at">y.position =</span> <span class="fu">c</span>(<span class="dv">20</span>,<span class="dv">20</span>),</span>
<span id="cb114-45"><a href="#cb114-45" aria-hidden="true" tabindex="-1"></a>                     <span class="at">size =</span> <span class="fl">2.5</span>,</span>
<span id="cb114-46"><a href="#cb114-46" aria-hidden="true" tabindex="-1"></a>                     <span class="at">tip.length =</span> <span class="fl">0.01</span>) <span class="sc">+</span></span>
<span id="cb114-47"><a href="#cb114-47" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb114-48"><a href="#cb114-48" aria-hidden="true" tabindex="-1"></a>  <span class="co"># additive line + interaction p bracket</span></span>
<span id="cb114-49"><a href="#cb114-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_segment</span>(<span class="at">x =</span> <span class="fl">3.85</span>,</span>
<span id="cb114-50"><a href="#cb114-50" aria-hidden="true" tabindex="-1"></a>               <span class="at">y =</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>] <span class="sc">+</span> b[<span class="dv">3</span>],</span>
<span id="cb114-51"><a href="#cb114-51" aria-hidden="true" tabindex="-1"></a>               <span class="at">xend =</span> <span class="fl">4.15</span>,</span>
<span id="cb114-52"><a href="#cb114-52" aria-hidden="true" tabindex="-1"></a>               <span class="at">yend =</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>] <span class="sc">+</span> b[<span class="dv">3</span>],</span>
<span id="cb114-53"><a href="#cb114-53" aria-hidden="true" tabindex="-1"></a>               <span class="at">linetype =</span> <span class="st">"dashed"</span>,</span>
<span id="cb114-54"><a href="#cb114-54" aria-hidden="true" tabindex="-1"></a>               <span class="at">color =</span> <span class="st">"gray"</span>) <span class="sc">+</span></span>
<span id="cb114-55"><a href="#cb114-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bracket</span>(</span>
<span id="cb114-56"><a href="#cb114-56" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="fl">4.2</span>,</span>
<span id="cb114-57"><a href="#cb114-57" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>] <span class="sc">+</span> b[<span class="dv">3</span>],</span>
<span id="cb114-58"><a href="#cb114-58" aria-hidden="true" tabindex="-1"></a>    <span class="at">yend =</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>] <span class="sc">+</span> b[<span class="dv">3</span>] <span class="sc">+</span> b[<span class="dv">4</span>],</span>
<span id="cb114-59"><a href="#cb114-59" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="fu">paste0</span>(<span class="st">"interaction</span><span class="sc">\n</span><span class="st">p = "</span>, p_ixn),</span>
<span id="cb114-60"><a href="#cb114-60" aria-hidden="true" tabindex="-1"></a>    <span class="at">text.size =</span> <span class="dv">3</span>,</span>
<span id="cb114-61"><a href="#cb114-61" aria-hidden="true" tabindex="-1"></a>    <span class="at">text.hjust =</span> <span class="dv">0</span>,</span>
<span id="cb114-62"><a href="#cb114-62" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb114-63"><a href="#cb114-63" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb114-64"><a href="#cb114-64" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"GC (%)"</span>) <span class="sc">+</span></span>
<span id="cb114-65"><a href="#cb114-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> pal_okabe_ito_blue) <span class="sc">+</span></span>
<span id="cb114-66"><a href="#cb114-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_pubr</span>() <span class="sc">+</span></span>
<span id="cb114-67"><a href="#cb114-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">xlim =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="fl">4.1</span>)) <span class="sc">+</span></span>
<span id="cb114-68"><a href="#cb114-68" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb114-69"><a href="#cb114-69" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(), <span class="co"># no x-axis title</span></span>
<span id="cb114-70"><a href="#cb114-70" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">"none"</span></span>
<span id="cb114-71"><a href="#cb114-71" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span> </span>
<span id="cb114-72"><a href="#cb114-72" aria-hidden="true" tabindex="-1"></a>  <span class="cn">NULL</span></span>
<span id="cb114-73"><a href="#cb114-73" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb114-74"><a href="#cb114-74" aria-hidden="true" tabindex="-1"></a>exp1g_plot_a <span class="ot">&lt;-</span> <span class="fu">factor_wrap</span>(exp1g_plot_a)</span>
<span id="cb114-75"><a href="#cb114-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-76"><a href="#cb114-76" aria-hidden="true" tabindex="-1"></a><span class="co">#exp1g_plot_a</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb115"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb115-1"><a href="#cb115-1" aria-hidden="true" tabindex="-1"></a><span class="co"># get coefficients of model</span></span>
<span id="cb115-2"><a href="#cb115-2" aria-hidden="true" tabindex="-1"></a>b <span class="ot">&lt;-</span> exp1g_m1_coef[, <span class="st">"Estimate"</span>]</span>
<span id="cb115-3"><a href="#cb115-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb115-4"><a href="#cb115-4" aria-hidden="true" tabindex="-1"></a><span class="co"># get interaction p</span></span>
<span id="cb115-5"><a href="#cb115-5" aria-hidden="true" tabindex="-1"></a>p_ixn <span class="ot">&lt;-</span> exp1g_m1_planned_dt[contrast <span class="sc">==</span> <span class="st">"Interaction"</span>, pretty_p]</span>
<span id="cb115-6"><a href="#cb115-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb115-7"><a href="#cb115-7" aria-hidden="true" tabindex="-1"></a>dodge_width <span class="ot">=</span> <span class="fl">0.6</span></span>
<span id="cb115-8"><a href="#cb115-8" aria-hidden="true" tabindex="-1"></a>exp1g_plot_b <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> exp1g,</span>
<span id="cb115-9"><a href="#cb115-9" aria-hidden="true" tabindex="-1"></a>              <span class="fu">aes</span>(<span class="at">x =</span> treatment,</span>
<span id="cb115-10"><a href="#cb115-10" aria-hidden="true" tabindex="-1"></a>                  <span class="at">y =</span> gc,</span>
<span id="cb115-11"><a href="#cb115-11" aria-hidden="true" tabindex="-1"></a>                  <span class="at">group =</span> experiment_id)) <span class="sc">+</span></span>
<span id="cb115-12"><a href="#cb115-12" aria-hidden="true" tabindex="-1"></a>  <span class="co"># modeled experiment by treatment means</span></span>
<span id="cb115-13"><a href="#cb115-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> exp1g_m1_emm2,</span>
<span id="cb115-14"><a href="#cb115-14" aria-hidden="true" tabindex="-1"></a>            <span class="fu">aes</span>(<span class="at">y =</span> group_mean,</span>
<span id="cb115-15"><a href="#cb115-15" aria-hidden="true" tabindex="-1"></a>                <span class="at">color =</span> experiment_id),</span>
<span id="cb115-16"><a href="#cb115-16" aria-hidden="true" tabindex="-1"></a>            <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> dodge_width),</span>
<span id="cb115-17"><a href="#cb115-17" aria-hidden="true" tabindex="-1"></a>            <span class="at">alpha =</span> <span class="dv">1</span>,</span>
<span id="cb115-18"><a href="#cb115-18" aria-hidden="true" tabindex="-1"></a>            <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb115-19"><a href="#cb115-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">data =</span> exp1g_m1_emm2,</span>
<span id="cb115-20"><a href="#cb115-20" aria-hidden="true" tabindex="-1"></a>            <span class="fu">aes</span>(<span class="at">y =</span> group_mean,</span>
<span id="cb115-21"><a href="#cb115-21" aria-hidden="true" tabindex="-1"></a>                <span class="at">group =</span> experiment_id,</span>
<span id="cb115-22"><a href="#cb115-22" aria-hidden="true" tabindex="-1"></a>                <span class="at">color =</span> experiment_id),</span>
<span id="cb115-23"><a href="#cb115-23" aria-hidden="true" tabindex="-1"></a>            <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> dodge_width)) <span class="sc">+</span></span>
<span id="cb115-24"><a href="#cb115-24" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb115-25"><a href="#cb115-25" aria-hidden="true" tabindex="-1"></a>  <span class="co"># raw data</span></span>
<span id="cb115-26"><a href="#cb115-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> dodge_width),</span>
<span id="cb115-27"><a href="#cb115-27" aria-hidden="true" tabindex="-1"></a>             <span class="at">color =</span> <span class="st">"gray80"</span></span>
<span id="cb115-28"><a href="#cb115-28" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb115-29"><a href="#cb115-29" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb115-30"><a href="#cb115-30" aria-hidden="true" tabindex="-1"></a>  <span class="co"># modeled treatment means</span></span>
<span id="cb115-31"><a href="#cb115-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> exp1g_m1_emm_dt,</span>
<span id="cb115-32"><a href="#cb115-32" aria-hidden="true" tabindex="-1"></a>             <span class="fu">aes</span>(<span class="at">y =</span> emmean),</span>
<span id="cb115-33"><a href="#cb115-33" aria-hidden="true" tabindex="-1"></a>             <span class="at">color =</span> <span class="st">"black"</span>,</span>
<span id="cb115-34"><a href="#cb115-34" aria-hidden="true" tabindex="-1"></a>             <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb115-35"><a href="#cb115-35" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_errorbar(data = exp1g_m1_emm_dt,</span></span>
<span id="cb115-36"><a href="#cb115-36" aria-hidden="true" tabindex="-1"></a>  <span class="co">#            aes(y = emmean,</span></span>
<span id="cb115-37"><a href="#cb115-37" aria-hidden="true" tabindex="-1"></a>  <span class="co">#                ymin = lower.CL,</span></span>
<span id="cb115-38"><a href="#cb115-38" aria-hidden="true" tabindex="-1"></a>  <span class="co">#                ymax = upper.CL),</span></span>
<span id="cb115-39"><a href="#cb115-39" aria-hidden="true" tabindex="-1"></a>  <span class="co">#             color = "black",</span></span>
<span id="cb115-40"><a href="#cb115-40" aria-hidden="true" tabindex="-1"></a>  <span class="co">#           width = .05) +</span></span>
<span id="cb115-41"><a href="#cb115-41" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb115-42"><a href="#cb115-42" aria-hidden="true" tabindex="-1"></a>  <span class="co"># pvalue brackets</span></span>
<span id="cb115-43"><a href="#cb115-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_pvalue_manual</span>(exp1g_m1_planned_dt[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>],</span>
<span id="cb115-44"><a href="#cb115-44" aria-hidden="true" tabindex="-1"></a>                     <span class="at">label =</span> <span class="st">"pretty_p"</span>,</span>
<span id="cb115-45"><a href="#cb115-45" aria-hidden="true" tabindex="-1"></a>                     <span class="at">y.position =</span> <span class="fu">c</span>(<span class="dv">20</span>,<span class="dv">20</span>),</span>
<span id="cb115-46"><a href="#cb115-46" aria-hidden="true" tabindex="-1"></a>                     <span class="at">size =</span> <span class="fl">2.5</span>,</span>
<span id="cb115-47"><a href="#cb115-47" aria-hidden="true" tabindex="-1"></a>                     <span class="at">tip.length =</span> <span class="fl">0.01</span>) <span class="sc">+</span></span>
<span id="cb115-48"><a href="#cb115-48" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb115-49"><a href="#cb115-49" aria-hidden="true" tabindex="-1"></a>  <span class="co"># additive line + interaction p bracket</span></span>
<span id="cb115-50"><a href="#cb115-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_segment</span>(<span class="at">x =</span> <span class="dv">4</span> <span class="sc">-</span> dodge_width<span class="sc">/</span><span class="dv">2</span>,</span>
<span id="cb115-51"><a href="#cb115-51" aria-hidden="true" tabindex="-1"></a>               <span class="at">y =</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>] <span class="sc">+</span> b[<span class="dv">3</span>],</span>
<span id="cb115-52"><a href="#cb115-52" aria-hidden="true" tabindex="-1"></a>               <span class="at">xend =</span> <span class="dv">4</span> <span class="sc">+</span> dodge_width<span class="sc">/</span><span class="dv">2</span>,</span>
<span id="cb115-53"><a href="#cb115-53" aria-hidden="true" tabindex="-1"></a>               <span class="at">yend =</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>] <span class="sc">+</span> b[<span class="dv">3</span>],</span>
<span id="cb115-54"><a href="#cb115-54" aria-hidden="true" tabindex="-1"></a>               <span class="at">linetype =</span> <span class="st">"dashed"</span>,</span>
<span id="cb115-55"><a href="#cb115-55" aria-hidden="true" tabindex="-1"></a>               <span class="at">color =</span> <span class="st">"gray"</span>) <span class="sc">+</span></span>
<span id="cb115-56"><a href="#cb115-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bracket</span>(</span>
<span id="cb115-57"><a href="#cb115-57" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="dv">4</span> <span class="sc">+</span> dodge_width<span class="sc">/</span><span class="fl">1.9</span>,</span>
<span id="cb115-58"><a href="#cb115-58" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>] <span class="sc">+</span> b[<span class="dv">3</span>],</span>
<span id="cb115-59"><a href="#cb115-59" aria-hidden="true" tabindex="-1"></a>    <span class="at">yend =</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>] <span class="sc">+</span> b[<span class="dv">3</span>] <span class="sc">+</span> b[<span class="dv">4</span>],</span>
<span id="cb115-60"><a href="#cb115-60" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="fu">paste0</span>(<span class="st">"interaction</span><span class="sc">\n</span><span class="st">p = "</span>, p_ixn),</span>
<span id="cb115-61"><a href="#cb115-61" aria-hidden="true" tabindex="-1"></a>    <span class="at">text.size =</span> <span class="dv">3</span>,</span>
<span id="cb115-62"><a href="#cb115-62" aria-hidden="true" tabindex="-1"></a>    <span class="at">text.hjust =</span> <span class="dv">0</span>,</span>
<span id="cb115-63"><a href="#cb115-63" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb115-64"><a href="#cb115-64" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb115-65"><a href="#cb115-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"GC (%)"</span>) <span class="sc">+</span></span>
<span id="cb115-66"><a href="#cb115-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> pal_okabe_ito_blue) <span class="sc">+</span></span>
<span id="cb115-67"><a href="#cb115-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_pubr</span>() <span class="sc">+</span></span>
<span id="cb115-68"><a href="#cb115-68" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">xlim =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="fl">4.2</span>)) <span class="sc">+</span></span>
<span id="cb115-69"><a href="#cb115-69" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb115-70"><a href="#cb115-70" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(), <span class="co"># no x-axis title</span></span>
<span id="cb115-71"><a href="#cb115-71" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">"none"</span></span>
<span id="cb115-72"><a href="#cb115-72" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span> </span>
<span id="cb115-73"><a href="#cb115-73" aria-hidden="true" tabindex="-1"></a>  <span class="cn">NULL</span></span>
<span id="cb115-74"><a href="#cb115-74" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb115-75"><a href="#cb115-75" aria-hidden="true" tabindex="-1"></a>exp1g_plot_b <span class="ot">&lt;-</span> <span class="fu">factor_wrap</span>(exp1g_plot_b)</span>
<span id="cb115-76"><a href="#cb115-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb115-77"><a href="#cb115-77" aria-hidden="true" tabindex="-1"></a><span class="co"># exp1g_plot_b</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="lmm_files/figure-html/unnamed-chunk-61-1.png" class="img-fluid figure-img" width="1066"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="repeated-measures-anova-randomized-complete-block-with-no-subsampling" class="level3" data-number="14.12.2">
<h3 data-number="14.12.2" class="anchored" data-anchor-id="repeated-measures-anova-randomized-complete-block-with-no-subsampling"><span class="header-section-number">14.12.2</span> Repeated measures ANOVA (randomized complete block with no subsampling)</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb116"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb116-1"><a href="#cb116-1" aria-hidden="true" tabindex="-1"></a><span class="co"># this is the rm-ANOVA</span></span>
<span id="cb116-2"><a href="#cb116-2" aria-hidden="true" tabindex="-1"></a>m1 <span class="ot">&lt;-</span> <span class="fu">aov_4</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb116-3"><a href="#cb116-3" aria-hidden="true" tabindex="-1"></a>              (treatment <span class="sc">*</span> activity <span class="sc">|</span> donor),</span>
<span id="cb116-4"><a href="#cb116-4" aria-hidden="true" tabindex="-1"></a>            <span class="at">data =</span> exp5c)</span>
<span id="cb116-5"><a href="#cb116-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-6"><a href="#cb116-6" aria-hidden="true" tabindex="-1"></a><span class="co"># lmm equivalent</span></span>
<span id="cb116-7"><a href="#cb116-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-8"><a href="#cb116-8" aria-hidden="true" tabindex="-1"></a>m2 <span class="ot">&lt;-</span> <span class="fu">lmer</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb116-9"><a href="#cb116-9" aria-hidden="true" tabindex="-1"></a>         (<span class="dv">1</span> <span class="sc">|</span> donor) <span class="sc">+</span></span>
<span id="cb116-10"><a href="#cb116-10" aria-hidden="true" tabindex="-1"></a>         (<span class="dv">1</span> <span class="sc">|</span> donor<span class="sc">:</span>treatment) <span class="sc">+</span></span>
<span id="cb116-11"><a href="#cb116-11" aria-hidden="true" tabindex="-1"></a>         (<span class="dv">1</span> <span class="sc">|</span> donor<span class="sc">:</span>activity),</span>
<span id="cb116-12"><a href="#cb116-12" aria-hidden="true" tabindex="-1"></a>       <span class="at">data =</span> exp5c)</span>
<span id="cb116-13"><a href="#cb116-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-14"><a href="#cb116-14" aria-hidden="true" tabindex="-1"></a><span class="co"># random intercept and slope model that is *not* equivalent</span></span>
<span id="cb116-15"><a href="#cb116-15" aria-hidden="true" tabindex="-1"></a><span class="co"># this isn't solvable because there is no subsampling</span></span>
<span id="cb116-16"><a href="#cb116-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-17"><a href="#cb116-17" aria-hidden="true" tabindex="-1"></a>m3 <span class="ot">&lt;-</span> <span class="fu">lmer</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb116-18"><a href="#cb116-18" aria-hidden="true" tabindex="-1"></a>              (treatment <span class="sc">*</span> activity <span class="sc">|</span> donor),</span>
<span id="cb116-19"><a href="#cb116-19" aria-hidden="true" tabindex="-1"></a>            <span class="at">data =</span> exp5c)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Notes</p>
<ol type="1">
<li><code>afex</code> computes the repeated measures anova model using both <code>aov</code> (classical univariate repeated measures ANOVA) and using <code>lm</code> with multiple response variables (the multivariate repeated measures ANOVA). As of this writing, the output from <code>aov</code> is the default.</li>
<li>Given only one measure for each donor within a <span class="math inline">\(\texttt{treatment} \times \texttt{activity}\)</span> combination, the linear mixed model <code>m2</code> is equivalent to the univariate repeated measures model m1.</li>
<li>The model formula in <code>m1</code> looks like that in the linear mixed model <code>m3</code> but the two models are not equivalent.</li>
</ol>
<section id="univariate-vs.-multivariate-repeated-measures-anova" class="level4" data-number="14.12.2.1">
<h4 data-number="14.12.2.1" class="anchored" data-anchor-id="univariate-vs.-multivariate-repeated-measures-anova"><span class="header-section-number">14.12.2.1</span> univariate vs.&nbsp;multivariate repeated measures ANOVA</h4>
<p>Use the multivariate model unless you want to replicate the result of someone who used a univariate model. By default, <code>aov_4</code> computes only the multivariate model (prior to version xxx, the default was to compute both models).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb117"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb117-1"><a href="#cb117-1" aria-hidden="true" tabindex="-1"></a><span class="co"># default mode -- should be multivariate</span></span>
<span id="cb117-2"><a href="#cb117-2" aria-hidden="true" tabindex="-1"></a>exp5c_aov <span class="ot">&lt;-</span> <span class="fu">aov_4</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb117-3"><a href="#cb117-3" aria-hidden="true" tabindex="-1"></a>                    (treatment <span class="sc">*</span> activity <span class="sc">|</span> donor),</span>
<span id="cb117-4"><a href="#cb117-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> exp5c)</span>
<span id="cb117-5"><a href="#cb117-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb117-6"><a href="#cb117-6" aria-hidden="true" tabindex="-1"></a><span class="co"># force aov_4 to compute univariate model</span></span>
<span id="cb117-7"><a href="#cb117-7" aria-hidden="true" tabindex="-1"></a>exp5c_aov1 <span class="ot">&lt;-</span> <span class="fu">aov_4</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb117-8"><a href="#cb117-8" aria-hidden="true" tabindex="-1"></a>                    (treatment <span class="sc">*</span> activity <span class="sc">|</span> donor),</span>
<span id="cb117-9"><a href="#cb117-9" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> exp5c,</span>
<span id="cb117-10"><a href="#cb117-10" aria-hidden="true" tabindex="-1"></a>                  <span class="at">include_aov =</span> <span class="cn">TRUE</span>)</span>
<span id="cb117-11"><a href="#cb117-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb117-12"><a href="#cb117-12" aria-hidden="true" tabindex="-1"></a><span class="co"># explicitly exclude the univariate model</span></span>
<span id="cb117-13"><a href="#cb117-13" aria-hidden="true" tabindex="-1"></a>exp5c_aov2 <span class="ot">&lt;-</span> <span class="fu">aov_4</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb117-14"><a href="#cb117-14" aria-hidden="true" tabindex="-1"></a>                    (treatment <span class="sc">*</span> activity <span class="sc">|</span> donor),</span>
<span id="cb117-15"><a href="#cb117-15" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> exp5c,</span>
<span id="cb117-16"><a href="#cb117-16" aria-hidden="true" tabindex="-1"></a>                  <span class="at">include_aov =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Notes</p>
<ol type="1">
<li>The <code>include_aov = TRUE</code> argument forces output from <code>aov_4</code> to include the univariate model.</li>
</ol>
<p><strong>contrasts from multivariate model</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb118"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb118-1"><a href="#cb118-1" aria-hidden="true" tabindex="-1"></a><span class="co"># These three should give equivalent results</span></span>
<span id="cb118-2"><a href="#cb118-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb118-3"><a href="#cb118-3" aria-hidden="true" tabindex="-1"></a><span class="co"># exp5c_aov was fit with the default -- multivariate model only</span></span>
<span id="cb118-4"><a href="#cb118-4" aria-hidden="true" tabindex="-1"></a><span class="fu">emmeans</span>(exp5c_aov,</span>
<span id="cb118-5"><a href="#cb118-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">"treatment"</span>, <span class="st">"activity"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb118-6"><a href="#cb118-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">contrast</span>(<span class="at">method =</span> <span class="st">"revpairwise"</span>,</span>
<span id="cb118-7"><a href="#cb118-7" aria-hidden="true" tabindex="-1"></a>           <span class="at">simple =</span> <span class="st">"each"</span>,</span>
<span id="cb118-8"><a href="#cb118-8" aria-hidden="true" tabindex="-1"></a>           <span class="at">combine =</span> <span class="cn">TRUE</span>,</span>
<span id="cb118-9"><a href="#cb118-9" aria-hidden="true" tabindex="-1"></a>           <span class="at">adjust =</span> <span class="st">"none"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> activity treatment contrast      estimate     SE df t.ratio p.value
 Basal    .         siNR4A3 - Scr   0.0813 0.1012  5   0.804  0.4581
 EPS      .         siNR4A3 - Scr  -0.1858 0.0359  5  -5.180  0.0035
 .        Scr       EPS - Basal     0.1510 0.0395  5   3.821  0.0124
 .        siNR4A3   EPS - Basal    -0.1161 0.0626  5  -1.855  0.1228</code></pre>
</div>
<div class="sourceCode cell-code" id="cb120"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb120-1"><a href="#cb120-1" aria-hidden="true" tabindex="-1"></a><span class="co"># exp5c_aov included the univariate model but the default is still the multivariate model</span></span>
<span id="cb120-2"><a href="#cb120-2" aria-hidden="true" tabindex="-1"></a><span class="fu">emmeans</span>(exp5c_aov1,</span>
<span id="cb120-3"><a href="#cb120-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">"treatment"</span>, <span class="st">"activity"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb120-4"><a href="#cb120-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">contrast</span>(<span class="at">method =</span> <span class="st">"revpairwise"</span>,</span>
<span id="cb120-5"><a href="#cb120-5" aria-hidden="true" tabindex="-1"></a>           <span class="at">simple =</span> <span class="st">"each"</span>,</span>
<span id="cb120-6"><a href="#cb120-6" aria-hidden="true" tabindex="-1"></a>           <span class="at">combine =</span> <span class="cn">TRUE</span>,</span>
<span id="cb120-7"><a href="#cb120-7" aria-hidden="true" tabindex="-1"></a>           <span class="at">adjust =</span> <span class="st">"none"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> activity treatment contrast      estimate     SE df t.ratio p.value
 Basal    .         siNR4A3 - Scr   0.0813 0.1012  5   0.804  0.4581
 EPS      .         siNR4A3 - Scr  -0.1858 0.0359  5  -5.180  0.0035
 .        Scr       EPS - Basal     0.1510 0.0395  5   3.821  0.0124
 .        siNR4A3   EPS - Basal    -0.1161 0.0626  5  -1.855  0.1228</code></pre>
</div>
<div class="sourceCode cell-code" id="cb122"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb122-1"><a href="#cb122-1" aria-hidden="true" tabindex="-1"></a><span class="co"># passing `model = "multivariate"` makes the model choice transparent</span></span>
<span id="cb122-2"><a href="#cb122-2" aria-hidden="true" tabindex="-1"></a><span class="fu">emmeans</span>(exp5c_aov1,</span>
<span id="cb122-3"><a href="#cb122-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">"treatment"</span>, <span class="st">"activity"</span>),</span>
<span id="cb122-4"><a href="#cb122-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">model =</span> <span class="st">"multivariate"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb122-5"><a href="#cb122-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">contrast</span>(<span class="at">method =</span> <span class="st">"revpairwise"</span>,</span>
<span id="cb122-6"><a href="#cb122-6" aria-hidden="true" tabindex="-1"></a>           <span class="at">simple =</span> <span class="st">"each"</span>,</span>
<span id="cb122-7"><a href="#cb122-7" aria-hidden="true" tabindex="-1"></a>           <span class="at">combine =</span> <span class="cn">TRUE</span>,</span>
<span id="cb122-8"><a href="#cb122-8" aria-hidden="true" tabindex="-1"></a>           <span class="at">adjust =</span> <span class="st">"none"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> activity treatment contrast      estimate     SE df t.ratio p.value
 Basal    .         siNR4A3 - Scr   0.0813 0.1012  5   0.804  0.4581
 EPS      .         siNR4A3 - Scr  -0.1858 0.0359  5  -5.180  0.0035
 .        Scr       EPS - Basal     0.1510 0.0395  5   3.821  0.0124
 .        siNR4A3   EPS - Basal    -0.1161 0.0626  5  -1.855  0.1228</code></pre>
</div>
</div>
<p>Notes</p>
<ol type="1">
<li>If our rmANOVA model is fit with the default specification (<code>exp5c_aov1</code>), we can get the multivariate output using the <code>model = "multivariate"</code> argument within <code>emmeans</code> (not the <code>contrast</code> function!). Or, if our fit excluded the univariate model (<code>exp5c_aov2</code>), then we don’t need the <code>model</code> argument.</li>
</ol>
<p><strong>contrasts from univariate model</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb124"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb124-1"><a href="#cb124-1" aria-hidden="true" tabindex="-1"></a><span class="co"># get the univariate model results using $aov</span></span>
<span id="cb124-2"><a href="#cb124-2" aria-hidden="true" tabindex="-1"></a><span class="fu">emmeans</span>(exp5c_aov1<span class="sc">$</span>aov, <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">"treatment"</span>, <span class="st">"activity"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb124-3"><a href="#cb124-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">contrast</span>(<span class="at">method =</span> <span class="st">"revpairwise"</span>,</span>
<span id="cb124-4"><a href="#cb124-4" aria-hidden="true" tabindex="-1"></a>           <span class="at">simple =</span> <span class="st">"each"</span>,</span>
<span id="cb124-5"><a href="#cb124-5" aria-hidden="true" tabindex="-1"></a>           <span class="at">combine =</span> <span class="cn">TRUE</span>,</span>
<span id="cb124-6"><a href="#cb124-6" aria-hidden="true" tabindex="-1"></a>           <span class="at">adjust =</span> <span class="st">"none"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> activity treatment contrast      estimate     SE   df t.ratio p.value
 Basal    .         siNR4A3 - Scr   0.0813 0.0759 8.60   1.071  0.3132
 EPS      .         siNR4A3 - Scr  -0.1858 0.0759 8.60  -2.448  0.0380
 .        Scr       EPS - Basal     0.1510 0.0524 9.39   2.884  0.0173
 .        siNR4A3   EPS - Basal    -0.1161 0.0524 9.39  -2.218  0.0525</code></pre>
</div>
</div>
<p>Notes</p>
<ol type="1">
<li>Passing <code>exp5c_aov1</code> to <code>emmeans</code> will return the contrasts from the multivariate. change this to the univariate model by passing <code>exp5c_aov1$aov</code>.</li>
</ol>
</section>
<section id="the-anova-table" class="level4" data-number="14.12.2.2">
<h4 data-number="14.12.2.2" class="anchored" data-anchor-id="the-anova-table"><span class="header-section-number">14.12.2.2</span> The ANOVA table</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb126"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb126-1"><a href="#cb126-1" aria-hidden="true" tabindex="-1"></a><span class="fu">nice</span>(exp5c_aov1, <span class="at">correction =</span> <span class="st">"GG"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Anova Table (Type 3 tests)

Response: glucose_uptake
              Effect   df  MSE       F   ges p.value
1          treatment 1, 5 0.02    0.68  .006    .448
2           activity 1, 5 0.01    0.30 &lt;.001    .609
3 treatment:activity 1, 5 0.01 10.37 *  .037    .023
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '+' 0.1 ' ' 1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb128"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb128-1"><a href="#cb128-1" aria-hidden="true" tabindex="-1"></a><span class="fu">nice</span>(exp5c_aov1, <span class="at">correction =</span> <span class="st">"none"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Anova Table (Type 3 tests)

Response: glucose_uptake
              Effect   df  MSE       F   ges p.value
1          treatment 1, 5 0.02    0.68  .006    .448
2           activity 1, 5 0.01    0.30 &lt;.001    .609
3 treatment:activity 1, 5 0.01 10.37 *  .037    .023
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '+' 0.1 ' ' 1</code></pre>
</div>
</div>
<p>Notes</p>
<ol type="1">
<li>The Greenhouse-Geiger (“GG”) correction is the default. While the <code>correction = "GG"</code> argument is not needed, it makes the script more transparent.</li>
<li>For <em>these data</em> the Greenhouse-Geiger correction doesn’t make a difference in the table.</li>
</ol>
</section>
</section>
</section>
<section id="hidden-code" class="level2" data-number="14.13">
<h2 data-number="14.13" class="anchored" data-anchor-id="hidden-code"><span class="header-section-number">14.13</span> Hidden code</h2>
<section id="import-exp5c" class="level3" data-number="14.13.1">
<h3 data-number="14.13.1" class="anchored" data-anchor-id="import-exp5c"><span class="header-section-number">14.13.1</span> Import exp5c</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb130"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb130-1"><a href="#cb130-1" aria-hidden="true" tabindex="-1"></a>data_from <span class="ot">&lt;-</span> <span class="st">"Transcriptomic profiling of skeletal muscle adaptations to exercise and inactivity"</span></span>
<span id="cb130-2"><a href="#cb130-2" aria-hidden="true" tabindex="-1"></a>file_name <span class="ot">&lt;-</span> <span class="st">"41467_2019_13869_MOESM6_ESM.xlsx"</span></span>
<span id="cb130-3"><a href="#cb130-3" aria-hidden="true" tabindex="-1"></a>file_path <span class="ot">&lt;-</span> <span class="fu">here</span>(data_folder, data_from, file_name)</span>
<span id="cb130-4"><a href="#cb130-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-5"><a href="#cb130-5" aria-hidden="true" tabindex="-1"></a>exp5c_wide <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(file_path,</span>
<span id="cb130-6"><a href="#cb130-6" aria-hidden="true" tabindex="-1"></a>                         <span class="at">sheet =</span> <span class="st">"Fig5c"</span>,</span>
<span id="cb130-7"><a href="#cb130-7" aria-hidden="true" tabindex="-1"></a>                         <span class="at">range =</span> <span class="st">"A2:M3"</span>,</span>
<span id="cb130-8"><a href="#cb130-8" aria-hidden="true" tabindex="-1"></a>                         <span class="at">col_names =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb130-9"><a href="#cb130-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.table</span>() <span class="sc">%&gt;%</span></span>
<span id="cb130-10"><a href="#cb130-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">transpose</span>(<span class="at">make.names =</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>New names:
• `` -&gt; `...1`
• `` -&gt; `...2`
• `` -&gt; `...3`
• `` -&gt; `...4`
• `` -&gt; `...5`
• `` -&gt; `...6`
• `` -&gt; `...7`
• `` -&gt; `...8`
• `` -&gt; `...9`
• `` -&gt; `...10`
• `` -&gt; `...11`
• `` -&gt; `...12`
• `` -&gt; `...13`</code></pre>
</div>
<div class="sourceCode cell-code" id="cb132"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb132-1"><a href="#cb132-1" aria-hidden="true" tabindex="-1"></a>activity_levels <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Basal"</span>, <span class="st">"EPS"</span>)</span>
<span id="cb132-2"><a href="#cb132-2" aria-hidden="true" tabindex="-1"></a>treatment_levels <span class="ot">&lt;-</span> <span class="fu">names</span>(exp5c_wide)</span>
<span id="cb132-3"><a href="#cb132-3" aria-hidden="true" tabindex="-1"></a>exp5c_wide[, activity <span class="sc">:</span><span class="er">=</span> <span class="fu">rep</span>(activity_levels, <span class="at">each =</span> <span class="dv">6</span>)]</span>
<span id="cb132-4"><a href="#cb132-4" aria-hidden="true" tabindex="-1"></a>exp5c_wide[, activity <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(activity, <span class="at">levels =</span> activity_levels)]</span>
<span id="cb132-5"><a href="#cb132-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb132-6"><a href="#cb132-6" aria-hidden="true" tabindex="-1"></a>exp5c <span class="ot">&lt;-</span> <span class="fu">melt</span>(exp5c_wide,</span>
<span id="cb132-7"><a href="#cb132-7" aria-hidden="true" tabindex="-1"></a>              <span class="at">id.vars =</span> <span class="st">"activity"</span>,</span>
<span id="cb132-8"><a href="#cb132-8" aria-hidden="true" tabindex="-1"></a>              <span class="at">variable.name =</span> <span class="st">"treatment"</span>,</span>
<span id="cb132-9"><a href="#cb132-9" aria-hidden="true" tabindex="-1"></a>              <span class="at">value.name =</span> <span class="st">"glucose_uptake"</span>)</span>
<span id="cb132-10"><a href="#cb132-10" aria-hidden="true" tabindex="-1"></a>exp5c[, treatment <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(treatment, <span class="at">levels =</span> treatment_levels)]</span>
<span id="cb132-11"><a href="#cb132-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb132-12"><a href="#cb132-12" aria-hidden="true" tabindex="-1"></a>exp5c[, donor <span class="sc">:</span><span class="er">=</span> <span class="fu">rep</span>(<span class="fu">paste0</span>(<span class="st">"donor_"</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>), <span class="dv">4</span>)]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="import-exp1g" class="level3" data-number="14.13.2">
<h3 data-number="14.13.2" class="anchored" data-anchor-id="import-exp1g"><span class="header-section-number">14.13.2</span> Import exp1g</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb133"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb133-1"><a href="#cb133-1" aria-hidden="true" tabindex="-1"></a>data_from <span class="ot">&lt;-</span> <span class="st">"A GPR174–CCL21 module imparts sexual dimorphism to humoral immunity"</span></span>
<span id="cb133-2"><a href="#cb133-2" aria-hidden="true" tabindex="-1"></a>file_name <span class="ot">&lt;-</span> <span class="st">"41586_2019_1873_MOESM3_ESM.xlsx"</span></span>
<span id="cb133-3"><a href="#cb133-3" aria-hidden="true" tabindex="-1"></a>file_path <span class="ot">&lt;-</span> <span class="fu">here</span>(data_folder, data_from, file_name)</span>
<span id="cb133-4"><a href="#cb133-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb133-5"><a href="#cb133-5" aria-hidden="true" tabindex="-1"></a>exp1g_wide <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(file_path,</span>
<span id="cb133-6"><a href="#cb133-6" aria-hidden="true" tabindex="-1"></a>                         <span class="at">sheet =</span> <span class="st">"Fig 1g"</span>,</span>
<span id="cb133-7"><a href="#cb133-7" aria-hidden="true" tabindex="-1"></a>                         <span class="at">range =</span> <span class="st">"B4:E25"</span>,</span>
<span id="cb133-8"><a href="#cb133-8" aria-hidden="true" tabindex="-1"></a>                         <span class="at">col_types =</span> <span class="fu">c</span>(<span class="st">"numeric"</span>),</span>
<span id="cb133-9"><a href="#cb133-9" aria-hidden="true" tabindex="-1"></a>                         <span class="at">col_names =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb133-10"><a href="#cb133-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.table</span>()</span>
<span id="cb133-11"><a href="#cb133-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb133-12"><a href="#cb133-12" aria-hidden="true" tabindex="-1"></a>genotype_levels <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Gpr174+"</span>, <span class="st">"Gpr174-"</span>)</span>
<span id="cb133-13"><a href="#cb133-13" aria-hidden="true" tabindex="-1"></a>sex_levels <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"M"</span>, <span class="st">"F"</span>)</span>
<span id="cb133-14"><a href="#cb133-14" aria-hidden="true" tabindex="-1"></a>g.by.s_levels <span class="ot">&lt;-</span> <span class="fu">do.call</span>(paste, <span class="fu">expand.grid</span>(genotype_levels, sex_levels))</span>
<span id="cb133-15"><a href="#cb133-15" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(exp1g_wide) <span class="ot">&lt;-</span> g.by.s_levels</span>
<span id="cb133-16"><a href="#cb133-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb133-17"><a href="#cb133-17" aria-hidden="true" tabindex="-1"></a>exp_levels <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"exp_"</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>)</span>
<span id="cb133-18"><a href="#cb133-18" aria-hidden="true" tabindex="-1"></a>exp1g_wide[, experiment_id <span class="sc">:</span><span class="er">=</span> <span class="fu">rep</span>(exp_levels, <span class="fu">c</span>(<span class="dv">5</span>,<span class="dv">6</span>,<span class="dv">6</span>,<span class="dv">5</span>))] <span class="co">#check!</span></span>
<span id="cb133-19"><a href="#cb133-19" aria-hidden="true" tabindex="-1"></a>exp1g_wide[, experiment_id <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(experiment_id)] <span class="co">#check!</span></span>
<span id="cb133-20"><a href="#cb133-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb133-21"><a href="#cb133-21" aria-hidden="true" tabindex="-1"></a>exp1g <span class="ot">&lt;-</span> <span class="fu">melt</span>(exp1g_wide,</span>
<span id="cb133-22"><a href="#cb133-22" aria-hidden="true" tabindex="-1"></a>              <span class="at">id.vars =</span> <span class="st">"experiment_id"</span>,</span>
<span id="cb133-23"><a href="#cb133-23" aria-hidden="true" tabindex="-1"></a>              <span class="at">measure.vars =</span> g.by.s_levels,</span>
<span id="cb133-24"><a href="#cb133-24" aria-hidden="true" tabindex="-1"></a>              <span class="at">variable.name =</span> <span class="st">"treatment"</span>,</span>
<span id="cb133-25"><a href="#cb133-25" aria-hidden="true" tabindex="-1"></a>              <span class="at">value.name =</span> <span class="st">"gc"</span>) <span class="sc">%&gt;%</span> <span class="co"># cell count</span></span>
<span id="cb133-26"><a href="#cb133-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">na.omit</span>()</span>
<span id="cb133-27"><a href="#cb133-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb133-28"><a href="#cb133-28" aria-hidden="true" tabindex="-1"></a>exp1g[, <span class="fu">c</span>(<span class="st">"genotype"</span>, <span class="st">"sex"</span>)<span class="sc">:</span><span class="er">=</span> <span class="fu">tstrsplit</span>(treatment,</span>
<span id="cb133-29"><a href="#cb133-29" aria-hidden="true" tabindex="-1"></a>                                             <span class="st">" "</span>,</span>
<span id="cb133-30"><a href="#cb133-30" aria-hidden="true" tabindex="-1"></a>                                             <span class="at">fixed =</span> <span class="cn">TRUE</span>)]</span>
<span id="cb133-31"><a href="#cb133-31" aria-hidden="true" tabindex="-1"></a>exp1g[, genotype <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(genotype,</span>
<span id="cb133-32"><a href="#cb133-32" aria-hidden="true" tabindex="-1"></a>                           <span class="at">levels =</span> genotype_levels)]</span>
<span id="cb133-33"><a href="#cb133-33" aria-hidden="true" tabindex="-1"></a>exp1g[, sex <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(sex,</span>
<span id="cb133-34"><a href="#cb133-34" aria-hidden="true" tabindex="-1"></a>                           <span class="at">levels =</span> sex_levels)]</span>
<span id="cb133-35"><a href="#cb133-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb133-36"><a href="#cb133-36" aria-hidden="true" tabindex="-1"></a>exp1g_means <span class="ot">&lt;-</span> exp1g[, .(<span class="at">gc =</span> <span class="fu">mean</span>(gc)),</span>
<span id="cb133-37"><a href="#cb133-37" aria-hidden="true" tabindex="-1"></a>                     by <span class="ot">=</span> .(treatment, genotype, sex, experiment_id)]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="../chapters/two-factors.html" class="pagination-link" aria-label="Linear models with two categorical $X$ -- Factorial linear models (&quot;two-way ANOVA&quot;)">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">13</span>&nbsp; <span class="chapter-title">Linear models with two categorical <span class="math inline">\(X\)</span> – Factorial linear models (“two-way ANOVA”)</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
  </div>
</nav>
</div> <!-- /content -->




</body></html>